diff --git a/setup.py b/setup.py
index 029584c..32d3558 100644
--- a/setup.py
+++ b/setup.py
@@ -31,7 +31,7 @@ CMAKE_BUILD_DIR = os.path.join(TOP_DIR, ".setuptools-cmake-build")
 
 WINDOWS = os.name == "nt"
 
-CMAKE = shutil.which("cmake3") or shutil.which("cmake")
+CMAKE = shutil.which("cmake")
 
 ################################################################################
 # Global variables for controlling the build variant
