diff --git a/setup.py b/setup.py
index 52399d2..542ca41 100644
--- a/setup.py
+++ b/setup.py
@@ -97,6 +97,10 @@ class CMakeBuild(build_ext):
             assert os.environ['TORCH_CUDA_ARCH_LIST'] is not None
             print(f'TORCH_CUDA_ARCH_LIST={os.environ["TORCH_CUDA_ARCH_LIST"]}')
 
+        if CMakeBuild.check_env_flag('USE_CUDNN', "0"):
+            cmake_args.append('-DCAFFE2_USE_CUDNN=ON')
+        if CMakeBuild.check_env_flag('USE_CUFILE', "0"):
+            cmake_args.append('-DCAFFE2_USE_CUFILE=ON')
         if CMakeBuild.check_env_flag('USE_MKL_BLAS'):
             include_dir = f'{sysconfig.get_path("data")}{os.sep}include'
             cmake_args.append(f'-DBLAS_INCLUDE_DIR={include_dir}')
@@ -158,6 +162,9 @@ setup(
     name='pyg_lib',
     version=__version__,
     install_requires=install_requires,
+    package_data={
+        'pyg_lib': ['csrc/**/*.h', 'csrc/**/*.hpp']
+    },
     packages=find_packages(),
     ext_modules=ext_modules,
     cmdclass=cmdclass,
--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,7 @@
             f'-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}',
             f'-DCMAKE_RUNTIME_OUTPUT_DIRECTORY={extdir}',
             f'-DCMAKE_BUILD_TYPE={self.build_type}',
-            f'-DCMAKE_PREFIX_PATH={torch.utils.cmake_prefix_path}',
+            f'-DCMAKE_PREFIX_PATH={torch.utils.cmake_prefix_path};{os.environ.get("CMAKE_PREFIX_PATH", "")}',
         ]
 
         if WITH_CUDA and not os.getenv('TORCH_CUDA_ARCH_LIST'):
