diff --git a/build/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp b/build/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp
index 528496c..e3fcca7 100644
--- a/build/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp
+++ b/build/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <cassert>
+#include <cstdint>
 #include <iostream>
 #include <memory>
 #include <sstream>
diff --git a/build/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp b/build/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp
index afe1553..62ef0d1 100644
--- a/build/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp
+++ b/build/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp
@@ -8,6 +8,7 @@
 #pragma once
 
 // System include(s)
+#include <climits>
 #include <memory>
 
 // Project include(s).
--- a/build/_deps/traccc-src/io/include/traccc/io/csv/dfe.hpp
+++ b/build/_deps/traccc-src/io/include/traccc/io/csv/dfe.hpp
@@ -8,6 +8,7 @@
 #include <algorithm>
 #include <array>
 #include <cassert>
+#include <cstdint>
 #include <fstream>
 #include <ostream>
 #include <sstream>
--- a/acts-46.8.1/Detray/io/include/detray/io/csv/dfe.hpp
+++ b/acts-46.8.1/Detray/io/include/detray/io/csv/dfe.hpp
@@ -38,7 +38,9 @@
 #include <algorithm>
 #include <array>
 #include <cassert>
+#include <cstdint>
 #include <fstream>
+#include <iomanip>
 #include <ostream>
 #include <sstream>
 #include <string>
--- a/acts-46.8.1/Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvInputOutput.hpp
+++ b/acts-46.8.1/Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvInputOutput.hpp
@@ -14,6 +14,7 @@
 #include <algorithm>
 #include <array>
 #include <cassert>
+#include <cstdint>
 #include <fstream>
 #include <limits>
 #include <ostream>
