diff --git a/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp b/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp
index 528496c..e3fcca7 100644
--- a/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp
+++ b/_deps/covfie-src/lib/core/covfie/core/utility/binary_io.hpp
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <cassert>
+#include <cstdint>
 #include <iostream>
 #include <memory>
 #include <sstream>
diff --git a/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp b/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp
index afe1553..62ef0d1 100644
--- a/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp
+++ b/_deps/traccc-src/core/include/traccc/gbts_seeding/gbts_seeding_config.hpp
@@ -8,6 +8,7 @@
 #pragma once
 
 // System include(s)
+#include <climits>
 #include <memory>
 
 // Project include(s).
