diff --git a/src/targets/gpu/lowering.cpp b/src/targets/gpu/lowering.cpp
--- a/src/targets/gpu/lowering.cpp
+++ b/src/targets/gpu/lowering.cpp
@@ -277,7 +277,11 @@
             // if the hardware is defaulted to use rocBLAS (such as gfx90).
             if(not has_fp8_inputs and
                ((string_value_of(MIGRAPHX_SET_GEMM_PROVIDER{}) == "rocblas") or
+#if MIGRAPHX_USE_HIPBLASLT
                 not hipblaslt_supported() or gpu::gfx_default_rocblas()))
+#else
+                true))
+#endif
             {
                 return mod->replace_instruction(
                     ins, rocblas_gemm<Op>{Op{}, 1, 0, compute_fp32}, refs);
