diff --git a/hipamd/include/hip/amd_detail/host_defines.h b/hipamd/include/hip/amd_detail/host_defines.h
index f0d1b57..8dc6817 100644
--- a/hipamd/include/hip/amd_detail/host_defines.h
+++ b/hipamd/include/hip/amd_detail/host_defines.h
@@ -353,7 +353,8 @@ typedef __hip_internal::int64_t __hip_int64_t;
 
 #define __global__
 
-#define __noinline__
+// See: https://github.com/ROCm/rocm-systems/issues/9897
+// #define __noinline__
 #define __forceinline__ inline
 
 #define __shared__
