diff --git a/shared/stinkytofu/src/transforms/asm/StinkyWmmaVgprReorderPass.cpp b/shared/stinkytofu/src/transforms/asm/StinkyWmmaVgprReorderPass.cpp
index 0a00d71..e25dd06 100644
--- a/shared/stinkytofu/src/transforms/asm/StinkyWmmaVgprReorderPass.cpp
+++ b/shared/stinkytofu/src/transforms/asm/StinkyWmmaVgprReorderPass.cpp
@@ -55,6 +55,7 @@
 #include <set>
 #include <unordered_map>
 #include <utility>
+#include <limits.h>
 
 #define DEBUG_TYPE "StinkyWmmaVgprReorderPass"
 
