--- a/projects/rocprofiler-systems/cmake/DyninstElfUtils.cmake
+++ b/projects/rocprofiler-systems/cmake/DyninstElfUtils.cmake
@@ -139,7 +139,7 @@
             ${CMAKE_COMMAND} -E env CC=${CMAKE_C_COMPILER}
             CFLAGS=-fPIC\ -O3\ -Wno-error=maybe-uninitialized CXX=${CMAKE_CXX_COMPILER}
             CXXFLAGS=-fPIC\ -O3\ -Wno-error=maybe-uninitialized
-            [=[LDFLAGS=-Wl,-rpath='$$ORIGIN' -pthread]=] <SOURCE_DIR>/configure
+            <SOURCE_DIR>/configure
             --enable-install-elfh --prefix=${_eu_root} --disable-libdebuginfod
             --disable-debuginfod --enable-thread-safety --disable-nls
             ${ElfUtils_CONFIG_OPTIONS} --libdir=${_eu_root}/lib
