diff --git a/compile.mk b/compile.mk
index 4201a4a..8c96177 100755
--- a/compile.mk
+++ b/compile.mk
@@ -15,11 +15,11 @@ ifeq ($(shell uname),Darwin)
 	 -ffast-math
   endif
   ifeq  ($(shell uname -m),i386) 
-     FFF = ifort -noautomatic -warn
-     FF90 = ifort -noautomatic -warn
+#    FFF = ifort -noautomatic -warn
+#    FF90 = ifort -noautomatic -warn
 #     FFF= g77 -O1 -Wall -fno-automatic -Wno-globals
-#     FFF = gfortran -fnoautomatic
-#	FF90 = gfortran
+     FFF = gfortran -O1 -Wall -fno-automatic -fno-backslash -ffast-math
+     FF90 = gfortran
   endif
 endif
 # Unix-ALPHA fortran
diff --git a/compile.mk b/compile.mk
index 34d704a..8011a1d 100755
--- a/compile.mk
+++ b/compile.mk
@@ -14,7 +14,7 @@ ifeq ($(shell uname),Darwin)
      FFF = g77 -O1 -Wall -fno-automatic -Wno-globals -fno-backslash \
 	 -ffast-math
   endif
-  ifeq  ($(shell uname -m),i386) 
+  ifeq  ($(shell uname -p),i386) 
 #    FFF = ifort -noautomatic -warn
 #    FF90 = ifort -noautomatic -warn
 #     FFF= g77 -O1 -Wall -fno-automatic -Wno-globals
