--- modules/ssl/ssl_engine_vars.c.orig	2010-07-24 12:15:59.000000000 +0200
+++ modules/ssl/ssl_engine_vars.c	2010-07-24 12:17:15.000000000 +0200
@@ -604,7 +604,7 @@
         result = "GENEROUS";
     else
         /* client verification failed */
-        result = apr_psprintf(p, "FAILED:%s", verr);
+        result = apr_psprintf(p, "FAILED:%s", verr ? verr : X509_verify_cert_error_string(vrc));
 
     if (xs)
         X509_free(xs);
