diff --git a/tools/genutil.h b/tools/genutil.h
index 759924a..f16c161 100644
--- a/tools/genutil.h
+++ b/tools/genutil.h
@@ -13,11 +13,7 @@ bool endswith(const char *str, const char *suffix)
 }
 
 static
-#if defined(__APPLE__) || defined(__FREEBSD__)
-int selectRPMs(struct dirent *ent)
-#else
 int selectRPMs(const struct dirent *ent)
-#endif
 {
    const char *d = ent->d_name;
    return (*d != '.' && endswith(d, ".rpm"));
