diff --git a/apt-pkg/contrib/progress.h b/apt-pkg/contrib/progress.h
index fd32f13..6926ad7 100644
--- a/apt-pkg/contrib/progress.h
+++ b/apt-pkg/contrib/progress.h
@@ -24,6 +24,7 @@
 #include <map>
 #include <string>
 #include <sys/time.h>
+#include <unistd.h>
 
 using std::string;
 using std::map;
diff --git a/apt-pkg/pkgcache.h b/apt-pkg/pkgcache.h
index ef5f897..cc2cdb6 100644
--- a/apt-pkg/pkgcache.h
+++ b/apt-pkg/pkgcache.h
@@ -21,6 +21,7 @@
 
 #include <string>
 #include <time.h>
+#include <unistd.h>
 #include <apt-pkg/mmap.h>
 
 using std::string;
diff --git a/apt-pkg/srcrecords.h b/apt-pkg/srcrecords.h
index 924617c..802fd6b 100644
--- a/apt-pkg/srcrecords.h
+++ b/apt-pkg/srcrecords.h
@@ -14,7 +14,8 @@
 #define PKGLIB_SRCRECORDS_H
 
 #include <string>
-#include <vector>    
+#include <vector>
+#include <unistd.h>
 
 using std::string;
 using std::vector;
diff --git a/apt-pkg/tagfile.h b/apt-pkg/tagfile.h
index 8bccb96..18e435d 100644
--- a/apt-pkg/tagfile.h
+++ b/apt-pkg/tagfile.h
@@ -22,6 +22,7 @@
 
 #include <apt-pkg/fileutl.h>
 #include <stdio.h>
+#include <unistd.h>
     
 class pkgTagSection
 {
