diff --git a/buildlib/install-sh b/buildlib/install-sh
old mode 100644
new mode 100755
diff --git a/tools/genpkglist.cc b/tools/genpkglist.cc
index 34d78e4..612d0e6 100644
--- a/tools/genpkglist.cc
+++ b/tools/genpkglist.cc
@@ -13,6 +13,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <assert.h>
+#include <climits>
 
 #include <map>
 #include <iostream>
diff --git a/tools/gensrclist.cc b/tools/gensrclist.cc
index 06e6743..cb63197 100644
--- a/tools/gensrclist.cc
+++ b/tools/gensrclist.cc
@@ -13,6 +13,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <assert.h>
+#include <climits>
 
 #include <map>
 #include <list>
