
Index: src/couchapp/Others/_id
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/_id	(revision )
+++ src/couchapp/Others/_id	(revision )
@@ -0,0 +1,1 @@
+_design/Others
\ No newline at end of file
Index: src/couchapp/Others/views/sites/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/sites/map.js	(revision )
+++ src/couchapp/Others/views/sites/map.js	(revision )
@@ -0,0 +1,6 @@
+function(doc) {
+        if (doc.lfn){
+                emit(doc.source, 1);
+                emit(doc.destination, 1);
+                }
+        }
Index: src/couchapp/Others/views/sites/reduce.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/sites/reduce.js	(revision )
+++ src/couchapp/Others/views/sites/reduce.js	(revision )
@@ -0,0 +1,1 @@
+_sum
\ No newline at end of file
Index: src/couchapp/Others/views/forKill/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/forKill/map.js	(revision )
+++ src/couchapp/Others/views/forKill/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+	if (doc.workflow && (doc.state=='new' || doc.state=='acquired' || doc.state=='retry')){
+		emit(doc.workflow,doc._id);
+	}
+}
Index: src/couchapp/Others/views/forResub/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/forResub/map.js	(revision )
+++ src/couchapp/Others/views/forResub/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+	if (doc.workflow && (doc.state=='failed'||doc.state=='killed')){
+		emit(doc.workflow, doc._id);
+	}
+}
Index: src/couchapp/Others/views/getFilesToRetry/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/getFilesToRetry/map.js	(revision )
+++ src/couchapp/Others/views/getFilesToRetry/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+	if (doc.state == 'retry' && doc.lfn) {
+		emit(doc._id, doc.last_update);
+	}
+}
Index: src/couchapp/Others/views/UserByStartTime/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/UserByStartTime/map.js	(revision )
+++ src/couchapp/Others/views/UserByStartTime/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+        if (doc.state != 'failed' && doc.state != 'done' && doc.lfn) {
+                emit([doc.user, doc.group, doc.role, doc.start_time.split('.')[0]], 1);
+        }
+}
Index: src/couchapp/Others/views/LFNSiteByLastUpdate/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/LFNSiteByLastUpdate/map.js	(revision )
+++ src/couchapp/Others/views/LFNSiteByLastUpdate/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+	if(doc.source_lfn && doc.state == 'done'){
+		emit(doc.last_update, {"lfn": doc.source_lfn, "location": doc.source});
+	}
+}
Index: src/couchapp/Others/views/TransfersByFailuresReasons/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/TransfersByFailuresReasons/map.js	(revision )
+++ src/couchapp/Others/views/TransfersByFailuresReasons/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+	if(doc.state == 'failed'){
+		emit([doc.destination, doc.source, doc.user, doc.workflow, doc.end_time, doc.failure_reason], doc.lfn);
+	}
+}
Index: src/couchapp/Others/views/TransfersByFailuresReasons/reduce.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/views/TransfersByFailuresReasons/reduce.js	(revision )
+++ src/couchapp/Others/views/TransfersByFailuresReasons/reduce.js	(revision )
@@ -0,0 +1,1 @@
+_count
Index: src/couchapp/Others/language
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/language	(revision )
+++ src/couchapp/Others/language	(revision )
@@ -0,0 +1,1 @@
+javascript
\ No newline at end of file
Index: src/couchapp/Others/.couchapprc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/.couchapprc	(revision )
+++ src/couchapp/Others/.couchapprc	(revision )
@@ -0,0 +1,1 @@
+{}
\ No newline at end of file
Index: src/couchapp/Others/couchapp.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/couchapp.json	(revision )
+++ src/couchapp/Others/couchapp.json	(revision )
@@ -0,0 +1,4 @@
+{
+    "name": "Name of your CouchApp",
+    "description": "CouchApp"
+}
\ No newline at end of file
Index: src/couchapp/Others/validate_doc_update.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Others/validate_doc_update.js	(revision )
+++ src/couchapp/Others/validate_doc_update.js	(revision )
@@ -0,0 +1,56 @@
+function(newDoc, oldDoc, userCtx) {
+    // Determines the doc operation type
+    var DOCOPS = {
+        modif : 0,
+        creat : 1,
+        delet : 2
+    };
+    var docOp = oldDoc ? (newDoc._deleted === true ? DOCOPS.delet : DOCOPS.modif) : DOCOPS.creat;
+
+    // Function to get the user list of site/groups for the given role
+    var getRole = function(role) {
+        var roles = userCtx.roles;
+        for (i in roles) {
+            if ( typeof (roles[i]) == "object" && roles[i][0] === role)
+                return roles[i][1];
+            // Request comes from backend auth handler
+            if ( typeof (roles[i]) == "string" && roles[i] === role)
+                return [];
+            // Request comes from other handlers (i.e. host auth)
+        }
+        return null;
+    }
+    // Function to check if user has the role for a given group or site
+    var matchesRole = function(role, grpsite) {
+        var r = getRole(role);
+        if (r != null)
+            if (grpsite === "" || r.indexOf(grpsite) != -1)
+                return true;
+        return false;
+    }
+    // Gets whether the user is a global admin
+    // name=null means requests coming from the local replicator, so we must allow
+    // (the cms couch auth does not allow name=null, so it affects only internal
+    // replication requests)
+    var isGlobalAdm = (userCtx.name === null) ||
+                      matchesRole("_admin", "") ||
+                      matchesRole("-admin", "group:couchdb");
+
+    //---------------------------------
+    // Authorization rules for Myapp DB
+
+    // The following rule aplies for all operation types
+    var allowed = isGlobalAdm ||
+                  matchesRole("operator", "group:aso") ||
+                  matchesRole("web-service", "group:facops") ||
+                  ((newDoc._deleted === true || newDoc.user === userCtx.name) &&
+                  (!oldDoc || oldDoc.user === userCtx.name));
+
+    // Throw if user not validated
+    if (!allowed) {
+        log(toJSON(userCtx));
+        throw {
+            forbidden : "User not authorized for action."
+        };
+    }
+}
Index: src/couchapp/Acquired/views/get_acquired/map.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Acquired/views/get_acquired/map.js	(revision )
+++ src/couchapp/Acquired/views/get_acquired/map.js	(revision )
@@ -0,0 +1,5 @@
+function(doc) {
+        if (doc.state == 'acquired'&& doc.lfn) {
+		emit([doc.user, doc.group, doc.role, doc.destination, doc.source], doc.lfn);
+	}
+}
Index: src/couchapp/Acquired/views/get_acquired/reduce.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Acquired/views/get_acquired/reduce.js	(revision )
+++ src/couchapp/Acquired/views/get_acquired/reduce.js	(revision )
@@ -0,0 +1,1 @@
+_count
Index: src/couchapp/Acquired/language
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/couchapp/Acquired/language	(revision )
+++ src/couchapp/Acquired/language	(revision )
@@ -0,0 +1,1 @@
+javascript
\ No newline at end of file
Index: src/couchapp/AsyncTransfer/views/get_acquired/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/get_acquired/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/get_acquired/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-        if (doc.state == 'acquired'&& doc.lfn) {
-		emit([doc.user, doc.group, doc.role, doc.destination, doc.source], doc.lfn);
-	}
-}
Index: src/couchapp/AsyncTransfer/views/get_acquired/reduce.js
===================================================================
--- src/couchapp/AsyncTransfer/views/get_acquired/reduce.js	(revision )
+++ src/couchapp/AsyncTransfer/views/get_acquired/reduce.js	(revision )
@@ -1,1 +0,0 @@
-_count
Index: src/couchapp/AsyncTransfer/views/TransfersByFailuresReasons/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/TransfersByFailuresReasons/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/TransfersByFailuresReasons/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-	if(doc.state == 'failed'){
-		emit([doc.destination, doc.source, doc.user, doc.workflow, doc.end_time, doc.failure_reason], doc.lfn);
-	}
-}
Index: src/couchapp/AsyncTransfer/views/TransfersByFailuresReasons/reduce.js
===================================================================
--- src/couchapp/AsyncTransfer/views/TransfersByFailuresReasons/reduce.js	(revision )
+++ src/couchapp/AsyncTransfer/views/TransfersByFailuresReasons/reduce.js	(revision )
@@ -1,1 +0,0 @@
-_count
Index: src/couchapp/AsyncTransfer/views/forKill/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/forKill/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/forKill/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-	if (doc.workflow && (doc.state=='new' || doc.state=='acquired' || doc.state=='retry')){
-		emit(doc.workflow,doc._id);
-	}
-}
Index: src/couchapp/AsyncTransfer/views/forResub/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/forResub/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/forResub/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-	if (doc.workflow && (doc.state=='failed'||doc.state=='killed')){
-		emit(doc.workflow, doc._id);
-	}
-}
Index: src/couchapp/AsyncTransfer/views/sites/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/sites/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/sites/map.js	(revision )
@@ -1,6 +0,0 @@
-function(doc) {
-        if (doc.lfn){
-                emit(doc.source, 1);
-                emit(doc.destination, 1);
-                }
-        }
Index: src/couchapp/AsyncTransfer/views/sites/reduce.js
===================================================================
--- src/couchapp/AsyncTransfer/views/sites/reduce.js	(revision )
+++ src/couchapp/AsyncTransfer/views/sites/reduce.js	(revision )
@@ -1,1 +0,0 @@
-_sum
\ No newline at end of file
Index: src/couchapp/AsyncTransfer/views/UserByStartTime/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/UserByStartTime/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/UserByStartTime/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-        if (doc.state != 'failed' && doc.state != 'done' && doc.lfn) {
-                emit([doc.user, doc.group, doc.role, doc.start_time.split('.')[0]], 1);
-        }
-}
Index: src/couchapp/AsyncTransfer/views/getFilesToRetry/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/getFilesToRetry/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/getFilesToRetry/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-	if (doc.state == 'retry' && doc.lfn) {
-		emit(doc._id, doc.last_update);
-	}
-}
Index: src/couchapp/AsyncTransfer/views/LFNSiteByLastUpdate/map.js
===================================================================
--- src/couchapp/AsyncTransfer/views/LFNSiteByLastUpdate/map.js	(revision )
+++ src/couchapp/AsyncTransfer/views/LFNSiteByLastUpdate/map.js	(revision )
@@ -1,5 +0,0 @@
-function(doc) {
-	if(doc.source_lfn && doc.state == 'done'){
-		emit(doc.last_update, {"lfn": doc.source_lfn, "location": doc.source});
-	}
-}

