--- setup_dependencies.py       2011-06-30 10:51:06.000000000 +0200
+++ setup_dependencies.py       2011-06-30 10:53:06.000000000 +0200
@@ -70,4 +70,12 @@ dependencies = {
                                 'src/javascript/external/graphael',
                                 'src/templates/WMCore/WebTools/WMBS',],
                         },
+                'asyncstageout':{
+                        'packages': ['WMCore.Agent+', 'WMCore.Storage',
+                                    'WMCore.Credential','WMCore.WorkerThreads',
+                                    'WMCore.Services.PhEDEx+','WMCore.ACDC'],
+                        'modules': ['WMQuality.TestInitCouchApp'],
+                        'systems': ['wmc-database'],
+                        'statics': ['src/couchapps/Agent+'],
+                        },
                }
