--- setup_dependencies.py
+++ setup_dependencies.py
@@ -87,10 +87,12 @@ dependencies = {
                                     'src/templates/WMCore/WebTools/WMBS'],
                         },
                 'asyncstageout':{
-                        'packages': ['WMCore.Agent+', 'WMCore.Storage+',
+                        'packages': ['WMCore.Agent+', 'WMCore.Storage+','WMComponent.DBSUpload',
                                     'WMCore.Credential', 'WMCore.WorkerThreads',
-                                    'WMCore.Services.PhEDEx+', 'WMCore.ACDC', 'WMCore.Alerts+'],
-                        'modules': ['WMQuality.TestInitCouchApp'],
+                                    'WMCore.Services.PhEDEx+', 'WMCore.ACDC', 'WMCore.Alerts+',
+                                    'WMCore.Services.UserFileCache'],
+                        'modules': ['WMQuality.TestInitCouchApp','WMCore.Services.Service',
+                                    'WMCore.Services.pycurl_manager'],
                         'systems': ['wmc-database'],
                         'statics': ['src/couchapps/Agent+'],
                         },
