diff -Naur ATLAS-org/makes/Make.lib ATLAS/makes/Make.lib
--- ATLAS-org/makes/Make.lib	2014-07-10 18:22:01.000000001 +0200
+++ ATLAS/makes/Make.lib	2015-05-01 22:36:37.000000001 +0200
@@ -33,6 +33,11 @@
 	rm -rf $(ARCH)
 	$(BZIP) --best $(tarnam).tar
 
+%.so : %.a
+	$(MAKE) TRYALL outso=$@ \
+                libas=$< \
+                LIBINSTdir="$(LIBINSTdir)"
+
 # ===================================================================
 # The following commands are to build dynamic/shared objects on Linux
 # using the gnu gcc or ld
diff -Naur ATLAS-org/Make.top ATLAS/Make.top
--- ATLAS-org/Make.top	2014-07-10 18:22:00.000000001 +0200
+++ ATLAS/Make.top	2015-05-02 00:08:35.000000001 +0200
@@ -655,21 +655,11 @@
 	cp $(INCAdir)/* $(INSTdir)/atlas/.
 	- chmod 0644 $(INSTdir)/atlas/*
 install_lib:
-	cp $(LIBdir)/libatlas.a $(INSTdir)/.
-	cp $(LIBdir)/libcblas.a $(INSTdir)/.
-	cp $(LIBdir)/liblapack.a $(INSTdir)/.
-	chmod 0644 $(INSTdir)/libatlas.a  $(INSTdir)/liblapack.a \
-                   $(INSTdir)/libcblas.a 
-	- cp $(LIBdir)/libf77blas.a $(INSTdir)/.
-	- chmod 0644 $(INSTdir)/libf77blas.a
-	- cp $(LIBdir)/libptcblas.a $(INSTdir)/.
-	- chmod 0644 $(INSTdir)/libptcblas.a 
-	- cp $(LIBdir)/libptf77blas.a $(INSTdir)/.
-	- chmod 0644 $(INSTdir)/libptf77blas.a
-	- cp $(LIBdir)/libsatlas.dylib $(INSTdir)/.
-	- cp $(LIBdir)/libtatlas.dylib $(INSTdir)/.
-	- cp $(LIBdir)/libsatlas.dll $(INSTdir)/.
-	- cp $(LIBdir)/libtatlas.dll $(INSTdir)/.
-	- cp $(LIBdir)/libsatlas.so $(INSTdir)/.
-	- cp $(LIBdir)/libtatlas.so $(INSTdir)/.
+	cp $(LIBdir)/libatlas.so $(INSTdir)/.
+	cp $(LIBdir)/libf77blas.so $(INSTdir)/.
+	cp $(LIBdir)/libcblas.so $(INSTdir)/.
+	cp $(LIBdir)/liblapack.so $(INSTdir)/.
+	cp $(LIBdir)/libptf77blas.so $(INSTdir)/.
+	cp $(LIBdir)/libptcblas.so $(INSTdir)/.
+	cp $(LIBdir)/libptlapack.so $(INSTdir)/.
 
