diff --git a/scripts/bootstrap/compile.sh b/scripts/bootstrap/compile.sh
index dee6ce2..a562b86 100755
--- a/scripts/bootstrap/compile.sh
+++ b/scripts/bootstrap/compile.sh
@@ -295,7 +295,7 @@ function build_jni() {
 
   *)
     # We don't need JNI on other platforms.
-    JNI_FLAGS="-Dio.bazel.EnableJni=0"
+    JNI_FLAGS="-Dio.bazel.EnableJni=0 -Xmx512m"
     ;;
   esac
 }
@@ -331,7 +331,7 @@ function run_bazel_jar() {
       ${BAZEL_DIR_STARTUP_OPTIONS} \
       ${BAZEL_BOOTSTRAP_STARTUP_OPTIONS:-} \
       $command \
-      --ignore_unsupported_sandboxing \
+      --ignore_unsupported_sandboxing --jobs=8 \
       --startup_time=329 --extract_data_time=523 \
       --rc_source=/dev/null --isatty=1 \
       --ignore_client_env \
