diff --git a/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java b/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java
index 4426ad4c5a..7c64f42381 100644
--- a/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java
+++ b/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java
@@ -112,7 +112,7 @@ public class LocalSpawnRunner implements SpawnRunner {
     this.localExecutionOptions = Preconditions.checkNotNull(localExecutionOptions);
     this.hostName = NetUtil.getCachedShortHostName();
     this.resourceManager = resourceManager;
-    this.useProcessWrapper = useProcessWrapper;
+    this.useProcessWrapper = false;
     this.localEnvProvider = localEnvProvider;
     this.binTools = binTools;
   }
