diff --git a/tools/cpp/unix_cc_configure.bzl b/tools/cpp/unix_cc_configure.bzl
index ccb18a431a..1858574756 100644
--- a/tools/cpp/unix_cc_configure.bzl
+++ b/tools/cpp/unix_cc_configure.bzl
@@ -200,6 +200,8 @@ def _find_gold_linker_path(repository_ctx, cc):
         for flag in line.split(" "):
             if flag.find("gold") == -1:
                 continue
+            if flag.find("enable-gold") > -1 or flag.find("with-plugin-ld") > -1:
+                continue
 
             # flag is '-fuse-ld=gold' for GCC or "/usr/lib/ld.gold" for Clang
             # strip space, single quote, and double quotes
