diff --git a/third_party/zlib/BUILD b/third_party/zlib/BUILD
index f1aa936..7317aff 100755
--- a/third_party/zlib/BUILD
+++ b/third_party/zlib/BUILD
@@ -43,6 +43,7 @@ distrib_cc_library(
     # Use -Dverbose=-1 to turn off zlib's trace logging. (#3280)
     copts = [
         "-w",
+	"-Wno-error=implicit-function-declaration",
         "-Dverbose=-1",
     ],
     enable_distributions = ["debian"],
