diff --git a/src/chttpd.erl b/src/chttpd.erl
index d104811..1186a92 100644
--- a/src/chttpd.erl
+++ b/src/chttpd.erl
@@ -136,7 +136,6 @@ stop() ->
 
 handle_request(MochiReq) ->
     Begin = now(),
-
     case couch_config:get("chttpd", "socket_options") of
     undefined ->
         ok;
@@ -146,7 +145,9 @@ handle_request(MochiReq) ->
     end,
 
     AuthenticationFuns = [
-        fun couch_httpd_auth:cookie_authentication_handler/1,
+       fun couch_cms_auth:cms_host_authentication_handler/1,
+       fun couch_cms_auth:cms_backend_authentication_handler/1,
+       fun couch_httpd_auth:cookie_authentication_handler/1,
         fun couch_httpd_auth:default_authentication_handler/1
     ],
 
@@ -198,13 +199,13 @@ handle_request(MochiReq) ->
 
     HttpReq = #httpd{
         mochi_req = MochiReq,
+        peer = Peer,
         method = Method,
         path_parts = [list_to_binary(chttpd:unquote(Part))
                 || Part <- string:tokens(Path, "/")],
         db_url_handlers = db_url_handlers(),
         design_url_handlers = design_url_handlers()
     },
-
     % put small token on heap to keep requests synced to backend calls
     erlang:put(nonce, couch_util:to_hex(crypto:rand_bytes(4))),
 

