--- couchjs/c_src/SConscript	2014-01-13 17:32:48.000000000 +0100
+++ couchjs/c_src/SConscript	2014-01-13 17:35:49.000000000 +0100
@@ -36,16 +36,9 @@
	env['LIB_DL'] = env['LIB_RT'] = ''
	env['LIB_COMPAT'] = 'compat'

-if os.path.exists('/usr/bin/pkg-config'):
-    for pkg in ["mozilla-js-185", "mozilla-js"]:
-        (s1, output) = runcmd("/usr/bin/pkg-config %s --cflags" % pkg)
-        if s1 == 0:
-            env.Append(CCFLAGS=" " + output)
-        (s2, output) = runcmd("/usr/bin/pkg-config %s --libs-only-L" % pkg)
-        if s2 == 0:
-            env.Append(LINKFLAGS=output)
-        if s1 == 0 or s2 == 0:
-            break
+for key in ['spider', 'curl']:
+    env.Append( CPPPATH=[ARGUMENTS.get('%sInclude' % key, '')] )
+    env.Append( LIBPATH=[ARGUMENTS.get('%sLib' % key, '')] )

 if not env.GetOption('clean'):
     conf = Configure(env, config_h='config.h')
