diff --git a/my_programs/Makefile.am b/my_programs/Makefile.am
index 6fef244..c35abe4 100644
--- a/my_programs/Makefile.am
+++ b/my_programs/Makefile.am
@@ -65,13 +65,13 @@ LINKER_ARGS += -lmpfr -lgmpxx -lgmp
 CXXFLAGS += -DBH_USE_GMP
 endif 
 
-include $(DEPDIR)/*.Po
-
 %.o:	$(abs_srcdir)/%.cpp 
 	test -f $(DEPDIR)/$*.Po || mkdir -p .deps ; echo "#dummy" > $(DEPDIR)/$*.Po 
 	$(CXX) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS) $(INCLUDE) -o $@  $< 
 	mv -f $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
 
+#include $(DEPDIR)/*.Po
+
 LINKER_FLAGS=$(AM_LDFLAGS) 
 if GMP_PATH_SET 
    LINKER_FLAGS += -L$(AM_GMP_PATH)
diff --git a/configure.ac b/configure.ac
index 2099377..6396f8b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,7 +1,8 @@
 AC_INIT([BlackHat], [0.9.9], [daniel.maitre@durham.ac.uk])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 AM_DISABLE_STATIC
 AC_PROG_CXX
+AM_PROG_AR
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
 AC_CHECK_FUNCS([pow log sqrt])
