--- blackhat-0.9.9/src/cut_Darren.h	2012-10-18 21:49:36.000000000 +0100
+++ blackhat-0.9.9-new/src/cut_Darren.h	2014-01-31 12:05:12.675707975 +0000
@@ -489,7 +489,7 @@ protected:
 private:
         //no copy
         triangle_Darren(const triangle_Darren& cd);
-        triangle_Darren operator=(const triangle_Darren& cd);
+        triangle_Darren& operator=(const triangle_Darren& cd);
         void init();
 };
 
# bla bla
