--- orig.boost_1_37_0/boost/type_traits/intrinsics.hpp	2008-10-15 11:29:29.000000000 +0200
+++ boost_1_37_0/boost/type_traits/intrinsics.hpp	2009-01-27 15:51:41.000000000 +0100
@@ -126,7 +126,7 @@
 #   define BOOST_HAS_TYPE_TRAITS_INTRINSICS
 #endif
 
-#if defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 3)))
+#if defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 3) && !defined(__GCCXML__)))
 #   include <boost/type_traits/is_same.hpp>
 #   include <boost/type_traits/is_reference.hpp>
 #   include <boost/type_traits/is_volatile.hpp>
