diff -Naur orig.boost-build/jam_src/build.jam boost-build/jam_src/build.jam
--- orig.boost-build/jam_src/build.jam	2007-09-30 21:12:47.000000000 +0200
+++ boost-build/jam_src/build.jam	2007-11-05 23:19:07.000000000 +0100
@@ -174,7 +174,7 @@
 ## GCC 2.x, 3.x, 4.x
 toolset gcc gcc : "-o " : -D
     : -pedantic
-    [ opt --release : [ opt --symbols : -g : -s ] -O3 ]
+    [ opt --release : [ opt --symbols : -g : -s ] -O3 -fno-strict-aliasing ]
     [ opt --debug : -g -O0 -fno-inline ]
     -I$(--python-include) -I$(--extra-include) -Wno-long-long
     : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
