diff --git a/boost/lexical_cast.hpp b/boost/lexical_cast.hpp
index 5a3d4f0..1724c6e 100644
--- a/boost/lexical_cast.hpp
+++ b/boost/lexical_cast.hpp
@@ -1294,7 +1294,7 @@ namespace boost
             bool shl_double(double val,T* out)
             {   using namespace std;
                 if (put_inf_nan(start,finish,val)) return true;
-                finish = start + sprintf(out,"%.*lg", static_cast<int>(boost::detail::lcast_get_precision<double >()), val );
+                finish = start + sprintf(out,"%.*g", static_cast<int>(boost::detail::lcast_get_precision<double >()), val );
                 return finish > start;
             }
 #ifndef __MINGW32__
