diff --git a/boost/regex/v4/match_results.hpp b/boost/regex/v4/match_results.hpp
index ca9898f..2eb4782 100644
--- a/boost/regex/v4/match_results.hpp
+++ b/boost/regex/v4/match_results.hpp
@@ -212,6 +212,9 @@ public:
    {
       if(m_is_singular && m_subs.empty())
          raise_logic_error();
+      if (sub > (INT_MAX-2))
+        return m_null;
+
       sub += 2;
       if(sub < (int)m_subs.size() && (sub >= 0))
       {
