diff --git a/boost/exception/detail/error_info_impl.hpp b/boost/exception/detail/error_info_impl.hpp
index 883d313..f432aab 100644
--- a/boost/exception/detail/error_info_impl.hpp
+++ b/boost/exception/detail/error_info_impl.hpp
@@ -30,7 +30,7 @@ boost
 
             protected:
 
-            ~error_info_base() throw()
+            virtual ~error_info_base() throw()
                 {
                 }
             };
