diff --git a/boost/config/user.hpp b/boost/config/user.hpp
index fbe10bf..bb95c39 100644
--- a/boost/config/user.hpp
+++ b/boost/config/user.hpp
@@ -138,3 +138,7 @@
 // dictionary (CMSSW, C++11). The following forces to use C++03 hack even if C++11 is
 // enabled.
 #define BOOST_NO_CXX11_EXPLICIT_CONVERSION_OPERATORS
+
+// Do not use inline assembly for atomic operations until ROOT6 can handle inline
+// assembly. This could happen with MCJIT support (middle of 2015).
+#define BOOST_SP_USE_SPINLOCK
# bla bla
