diff --git a/src/dcasrn.F b/src/dcasrn.F
index f707c72..61be44c 100644
--- a/src/dcasrn.F
+++ b/src/dcasrn.F
@@ -1,20 +1,3 @@
-      DOUBLE PRECISION FUNCTION dcasrn(i)
-
-      DOUBLE PRECISION DR(1)
-      DOUBLE PRECISION TWOM24
-      PARAMETER (TWOM24 = 2.D0**(-24))
-      Real Rvec(2)
-	Integer i
-      CALL RANLUX(RVEC,2)
-      DR(1) = DBLE (RVEC(1)) + DBLE (RVEC(2)) * TWOM24
-c      CALL RM48 (DR,1)
-      dcasrn = DR(1)
-
-
-      RETURN
-      END
-
-
       SUBROUTINE dcasrnv(RVEC,LENV)
 
       Integer lenv,I
diff --git a/src/Makefile.in b/src/Makefile.in
index 2f2d255..5aed8f2 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -37,7 +37,7 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = cascade$(EXEEXT)
+bin_PROGRAMS =
 subdir = src
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
# bla bla
