--- orig.cascade/2.2.04/src/casini.F	2010-10-31 15:20:03.000000000 +0100
+++ cascade/2.2.04/src/casini.F	2011-11-12 16:33:02.000000000 +0100
@@ -59,9 +59,9 @@
 c IGLU=5 --> GBW
 c IGLU=6 --> KMR
 c IGLU=6 --> RS
-      call getenv('PDFPATH',pdfpath)
+      call getenv('CASCADE_PDFPATH',pdfpath)
       if(pdfpath.ne.'') then
-         call getenv('PDFPATH',pdfpath)
+         call getenv('CASCADE_PDFPATH',pdfpath)
       else
          PDFPATH='../share/'
       endif
# bla bla
