--- orig.cascade/2.2.0/Makefile.in	2010-08-05 18:13:21.000000000 +0200
+++ cascade/2.2.0/Makefile.in	2010-09-30 15:49:04.000000000 +0200
@@ -195,7 +195,7 @@
 #endif
 MAYBE_OPT = 
 #SUBDIRS = include bases51 misc src $(MAYBE_OPT) data manual
-SUBDIRS = include misc bases51 src $(MAYBE_OPT) data manual
+SUBDIRS = include misc bases51 src $(MAYBE_OPT) data
 
 #dist_doc_DATA = README
 EXTRA_DIST = README history.txt
# bla bla
