diff -arupN castor-2.1.9.orig/rtcopy/Imakefile castor-2.1.9/rtcopy/Imakefile
--- castor-2.1.9.orig/rtcopy/Imakefile	2012-02-28 18:36:38.000000000 +0100
+++ castor-2.1.9/rtcopy/Imakefile	2012-02-29 09:27:44.000000000 +0100
@@ -89,6 +89,7 @@ DFLAGS = -DBIN=\"$(BIN)\" -DRTCOPY_LOGFI
 CFLAGS = $(CCOPTIM) -I../h -I.. $(THREADFLAGS) $(DFLAGS) $(SECURITYFLAG)
 
 DependsOnLibrary(common,castorcommon)
+DependsOnLibrary(rfio,castorrfio)
 DependsOnLibrary(dlf,castordlf)
 DependsOnLibrary(castor,castorclient)
 DependsOnLibrary(ns,castorns)
