diff --git a/gridftp2/external/Imakefile b/gridftp2/external/Imakefile
index 049e172..dfccf07 100644
--- a/gridftp2/external/Imakefile
+++ b/gridftp2/external/Imakefile
@@ -24,7 +24,9 @@ COMM
 COMM @author castor-dev@cern.ch
 COMM
 
-MOPTION=-m32
+ifeq ($(shell uname -m), i386)
+  MOPTION=-m32
+endif
 ifeq ($(shell uname -m), x86_64)
   MOPTION=-m64
 endif
diff --git a/gridftp2/internal/Imakefile b/gridftp2/internal/Imakefile
index 4e1378e..30dbdfd 100644
--- a/gridftp2/internal/Imakefile
+++ b/gridftp2/internal/Imakefile
@@ -24,7 +24,9 @@ COMM
 COMM @author castor-dev@cern.ch
 COMM
 
-MOPTION=-m32
+ifeq ($(shell uname -m), i386)
+  MOPTION=-m32
+endif
 ifeq ($(shell uname -m), x86_64)
   MOPTION=-m64
 endif
diff --git a/gridftp2/xroot/Imakefile b/gridftp2/xroot/Imakefile
index d915dcb..29a523d 100644
--- a/gridftp2/xroot/Imakefile
+++ b/gridftp2/xroot/Imakefile
@@ -24,7 +24,9 @@ COMM
 COMM @author castor-dev@cern.ch
 COMM
 
-MOPTION=-m32
+ifeq ($(shell uname -m), i386)
+  MOPTION=-m32
+endif
 ifeq ($(shell uname -m), x86_64)
   MOPTION=-m64
 endif
# bla bla
