diff -arupN castor-2.1.9.orig/ns/Imakefile castor-2.1.9/ns/Imakefile
--- castor-2.1.9.orig/ns/Imakefile	2012-02-28 18:36:38.000000000 +0100
+++ castor-2.1.9/ns/Imakefile	2012-02-29 09:52:23.000000000 +0100
@@ -34,6 +34,8 @@ LIBCSEC += $(GLOBUS_LIBS) $(LIB_CRYPTO)
 #endif
 LIBCSEC += -ldl
 
+LDFLAGS += -ldl
+
 DFLAGS = -DBIN=\"$(BIN)\" -DCNS_HOST=\"$(CNS_HOST)\" \
          -DSCNS_HOST=\"$(SCNS_HOST)\" \
          -DCNSHOSTPFX=\"$(CNSHOSTPFX)\" \
