diff --git a/castor-2.1.9/rfio/open.c b/castor-2.1.9/rfio/open.c
index d04975e..050dd6e 100644
--- a/castor-2.1.9/rfio/open.c
+++ b/castor-2.1.9/rfio/open.c
@@ -235,7 +235,7 @@ int rfio_open_ext(filepath, flags, mode,uid,gid,passwd,reqhost,vmstr)
   int      rt ;  /* daemon in site(0) or not (1) */
   int    bufsize ;  /* socket buffer size   */
   struct sockaddr_in      to;
-#if (defined(_AIX) || defined(linux))
+#if (defined(_AIX) || defined(linux) || defined(__APPLE__))
   socklen_t tolen;
 #else
   int     tolen;
diff --git a/castor-2.1.9/rfio/open64.c b/castor-2.1.9/rfio/open64.c
index a52900e..4a093ac 100644
--- a/castor-2.1.9/rfio/open64.c
+++ b/castor-2.1.9/rfio/open64.c
@@ -125,7 +125,7 @@ int rfio_open64_ext(filepath, flags, mode,uid,gid,passwd,reqhost)
   int     rt ;  /* daemon in site(0) or not (1) */
   int     bufsize ;  /* socket buffer size   */
   struct  sockaddr_in to;
-#if (defined(_AIX) || defined(linux))
+#if (defined(_AIX) || defined(linux) || defined(__APPLE__))
   socklen_t tolen;
 #else
   int     tolen;
diff --git a/castor-2.1.9/castor/db/newora/OraCnvSvc.cpp b/castor-2.1.9/castor/db/newora/OraCnvSvc.cpp
index 9e4fc9b..cc6beb2 100644
--- a/castor-2.1.9/castor/db/newora/OraCnvSvc.cpp
+++ b/castor-2.1.9/castor/db/newora/OraCnvSvc.cpp
@@ -98,6 +98,21 @@ const unsigned int castor::db::ora::OraCnvSvc::getPhysRepType() const {
   return castor::REP_ORACLE;
 }
 
+#if defined (__APPLE__) 
+#include <mach/mach.h> 
+#include <sys/resource.h> 
+#endif 
+static int gettid() 
+{ 
+#if defined(__APPLE__) 
+return mach_thread_self(); 
+#elif defined(linux) 
+ return syscall(__NR_gettid); 
+#else 
+return -ENOSYS; 
+#endif 
+} 
+
 //------------------------------------------------------------------------------
 // getConnection
 //------------------------------------------------------------------------------
@@ -224,7 +239,7 @@ oracle::occi::Connection* castor::db::ora::OraCnvSvc::getConnection()
     // for logging/debugging purposes, we set an identifier for this session
     std::ostringstream ss;
     ss << "BEGIN DBMS_APPLICATION_INFO.SET_CLIENT_INFO('CASTOR pid="
-       << getpid() << " tid=" << syscall(__NR_gettid) << "'); END;";    // gettid() is not defined???
+       << getpid() << " tid=" << gettid() << "'); END;";    // gettid() is not defined???
     stmt = m_connection->createStatement(ss.str());
     stmt->executeUpdate();
     m_connection->terminateStatement(stmt);
diff --git a/castor-2.1.9/dlf/dlf_lib.c b/castor-2.1.9/dlf/dlf_lib.c
index ab00541..ea3ad26 100644
--- a/castor-2.1.9/dlf/dlf_lib.c
+++ b/castor-2.1.9/dlf/dlf_lib.c
@@ -368,6 +368,21 @@ int DLL_DECL dlf_write(Cuuid_t reqid,
   return (rv);
 }
 
+#if defined (__APPLE__) 
+#include <mach/mach.h> 
+#include <sys/resource.h> 
+#endif 
+static int gettid() 
+{ 
+#if defined(__APPLE__) 
+return mach_thread_self(); 
+#elif defined(linux) 
+ return syscall(__NR_gettid); 
+#else 
+return -ENOSYS; 
+#endif 
+} 
+
 
 /*---------------------------------------------------------------------------
  * dlf_writep
@@ -426,7 +441,7 @@ int DLL_DECL dlf_writep(Cuuid_t reqid,
 
   buffer[0] = '\0';
   len = snprintf(buffer, maxmsglen - len, "LVL=%s TID=%d MSG=\"%s\" ",
-                 prioritylist[priority].text, (int)syscall(__NR_gettid),
+                 prioritylist[priority].text, gettid(),
                  messages[msgno]);
 
   /* Append the request uuid if defined */
diff --git a/castor-2.1.9/castor/io/biniostream.h b/castor-2.1.9/castor/io/biniostream.h
index 15771f1..10f9b8f 100644
--- a/castor-2.1.9/castor/io/biniostream.h
+++ b/castor-2.1.9/castor/io/biniostream.h
@@ -28,7 +28,12 @@
 #define IO_BINIOSTREAM_H 1
 
 #include <sstream>
+#if defined(__APPLE__)
+#include <machine/endian.h>
+#include <machine/byte_order.h>
+#else
 #include <byteswap.h>
+#endif
 #include "osdep.h"
 #include "castor/exception/OutOfMemory.hpp"
 #include <string.h>
diff --git a/castor-2.1.9/tape/getcompstat.c b/castor-2.1.9/tape/getcompstat.c
index 2214e32..41ffb49 100644
--- a/castor-2.1.9/tape/getcompstat.c
+++ b/castor-2.1.9/tape/getcompstat.c
@@ -24,7 +24,7 @@ char *path;
 char *devtype;
 COMPRESSION_STATS *comp_stats;
 {
-#if defined(ADSTAR) || defined(SOLARIS25) || defined(sgi) || defined(hpux) || (defined(__osf__) && defined(__alpha)) || defined(linux)
+#if defined(ADSTAR) || defined(SOLARIS25) || defined(sgi) || defined(hpux) || (defined(__osf__) && defined(__alpha)) || defined(linux) || defined(__APPLE__)
 	unsigned long kbytes_from_host;
 	unsigned long kbytes_to_tape;
 	unsigned long kbytes_from_tape;
diff --git a/castor-2.1.9/tape/getcompstat.c b/castor-2.1.9/tape/getcompstat.c
index 41ffb49..83e701b 100644
--- a/castor-2.1.9/tape/getcompstat.c
+++ b/castor-2.1.9/tape/getcompstat.c
@@ -245,7 +245,7 @@ int tapefd;
 char *path;
 char *devtype;
 {
-#if defined(ADSTAR) || defined(SOLARIS25) || defined(sgi) || defined(hpux) || (defined(__osf__) && defined(__alpha)) || defined(linux)
+#if defined(ADSTAR) || defined(SOLARIS25) || defined(sgi) || defined(hpux) || (defined(__osf__) && defined(__alpha)) || defined(linux) || defined(__APPLE__)
 #if defined(ADSTAR)
 	struct sc_iocmd sc_iocmd;
 
diff --git a/castor-2.1.9/vmgr/vmgr_modifytape.c b/castor-2.1.9/vmgr/vmgr_modifytape.c
index 33cc60a..3b1240a 100644
--- a/castor-2.1.9/vmgr/vmgr_modifytape.c
+++ b/castor-2.1.9/vmgr/vmgr_modifytape.c
@@ -21,7 +21,7 @@
 int vmgr_modifytape(const char *vid, char *vsn, char *library, char *density, char *lbltype, char *manufacturer, char *sn, char *poolname, int status)
 {
 	int c;
-	char func[15];
+	char func[16];
 	gid_t gid;
 	int msglen;
 	char *q;
diff --git a/castor-2.1.9/tape/rwndtape.c b/castor-2.1.9/tape/rwndtape.c
index 3cd0905..3d9feb2 100644
--- a/castor-2.1.9/tape/rwndtape.c
+++ b/castor-2.1.9/tape/rwndtape.c
@@ -20,7 +20,7 @@
 #else
 #if defined(_WIN32)
 #include <windows.h>
-#else
+#elif !defined(__APPLE__)
 #include <sys/ioctl.h>
 #include <sys/mtio.h>
 #endif
@@ -36,6 +36,7 @@ int tapefd;
 #endif
 char *path;
 {
+#if !defined(__APPLE__)
 	char func[16];
 #if !defined(_WIN32)
 	struct mtop mtop;
@@ -52,5 +53,6 @@ char *path;
 		serrno = rpttperror (func, tapefd, path, "ioctl"); 
 		RETURN (-1);
 	}
+#endif
 	RETURN (0);
 }
diff --git a/castor-2.1.9/tape/skiptape.c b/castor-2.1.9/tape/skiptape.c
index 16c747d..711ee01 100644
--- a/castor-2.1.9/tape/skiptape.c
+++ b/castor-2.1.9/tape/skiptape.c
@@ -5,11 +5,13 @@
 #include <errno.h>
 #include <string.h>
 #include <stdio.h>
+#if defined(linux)
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/mtio.h>
 #include <linux/version.h>
 #include <sys/utsname.h>
+#endif
 #include "Ctape.h"
 #include "Ctape_api.h"
 #include "serrno.h"
@@ -17,6 +19,7 @@
 extern int mt_rescnt;
 
 int skiptpfff(int tapefd, char *path, int n) {
+#if defined(linux)
 
 	/* skip n files forward */
 
@@ -59,10 +62,13 @@ int skiptpfff(int tapefd, char *path, int n) {
 		tobeskipped -= count;
 	}
 	RETURN (0);
+#else
+	return -1;
+#endif
 }
 
 int skiptpff(int tapefd, char *path, int n) {
-
+#if defined(linux)
 	/* skip n files forward */
 
 	int count;
@@ -83,10 +89,13 @@ int skiptpff(int tapefd, char *path, int n) {
 		tobeskipped -= count;
 	}
 	RETURN (0);
+#else
+	return -1;
+#endif
 }
 
 int skiptpfb(int tapefd, char *path, int n) {
-
+#if defined(linux)
         /* skip n files backward */
 
 	int count;
@@ -107,10 +116,13 @@ int skiptpfb(int tapefd, char *path, int n) {
 		tobeskipped -= count;
 	}
 	RETURN (0);
+#else
+	return -1;
+#endif
 }
 
 int skiptprf(int tapefd, char *path, int n, int silent) {
-
+#if defined(linux)
 	/* skip n blocks forward */
 
 	int count;
@@ -133,10 +145,13 @@ int skiptprf(int tapefd, char *path, int n, int silent) {
 		tobeskipped -= count;
 	}
 	RETURN (0);
+#else
+	return -1;
+#endif
 }
 
 int skiptprb(int tapefd, char *path, int n) {
-
+#if defined(linux)
         /* skip n blocks backward */
 
 	int count;
@@ -157,10 +172,13 @@ int skiptprb(int tapefd, char *path, int n) {
 		tobeskipped -= count;
 	}
 	RETURN (0);
+#else
+	return -1;
+#endif
 }
 
 int skip2eod(int tapefd, char *path) {
-        
+#if defined(linux) 
         /* Skip to the end of data. */
 
         char func[16];
@@ -175,4 +193,7 @@ int skip2eod(int tapefd, char *path) {
 		RETURN (-1);
 	}
 	RETURN (0);
+#else
+	return -1;
+#endif
 }
diff --git a/castor-2.1.9/tape/unldtape.c b/castor-2.1.9/tape/unldtape.c
index 93c9b94..abb1686 100644
--- a/castor-2.1.9/tape/unldtape.c
+++ b/castor-2.1.9/tape/unldtape.c
@@ -36,6 +36,7 @@ int tapefd;
 #endif
 char *path;
 {
+#if !defined(__APPLE__)
 	char func[16];
 #if !defined(_WIN32)
 	struct mtop mtop;
@@ -57,5 +58,6 @@ char *path;
 		serrno = rpttperror (func, tapefd, path, "ioctl");
 		RETURN (-1);
 	}
+#endif // !defined(__APPLE__)
 	RETURN (0);
 }
diff --git a/castor-2.1.9/tape/wrttpmrk.c b/castor-2.1.9/tape/wrttpmrk.c
index e04e9df..6433554 100644
--- a/castor-2.1.9/tape/wrttpmrk.c
+++ b/castor-2.1.9/tape/wrttpmrk.c
@@ -37,6 +37,7 @@ int tapefd;
 char *path;
 int n;
 {
+#if !defined(__APPLE__)
 	char func[16];
 #if !defined(_WIN32)
 	struct mtop mtop;
@@ -53,5 +54,6 @@ int n;
 		serrno = rpttperror (func, tapefd, path, "ioctl");
 		RETURN (-1);
 	}
+#endif // !defined(__APPLE__)
 	RETURN (0);
 }
diff --git a/castor-2.1.9/tape/readlbl.c b/castor-2.1.9/tape/readlbl.c
index 8a73801..c68f3e5 100644
--- a/castor-2.1.9/tape/readlbl.c
+++ b/castor-2.1.9/tape/readlbl.c
@@ -16,8 +16,10 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdio.h>
+#if !defined(__APPLE__)
 #include <sys/mtio.h>
 #include <sys/utsname.h>
+#endif
 #include "Ctape.h"
 #include "Ctape_api.h"
 #include "serrno.h"
@@ -30,6 +32,7 @@ int tapefd;
 char *path;
 char *lblbuf;
 {
+#if !defined(__APPLE__)
 	/* int errcat; */
 	char func[16];
 	char *msgaddr;
@@ -142,4 +145,7 @@ char *lblbuf;
                 RETURN (1); 
         }
 	RETURN (0);
+#else
+	RETURN (-1);
+#endif
 }
diff --git a/castor-2.1.9/tape/tperror.c b/castor-2.1.9/tape/tperror.c
index 1d38749..a12c372 100644
--- a/castor-2.1.9/tape/tperror.c
+++ b/castor-2.1.9/tape/tperror.c
@@ -21,7 +21,9 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
+#if !defined(__APPLE__)
 #include <sys/ioctl.h>
+#endif
 #include "Ctape.h"
 #include "Ctape_api.h"
 #include "serrno.h"
diff --git a/castor-2.1.9/tape/skiptape.c b/castor-2.1.9/tape/skiptape.c
index 711ee01..6c79d99 100644
--- a/castor-2.1.9/tape/skiptape.c
+++ b/castor-2.1.9/tape/skiptape.c
@@ -11,12 +11,12 @@
 #include <sys/mtio.h>
 #include <linux/version.h>
 #include <sys/utsname.h>
-#endif
 #include "Ctape.h"
 #include "Ctape_api.h"
 #include "serrno.h"
 
 extern int mt_rescnt;
+#endif
 
 int skiptpfff(int tapefd, char *path, int n) {
 #if defined(linux)
diff --git a/castor-2.1.9/tape/writelbl.c b/castor-2.1.9/tape/writelbl.c
index ab23c1f..6a891f0 100644
--- a/castor-2.1.9/tape/writelbl.c
+++ b/castor-2.1.9/tape/writelbl.c
@@ -14,15 +14,18 @@
 #include <errno.h>
 #include <string.h>
 #include <unistd.h>
+#if !defined(__APPLE__)
 #include <sys/types.h>
 #include "Ctape.h"
 #include "Ctape_api.h"
+#endif
 #include "serrno.h"
 int writelbl(tapefd, path, lblbuf)
 int tapefd;
 char *path;
 char *lblbuf;
 {
+#if !defined(__APPLE__)
 	char func[16];
 	int n;
 
@@ -32,4 +35,7 @@ char *lblbuf;
 		RETURN (-1);
 	}
 	RETURN (n);
+#else
+	return -1;
+#endif
 }
diff --git a/castor-2.1.9/tape/tperror.c b/castor-2.1.9/tape/tperror.c
index a12c372..05484b8 100644
--- a/castor-2.1.9/tape/tperror.c
+++ b/castor-2.1.9/tape/tperror.c
@@ -28,7 +28,9 @@
 #include "Ctape_api.h"
 #include "serrno.h"
 
+#if defined(linux)
 #include <linux/mtio.h> 
+#endif
 #ifndef MTIOCSENSE
 #include "mtio_add.h"
 #endif
@@ -74,6 +76,7 @@ int tapefd;
 char *path;
 char **msgaddr;
 {
+#if !defined(__APPLE__)
 #ifndef NOTRACE
 	extern char *devtype;
 #else
@@ -155,6 +158,9 @@ char **msgaddr;
 
 	errno = save_errno;
 	RETURN (rc);
+#else
+	return -1;
+#endif
 }
 
 
diff --git a/castor-2.1.9/tape/wrttpmrk.c b/castor-2.1.9/tape/wrttpmrk.c
index 6433554..8514a29 100644
--- a/castor-2.1.9/tape/wrttpmrk.c
+++ b/castor-2.1.9/tape/wrttpmrk.c
@@ -20,7 +20,7 @@
 #else
 #if defined(_WIN32)
 #include <windows.h>
-#else
+#elif !defined(__APPLE__)
 #include <sys/ioctl.h>
 #include <sys/mtio.h>
 #endif
diff --git a/castor-2.1.9/h/mtio_add.h b/castor-2.1.9/h/mtio_add.h
index 13c15fd..9bad12f 100644
--- a/castor-2.1.9/h/mtio_add.h
+++ b/castor-2.1.9/h/mtio_add.h
@@ -21,6 +21,12 @@
 #ifndef _MTIO_ADD_H
 #define _MTIO_ADD_H
 
+#if defined(__APPLE__)
+#include <stdint.h>
+#define __u16 uint16_t
+#define __u8  uint8_t
+#define __s32 uint32_t
+#endif
 /* structure for MTIOCSENSE - mag tape get sense data */
 #define MTIOCSENSE_SENSE_LENGTH 252
 #define MTIOCSENSE_CDB_LENGTH   16
diff --git a/castor-2.1.9/tape/tperror.c b/castor-2.1.9/tape/tperror.c
index 05484b8..ece28fc 100644
--- a/castor-2.1.9/tape/tperror.c
+++ b/castor-2.1.9/tape/tperror.c
@@ -35,6 +35,7 @@
 #include "mtio_add.h"
 #endif
 
+#if !defined(__APPLE__)
 static char bbot[] = "BOT hit";
 struct sk_info {
 	char *text;
@@ -70,7 +71,7 @@ int mt_rescnt;
 static char tp_err_msgbuf[TPERRMSGBUSZ];
 
 int get_sk_msg(char *, int, int, int, char **);
-
+#endif
 int gettperror(tapefd, path, msgaddr)
 int tapefd;
 char *path;
diff --git a/castor-2.1.9/tape/tperror.c b/castor-2.1.9/tape/tperror.c
index ece28fc..24b09b0 100644
--- a/castor-2.1.9/tape/tperror.c
+++ b/castor-2.1.9/tape/tperror.c
@@ -172,6 +172,7 @@ int asc;
 int ascq;
 char **msgaddr;
 {
+#if !defined(__APPLE__)
 	int rc;
 	if (key >= 0 && key < 15) {
 		if (asc == 0 && ascq == 0) {
@@ -194,6 +195,9 @@ char **msgaddr;
 		rc = -1;
 	}
 	return (rc);
+#else
+	return (-1);
+#endif
 }
 
 int rpttperror(func, tapefd, path, cmd)
@@ -202,6 +206,7 @@ int tapefd;
 char *path;
 char *cmd;
 {
+#if !defined(__APPLE__)
 	char *msgaddr;
 	int rc;
 
@@ -213,4 +218,7 @@ char *cmd;
 	}
 	usrmsg (func, TP042, path, cmd, msgaddr);
 	return (rc);
+#else
+	return (-1);
+#endif
 }
diff --git a/castor-2.1.9/common/sendscsicmd.c b/castor-2.1.9/common/sendscsicmd.c
index 0234960..fb28c63 100644
--- a/castor-2.1.9/common/sendscsicmd.c
+++ b/castor-2.1.9/common/sendscsicmd.c
@@ -12,7 +12,7 @@
  *		 0	if successful with no data transfer
  *		>0	number of bytes transferred
  */
-
+#if !defined(__APPLE__)
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <stdio.h>
@@ -350,3 +350,27 @@ char **msgaddr;
 		}
 	}
 }
+#else
+// dummy implementation for unsupported platforms.
+// Maybe the if should just be inverted.
+void find_sgpath(char *sgpath, int maj, int min) {}
+int send_scsi_cmd (tapefd, path, do_not_open, cdb, cdblen, buffer, buflen, sense, senselen, timeout, flags, nb_sense_ret, msgaddr)
+int tapefd;
+char *path;
+int do_not_open;
+unsigned char *cdb;
+int cdblen;
+unsigned char *buffer;
+int buflen;
+char *sense;
+int senselen;
+int timeout;    /* in milliseconds */
+int flags;
+int *nb_sense_ret;
+char **msgaddr;
+{return -1;}
+void get_ss_msg(scsi_status, msgaddr)
+int scsi_status;
+char **msgaddr;
+{}
+#endif
diff --git a/castor-2.1.9/config/Imake.rules b/castor-2.1.9/config/Imake.rules
index 7d327c5..da40eb6 100644
--- a/castor-2.1.9/config/Imake.rules
+++ b/castor-2.1.9/config/Imake.rules
@@ -169,7 +169,7 @@ ProgramTarget(program,objects,deplibs,libs,mode,tape,tape)
 #define BuildRPathcastorns commonRPath securityRPath upvRPath
 #define BuildRPathcastorclient commonRPath securityRPath upvRPath dlfRPath nsRPath
 #define BuildRPathcastorexpert commonRPath upvRPath nsRPath vmgrRPath vdqmRPath
-#define BuildRPathcastorrtcopy commonRPath dlfRPath castorRPath 
+#define BuildRPathcastorrtcopy commonRPath rfioRPath dlfRPath castorRPath 
 #define BuildRPathcastorrfio commonRPath securityRPath castorRPath dlfRPath nsRPath
 #define BuildRPathcastortape commonRPath securityRPath rfioRPath vmgrRPath upvRPath castorRPath vdqmRPath nsRPath
 #define BuildRPathcastormonitor commonRPath castorRPath
diff --git a/castor-2.1.9/Imakefile b/castor-2.1.9/Imakefile
index 83e0d51..e3aba0b 100644
--- a/castor-2.1.9/Imakefile
+++ b/castor-2.1.9/Imakefile
@@ -34,7 +34,7 @@ doxygen: tools/doxyconfig
 .PHONY : client installclient tape installtape
 
 client: 
-	GotoSubdirs(upv ns client rfio vdqm vmgr security tape rtcopy shlib)
+	GotoSubdirs(upv ns client rfio vdqm vmgr security tape)
 
 installclient: checkDESTDIR checkVERSION client exportman
 	GotoSubdirs(upv ns client rfio vdqm vmgr castor common tape rtcopy h shlib)
diff --git a/castor-2.1.9/Imakefile b/castor-2.1.9/Imakefile
index e3aba0b..0a30303 100644
--- a/castor-2.1.9/Imakefile
+++ b/castor-2.1.9/Imakefile
@@ -7,7 +7,7 @@ COMM
 COMM		Make CASTOR libraries and executables
 
 SUBDIRS = common dlf security upv ns rfio logprocessor \
-	  expert rmc rtcopy client tape vdqm vmgr \
+	  rmc client tape vdqm vmgr \
 	  hsmtools h castor gridftp2 shlib clips test
 
 clean: imakeclean
diff --git a/castor-2.1.9/shlib/Imakefile b/castor-2.1.9/shlib/Imakefile
index 1fb079d..7121b6f 100644
--- a/castor-2.1.9/shlib/Imakefile
+++ b/castor-2.1.9/shlib/Imakefile
@@ -15,7 +15,5 @@ DependsOnLibrary(tape,castortape)
 DependsOnLibrary(rfio,castorrfio)
 DependsOnLibrary(castor,castorclient)
 DependsOnLibrary(vdqm,castorvdqm)
-DependsOnLibrary(rtcopy,castorrtcopy)
 DependsOnLibrary(vmgr,castorvmgr)
-DependsOnLibrary(expert,castorexpert)
 ClientSharedLibraryTarget(shift,,$(DEPLIB),$(LIBS))
# bla bla
