diff --git a/client/src/stager/Imakefile b/client/src/stager/Imakefile
index 4fc670f..bc73c63 100644
--- a/client/src/stager/Imakefile
+++ b/client/src/stager/Imakefile
@@ -3,6 +3,8 @@ COMM  @(#)Imakefile,v 1.52 2005/11/25 14:27:22 CERN IT-ADC/CA Jean-Damien Durand
 COMM
 COMM  Make stager clients.
 
+AddLdFlags(UUIDLIBNAME)
+
 CPPFLAGS += -DSTAGEMAP=StageMapFile
 AddLdFlags(-lstdc++)
 
diff --git a/common/Imakefile b/common/Imakefile
index 95b8ec7..6c975aa 100644
--- a/common/Imakefile
+++ b/common/Imakefile
@@ -7,7 +7,7 @@ COMM
 COMM		Make common routines library
 
 CPPFLAGS += -DPATH_CONFIG=CastorConfigFile -DLOCALHOSTSFILE=LocalHostsList -DMAPPING_FILE=MappingFile -DRTHOSTSFILE=RemoteHostsList -DADMIN_HOSTS=AdminHosts
-AddLdFlags(-l UUIDLIBNAME -lstdc++)
+AddLdFlags(UUIDLIBNAME -lstdc++)
 
 COMMONLIB_OBJS = $(SYSDEP_OBJS) \
 		 compat.o getacct.o getacctent.o getconfent.o \
diff --git a/config/apple.cf b/config/apple.cf
index a5af00d..bc5cafd 100644
--- a/config/apple.cf
+++ b/config/apple.cf
@@ -10,5 +10,5 @@ COMM  Default rules for Make          APPLE
 #define SONAMELINKEROPTION -install_name
 #define DYLIBLINKEROPTION -dynamiclib
 #define NOUNDEFLINKEROPTION 
-#define UUIDLIBNAME System
+#define UUIDLIBNAME -lSystem
 LDFLAGS += -Wl,-current_version,$(subst .,,$(MAJOR_CASTOR_VERSION)).$(MINOR_CASTOR_VERSION)
diff --git a/config/linux.cf b/config/linux.cf
index c65765b..088ac33 100644
--- a/config/linux.cf
+++ b/config/linux.cf
@@ -12,5 +12,5 @@ COMM  Default rules for Make          Linux
 #define SONAMELINKEROPTION -soname
 #define DYLIBLINKEROPTION -shared
 #define NOUNDEFLINKEROPTION -Wl,-no-undefined
-#define UUIDLIBNAME uuid
+#define UUIDLIBNAME -luuid
 LDFLAGS += -Wl,--defsym,CastorVersion_v$(MAJOR_CASTOR_VERSION).$(MINOR_CASTOR_VERSION)=0
diff --git a/ns/Imakefile b/ns/Imakefile
index 5244545..49ebd01 100644
--- a/ns/Imakefile
+++ b/ns/Imakefile
@@ -7,7 +7,7 @@ COMM       @(#)Imakefile,v 1.51 2004/06/09 16:10:35 CERN IT-PDP/DM Jean-Philippe
 COMM    Make name server programs.
 
 NSLIB = DepSharedLibraryTargetName(ns,castorns)
-AddLdFlags(-l UUIDLIBNAME)
+AddLdFlags(UUIDLIBNAME)
 
 #if UseGSI
 NSLIBCSEC += -lcrypto
diff --git a/rfio/Imakefile b/rfio/Imakefile
index 7a73008..152d10b 100644
--- a/rfio/Imakefile
+++ b/rfio/Imakefile
@@ -11,6 +11,8 @@ COMM Define the file indicating that no more RFIO requests
 COMM should be issued.
 COMM
 
+AddLdFlags(UUIDLIBNAME)
+
 CPPFLAGS += -DSACCT -DNOMORERFIO=RfioNoMore -DRFIOLOGFILE=RfioLogFile \
 	    -DCNS_ROOT=NsRoot $(STAGERSUPERUSER) $(STAGERSUPERGROUP)
 CFLAGS += -DUSE_XFSPREALLOC
diff --git a/upv/Imakefile b/upv/Imakefile
index f922502..ba8055d 100644
--- a/upv/Imakefile
+++ b/upv/Imakefile
@@ -6,6 +6,8 @@ COMM       @(#)Imakefile,v 1.3 2005/01/23 08:49:33 CERN IT-PDP/DM Ben Couturier
 
 COMM    Make UPV programs.
 
+AddLdFlags(UUIDLIBNAME)
+
 #if BuildSecureCupv
 CPPFLAGS += -DUPVCSEC
 #endif
diff --git a/vdqm/Imakefile b/vdqm/Imakefile
index ec1c611..3c3a37e 100644
--- a/vdqm/Imakefile
+++ b/vdqm/Imakefile
@@ -23,6 +23,8 @@ COMM Makefile for the VDQM API library and client
 COMM
 COMM @author Castor Dev team, castor-dev@cern.ch
 
+AddLdFlags(UUIDLIBNAME)
+
 #if BuildSecureVdqm
 CPPFLAGS += -DVDQMCSEC
 LDLIBS += -ldl
diff --git a/vmgr/Imakefile b/vmgr/Imakefile
index e620887..8266215 100644
--- a/vmgr/Imakefile
+++ b/vmgr/Imakefile
@@ -5,6 +5,8 @@ COMM
 
 COMM    Make volume manager programs.
 
+AddLdFlags(UUIDLIBNAME)
+
 #if BuildSecureVmgr
 CPPFLAGS += -DVMGRCSEC
 AddLdFlags(-ldl)
# bla bla
