diff --git a/src/catch2/internal/catch_test_macro_impl.hpp b/src/catch2/internal/catch_test_macro_impl.hpp
index ccd5bb3..b233a7f 100644
--- a/src/catch2/internal/catch_test_macro_impl.hpp
+++ b/src/catch2/internal/catch_test_macro_impl.hpp
@@ -17,7 +17,7 @@
 
 // We need this suppression to leak, because it took until GCC 10
 // for the front end to handle local suppression via _Pragma properly
-#if defined(__GNUC__) && !defined(__clang__) && !defined(__ICC) && __GNUC__ <= 9
+#if defined(__GNUC__) && !defined(__clang__) && !defined(__ICC) && __GNUC__ >= 14
   #pragma GCC diagnostic ignored "-Wparentheses"
 #endif
 
