diff --git a/CepGenAddOns/CMakeLists.txt b/CepGenAddOns/CMakeLists.txt
index 9b7db4e..4bf730a 100644
--- a/CepGenAddOns/CMakeLists.txt
+++ b/CepGenAddOns/CMakeLists.txt
@@ -17,7 +17,6 @@ add_subdirectory(Pythia6Wrapper)
 add_subdirectory(Pythia8Wrapper)
 add_subdirectory(REvolverWrapper)
 add_subdirectory(RivetWrapper)
-add_subdirectory(ROOTWrapper)
 add_subdirectory(TopdrawerWrapper)
 
 add_subdirectory(Functionals)
