diff --git a/charybdis/1.003/Makefile b/charybdis/1.003/Makefile
index d4cbdd0..a9f0690 100644
--- a/charybdis/1.003/Makefile
+++ b/charybdis/1.003/Makefile
@@ -53,10 +53,6 @@ libs:
 	export LIBNAME ; export DIRNAME ; \
 	$(MAKE) -f Makearchive.subdir
 
-	@LIBNAME=$(LIBDIR)/libcharybdis.so ; DIRNAME=src ; \
-	export LIBNAME ; export DIRNAME ; \
-	$(MAKE) -f Makeshared.subdir
-
 clean:
 	rm -f *~; rm -f \#*;
 	rm -rf $(TMPDIR)
# bla bla
