diff --git a/tests/test_clang_util.cc b/tests/test_clang_util.cc
index 3197efa..2117824 100644
--- a/tests/test_clang_util.cc
+++ b/tests/test_clang_util.cc
@@ -144,14 +144,14 @@ TEST_CASE("Test diagnostic_consumer")
     auto fs = llvm::vfs::getRealFileSystem();
     FileSystemOptions file_opts;
     FileManager file_mgr(file_opts, fs);
-    auto diagnostic_engine = DiagnosticsEngine(
-        llvm::IntrusiveRefCntPtr<DiagnosticIDs>(new DiagnosticIDs()),
-        new DiagnosticOptions());
+    auto diagIDs1 = llvm::makeIntrusiveRefCnt<clang::DiagnosticIDs>();
+    clang::DiagnosticOptions diagOpts1;
+    auto diagnostic_engine = clang::DiagnosticsEngine(diagIDs1, diagOpts1);
     SourceManager source_mgr(diagnostic_engine, file_mgr);
 
-    DiagnosticsEngine diag_engine(
-        llvm::IntrusiveRefCntPtr<DiagnosticIDs>(new DiagnosticIDs()),
-        new DiagnosticOptions());
+    auto diagIDs2 = llvm::makeIntrusiveRefCnt<clang::DiagnosticIDs>();
+    clang::DiagnosticOptions diagOpts2;
+    clang::DiagnosticsEngine diag_engine(diagIDs2, diagOpts2);
 
     diag_engine.setClient(&consumer, false);
     auto id_note = diag_engine.getCustomDiagID(
@@ -198,4 +198,4 @@ TEST_CASE("Test diagnostic_consumer")
     CHECK(consumer.diagnostics.size() == 4);
     CHECK(consumer.diagnostics[3].level == DiagnosticsEngine::Fatal);
     CHECK(consumer.diagnostics[3].description == "test fatal message");
-}
\ No newline at end of file
+}
