diff --git a/src/zip/TarInputStream.cpp b/src/zip/TarInputStream.cpp
index 31218af..21b5fec 100644
--- a/src/zip/TarInputStream.cpp
+++ b/src/zip/TarInputStream.cpp
@@ -452,7 +452,7 @@ TarInputStream::readExtendedAttrs (TarMember &xdata, uint8_t &xfields, IOOffset
 	   throw TarError (TarError::HEADER_VALUE_ERROR);
 
 	// Find property name end at the first equal sign.
-	const char *propname = ++end;
+	char *propname = ++end;
 	if (! (end = strchr(propname, '=')) || end - p > reclen)
 	    throw TarError (TarError::HEADER_VALUE_ERROR);
 	ptrdiff_t proplen = end++ - propname;
# bla bla
