diff -Naur orig.classlib-3.1.2/classlib/iotools/ReadWriteBuffer.h classlib-3.1.2/classlib/iotools/ReadWriteBuffer.h
--- orig.classlib-3.1.2/classlib/iotools/ReadWriteBuffer.h	2010-10-22 20:09:35.000000000 +0200
+++ classlib-3.1.2/classlib/iotools/ReadWriteBuffer.h	2011-03-05 21:30:03.000000000 +0100
@@ -2,7 +2,7 @@
 # define CLASSLIB_READ_WRITE_BUFFER_H
 # include "classlib/sysapi/system.h"
 # include "classlib/sysapi/IOTypes.h"
-
+# include <stddef.h>
 namespace lat {
 
 /** Buffer for reads and writes to a seekable backing store.
diff --git a/src/utils/Signal.cpp b/src/utils/Signal.cpp
index 7fcef14..72b226e 100644
--- a/src/utils/Signal.cpp
+++ b/src/utils/Signal.cpp
@@ -318,7 +318,7 @@ Signal::queue (pid_t process, int sig, int value /* = 0 */)
     process = 0; // silence unused warning
     sig = 0;     // silence unused warning
     value = 0;   // silence unused warning
-    return 0;
+    return process + sig + (long long) value;
 #endif
 }

@@ -335,7 +335,7 @@ Signal::queue (pid_t process, int sig, void *value)
     process = 0; // silence unused warning
     sig = 0;     // silence unused warning
     value = 0;   // silence unused warning
-    return 0;
+    return process + sig + (long long) value;
 #endif
 }

@@ -350,7 +350,7 @@ Signal::queue (int sig, int value /* = 0 */)
 #else
     sig = 0;   // silence unused warning
     value = 0; // silence unused warning
-    return 0;
+    return sig + (long long) value;
 #endif
 }

@@ -365,7 +365,7 @@ Signal::queue (int sig, void *value)
 #else
     sig = 0;   // silence unused warning
     value = 0; // silence unused warning
-    return 0;
+    return sig + (long long)value;
 #endif
 }

@@ -452,7 +452,7 @@ Signal::wait (const sigset_t *mask,
     mask = 0;  // silence unused warning
     info = 0;  // silence unused warning
     msecs = 0; // silence unused warning
-    return 0;
+    return (long long) mask + (long long)info + msecs;
 #endif
 }
 
diff --git a/classlib/iobase/DirIterator.h b/classlib/iobase/DirIterator.h
index 623f1e1..9b52cb2 100644
--- a/classlib/iobase/DirIterator.h
+++ b/classlib/iobase/DirIterator.h
@@ -49,7 +49,7 @@ class Filename;
 
     The end indicator for the iteration range is the default
     constructed iterator.  */
-class DirIterator : public std::iterator<std::input_iterator_tag, std::string, ptrdiff_t>
+class DirIterator : public std::iterator<std::input_iterator_tag, std::string, std::ptrdiff_t>
 {
 public:
     typedef std::input_iterator_tag	iterator_category;
# bla bla
