diff --git a/CLHEP/Matrix/Matrix/DiagMatrix.h b/CLHEP/Matrix/Matrix/DiagMatrix.h
index 0a2929a..4cb9b40 100755
--- a/CLHEP/Matrix/Matrix/DiagMatrix.h
+++ b/CLHEP/Matrix/Matrix/DiagMatrix.h
@@ -56,8 +56,8 @@ public:
    virtual ~HepDiagMatrix();
    // Destructor.
 
-   inline int num_row() const;
-   inline int num_col() const;
+   int num_row() const;
+   int num_col() const;
    // Returns the number of rows/columns. (Should be equal.)
 
    double &operator()(int row, int col);
