diff --git a/2.1.4.1/CLHEP/Matrix/Matrix/DiagMatrix.icc b/2.1.4.1/CLHEP/Matrix/Matrix/DiagMatrix.icc
index 4d466c4..79cbef2 100755
--- a/2.1.4.1/CLHEP/Matrix/Matrix/DiagMatrix.icc
+++ b/2.1.4.1/CLHEP/Matrix/Matrix/DiagMatrix.icc
@@ -12,10 +12,6 @@ inline HepDiagMatrix::HepDiagMatrix()
    : m(0), nrow(0)
 {}
 
-inline int HepDiagMatrix::num_row() const { return nrow;}
-inline int HepDiagMatrix::num_col() const  { return nrow;}
-inline int HepDiagMatrix::num_size() const  { return nrow;}
-
 inline double & HepDiagMatrix::fast(int row,int col)
 {
 #ifdef MATRIX_BOUND_CHECK
@@ -47,16 +43,6 @@ inline const double & HepDiagMatrix::fast(int row,int col) const
   }
 }
 
-inline double & HepDiagMatrix::operator()(int row, int col)
-{
-   return fast(col,row);
-}
-
-inline const double & HepDiagMatrix::operator()(int row, int col) const 
-{ 
-   return fast(col,row);
-}
-
 inline void HepDiagMatrix::assign(const HepDiagMatrix &hm2) {(*this)=hm2;}
 
 inline HepDiagMatrix HepDiagMatrix::T() const {return HepDiagMatrix(*this);}
diff --git a/2.1.4.1/CLHEP/Matrix/Matrix/GenMatrix.h b/2.1.4.1/CLHEP/Matrix/Matrix/GenMatrix.h
index 247d8f6..cf846e3 100755
--- a/2.1.4.1/CLHEP/Matrix/Matrix/GenMatrix.h
+++ b/2.1.4.1/CLHEP/Matrix/Matrix/GenMatrix.h
@@ -36,7 +36,7 @@ class HepGenMatrix;
 class HepGenMatrix {
  
 public:
-   virtual ~HepGenMatrix() {}
+   virtual ~HepGenMatrix();
 
 
 #ifdef DISABLE_ALLOC   // disable this non-compliant allocator
diff --git a/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.h b/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.h
index b8408d3..ece7669 100755
--- a/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.h
+++ b/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.h
@@ -109,8 +109,8 @@ public:
    virtual ~HepSymMatrix();
    // Destructor.
 
-   inline int num_row() const;
-   inline int num_col() const;
+   int num_row() const;
+   int num_col() const;
    // Returns number of rows/columns.
 
    const double & operator()(int row, int col) const; 
diff --git a/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.icc b/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.icc
index 6ac3113..415320b 100755
--- a/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.icc
+++ b/2.1.4.1/CLHEP/Matrix/Matrix/SymMatrix.icc
@@ -13,10 +13,6 @@ inline HepSymMatrix::HepSymMatrix()
   : m(0), nrow(0), size_(0)
 {}
 
-inline int HepSymMatrix::num_row() const { return nrow;}
-inline int HepSymMatrix::num_col() const  { return nrow;}
-inline int HepSymMatrix::num_size() const  { return size_;}
-
 inline double & HepSymMatrix::fast(int row,int col)
 {
 #ifdef MATRIX_BOUND_CHECK
@@ -34,11 +30,6 @@ inline const double & HepSymMatrix::fast(int row,int col) const
   return *(m.begin()+(row*(row-1))/2+(col-1));
 }
 
-inline double & HepSymMatrix::operator()(int row, int col)
-    {return (row>=col? fast(row,col) : fast(col,row));}
-inline const double & HepSymMatrix::operator()(int row, int col) const 
-    {return (row>=col? fast(row,col) : fast(col,row));}
-
 inline void HepSymMatrix::assign(const HepSymMatrix &hm2) 
   {(*this)=hm2;}
 
diff --git a/2.1.4.1/CLHEP/Matrix/src/DiagMatrix.cc b/2.1.4.1/CLHEP/Matrix/src/DiagMatrix.cc
index f1a844c..09c7218 100755
--- a/2.1.4.1/CLHEP/Matrix/src/DiagMatrix.cc
+++ b/2.1.4.1/CLHEP/Matrix/src/DiagMatrix.cc
@@ -24,6 +24,20 @@
 
 namespace CLHEP {
 
+int HepDiagMatrix::num_row() const { return nrow;}
+int HepDiagMatrix::num_col() const  { return nrow;}
+int HepDiagMatrix::num_size() const  { return nrow;}
+
+double & HepDiagMatrix::operator()(int row, int col)
+{
+   return fast(col,row);
+}
+
+const double & HepDiagMatrix::operator()(int row, int col) const
+{
+   return fast(col,row);
+}
+
 // Simple operation for all elements
 
 #define SIMPLE_UOP(OPER)          \
diff --git a/2.1.4.1/CLHEP/Matrix/src/GenMatrix.cc b/2.1.4.1/CLHEP/Matrix/src/GenMatrix.cc
index f58f4b9..7030546 100755
--- a/2.1.4.1/CLHEP/Matrix/src/GenMatrix.cc
+++ b/2.1.4.1/CLHEP/Matrix/src/GenMatrix.cc
@@ -24,6 +24,9 @@
 
 namespace CLHEP {
 
+HepGenMatrix::~HepGenMatrix()
+{}
+
 #ifdef HEP_THIS_FUNCTION_IS_NOT_NEEDED
 static void delete_array(double *hm)
 {
diff --git a/2.1.4.1/CLHEP/Matrix/src/SymMatrix.cc b/2.1.4.1/CLHEP/Matrix/src/SymMatrix.cc
index b5c6fac..dfd2b0e 100755
--- a/2.1.4.1/CLHEP/Matrix/src/SymMatrix.cc
+++ b/2.1.4.1/CLHEP/Matrix/src/SymMatrix.cc
@@ -24,6 +24,15 @@
 
 namespace CLHEP {
 
+int HepSymMatrix::num_row() const { return nrow;}
+int HepSymMatrix::num_col() const  { return nrow;}
+int HepSymMatrix::num_size() const  { return size_;}
+
+double & HepSymMatrix::operator()(int row, int col)
+    {return (row>=col? fast(row,col) : fast(col,row));}
+const double & HepSymMatrix::operator()(int row, int col) const
+    {return (row>=col? fast(row,col) : fast(col,row));}
+
 // Simple operation for all elements
 
 #define SIMPLE_UOP(OPER)          \
# bla bla
