diff --git a/CLHEP/Units/Units/PhysicalConstants.h b/CLHEP/Units/Units/PhysicalConstants.h
index b25e5ff..448db37 100755
--- a/CLHEP/Units/Units/PhysicalConstants.h
+++ b/CLHEP/Units/Units/PhysicalConstants.h
@@ -44,6 +44,9 @@
 
 namespace CLHEP {
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
+
 //
 //
 //
@@ -131,6 +134,8 @@ static const double kGasThreshold   = 10.*mg/cm3;
 //
 static const double universe_mean_density = 1.e-25*g/cm3;
 
+#pragma GCC diagnostic pop
+
 }  // namespace CLHEP
 
 #ifdef ENABLE_BACKWARDS_COMPATIBILITY
# bla bla
