diff --git a/Utilities/cmcurl/CMakeLists.txt b/Utilities/cmcurl/CMakeLists.txt
index d7ff9aa..1e00622 100644
--- a/Utilities/cmcurl/CMakeLists.txt
+++ b/Utilities/cmcurl/CMakeLists.txt
@@ -692,22 +692,6 @@ ELSE(RETSIGTYPE_TEST)
   SET(RETSIGTYPE int)
 ENDIF(RETSIGTYPE_TEST)
 
-IF(CMAKE_COMPILER_IS_GNUCC AND APPLE)
-  INCLUDE(CheckCCompilerFlag)
-  CHECK_C_COMPILER_FLAG(-Wno-long-double HAVE_C_FLAG_Wno_long_double)
-  IF(HAVE_C_FLAG_Wno_long_double)
-    # The Mac version of GCC warns about use of long double.  Disable it.
-    GET_SOURCE_FILE_PROPERTY(MPRINTF_COMPILE_FLAGS mprintf.c COMPILE_FLAGS)
-    IF(MPRINTF_COMPILE_FLAGS)
-      SET(MPRINTF_COMPILE_FLAGS "${MPRINTF_COMPILE_FLAGS} -Wno-long-double")
-    ELSE(MPRINTF_COMPILE_FLAGS)
-      SET(MPRINTF_COMPILE_FLAGS "-Wno-long-double")
-    ENDIF(MPRINTF_COMPILE_FLAGS)
-    SET_SOURCE_FILES_PROPERTIES(mprintf.c PROPERTIES
-      COMPILE_FLAGS ${MPRINTF_COMPILE_FLAGS})
-  ENDIF(HAVE_C_FLAG_Wno_long_double)
-ENDIF(CMAKE_COMPILER_IS_GNUCC AND APPLE)
-
 INCLUDE(CMake/OtherTests.cmake)
 
 # The rest of the build
# bla bla
