diff -Naur cmake-1.8.3.orig/Source/cmDynamicLoader.cxx cmake-1.8.3/Source/cmDynamicLoader.cxx
--- cmake-1.8.3.orig/Source/cmDynamicLoader.cxx	2003-08-27 22:50:00.000000000 +0200
+++ cmake-1.8.3/Source/cmDynamicLoader.cxx	2006-06-01 00:38:45.000000000 +0200
@@ -184,7 +184,7 @@
   // we have to use lib because the macro may not...
   (void)lib;
 
-  NSUnLinkModule(lib, FALSE);
+  NSUnLinkModule((__NSModule*)lib, FALSE);
   return 0;
 }
 
# bla bla
