--- include/Inventor/nodes/SoSubNode.h	2006-05-26 14:07:08.000000000 +0200
+++ include/Inventor/nodes/SoSubNode.h.new	2006-05-26 14:16:03.000000000 +0200
@@ -98,9 +98,6 @@
   return (const SoFieldData **)(&_class_::fieldData); \
 } \
  \
-unsigned int _class_::classinstances = 0; \
-const SoFieldData ** _class_::parentFieldData = NULL; \
-SoFieldData * _class_::fieldData = NULL; \
  \
 const SoFieldData * \
 _class_::getFieldData(void) const \
@@ -113,7 +110,10 @@
 { \
   delete _class_::fieldData; \
   _class_::fieldData = NULL; \
-}
+} \
+unsigned int _class_::classinstances = 0; \
+const SoFieldData ** _class_::parentFieldData = NULL; \
+SoFieldData * _class_::fieldData = NULL
 
 // FIXME: document. 20000103 mortene.
 #define SO_NODE_SOURCE(_class_) \
