diff -arupN src.orig/PyCoral/src/Attribute.cpp src/PyCoral/src/Attribute.cpp
--- src.orig/PyCoral/src/Attribute.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/Attribute.cpp	2012-07-04 12:14:01.000000000 +0200
@@ -11,6 +11,12 @@
 #include "Exception.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 
 static int Attribute_init( PyObject* self, PyObject* args , PyObject* /* kwds */);
diff -arupN src.orig/PyCoral/src/IColumn.cpp src/PyCoral/src/IColumn.cpp
--- src.orig/PyCoral/src/IColumn.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/IColumn.cpp	2012-07-04 12:12:31.000000000 +0200
@@ -3,6 +3,12 @@
 #include "RelationalAccess/IColumn.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int IColumn_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void IColumn_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/IConnectionServiceConfiguration.cpp src/PyCoral/src/IConnectionServiceConfiguration.cpp
--- src.orig/PyCoral/src/IConnectionServiceConfiguration.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/IConnectionServiceConfiguration.cpp	2012-07-04 12:10:47.000000000 +0200
@@ -3,6 +3,12 @@
 #include "Exception.h"
 #include <exception>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int IConnectionServiceConfiguration_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void IConnectionServiceConfiguration_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/ICursor.cpp src/PyCoral/src/ICursor.cpp
--- src.orig/PyCoral/src/ICursor.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/ICursor.cpp	2012-07-04 12:13:32.000000000 +0200
@@ -5,6 +5,12 @@
 #include "RelationalAccess/ICursor.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int ICursor_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void ICursor_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/IIndex.cpp src/PyCoral/src/IIndex.cpp
--- src.orig/PyCoral/src/IIndex.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/IIndex.cpp	2012-07-04 12:11:40.000000000 +0200
@@ -3,6 +3,12 @@
 #include "RelationalAccess/IIndex.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int IIndex_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void IIndex_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/ISchema.cpp src/PyCoral/src/ISchema.cpp
--- src.orig/PyCoral/src/ISchema.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/ISchema.cpp	2012-07-04 12:13:07.000000000 +0200
@@ -11,6 +11,12 @@
 #include "PyCoral/AttributeList.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int ISchema_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void ISchema_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/ITableDescription.cpp src/PyCoral/src/ITableDescription.cpp
--- src.orig/PyCoral/src/ITableDescription.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/ITableDescription.cpp	2012-07-04 12:14:21.000000000 +0200
@@ -9,6 +9,12 @@
 #include "RelationalAccess/ITableDescription.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int ITableDescription_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void ITableDescription_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/ITableSchemaEditor.cpp src/PyCoral/src/ITableSchemaEditor.cpp
--- src.orig/PyCoral/src/ITableSchemaEditor.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/ITableSchemaEditor.cpp	2012-07-04 12:14:44.000000000 +0200
@@ -7,6 +7,12 @@
 #include "Exception.h"
 #include "ITableSchemaEditor.h"
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int ITableSchemaEditor_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void ITableSchemaEditor_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/ITransaction.cpp src/PyCoral/src/ITransaction.cpp
--- src.orig/PyCoral/src/ITransaction.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/ITransaction.cpp	2012-07-04 12:16:10.000000000 +0200
@@ -3,6 +3,12 @@
 #include "RelationalAccess/ITransaction.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int ITransaction_init( PyObject* self, PyObject* args, PyObject* kwds );
 static void ITransaction_dealloc( PyObject* self );
diff -arupN src.orig/PyCoral/src/IWebCacheInfo.cpp src/PyCoral/src/IWebCacheInfo.cpp
--- src.orig/PyCoral/src/IWebCacheInfo.cpp	2011-03-22 11:53:01.000000000 +0100
+++ src/PyCoral/src/IWebCacheInfo.cpp	2012-07-04 12:15:48.000000000 +0200
@@ -3,6 +3,12 @@
 #include "RelationalAccess/IWebCacheInfo.h"
 #include <sstream>
 
+// Get rid of 'dereferencing type-punned pointer will break strict-aliasing rules'
+// warnings caused by Py_RETURN_TRUE/FALSE.
+#if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2))
+  #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
 // Forward declaration of the methods
 static int IWebCacheInfo_init( PyObject* self, PyObject* /* args */, PyObject* /* kwds */);
 static void IWebCacheInfo_dealloc( PyObject* self );
# bla bla
