diff --git a/src/CoralKernel/src/PluginManager.cpp b/src/CoralKernel/src/PluginManager.cpp
index 3ea645c..78b1720 100755
--- a/src/CoralKernel/src/PluginManager.cpp
+++ b/src/CoralKernel/src/PluginManager.cpp
@@ -300,7 +300,7 @@ coral::PluginManager::loadLibrary( const std::string& libraryName )
   }
 #ifdef __APPLE__
   //void* handle = ::dlopen( ( "liblcg_" + libraryName + ".so" ).c_str(), RTLD_GLOBAL | RTLD_NOW ); // attempt to fix bug #81005
-  void* handle = ::dlopen( ( "liblcg_" + libraryName + ".so" ).c_str(), RTLD_GLOBAL | RTLD_LAZY );
+  void* handle = ::dlopen( ( "liblcg_" + libraryName + ".dylib" ).c_str(), RTLD_GLOBAL | RTLD_LAZY );
 #else
   void* handle = ::dlopen( ( "liblcg_" + libraryName + ".so" ).c_str(), RTLD_GLOBAL | RTLD_LAZY );
 #endif
# bla bla
