--- orig.src/CoralBase/CoralBase/SearchPath.h	2008-03-14 01:06:15.000000000 +0100
+++ src/CoralBase/CoralBase/SearchPath.h	2010-11-05 15:04:27.000000000 +0100
@@ -37,6 +37,7 @@
     typedef DirList::const_reverse_iterator	const_reverse_iterator;
 
     typedef DirList::size_type			  size_type;
+    typedef DirList::value_type			  value_type;
     typedef DirList::reference			  reference;
     typedef DirList::const_reference	const_reference;
 
