--- orig.src/FrontierAccess/src/NamedInputParametersParser.cpp	2010-07-29 19:00:04.000000000 +0200
+++ src/FrontierAccess/src/NamedInputParametersParser.cpp	2010-11-05 16:03:58.000000000 +0100
@@ -45,7 +45,7 @@
             inQuotes = true;
             bquote   = i;
             ranges.push_back(
-              std::make_pair < std::string::size_type, std::string::size_type > (
+              std::pair < std::string::size_type, std::string::size_type > (
                 nbquote,
                 nequote-nbquote-1
                 )
@@ -70,7 +70,7 @@
       }
 
       ranges.push_back(
-        std::make_pair < std::string::size_type, std::string::size_type > (
+        std::pair < std::string::size_type, std::string::size_type > (
           nbquote,
           nequote-nbquote
           )
@@ -140,7 +140,7 @@
 //                     << " found bound variable "                        << std::endl
 //                     << ">>" << fragment.substr( sbeg,slen ) << "<< at position " << sbeg << " " << slen << " chars long and real name " << realName << std::endl;          
         
-          m_bvars.insert( std::make_pair< std::string, std::pair<std::string::size_type, std::string::size_type > > ( realName, std::make_pair< std::string::size_type, std::string::size_type > ( sbeg, slen ) ) );
+          m_bvars.insert( std::pair< std::string, std::pair<std::string::size_type, std::string::size_type > > ( realName, std::pair< std::string::size_type, std::string::size_type > ( sbeg, slen ) ) );
           
 //           fragment.erase( sbeg, slen );      
           
