diff --git a/src/CoralKernel/src/PluginManager.cpp b/src/CoralKernel/src/PluginManager.cpp
index 2c40531..815d4ad 100755
--- a/src/CoralKernel/src/PluginManager.cpp
+++ b/src/CoralKernel/src/PluginManager.cpp
@@ -302,7 +302,11 @@ coral::PluginManager::loadLibrary( const std::string& libraryName )
       else log << coral::Warning << "Could not preload the globus 64bit gssapi library (workaround for bug #70641)" << coral::MessageStream::endmsg;
     }
   }
+#ifdef __APPLE__
+  void* handle = ::dlopen( ( "liblcg_" + libraryName + ".dylib" ).c_str(), RTLD_GLOBAL | RTLD_LAZY );
+#else
   void* handle = ::dlopen( ( "liblcg_" + libraryName + ".so" ).c_str(), RTLD_GLOBAL | RTLD_LAZY );
+#endif
   if ( handle == 0 ) {
     throw std::runtime_error( "Could not load the shared library \"" + libraryName + "\" : " + std::string( ::dlerror() ) );
   }
