--- src/couchdb/Makefile.am	2010-09-08 16:31:30.000000000 +0200
+++ src/couchdb/Makefile.am	2010-09-07 19:56:21.000000000 +0200
@@ -32,6 +32,7 @@
     couch_auth_cache.erl \
     couch_btree.erl \
     couch_changes.erl \
+    couch_cms_auth.erl \
     couch_config.erl \
     couch_config_writer.erl \
     couch_db.erl \
@@ -90,6 +91,7 @@
     couch_auth_cache.beam \
     couch_btree.beam \
     couch_changes.beam \
+    couch_cms_auth.beam \
     couch_config.beam \
     couch_config_writer.beam \
     couch_db.beam \
