Index: src/couchdb/couch_rep_changes_feed.erl
===================================================================
--- src/couchdb/couch_rep_changes_feed.erl	(revision 1199356)
+++ src/couchdb/couch_rep_changes_feed.erl	(working copy)
@@ -154,7 +154,7 @@
         end;
     {ibrowse_async_headers, ReqId, Code, _} ->
         {stop, {changes_error_code, list_to_integer(Code)}}
-    after 10000 ->
+    after 30000 ->
         {stop, changes_timeout}
     end;
 
