Index: src/couchdb/couch_db_updater.erl
===================================================================
--- src/couchdb/couch_db_updater.erl	2011-03-28 16:00:53.000000000 +0200
+++ src/couchdb/couch_db_updater.erl	2013-06-12 08:40:50.000000000 +0200
@@ -151,7 +151,7 @@
         ?LOG_INFO("Starting compaction for db \"~s\"", [Db#db.name]),
         Pid = spawn_link(fun() -> start_copy_compact(Db) end),
         Db2 = Db#db{compactor_pid=Pid},
-        ok = gen_server:call(Db#db.main_pid, {db_updated, Db2}),
+        ok = gen_server:call(Db#db.main_pid, {db_updated, Db2}, infinity),
         {reply, ok, Db2};
     _ ->
         % compact currently running, this is a no-op

