--- src/couchdb/couch_rep_changes_feed.erl	2012-03-15 16:54:00.000000000 +0100
+++ src/couchdb/couch_rep_changes_feed.erl	2012-03-16 10:03:41.000000000 +0100
@@ -267,7 +269,7 @@
     handle_messages(Messages, State);
 
 handle_info({ibrowse_async_response_end, Id}, #state{reqid=Id} = State) ->
-    handle_feed_completion(State);
+    handle_retry(State);
 
 handle_info({'EXIT', From, normal}, #state{changes_loop=From} = State) ->
     handle_feed_completion(State);
