--- src/couchdb/couch_rep_httpc.erl	2011-04-13 13:06:16.000000000 +0200
+++ src/couchdb/couch_rep_httpc.erl	2011-07-21 10:42:21.000000000 +0200
@@ -287,10 +287,13 @@
 ssl_options(#http_db{url = Url}) ->
     case ibrowse_lib:parse_url(Url) of
     #url{protocol = https} ->
+        CertFile=couch_config:get("ssl", "cert_file", nil),
+        KeyFile=couch_config:get("ssl", "key_file", nil),
+        CaCertFile=couch_config:get("ssl", "cacert_file", nil),
         Depth = list_to_integer(
             couch_config:get("replicator", "ssl_certificate_max_depth", "3")
         ),
-        SslOpts = [{depth, Depth} |
+        SslOpts = [{certfile, CertFile}, {keyfile, KeyFile}, {cacertfile, CaCertFile}, {depth, Depth} |
         case couch_config:get("replicator", "verify_ssl_certificates") of
         "true" ->
             ssl_verify_options(true);
