From 073549685099b6c7c55db251133403fdcfb4d965 Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <davidlt@cern.ch>
Date: Sat, 19 Oct 2013 20:26:47 +0200
Subject: [PATCH 1/2] Protect gets with HAVE_RAW_DECL_GETS in stdio.in.h

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>
---
 gnu/stdio.in.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gnu/stdio.in.h b/gnu/stdio.in.h
index 88b368b..44f1087 100644
--- a/gnu/stdio.in.h
+++ b/gnu/stdio.in.h
@@ -139,7 +139,9 @@ _GL_WARN_ON_USE (fflush, "fflush is not always POSIX compliant - "
    so any use of gets warrants an unconditional warning.  Assume it is
    always declared, since it is required by C89.  */
 #undef gets
+#if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
+#endif
 
 #if @GNULIB_FOPEN@
 # if @REPLACE_FOPEN@
-- 
1.8.0

# bla bla
