From 59cac0936a79f8c7a6d00e5681da728b32cecc8a Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <davidlt@cern.ch>
Date: Sat, 19 Oct 2013 20:31:40 +0200
Subject: [PATCH 2/2] Fix invalid redefinition of stat()

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>
---
 src/filetypes.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/filetypes.h b/src/filetypes.h
index f80faab..173ddfc 100644
--- a/src/filetypes.h
+++ b/src/filetypes.h
@@ -81,5 +81,9 @@
 #ifndef S_ISLNK
 #define lstat stat
 #endif
+#ifndef lstat
 int lstat ();
+#endif
+#ifndef stat
 int stat ();
+#endif
-- 
1.8.0

# bla bla
