--- setup_dependencies.py
+++ setup_dependencies.py
@@ -106,6 +106,12 @@ dependencies = {
                         'modules': ['WMCore.FwkJobReport.FileInfo', 'WMCore.Services.Requests',
                                     'WMCore.Services.Service', 'WMCore.Services.pycurl_manager'],
                         },
+                'crabinterface':{
+                        'packages' : ['WMCore.WMSpec+', 'WMCore.Services.SiteDB+', 'WMCore.Services.JSONParser+',
+                                      'WMCore.Services.WorkQueue', 'WMCore.HTTPFrontEnd', 'WMCore.RequestManager+',
+                                      'WMCore.HTTPFrontEnd.RequestManager' ],
+                        'systems': ['wmc-database'],
+                        },
                 'wmclient':{
                         'systems': ['wmc-runtime', 'wmc-database']
                         },
