--- htcondor_make_runtime.sh.orig	2020-11-16 13:44:04.093228726 +0100
+++ htcondor_make_runtime.sh	2020-11-16 13:46:05.390384346 +0100
@@ -56,7 +56,7 @@
     zip -rq $STARTDIR/CRAB3.zip RESTInteractions.py HTCondorUtils.py HTCondorLocator.py TaskWorker CRABInterface  TransferInterface -x \*.pyc || exit 3
     popd
 
-    pushd $VO_CMS_SW_DIR/$SCRAM_ARCH/external/cherrypy/*/lib/python2.7/site-packages
+    pushd $VO_CMS_SW_DIR/$SCRAM_ARCH/external/cherrypy/*/lib/python2.7/site-packages/CherryPy-5.4.0.post*/
     zip -rq $STARTDIR/CRAB3.zip cherrypy -x \*.pyc
     popd
 
