--- setup_dependencies.py
+++ setup_dependencies.py
@@ -101,10 +101,11 @@ dependencies = {
                         },
                 'crabclient':{
                         'packages': ['WMCore.Wrappers+', 'WMCore.Credential', 'PSetTweaks',
-                                     'WMCore.Services.UserFileCache'],
+                                     'WMCore.Services.UserFileCache', 'WMCore.GroupUser'],
                         'systems': ['wmc-base'],
-                        'modules': ['WMCore.FwkJobReport.FileInfo', 'WMCore.Services.Requests',
-                                    'WMCore.Services.Service', 'WMCore.Services.pycurl_manager'],
+                        'modules': ['WMCore.Database.CMSCouch', 'WMCore.FwkJobReport.FileInfo',
+                                    'WMCore.Services.Requests', 'WMCore.Services.Service',
+                                    'WMCore.Services.pycurl_manager', 'WMCore.Cache.WMConfigCache'],
                         },
                 'wmclient':{
                         'systems': ['wmc-runtime', 'wmc-database']
