--- setup_dependencies.py
+++ setup_dependencies.py
@@ -94,6 +94,11 @@ dependencies = {
                         'packages': ['WMCore.Services+'],
                         'systems': ['wmagent', 'reqmgr']
                         },
+                'crabclient':{
+                        'packages': ['WMCore.Credential'],
+                        'systems': ['wmc-base'],
+                        'modules': ['WMCore.FwkJobReport.FileInfo', 'WMCore.Services.Requests']
+                        },
                 'wmclient':{
                         'systems': ['wmc-runtime', 'wmc-database']
                         },
