--- setup_dependencies.py
+++ setup_dependencies.py
@@ -95,6 +95,19 @@ dependencies = {
                         'statics': ['src/couchapps/Agent+'],
                         },
                 'crabserver':{
+                        'packages' : ['WMCore.WMSpec+','WMCore.Services.SiteDB+', 'WMCore.Services.JSONParser+',
+                                      'WMCore.Services.WorkQueue', 'WMCore.HTTPFrontEnd', 'WMCore.RequestManager+',
+                                      'WMCore.HTTPFrontEnd.RequestManager', 'WMCore.Services.PhEDEx+', 'WMCore.WMStats',
+                                      'WMCore.Services.WMStats', 'WMCore.Services.Dashboard'],
+                        'systems': ['wmc-rest', 'wmc-database'],
+                        'statics': ['src/couchapps/WMStat+'],
+                        },
+                'wmc-rest':{
+                       'bin': ['wmc-dist-patch', 'wmc-dist-unpatch','wmc-httpd'],
+                        'packages' : ['WMCore.REST'],
+                        'modules': ['WMCore.Configuration'],
+                       },
+                'crab3':{
                         'packages': ['WMCore.WMSpec', 'WMCore.ACDC',
                                      'WMCore.Storage+', 'WMCore.HTTPFrontEnd.RequestManager+',
                                      'WMCore.RequestManager+', 'WMComponent.DBSUpload',
@@ -124,7 +137,6 @@ dependencies = {
                 #        'statics': ['src/html/GlobalMonitor',
                 #                    'src/javascript/WMCore/WebTools/GlobalMonitor+']
                 #        },
-                
                 'reqmon':{
                         'statics': ['src/couchapps/WMStats+'],
                         },
