diff --git a/src/python/WMCore/Credential/Proxy.py b/src/python/WMCore/Credential/Proxy.py
index 5b6feee..6cfaa65 100644
--- a/src/python/WMCore/Credential/Proxy.py
+++ b/src/python/WMCore/Credential/Proxy.py
@@ -130,6 +130,7 @@ class Proxy(Credential):
         self.myproxyMinTime = args.get( "myproxyMinTime", 4) #threshold used in checkProxy
         self.myproxyAccount = args.get( "myproxyAccount", "") #to be used when computing myproxy account (-l option)
         self.rfcCompliant = args.get( "rfcCompliant", True) #to be used when computing myproxy account (-l option)
+        self.trustedRetrievers = None
 
         # User vo paramaters
         self.vo = 'cms'
