--- setup_dependencies.py
+++ setup_dependencies.py
@@ -98,6 +98,14 @@ dependencies = {
                                      'WMCore.RequestManager+'],
                         'systems': ['wmc-web']
                         },
+                'crabclient':{
+                        'packages': ['WMCore.Wrappers+', 'WMCore.Credential', 'PSetTweaks',
+                                     'WMCore.Services.UserFileCache'],
+                        'systems': ['wmc-base'],
+                        'modules': ['WMCore.FwkJobReport.FileInfo', 'WMCore.Services.Requests',
+                                    'WMCore.Services.Service', 'WMCore.Services.pycurl_manager']
+                        },
+
                 'wmclient':{
                         'systems': ['wmc-runtime', 'wmc-database']
                         },
