--- setup_dependencies.py.orig	2013-12-19 22:33:38.000000000 +0100
+++ setup_dependencies.py	2013-12-19 22:50:03.000000000 +0100
@@ -130,9 +130,9 @@
                         },
                 'crabtaskworker':{
                         'packages':['WMCore..WorkQueue', 'WMCore.Credential', 'WMCore.Algorithms+', 'WMCore.WMSpec+',
-                                     'WMCore.JobSplitting', 'WMCore.Services.SiteDB+', 'WMCore.Services.DBS+', 'WMCore.Services.UserFileCache+'],
+                                     'WMCore.JobSplitting', 'WMCore.Services.SiteDB+', 'WMCore.Services.DBS+', 'WMCore.Services.UserFileCache+', 'WMCore.Services.PhEDEx+'],
                         'modules': ['WMCore.WMBS.File', 'WMCore.WMBS.WMBSBase', 'WMCore.WMBS.__init__'],
-                        'systems': ['wmc-database'],
+                        'systems': ['wmc-database', 'wmc-runtime'],
                         },
                 'wmclient':{
                         'systems': ['wmc-runtime', 'wmc-database']
