--- setup_dependencies.py
+++ setup_dependencies.py
@@ -114,27 +114,33 @@ 
                         'statics': ['src/couchapps/Agent+'],
                         },
                 'crabserver':{
-                        'packages': ['WMCore.WMSpec', 'WMCore.ACDC',
+                        'packages':['WMCore.WMSpec', 'WMCore.ACDC',
                                      'WMCore.Storage+', 'WMCore.HTTPFrontEnd.RequestManager+',
                                      'WMCore.RequestManager+', 'WMComponent.DBSUpload',
                                      'WMCore.ProcessPool'],
                         'systems': ['wmc-web'],
                         },
                 'crabclient':{
-                        'packages': ['WMCore.Wrappers+', 'WMCore.Credential', 'PSetTweaks',
+                        'packages':['WMCore.Wrappers+', 'WMCore.Credential', 'PSetTweaks',
                                      'WMCore.Services.UserFileCache'],
                         'systems': ['wmc-base'],
                         'modules': ['WMCore.FwkJobReport.FileInfo', 'WMCore.Services.Requests',
                                     'WMCore.Services.Service', 'WMCore.Services.pycurl_manager'],
                         },
+                'crabtaskworker':{
+                        'packages':['WMCore..WorkQueue', 'WMCore.Credential', 'WMCore.Algorithms+',
+                                     'WMCore.JobSplitting', 'WMCore.Services.SiteDB+', 'WMCore.Services.DBS+'],
+                        'modules': ['WMCore.WMBS.File', 'WMCore.WMBS.WMBSBase', 'WMCore.WMBS.__init__'],
+                        'systems': ['wmc-database'],
+                        },
                 'wmclient':{
                         'systems': ['wmc-runtime', 'wmc-database']
                         },
                 'reqmon':{
-                        'statics': ['src/couchapps/WMStats+', 
+                        'statics': ['src/couchapps/WMStats+',
                                     'src/couchapps/WorkloadSummary+'],
                         },
-                'alertscollector': 
+                'alertscollector':
                 {
                         'statics': ['src/couchapps/AlertsCollector+'],
                 },
