diff --git a/Clients/Python/DBSAPI/dbsHttpService.py b/Clients/Python/DBSAPI/dbsHttpService.py
index 699801c..bffd325 100644
--- a/Clients/Python/DBSAPI/dbsHttpService.py
+++ b/Clients/Python/DBSAPI/dbsHttpService.py
@@ -253,7 +253,7 @@ class DbsHttpService:
        if typ == 'POST':
           result = self.conn.request(typ, request_string, params, headers)  
        else:
-          result = self.conn.request(typ, request_string, {}, headers )
+          result = self.conn.request(typ, request_string, "", headers )
 
        response = self.conn.getresponse() 
 
