diff --git a/DDCore/src/Primitives.cpp b/DDCore/src/Primitives.cpp
index b319624..bafb16a 100644
--- a/DDCore/src/Primitives.cpp
+++ b/DDCore/src/Primitives.cpp
@@ -25,7 +25,7 @@
 #include <cstring>
 #include <map>
 
-#if defined(__linux) || defined(__APPLE__)
+#if defined(__linux) || defined(__APPLE__) || defined(__powerpc64__)
 #include <cxxabi.h>
 #ifndef __APPLE__
 typedef abi::__class_type_info class_t;
