diff --git a/cmake/DD4hepBuild.cmake b/cmake/DD4hepBuild.cmake
index 174a22a..e2b8c3a 100644
--- a/cmake/DD4hepBuild.cmake
+++ b/cmake/DD4hepBuild.cmake
@@ -613,7 +613,7 @@ function(dd4hep_add_plugin binary)
   dd4hep_debug("|++++> USES ${ARG_USES}")
   dd4hep_debug("|++++> INCLUDES ${ARG_INCLUDES}")
   dd4hep_debug("|++++> DEFINITIONS ${ARG_DEFINITIONS}")
-  add_library(${binary} SHARED ${SOURCES} ${ARG_GENERATED})
+  add_library(${binary} STATIC ${SOURCES} ${ARG_GENERATED})
   target_link_libraries(${binary} PUBLIC ${ARG_USES})
   target_include_directories(${binary} PUBLIC ${ARG_INCLUDES})
   target_compile_definitions(${binary} PUBLIC ${ARG_DEFINITIONS})
@@ -623,7 +623,7 @@ function(dd4hep_add_plugin binary)
     if(CMAKE_INSTALL_LIBDIR)
       set(install_destination ${CMAKE_INSTALL_LIBDIR})
     endif()
-    install(TARGETS ${binary} LIBRARY DESTINATION ${install_destination})
+    install(TARGETS ${binary} ARCHIVE DESTINATION ${install_destination})
   endif()
 endfunction(dd4hep_add_plugin)
 
