--- orig.LCG-DM-1.7.4/shlib/Imakefile	2010-06-10 17:25:25.000000000 +0200
+++ LCG-DM-1.7.4/shlib/Imakefile	2010-08-27 13:11:06.000000000 +0200
@@ -60,7 +60,7 @@
 	@mkdir tmp
 	@for i in $^ ;\
 	do (cd tmp; case $$i in *.a) $(AR) x ../$$i; rm -f $(SYMTABNAM) ;; esac) done
-	(cd tmp; ld $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) -L .. -llcgdm $(LIBCSEC))
+	(cd tmp; c++ $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) -L .. -llcgdm $(LIBCSEC))
 	@-rm -rf tmp
 
 liblcgdm.sl liblcgdm.so liblcgdm.dynlib: $(COMMONLIB) $(SECURITYLIB)
@@ -69,7 +69,7 @@
 	@mkdir tmp
 	@for i in $^ ;\
 	do (cd tmp; $(AR) x ../$$i; rm -f $(SYMTABNAM)) done
-	(cd tmp; ld $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) $(LIBCSEC))
+	(cd tmp; c++ $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) $(LIBCSEC))
 	@-rm -rf tmp
 
 libdpm.dll: $(RFIOLIB) $(NSLIB) $(DPMLIB)
# bla bla
