diff --git a/LCG-DM-1.8.0/config/darwin.cf b/LCG-DM-1.8.0/config/darwin.cf
index ab0bf06..8cf4e0b 100644
--- a/LCG-DM-1.8.0/config/darwin.cf
+++ b/LCG-DM-1.8.0/config/darwin.cf
@@ -12,8 +12,8 @@ GSOAP_PLATFORM = macosx
 MTCCFLAGS = -DCTHREAD_POSIX -D_THREAD_SAFE -D_REENTRANT
 MTLDLIBS =
 MTLDFLAGS =
-SHLIBLDFLAGS = -undefined dynamic_lookup -dylib -single_module
-SHLIBREQLIBS = -lc /usr/lib/dylib1.o
+SHLIBLDFLAGS = -undefined dynamic_lookup -shared
+SHLIBREQLIBS = -lc
 FC = gfortran
 
 #include <darwin.rules>
# bla bla
