diff --git a/Imakefile b/Imakefile
index 323f9e8..bdef278 100644
--- a/LCG-DM-1.7.4/shlib/Imakefile
+++ b/LCG-DM-1.7.4/shlib/Imakefile
@@ -54,13 +54,13 @@ exportshr:
 libdpm.sl libdpm.so libdpm.dylib: $(RFIOLIB) $(NSLIB) $(DPMLIB)
 liblfc.sl liblfc.so liblfc.dylib: $(NSLIB)
 
-%.sl %.so %.dynlib: SharedLibraryTargetName(lcgdm)
+%.sl %.so %.dylib: SharedLibraryTargetName(lcgdm)
 	@echo " making $@ in `pwd`"
 	@-rm -rf tmp
 	@mkdir tmp
 	@for i in $^ ;\
 	do (cd tmp; case $$i in *.a) $(AR) x ../$$i; rm -f $(SYMTABNAM) ;; esac) done
-	(cd tmp; c++ $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) -L .. -llcgdm $(LIBCSEC))
+	(cd tmp; c++ $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) -L.. $(LIBCSEC))
 	@-rm -rf tmp
 
 liblcgdm.sl liblcgdm.so liblcgdm.dynlib: $(COMMONLIB) $(SECURITYLIB)
diff --git a/LCG-DM-1.7.4/shlib/Imakefile b/LCG-DM-1.7.4/shlib/Imakefile
index bdef278..1f1a1e0 100644
--- a/LCG-DM-1.7.4/shlib/Imakefile
+++ b/LCG-DM-1.7.4/shlib/Imakefile
@@ -63,7 +63,7 @@ liblfc.sl liblfc.so liblfc.dylib: $(NSLIB)
 	(cd tmp; c++ $(SHLIBLDFLAGS) -o ../$@ *.o $(SHLIBREQLIBS) -L.. $(LIBCSEC))
 	@-rm -rf tmp
 
-liblcgdm.sl liblcgdm.so liblcgdm.dynlib: $(COMMONLIB) $(SECURITYLIB)
+liblcgdm.sl liblcgdm.so liblcgdm.dylib: $(COMMONLIB) $(SECURITYLIB)
 	@echo " making $@ in `pwd`"
 	@-rm -rf tmp
 	@mkdir tmp
# bla bla
