diff -Naur dqmgui-1.0.1_R6.24.07/plugins/PCLPixelAlignmentRenderPlugin.cc ../dqmgui-1.0.1_R6.24.07/plugins/PCLPixelAlignmentRenderPlugin.cc
--- dqmgui-1.0.1_R6.24.07/plugins/PCLPixelAlignmentRenderPlugin.cc	2021-09-10 23:22:46.000000000 +0200
+++ ../dqmgui-1.0.1_R6.24.07/plugins/PCLPixelAlignmentRenderPlugin.cc	2022-06-01 15:52:10.488008246 +0200
@@ -12,6 +12,7 @@
 #include "TList.h"
 #include "TLine.h"
 #include <cassert>
+#include <array>
 
 class PCLPixelAlignmentRenderPlugin : public DQMRenderPlugin {
   std::array<double, 6> sigCut_;
diff -Naur dqmgui-1.0.1_R6.24.07/plugins/RPCRenderPlugin.cc ../dqmgui-1.0.1_R6.24.07/plugins/RPCRenderPlugin.cc
--- dqmgui-1.0.1_R6.24.07/plugins/RPCRenderPlugin.cc	2021-09-10 23:22:46.000000000 +0200
+++ ../dqmgui-1.0.1_R6.24.07/plugins/RPCRenderPlugin.cc	2022-06-01 15:52:32.232906821 +0200
@@ -6,6 +6,7 @@
 #include "TColor.h"
 #include <iostream>
 #include <cassert>
+#include <array>
 #include "TLine.h"
 #include "TText.h"
 #include "TPaletteAxis.h"
