--- lib/ssl/src/ssl_connection.erl	2014-05-23 09:00:51.000000000 +0200
+++ lib/ssl/src/ssl_connection.erl	2014-05-23 09:01:31.000000000 +0200
@@ -1080,7 +1080,7 @@
 
 init_certificates(undefined, CertDbRef, CertDbHandle, CacheHandle, CertFile, client) ->
     try 
-	[OwnCert] = ssl_certificate:file_to_certificats(CertFile, CertDbHandle),
+	[OwnCert|T] = ssl_certificate:file_to_certificats(CertFile, CertDbHandle),
 	{ok, CertDbRef, CertDbHandle, CacheHandle, OwnCert}
     catch _Error:_Reason  ->
 	    {ok, CertDbRef, CertDbHandle, CacheHandle, undefined}
@@ -1088,7 +1088,7 @@
 
 init_certificates(undefined, CertDbRef, CertDbHandle, CacheRef, CertFile, server) ->
     try
-	[OwnCert] = ssl_certificate:file_to_certificats(CertFile, CertDbHandle),
+	[OwnCert|T] = ssl_certificate:file_to_certificats(CertFile, CertDbHandle),
 	{ok, CertDbRef, CertDbHandle, CacheRef, OwnCert}
     catch
 	Error:Reason ->
