diff --git a/9.1/EvtGenBase/EvtStreamInputIterator.hh b/9.1/EvtGenBase/EvtStreamInputIterator.hh
index 6f3d609..e81845f 100755
--- a/9.1/EvtGenBase/EvtStreamInputIterator.hh
+++ b/9.1/EvtGenBase/EvtStreamInputIterator.hh
@@ -23,7 +23,7 @@ public:
 
   typedef input_iterator_tag iterator_category;
   typedef Point              value_type;
-  typedef ptrdiff_t          difference_type;
+  typedef std::ptrdiff_t     difference_type;
   typedef const Point*       pointer;
   typedef const Point&       reference;
 
