--- orig.evtgenlhc/9.1/src/Pythia.F	2010-09-01 09:52:27.000000000 +0200
+++ evtgenlhc/9.1/src/Pythia.F	2010-09-01 10:00:19.000000000 +0200
@@ -236,6 +236,10 @@
       COMMON/PYJETS/N,NPAD,K(4000,5),P(4000,5),V(4000,5)
       
       INTEGER N,NPAD,K
+
+      INTEGER MINT
+      DOUBLE PRECISION VINT 
+      COMMON/PYINT1/MINT(400),VINT(400)
       
       INTEGER NDMAX
       PARAMETER (NDMAX=100)
@@ -281,11 +285,13 @@
           IP1=MSTJ(92)
           QMAX=SQRT(MAX(0.,(P(IP1,4)+P(IP1+1,4))**2-(P(IP1,1)+P(IP1+1,
      &    1))**2-(P(IP1,2)+P(IP1+1,2))**2-(P(IP1,3)+P(IP1+1,3))**2))
+          MINT(33) = 0
           CALL PYSHOW(IP1,IP1+1,QMAX)
           CALL PYPREP(IP1)
           MSTJ(92)=0
         ELSEIF(MSTJ(92).LT.0) THEN
           IP1=-MSTJ(92)
+          MINT(33) = 0
           CALL PYSHOW(IP1,-3,P(1,5))
           CALL PYPREP(IP1)
           MSTJ(92)=0
