diff --git a/evtgenlhc/9.1/configure b/evtgenlhc/9.1/configure
index fca5fe2..121d9f7 100755
--- a/evtgenlhc/9.1/configure
+++ b/evtgenlhc/9.1/configure
@@ -131,23 +131,25 @@ if [ $ARCH = Linux ]; then
   fi
 fi
 # Linux platform with gcc4: new Fortran90 compiler.
-if [ $ARCH = Linux-gcc4 ]; then
-  FC=gfortran
-  FFLAGS="${FFLAGS_OPT}"
-  CFLAGS="${CFLAGS_OPT}"
-  CXXFLAGS="${CXXFLAGS_OPT}"
-  FLIBS="-lgfortran -lgfortranbegin"
-  if [ ${COMPMODE} = OPT ]; then
+case $ARCH in 
+  *-gcc4)
+    FC=gfortran
     FFLAGS="${FFLAGS_OPT}"
     CFLAGS="${CFLAGS_OPT}"
     CXXFLAGS="${CXXFLAGS_OPT}"
-  fi
-  if [ ${COMPMODE} = DBG ]; then
-    FFLAGS="${FFLAGS_DBG}"
-    CFLAGS="${CFLAGS_DBG}"
-    CXXFLAGS="${CXXFLAGS_DBG}"
-  fi
-fi
+    FLIBS="-lgfortran -lgfortranbegin"
+    if [ ${COMPMODE} = OPT ]; then
+      FFLAGS="${FFLAGS_OPT}"
+      CFLAGS="${CFLAGS_OPT}"
+      CXXFLAGS="${CXXFLAGS_OPT}"
+    fi
+    if [ ${COMPMODE} = DBG ]; then
+      FFLAGS="${FFLAGS_DBG}"
+      CFLAGS="${CFLAGS_DBG}"
+      CXXFLAGS="${CXXFLAGS_DBG}"
+    fi
+  ;;
+esac
 
 #Platform & opt/dbg - independent flags and variables:
 
diff --git a/evtgenlhc/9.1/src/Makefile b/evtgenlhc/9.1/src/Makefile
index 8814c9c..be282df 100644
--- a/evtgenlhc/9.1/src/Makefile
+++ b/evtgenlhc/9.1/src/Makefile
@@ -23,7 +23,13 @@ lib_archive: $(LIB_ARCHIVE)
 $(LIB_SHARED) : $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) $(FLIBS)
+	case ${ARCH} in \
+	  Linux*) \
+	    $(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) $(FLIBS) ;; \
+	  Darwin*) \
+	    $(CXX) $(LDFLAGSSHARED) $(objects) -Wl,-commons -Wl,use_dylibs -L$(PHOTOS_ROOT)/lib -lphotos -L$(PYTHIA6_ROOT)/lib -lpythia6 -L$(CLHEP_ROOT)/lib -lCLHEP -L$(dir $(shell gfortran --print-file-name=libgfortran.a)) -o $@ -shared $(FLIBS) ;; \
+	esac
+
 
 $(LIB_ARCHIVE) : $(objects)
 	ar cru $(LIB_ARCHIVE) $(objects)
