diff -Naur orig.evtgenlhc/9.1/src/EvtIncoherentMixing.cpp evtgenlhc/9.1/src/EvtIncoherentMixing.cpp
--- orig.evtgenlhc/9.1/src/EvtIncoherentMixing.cpp	2007-10-11 15:17:32.000000000 +0200
+++ evtgenlhc/9.1/src/EvtIncoherentMixing.cpp	2008-11-09 23:25:26.000000000 +0100
@@ -8,6 +8,7 @@
 #include "EvtGenBase/EvtPDL.hh"
 #include "EvtGenBase/EvtId.hh"
 #include "EvtGenBase/EvtRandom.hh"
+#include <cstdlib>
 
 //-----------------------------------------------------------------------------
 // Implementation file for class : EvtIncoherentMixing
diff -Naur orig.evtgenlhc/9.1/src/EvtPartPropDb.cpp evtgenlhc/9.1/src/EvtPartPropDb.cpp
--- orig.evtgenlhc/9.1/src/EvtPartPropDb.cpp	2007-10-14 20:50:08.000000000 +0200
+++ evtgenlhc/9.1/src/EvtPartPropDb.cpp	2008-11-09 23:34:19.000000000 +0100
@@ -11,6 +11,7 @@
 #include <stdio.h>
 #include <fstream>
 #include <stdlib.h>
+#include <cstring>
 #include "EvtGenBase/EvtPartPropDb.hh"
 
 std::vector<std::string> EvtPartPropDb::_names;
