diff --git a/plugins/SISCone/siscone/siscone/siscone.cpp b/plugins/SISCone/siscone/siscone/siscone.cpp
index 914a966..16f71a6 100644
--- a/plugins/SISCone/siscone/siscone/siscone.cpp
+++ b/plugins/SISCone/siscone/siscone/siscone.cpp
@@ -66,7 +66,7 @@ Csiscone::~Csiscone(){
 }
 
 bool Csiscone::init_done=false;
-std::ostream* Csiscone::_banner_ostr = &cout;
+std::ostream* Csiscone::_banner_ostr = 0;
 
 /*
  * compute the jets from a given particle set doing multiple passes
@@ -96,21 +96,21 @@ int Csiscone::compute_jets(vector<Cmomentum> &_particles, double _radius, double
 
     // print the banner
     if (_banner_ostr != 0){
-      (*_banner_ostr) << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
-      (*_banner_ostr) << "#                    SISCone   version " << setw(28) << left << siscone_version() << "o" << endl;
-      (*_banner_ostr) << "#              http://projects.hepforge.org/siscone                o" << endl;
-      (*_banner_ostr) << "#                                                                  o" << endl;
-      (*_banner_ostr) << "# This is SISCone: the Seedless Infrared Safe Cone Jet Algorithm   o" << endl;
-      (*_banner_ostr) << "# SISCone was written by Gavin Salam and Gregory Soyez             o" << endl;
-      (*_banner_ostr) << "# It is released under the terms of the GNU General Public License o" << endl;
-      (*_banner_ostr) << "#                                                                  o" << endl;
-      (*_banner_ostr) << "# A description of the algorithm is available in the publication   o" << endl;
-      (*_banner_ostr) << "# JHEP 05 (2007) 086 [arXiv:0704.0292 (hep-ph)].                   o" << endl;
-      (*_banner_ostr) << "# Please cite it if you use SISCone.                               o" << endl;
-      (*_banner_ostr) << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
-      (*_banner_ostr) << endl;
-
-      _banner_ostr->flush();
+//      (*_banner_ostr) << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
+//      (*_banner_ostr) << "#                    SISCone   version " << setw(28) << left << siscone_version() << "o" << endl;
+//      (*_banner_ostr) << "#              http://projects.hepforge.org/siscone                o" << endl;
+//      (*_banner_ostr) << "#                                                                  o" << endl;
+//      (*_banner_ostr) << "# This is SISCone: the Seedless Infrared Safe Cone Jet Algorithm   o" << endl;
+//      (*_banner_ostr) << "# SISCone was written by Gavin Salam and Gregory Soyez             o" << endl;
+//      (*_banner_ostr) << "# It is released under the terms of the GNU General Public License o" << endl;
+//      (*_banner_ostr) << "#                                                                  o" << endl;
+//      (*_banner_ostr) << "# A description of the algorithm is available in the publication   o" << endl;
+//      (*_banner_ostr) << "# JHEP 05 (2007) 086 [arXiv:0704.0292 (hep-ph)].                   o" << endl;
+//      (*_banner_ostr) << "# Please cite it if you use SISCone.                               o" << endl;
+//      (*_banner_ostr) << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
+//      (*_banner_ostr) << endl;
+//
+//      _banner_ostr->flush();
     }
   }
 
