diff --git a/src/ClusterSequenceAreaBase.cc b/src/ClusterSequenceAreaBase.cc
index afc50e8..e388438 100644
--- a/src/ClusterSequenceAreaBase.cc
+++ b/src/ClusterSequenceAreaBase.cc
@@ -41,7 +41,7 @@ using namespace std;
 /// allow for warnings
 LimitedWarning ClusterSequenceAreaBase::_warnings;
 LimitedWarning ClusterSequenceAreaBase::_warnings_zero_area;
-LimitedWarning ClusterSequenceAreaBase::_warnings_empty_area;
+LimitedWarning ClusterSequenceAreaBase::_warnings_empty_area = 0;
 
 //----------------------------------------------------------------------
 /// return the total area, within the selector's range, that is free
