diff --git a/src/ClusterSequence.cc b/src/ClusterSequence.cc
index e146f1a..6130692 100644
--- a/src/ClusterSequence.cc
+++ b/src/ClusterSequence.cc
@@ -418,7 +418,6 @@ void ClusterSequence::_decant_options(const JetDefinition & jet_def_in,
 // transfer all relevant info into internal variables
 void ClusterSequence::_decant_options_partial() {
   // let the user know what's going on
-  _print_banner();
   
   _jet_algorithm = _jet_def.jet_algorithm();
   _Rparam = _jet_def.R();  _R2 = _Rparam*_Rparam; _invR2 = 1.0/_R2;
