diff --git a/src/RangeDefinition.cc b/src/RangeDefinition.cc
index 6fc6f31..4927745 100644
--- a/src/RangeDefinition.cc
+++ b/src/RangeDefinition.cc
@@ -61,7 +61,7 @@ void RangeDefinition::_numerical_total_area(double rapmax, int npoints) {
 // reasons and is not guaranteed to work in future releases of
 // FastJet.
 void RangeDefinition::_warn_deprecated() const{
-  _warnings_deprecated.warn("The use of RangeDefinition is deprecated since FastJet version 3.0 onwards. Please consider using Selector (defined in fastjet/Selector.hh) instead. There is no guarantee that support for RangeDefinition will be provided in future releases of FastJet.");
+//  _warnings_deprecated.warn("The use of RangeDefinition is deprecated since FastJet version 3.0 onwards. Please consider using Selector (defined in fastjet/Selector.hh) instead. There is no guarantee that support for RangeDefinition will be provided in future releases of FastJet.");
 }
 
 FASTJET_END_NAMESPACE
