--- orig.fastjet-2.3.4/plugins/SISCone/siscone/siscone/siscone.cpp	2008-04-03 15:28:41.000000000 +0200
+++ fastjet-2.3.4/plugins/SISCone/siscone/siscone/siscone.cpp	2008-11-09 14:34:31.000000000 +0100
@@ -85,19 +85,19 @@
     ranlux_init();
 
     // print the banner
-    cout << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
-    cout << "#                    SISCone   version " << setw(28) << left << siscone_version() << "o" << endl;
-    cout << "#              http://projects.hepforge.org/siscone                o" << endl;
-    cout << "#                                                                  o" << endl;
-    cout << "# This is SISCone: the Seedless Infrared Safe Cone Jet Algorithm   o" << endl;
-    cout << "# SISCone was written by Gavin Salam and Gregory Soyez             o" << endl;
-    cout << "# It is released under the terms of the GNU General Public License o" << endl;
-    cout << "#                                                                  o" << endl;
-    cout << "# A description of the algorithm is available in the publication   o" << endl;
-    cout << "# JHEP 05 (2007) 086 [arXiv:0704.0292 (hep-ph)].                   o" << endl;
-    cout << "# Please cite it if you use SISCone.                               o" << endl;
-    cout << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
-    cout << endl;
+    //cout << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
+    //cout << "#                    SISCone   version " << setw(28) << left << siscone_version() << "o" << endl;
+    //cout << "#              http://projects.hepforge.org/siscone                o" << endl;
+    //cout << "#                                                                  o" << endl;
+    //cout << "# This is SISCone: the Seedless Infrared Safe Cone Jet Algorithm   o" << endl;
+    //cout << "# SISCone was written by Gavin Salam and Gregory Soyez             o" << endl;
+    //cout << "# It is released under the terms of the GNU General Public License o" << endl;
+    //cout << "#                                                                  o" << endl;
+    //cout << "# A description of the algorithm is available in the publication   o" << endl;
+    //cout << "# JHEP 05 (2007) 086 [arXiv:0704.0292 (hep-ph)].                   o" << endl;
+    //cout << "# Please cite it if you use SISCone.                               o" << endl;
+    //cout << "#ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo" << endl;
+    //cout << endl;
     // do not do this again
     init_done=true;
   }
