--- src/ClusterSequenceAreaBase.cc_orig	2011-07-08 10:21:00.000000000 +0200
+++ src/ClusterSequenceAreaBase.cc	2011-07-08 10:46:48.000000000 +0200
@@ -43,7 +43,7 @@
 /// allow for warnings
 LimitedWarning ClusterSequenceAreaBase::_warnings;
 LimitedWarning ClusterSequenceAreaBase::_warnings_zero_area;
-LimitedWarning ClusterSequenceAreaBase::_warnings_empty_area;
+LimitedWarning ClusterSequenceAreaBase::_warnings_empty_area = 0;
 
 //----------------------------------------------------------------------
 /// return the total area, within range, that is free of jets.
