diff -Naur orig.fastjet-2.1.0/src/ClusterSequence.cc fastjet-2.1.0/src/ClusterSequence.cc
--- orig.fastjet-2.1.0/src/ClusterSequence.cc	2007-02-14 22:41:53.000000000 +0100
+++ fastjet-2.1.0/src/ClusterSequence.cc	2007-05-29 22:04:44.000000000 +0200
@@ -174,7 +174,6 @@
                                       const bool & writeout_combinations) {
 
   // let the user know what's going on
-  _print_banner();
 
   // make a local copy of the jet definition (for future use?)
   _jet_def = jet_def;
