diff -arupN fastjet-3.0.1.orig/include/fastjet/NNH.hh fastjet-3.0.1/include/fastjet/NNH.hh
--- fastjet-3.0.1.orig/include/fastjet/NNH.hh	2012-03-06 15:07:07.000000000 +0100
+++ fastjet-3.0.1/include/fastjet/NNH.hh	2012-03-06 15:23:20.000000000 +0100
@@ -199,7 +199,7 @@ template<class BJ, class I> void NNH<BJ,
   // initialise the basic jet info 
   for (int i = 0; i< n; i++) {
     //jetA->init(jets[i], i);
-    init_jet(jetA, jets[i], i);
+    this->init_jet(jetA, jets[i], i);
     where_is[i] = jetA;
     jetA++; // move on to next entry of briefjets
   }
@@ -276,7 +276,7 @@ template<class BJ, class I> void NNH<BJ,
 
   // initialise jetB based on the new jet
   //jetB->init(jet, index);
-  init_jet(jetB, jet, index);
+  this->init_jet(jetB, jet, index);
   // and record its position (making sure we have the space)
   if (index >= int(where_is.size())) where_is.resize(2*index);
   where_is[jetB->index()] = jetB;
diff -arupN fastjet-3.0.1.orig/plugins/D0RunIICone/ILConeAlgorithm.hpp fastjet-3.0.1/plugins/D0RunIICone/ILConeAlgorithm.hpp
--- fastjet-3.0.1.orig/plugins/D0RunIICone/ILConeAlgorithm.hpp	2012-03-06 15:07:07.000000000 +0100
+++ fastjet-3.0.1/plugins/D0RunIICone/ILConeAlgorithm.hpp	2012-03-06 15:09:11.000000000 +0100
@@ -302,7 +302,7 @@ private:
 	  //std::cout << " is_stable: item y=" << (*tk)->y() << " phi=" << (*tk)->phi() << " RD2=" << RD2((*tk)->y(),(*tk)->phi(),Yst,PHIst) << " " << Yst-radius << " " << Yst+radius << endl;
 	  if(RD2((*tk)->y(),(*tk)->phi(),Yst,PHIst) <= radius2) 
 	    {
-	      addItem(*tk);
+	      this->addItem(*tk);
 	    }
 	}       
 #else
@@ -315,7 +315,7 @@ private:
 	    //std::cout << "     item " << (*tk)->y() << " " << (*tk)->phi() << " " << RD2((*tk)->y(),(*tk)->phi(),Yst,PHIst) << " " << Yst-radius << " " << Yst+radius << endl;
 	    if(RD2(((*tk).second)->y(),((*tk).second)->phi(),Yst,PHIst) <= radius2) 
 	      {
-		addItem((*tk).second);
+		this->addItem((*tk).second);
 	      }
 	  }
 	
@@ -328,7 +328,7 @@ private:
 	    if(RD2((*tk)->y(),(*tk)->phi(),Yst,PHIst) <= radius2) 
 	       {
 		 //cout << "add item to *tk" << endl;
-		addItem(*tk);
+		this->addItem(*tk);
 	      }
 	  }
 #endif
