diff --git a/fftjet/ProximityClusteringTree.icc b/fftjet/ProximityClusteringTree.icc
index 4d6c29e..671a1c7 100644
--- a/fftjet/ProximityClusteringTree.icc
+++ b/fftjet/ProximityClusteringTree.icc
@@ -112,8 +112,15 @@ namespace fftjet {
         TreeNode topnode(TreeNodeId(0,0), dummyCluster);
 
         // Create the top level
+        #if defined(__GNUC__) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 4
+        # pragma GCC diagnostic push
+        # pragma GCC diagnostic ignored "-Wuninitialized"
+        #endif
         LevelInfo dummyInfo;
         TreeLevel *topLevel = new TreeLevel(DBL_MAX, dummyInfo, 1);
+        #if defined(__GNUC__) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 4
+        # pragma GCC diagnostic pop 
+        #endif
         topLevel->nodes.push_back(topnode);
         levels.push_back(topLevel);
     }
