diff -arupN frontier_client__2.8.5__src.orig/test-req.cc frontier_client__2.8.5__src/test-req.cc
--- frontier_client__2.8.5__src.orig/test-req.cc	2012-03-06 14:46:10.000000000 +0100
+++ frontier_client__2.8.5__src/test-req.cc	2012-03-06 14:46:36.000000000 +0100
@@ -20,6 +20,7 @@
 #include <stdexcept>
 #include <cstdlib>
 #include <cstring>
+#include <unistd.h>
 
 int do_main(int argc, char **argv);
 static std::string escape_list="\\\'";
# bla bla
