diff --git a/gcc/stor-layout.c b/gcc/stor-layout.c
index 3381fcf..e720d01 100644
--- a/gcc/stor-layout.c
+++ b/gcc/stor-layout.c
@@ -1618,7 +1618,9 @@ compute_record_mode (tree type)
 		   && integer_zerop (TYPE_SIZE (TREE_TYPE (field)))))
 	  || ! host_integerp (bit_position (field), 1)
 	  || DECL_SIZE (field) == 0
-	  || ! host_integerp (DECL_SIZE (field), 1))
+	  || ! host_integerp (DECL_SIZE (field), 1)
+	  || (TREE_CODE (TREE_TYPE (field)) == ARRAY_TYPE
+	      && tree_low_cst (DECL_SIZE (field), 1) == 0))
 	return;
 
       /* If this field is the whole struct, remember its mode so
# bla bla
