diff --git a/gcc/c-family/c.opt b/gcc/c-family/c.opt
index 4c4727f..6fc7278 100644
--- a/gcc/c-family/c.opt
+++ b/gcc/c-family/c.opt
@@ -201,7 +201,7 @@ E
 C ObjC C++ ObjC++ Undocumented Var(flag_preprocess_only)
 
 F
-C ObjC C++ ObjC++ Joined Separate MissingArgError(missing path after %qs)
+Driver C ObjC C++ ObjC++ Joined Separate MissingArgError(missing path after %qs)
 -F <dir>	Add <dir> to the end of the main framework include path
 
 H
diff --git a/gcc/config/darwin-driver.c b/gcc/config/darwin-driver.c
index 1eb9201..0d63953 100644
--- a/gcc/config/darwin-driver.c
+++ b/gcc/config/darwin-driver.c
@@ -161,6 +161,14 @@ darwin_driver_init (unsigned int *decoded_options_count,
 	continue;
       switch ((*decoded_options)[i].opt_index)
 	{
+#if DARWIN_X86
+        case OPT_arch:
+          if (!strcmp ((*decoded_options)[i].arg, "i386"))
+            generate_option (OPT_m32, NULL, 1, CL_DRIVER, &(*decoded_options)[i]);
+          else if (!strcmp ((*decoded_options)[i].arg, "x86_64"))
+            generate_option (OPT_m64, NULL, 1, CL_DRIVER, &(*decoded_options)[i]);
+        break;
+#endif
 	case OPT_filelist:
 	case OPT_framework:
 	  ++*decoded_options_count;
diff --git a/gcc/config/darwin.opt b/gcc/config/darwin.opt
index a31ee07..af2f005 100644
--- a/gcc/config/darwin.opt
+++ b/gcc/config/darwin.opt
@@ -31,6 +31,9 @@ Driver Alias(Zall_load)
 allowable_client
 Driver Separate Alias(Zallowable_client)
 
+arch
+Driver RejectNegative Separate 
+
 arch_errors_fatal
 Driver Alias(Zarch_errors_fatal)
 
# bla bla
