diff -Naur rh73_gcc32/share/gccxml-0.6/GCC/3.2/pthread.h rh73_gcc32_patch1/share/gccxml-0.6/GCC/3.2/pthread.h
--- rh73_gcc32/share/gccxml-0.6/GCC/3.2/pthread.h	Thu Jan  1 01:00:00 1970
+++ rh73_gcc32_patch1/share/gccxml-0.6/GCC/3.2/pthread.h	Thu Apr 22 18:34:13 2004
@@ -0,0 +1,18 @@
+#ifndef GCCXML_PTHREAD_H
+#define GCCXML_PTHREAD_H
+
+/* Some pthread.h files use __thread, which is a keyword for GCC 3.3.
+   Change the name for the duration of including pthread.h.  */
+#ifndef __thread
+# define __thread gccxml__thread
+# define gccxml__thread gccxml__thread
+#endif
+#include_next <pthread.h>
+#ifdef gccxml__thread
+# undef gccxml__thread
+# undef __thread
+#endif
+
+#endif
+
+
diff -Naur rh73_gcc32/share/gccxml-0.6/GCC/find_flags rh73_gcc32_patch1/share/gccxml-0.6/GCC/find_flags
--- rh73_gcc32/share/gccxml-0.6/GCC/find_flags	Mon Feb  9 21:01:53 2004
+++ rh73_gcc32_patch1/share/gccxml-0.6/GCC/find_flags	Thu Apr 22 19:00:40 2004
@@ -64,12 +64,21 @@
 if [ "$MAJOR_VERSION" -lt 3 ]; then
   INCLUDES="-iwrapper$SELFPATH/2.95 $INCLUDES"
   if [ "$MINOR_VERSION" = 96 ]; then
-  INCLUDES="-iwrapper$SELFPATH/2.96 $INCLUDES"
+    INCLUDES="-iwrapper$SELFPATH/2.96 $INCLUDES"
   fi
 elif [ "$MAJOR_VERSION" = 3 -a "$MINOR_VERSION" -ge 3 ]; then
+  INCLUDES="-iwrapper$SELFPATH/3.3 $INCLUDES"
   SPECIAL="-include $SELFPATH/3.3/gccxml_builtins.h"
+elif [ "$MAJOR_VERSION" = 3 -a "$MINOR_VERSION" = 2 ]; then
+  INCLUDES="-iwrapper$SELFPATH/3.2 $INCLUDES"
+elif [ "$MAJOR_VERSION" = 3 -a "$MINOR_VERSION" = 1 ]; then
+  INCLUDES="-iwrapper$SELFPATH/3.1 $INCLUDES"
+elif [ "$MAJOR_VERSION" = 3 -a "$MINOR_VERSION" = 0 ]; then
+  INCLUDES="-iwrapper$SELFPATH/3.0 $INCLUDES"
 fi
 
 # Format and print out the options.
-OPTIONS="$MACROS -D__declspec(x)= -D__attribute__(x)= $INCLUDES $SPECIAL"
+OPTIONS="$MACROS -D__declspec(x)=__attribute__((x)) $INCLUDES $SPECIAL"
 echo $OPTIONS
+
+
