diff --git a/gdb/python/python.c b/gdb/python/python.c
index c66efe4..e0b1bbc 100644
--- a/gdb/python/python.c
+++ b/gdb/python/python.c
@@ -1216,17 +1216,6 @@ message == an error message without a stack will be printed."),
 			&user_show_python_list);
 
 #ifdef HAVE_PYTHON
-#ifdef WITH_PYTHON_PATH
-  /* Work around problem where python gets confused about where it is,
-     and then can't find its libraries, etc.
-     NOTE: Python assumes the following layout:
-     /foo/bin/python
-     /foo/lib/pythonX.Y/...
-     This must be done before calling Py_Initialize.  */
-  Py_SetProgramName (concat (ldirname (python_libdir), SLASH_STRING, "bin",
-			     SLASH_STRING, "python", NULL));
-#endif
-
   Py_Initialize ();
   PyEval_InitThreads ();
 
