diff -arupN geant4.9.4.p03.orig/config/sys/Darwin-g++.gmk geant4.9.4.p03/config/sys/Darwin-g++.gmk
--- geant4.9.4.p03.orig/config/sys/Darwin-g++.gmk	2012-04-03 10:19:28.000000000 +0200
+++ geant4.9.4.p03/config/sys/Darwin-g++.gmk	2012-04-03 21:46:22.000000000 +0200
@@ -9,7 +9,7 @@
 #
 ifeq ($(G4SYSTEM),Darwin-g++)
   CXX       := g++
-  CXXFLAGS  := -Wall -ansi -pedantic -Wno-non-virtual-dtor -Wno-long-long
+  CXXFLAGS  := -Wall -ansi -pedantic -Wno-non-virtual-dtor -Wno-long-long $(CMS_CXXFLAGS)
 # CXXFLAGS  += -Wunused-parameter -Winline -Wunreachable-code
   CXXFLAGS  += -Wwrite-strings -Wpointer-arith -Woverloaded-virtual -pipe
   ifdef G4OPTIMISE
diff -arupN geant4.9.4.p03.orig/config/sys/Linux-g++.gmk geant4.9.4.p03/config/sys/Linux-g++.gmk
--- geant4.9.4.p03.orig/config/sys/Linux-g++.gmk	2012-04-03 10:19:28.000000000 +0200
+++ geant4.9.4.p03/config/sys/Linux-g++.gmk	2012-04-03 10:28:42.000000000 +0200
@@ -3,7 +3,7 @@
 #
 ifeq ($(G4SYSTEM),Linux-g++)
   CXX       := g++
-  CXXFLAGS  := -W -Wall -ansi -pedantic -Wno-non-virtual-dtor -Wno-long-long
+  CXXFLAGS  := -W -Wall -ansi -pedantic -Wno-non-virtual-dtor -Wno-long-long $(CMS_CXXFLAGS)
   CXXFLAGS  += -Wwrite-strings -Wpointer-arith -Woverloaded-virtual -pipe
 #
 # Uncomment the following options to activate Pentium4 chip specific
