diff -arupN geant4/source/processes/decay/include/G4Decay.hh geant40/source/processes/decay/include/G4Decay.hh
--- geant4/source/processes/decay/include/G4Decay.hh	2014-01-07 13:45:00.000000000 +0100
+++ geant40/source/processes/decay/include/G4Decay.hh	2014-01-07 14:27:41.000000000 +0100
@@ -202,7 +202,14 @@ inline  
 			     const G4Step&  aStep
 			    )
 {
-  return DecayIt(aTrack, aStep);
+  if((aTrack.GetTrackStatus() == fStopButAlive) ||
+     (aTrack.GetTrackStatus() == fStopAndKill)) {
+    fParticleChangeForDecay.Initialize(aTrack);
+    return &fParticleChangeForDecay;
+  } else {
+    return DecayIt(aTrack, aStep);
+
+  }
 }
 
 inline
