diff -Naur orig.geant4.8.2.p01/source/run/src/G4RunManagerKernel.cc geant4.8.2.p01/source/run/src/G4RunManagerKernel.cc
--- orig.geant4.8.2.p01/source/run/src/G4RunManagerKernel.cc	2007-02-23 10:24:19.000000000 +0100
+++ geant4.8.2.p01/source/run/src/G4RunManagerKernel.cc	2007-05-29 22:06:14.000000000 +0200
@@ -99,14 +99,6 @@
   versionString += vs;
   versionString += "   ";
   versionString += G4Date;
-  G4cout << G4endl
-    << "*************************************************************" << G4endl
-    << versionString << G4endl
-    << "                      Copyright : Geant4 Collaboration" << G4endl
-    << "                      Reference : NIM A 506 (2003), 250-303" << G4endl
-    << "                            WWW : http://cern.ch/geant4" << G4endl
-    << "*************************************************************" << G4endl
-    << G4endl;
 }
 
 G4RunManagerKernel::~G4RunManagerKernel()
# bla bla
