diff --git a/cmake/Modules/Geant4MakeRules_cxx.cmake b/cmake/Modules/Geant4MakeRules_cxx.cmake
index 3fc7030..2e9f5ab 100644
--- a/cmake/Modules/Geant4MakeRules_cxx.cmake
+++ b/cmake/Modules/Geant4MakeRules_cxx.cmake
@@ -158,7 +158,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   endif()
 
   # - Multithreading
-  set(GEANT4_MULTITHREADED_CXX_FLAGS "-ftls-model=initial-exec -pthread")
+  set(GEANT4_MULTITHREADED_CXX_FLAGS "-ftls-model=global-dynamic -pthread")
 endif()
 
 
diff --git a/config/sys/Darwin-clang.gmk b/config/sys/Darwin-clang.gmk
index ceeba4e..e62efc9 100644
--- a/config/sys/Darwin-clang.gmk
+++ b/config/sys/Darwin-clang.gmk
@@ -10,7 +10,7 @@ ifeq ($(G4SYSTEM),Darwin-clang)
   CXXFLAGS  += -Wwrite-strings -Wpointer-arith -Woverloaded-virtual -pipe
   CXXFLAGS  += -Wno-variadic-macros -Wshadow 
   ifdef G4MULTITHREADED
-    CXXFLAGS += -ftls-model=initial-exec -pthread
+    CXXFLAGS += -ftls-model=global-dynamic -pthread
   endif
   ifdef G4OPTIMISE
     CXXFLAGS  += -O2
diff --git a/config/sys/Linux-g++.gmk b/config/sys/Linux-g++.gmk
index 712e25e..9ce60ed 100644
--- a/config/sys/Linux-g++.gmk
+++ b/config/sys/Linux-g++.gmk
@@ -16,7 +16,7 @@ ifeq ($(G4SYSTEM),Linux-g++)
 #
 # CXXFLAGS  += -march=pentium4 -mfpmath=sse
   ifdef G4MULTITHREADED
-    CXXFLAGS += -ftls-model=initial-exec -pthread
+    CXXFLAGS += -ftls-model=global-dynamic -pthread
   endif
   ifdef G4OPTIMISE
     CXXFLAGS  += -O2
