diff --git a/gettext-runtime/gnulib-lib/progreloc.c b/gettext-runtime/gnulib-lib/progreloc.c
index bac7cd0..3596043 100644
--- a/gettext-runtime/gnulib-lib/progreloc.c
+++ b/gettext-runtime/gnulib-lib/progreloc.c
@@ -259,7 +259,6 @@ find_executable (const char *argv0)
     char *link = xreadlink (buf);
     if (link != NULL && link[0] != '[')
       return link;
-    }
     if (executable_fd < 0)
       executable_fd = open (buf, O_EXEC | O_CLOEXEC, 0);
   }
diff --git a/gettext-tools/gnulib-lib/progreloc.c b/gettext-tools/gnulib-lib/progreloc.c
index bac7cd0..3596043 100644
--- a/gettext-tools/gnulib-lib/progreloc.c
+++ b/gettext-tools/gnulib-lib/progreloc.c
@@ -259,7 +259,6 @@ find_executable (const char *argv0)
     char *link = xreadlink (buf);
     if (link != NULL && link[0] != '[')
       return link;
-    }
     if (executable_fd < 0)
       executable_fd = open (buf, O_EXEC | O_CLOEXEC, 0);
   }
