diff --git a/exec_cmd.c b/exec_cmd.c
index ce192a2..2414b2b 100644
--- a/exec_cmd.c
+++ b/exec_cmd.c
@@ -7,19 +7,21 @@
 static const char *argv_exec_path;
 
 #ifdef RUNTIME_PREFIX
-static const char *argv0_path;
+static const char *argv0_path = NULL;
 
 static const char *system_prefix(void)
 {
-	static const char *prefix;
+	static const char *prefix = NULL;
 
 	assert(argv0_path);
 	assert(is_absolute_path(argv0_path));
 
-	if (!prefix &&
-	    !(prefix = strip_path_suffix(argv0_path, GIT_EXEC_PATH)) &&
-	    !(prefix = strip_path_suffix(argv0_path, BINDIR)) &&
-	    !(prefix = strip_path_suffix(argv0_path, "git"))) {
+	if (!argv0_path ||
+	    !is_absolute_path(argv0_path) ||
+	    (!prefix &&
+	     !(prefix = strip_path_suffix(argv0_path, GIT_EXEC_PATH)) &&
+	     !(prefix = strip_path_suffix(argv0_path, BINDIR)) &&
+	     !(prefix = strip_path_suffix(argv0_path, "git")))) {
 		prefix = PREFIX;
 		trace_printf("RUNTIME_PREFIX requested, "
 				"but prefix computation failed.  "
