diff --git a/src/core/ext/filters/client_channel/lb_policy.h b/src/core/ext/filters/client_channel/lb_policy.h
index c9fc1429ac..0b575c7d0c 100644
--- a/src/core/ext/filters/client_channel/lb_policy.h
+++ b/src/core/ext/filters/client_channel/lb_policy.h
@@ -133,25 +133,28 @@ class LoadBalancingPolicy : public InternallyRefCounted<LoadBalancingPolicy> {
   /// Implemented by the client channel and used by the SubchannelPicker.
   class MetadataInterface {
    public:
-    class iterator
-        : public std::iterator<
-              std::input_iterator_tag,
-              std::pair<absl::string_view, absl::string_view>,  // value_type
-              std::ptrdiff_t,  // difference_type
-              std::pair<absl::string_view, absl::string_view>*,  // pointer
-              std::pair<absl::string_view, absl::string_view>&   // reference
-              > {
+    class iterator {
      public:
+      using iterator_category = std::input_iterator_tag;
+      using value_type = std::pair<absl::string_view, absl::string_view>;
+      using difference_type = std::ptrdiff_t;
+      using pointer = value_type*;
+      using reference = value_type&;
+
       iterator(const MetadataInterface* md, intptr_t handle)
           : md_(md), handle_(handle) {}
+
       iterator& operator++() {
         handle_ = md_->IteratorHandleNext(handle_);
         return *this;
       }
-      bool operator==(iterator other) const {
+
+      bool operator==(const iterator& other) const {
         return md_ == other.md_ && handle_ == other.handle_;
       }
-      bool operator!=(iterator other) const { return !(*this == other); }
+
+      bool operator!=(const iterator& other) const { return !(*this == other); }
+
       value_type operator*() const { return md_->IteratorHandleGet(handle_); }
 
      private:
