diff -Naur orig.gsl-1.10/cheb/gsl_chebyshev.h gsl-1.10/cheb/gsl_chebyshev.h
--- orig.gsl-1.10/cheb/gsl_chebyshev.h	2007-07-02 18:29:29.000000000 +0200
+++ gsl-1.10/cheb/gsl_chebyshev.h	2011-03-05 15:40:57.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include <gsl/gsl_math.h>
 #include <gsl/gsl_mode.h>
+#include <stddef.h>
 
 #undef __BEGIN_DECLS
 #undef __END_DECLS
diff -Naur orig.gsl-1.10/monte/gsl_monte.h gsl-1.10/monte/gsl_monte.h
--- orig.gsl-1.10/monte/gsl_monte.h	2007-07-02 18:29:29.000000000 +0200
+++ gsl-1.10/monte/gsl_monte.h	2011-03-05 16:04:05.000000000 +0100
@@ -17,6 +17,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
  */
 
+#include <stddef.h>
+
 /* Some things common to all the Monte-Carlo implementations */
 /* Author: MJB */
 
