diff --git a/src/H_TransportMatrices.cc b/src/H_TransportMatrices.cc
index ceb809d..fccd4fb 100644
--- a/src/H_TransportMatrices.cc
+++ b/src/H_TransportMatrices.cc
@@ -148,9 +148,9 @@ extern TMatrix rdipmat(const float l, const float k, const float eloss = 0., con
 	float * efmat = new float[MDIM*MDIM];
 	double simp = r*2*sin(l/(2*r))*sin(l/(2*r))/BE;
 	double psy = ke*l/2.;
-	float tefmat[MDIM*MDIM] = {1., tan(psy)*ke, 0., 0., 0., 0.,
+	float tefmat[MDIM*MDIM] = {1., (float)(tan(psy)*ke), 0., 0., 0., 0.,
 	                            0., 1., 0., 0., 0., 0.,
-	                            0., 0., 1., -tan(psy)*ke, 0., 0.,
+	                            0., 0., 1., (float)(-tan(psy)*ke), 0., 0.,
 	                            0., 0., 0., 1., 0., 0.,
 	                            0., 0., 0., 0., 1., 0.,
 	                            0., 0., 0., 0., 0., 1. };
@@ -159,7 +159,7 @@ extern TMatrix rdipmat(const float l, const float k, const float eloss = 0., con
 	             	           r*sin(l/r),cos(l/r),0.,0., 0., 0.,
    		           	           0.,0.,1.,0., 0., 0.,
    	            	           0.,0.,l,1., 0., 0., 
-   	                        simp, sin(l/r)/BE, 0., 0., 1., 0.,
+   	                        (float)simp, (float)(sin(l/r)/BE), 0., 0., 1., 0.,
    	                        0., 0., 0., 0., 0., 1. };
 	for(int i=0;i<MDIM*MDIM;i++) { 
 		mat[i] = tmat[i];
@@ -208,7 +208,7 @@ extern TMatrix sdipmat(const float l, const float k, const float eloss = 0., con
 		                      r*sin(l/r),cos(l/r),0.,0., 0., 0.,
 			                  0.,0.,1.,0., 0., 0.,
 			                  0.,0.,l,1., 0., 0.,
-			                  simp, sin(l/r)/BE, 0., 0., 1., 0.,
+			                  simp, (float)(sin(l/r)/BE), 0., 0., 1., 0.,
 			                  0., 0., 0., 0., 0., 1.
 				           };
 	if(!relative_energy) {
@@ -257,7 +257,7 @@ extern TMatrix hkickmat(const float l, const float k, const float eloss =0., con
                            0.,0.,1.,0.,0.,0.,
                            0.,0.,l ,1.,0.,0.,
                            0.,0.,0.,0.,1.,0.,
-                           l*tan(ke)/2.,ke, 0., 0., 0., 1. 
+                           (float)(l*tan(ke)/2.),ke, 0., 0., 0., 1. 
 	};
 
 	for(int i=0;i<MDIM*MDIM;i++) { mat[i] = tmat[i]; }
@@ -284,7 +284,7 @@ extern TMatrix vkickmat(const float l, const float k, const float eloss=0., cons
                            0.,0.,1.,0.,0.,0.,
                            0.,0.,l ,1.,0.,0.,
                            0.,0.,0.,0.,1.,0.,
-                           0.,0.,l*tan(ke)/2.,ke, 0., 1. 
+                           0.,0.,(float)(l*tan(ke)/2.),ke, 0., 1. 
 	};
 
 	for(int i=0;i<MDIM*MDIM;i++) { mat[i] = tmat[i]; }
