--- a/fio/PythiaWrapper.cc	2010-05-17 17:23:44.000000000 +0200
+++ b/fio/PythiaWrapper.cc	2011-10-24 08:15:38.000000000 +0200
@@ -15,14 +15,17 @@
 #else // Generic version, tested on Linux ecgs/gcc
 
 // declare the structs here to keep the shared library happy
-struct pin3 pyint3_;
-struct pin5 pyint5_;
-struct pin7 pyint7_;
-struct pin8 pyint8_;
-struct pin9 pyint9_;
-struct pssm pyssm_;
-struct prvnv pyrvnv_;
-struct prvpm pyrvpm_;
+// Do not declare them, since we are building archive libraries on linux as well.
+#if !defined(__APPLE__) && defined(__GNUC__) && (__GNUC__ <= 4) && (__GNUC_MINOR__ <= 3)
+  struct pin3 pyint3_;
+  struct pin5 pyint5_;
+  struct pin7 pyint7_;
+  struct pin8 pyint8_;
+  struct pin9 pyint9_;
+  struct pssm pyssm_;
+  struct prvnv pyrvnv_;
+  struct prvpm pyrvpm_;
+#endif
 
 #endif // Platform
 
# bla bla
