diff --git a/HepMC/GenEvent.h b/HepMC/GenEvent.h
index 719d24f..4293860 100644
--- a/HepMC/GenEvent.h
+++ b/HepMC/GenEvent.h
@@ -308,7 +308,7 @@ namespace HepMC {
 	/// HepMC::GenEvent::vertex_const_iterator
 	/// is used to iterate over all vertices in the event.
 	class vertex_const_iterator :
-	  public std::iterator<std::forward_iterator_tag,HepMC::GenVertex*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,HepMC::GenVertex*,std::ptrdiff_t>{
 	    // Iterates over all vertices in this event
 	public:
 	    /// constructor requiring vertex information
@@ -365,7 +365,7 @@ namespace HepMC {
 	/// HepMC::GenEvent::vertex_iterator
 	/// is used to iterate over all vertices in the event.
 	class vertex_iterator :
-	  public std::iterator<std::forward_iterator_tag,HepMC::GenVertex*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,HepMC::GenVertex*,std::ptrdiff_t>{
 	    // Iterates over all vertices in this event
 	public:
 	    /// constructor requiring vertex information
@@ -438,7 +438,7 @@ namespace HepMC {
 	/// HepMC::GenEvent::particle_const_iterator 
 	/// is used to iterate over all particles in the event.
 	class particle_const_iterator :
-	  public std::iterator<std::forward_iterator_tag,HepMC::GenParticle*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,HepMC::GenParticle*,std::ptrdiff_t>{
 	    // Iterates over all vertices in this event
 	public:
 	    /// iterate over particles
@@ -494,7 +494,7 @@ namespace HepMC {
 	/// HepMC::GenEvent::particle_iterator 
 	/// is used to iterate over all particles in the event.
  	class particle_iterator :
-	  public std::iterator<std::forward_iterator_tag,HepMC::GenParticle*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,HepMC::GenParticle*,std::ptrdiff_t>{
 	    // Iterates over all vertices in this event
 	public:
 	    /// iterate over particles
diff --git a/HepMC/GenVertex.h b/HepMC/GenVertex.h
index 7417332..340d5ae 100644
--- a/HepMC/GenVertex.h
+++ b/HepMC/GenVertex.h
@@ -176,7 +176,7 @@ namespace HepMC {
 	/// for the public iterators and is intended for internal use only.
 	/// The acceptable Iterator Ranges are: family, parents, children
 	class edge_iterator :
-	  public std::iterator<std::forward_iterator_tag,HepMC::GenParticle*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,HepMC::GenParticle*,std::ptrdiff_t>{
 	public:
 	    edge_iterator();
 	    /// used to set limits on the iteration
@@ -245,7 +245,7 @@ namespace HepMC {
 	/// vertex) and returning the vertices in POST ORDER traversal.
 	///
 	class vertex_iterator :
-	  public std::iterator<std::forward_iterator_tag,HepMC::GenVertex*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,HepMC::GenVertex*,std::ptrdiff_t>{
 	public:
 	    vertex_iterator();
 	    /// used to set limits on the iteration
@@ -321,7 +321,7 @@ namespace HepMC {
 	/// to exactly one vertex and so it is returned exactly once.
 	/// Is made friend so that it can access protected edge iterator
  	class particle_iterator :
-	  public std::iterator<std::forward_iterator_tag,GenParticle*,ptrdiff_t>{
+	  public std::iterator<std::forward_iterator_tag,GenParticle*,std::ptrdiff_t>{
 	public:
 	    particle_iterator();
 	    /// used to set limits on the iteration
# bla bla
