diff -Naur orig.HepPDT-2.03.00/src/HepPDT/Version.cc HepPDT-2.03.00/src/HepPDT/Version.cc
--- orig.HepPDT-2.03.00/src/HepPDT/Version.cc	2006-01-12 03:20:15.000000000 +0100
+++ HepPDT-2.03.00/src/HepPDT/Version.cc	2007-05-29 23:50:21.000000000 +0200
@@ -19,8 +19,6 @@
 
 void version( )
 {
-    std::cout << " --------------- HepPDT Version " << versionName()
-              << " --------------- " << std::endl;
 }
 
 void writeVersion( std::ostream & os )
# bla bla
