diff --git a/2.5.0/Decay/Baryon/BaryonFactorizedDecayer.cc b/2.5.0/Decay/Baryon/BaryonFactorizedDecayer.cc
index c95809a..a9b29af 100644
--- a/2.5.0/Decay/Baryon/BaryonFactorizedDecayer.cc
+++ b/2.5.0/Decay/Baryon/BaryonFactorizedDecayer.cc
@@ -737,7 +737,7 @@ void BaryonFactorizedDecayer::findModes(unsigned int imode,
       if(ix!=imode)
 	{
 	  // the particle mode
-	  if(particles[ix][0]->id()==id[0]&&particles[ix].size()==id.size())
+	  if(particles[ix].size()==id.size()&&particles[ix][0]->id()==id[0])
 	    {
 	      nfound=1;
 	      for(iy=0;iy<id.size();++iy){done[iy]=false;}
@@ -752,7 +752,7 @@ void BaryonFactorizedDecayer::findModes(unsigned int imode,
 	      if(nfound==id.size()){cc.push_back(false);loc.push_back(ix);}
 	    }
 	  // the charge conjugate mode
-	  if(particles[ix][0]->id()==idbar[0]&&particles[ix].size()==idbar.size())
+	  if(particles[ix].size()==idbar.size()&&particles[ix][0]->id()==idbar[0])
 	    {
 	      nfound=1;
 	      for(iy=0;iy<idbar.size();++iy){done[iy]=false;}
