--- orig.herwig/6.520/dummy/Makefile.am	2010-10-28 17:52:56.000000000 +0200
+++ herwig/6.520/dummy/Makefile.am	2010-10-28 17:54:15.000000000 +0200
@@ -3,20 +3,13 @@
 circes.f \
 circgg.f \
 decadd.f \
-dexay.f \
 eudini.f \
 fragmt.f \
 hvcbvi.f \
 hvhbvi.f \
 hwhvvj.f \
 ieupdg.f \
-inietc.f \
-inimas.f \
-iniphx.f \
-initdk.f \
 ipdgeu.f \
-phoini.f \
-photos.f \
 qqinit.f \
 timel.f \
 upevnt.f \
--- orig.herwig/6.520/dummy/Makefile.in	2010-10-28 17:52:56.000000000 +0200
+++ herwig/6.520/dummy/Makefile.in	2010-10-28 19:33:19.000000000 +0200
@@ -57,10 +57,9 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 libherwig_dummy_la_LIBADD =
-am__objects_1 = circee.lo circes.lo circgg.lo decadd.lo dexay.lo \
+am__objects_1 = circee.lo circes.lo circgg.lo decadd.lo \
 	eudini.lo fragmt.lo hvcbvi.lo hvhbvi.lo hwhvvj.lo ieupdg.lo \
-	inietc.lo inimas.lo iniphx.lo initdk.lo ipdgeu.lo phoini.lo \
-	photos.lo qqinit.lo timel.lo upevnt.lo upinit.lo
+	ipdgeu.lo qqinit.lo timel.lo upevnt.lo upinit.lo 
 am_libherwig_dummy_la_OBJECTS = $(am__objects_1)
 libherwig_dummy_la_OBJECTS = $(am_libherwig_dummy_la_OBJECTS)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)/config
@@ -176,20 +175,13 @@
 circes.f \
 circgg.f \
 decadd.f \
-dexay.f \
 eudini.f \
 fragmt.f \
 hvcbvi.f \
 hvhbvi.f \
 hwhvvj.f \
 ieupdg.f \
-inietc.f \
-inimas.f \
-iniphx.f \
-initdk.f \
 ipdgeu.f \
-phoini.f \
-photos.f \
 qqinit.f \
 timel.f \
 upevnt.f \
# bla bla
