diff --git a/m4/boost.m4 b/m4/boost.m4
index 96b4b54..c764665 100644
--- a/m4/boost.m4
+++ b/m4/boost.m4
@@ -376,8 +376,8 @@ AC_DEFUN([BOOST_FIND_LIB],
 AC_DEFUN([_BOOST_FIND_LIBS],
 [Boost_lib=no
   case "$3" in #(
-    (mt | mt-) boost_mt=-mt; boost_rtopt=;; #(
-    (mt* | mt-*) boost_mt=-mt; boost_rtopt=`expr "X$3" : 'Xmt-*\(.*\)'`;; #(
+    (mt | mt-) boost_mt=; boost_rtopt=;; #(
+    (mt* | mt-*) boost_mt=; boost_rtopt=`expr "X$3" : 'X-*\(.*\)'`;; #(
     (*) boost_mt=; boost_rtopt=$3;;
   esac
   if test $enable_static_boost = yes; then
@@ -392,7 +392,7 @@ AC_DEFUN([_BOOST_FIND_LIBS],
   esac
   # If the PREFERRED-RT-OPT are not empty, prepend a `-'.
   test -n "$boost_rtopt" && boost_rtopt="-$boost_rtopt"
-  $boost_guess_use_mt && boost_mt=-mt
+  $boost_guess_use_mt && boost_mt=
   # Look for the abs path the static archive.
   # $libext is computed by Libtool but let's make sure it's non empty.
   test -z "$libext" &&
@@ -428,7 +428,7 @@ dnl start the for loops).
 for boost_lib_ in $2; do
 for boost_tag_ in -$boost_cv_lib_tag ''; do
 for boost_ver_ in -$boost_cv_lib_version ''; do
-for boost_mt_ in $boost_mt -mt ''; do
+for boost_mt_ in $boost_mt ''; do
 for boost_rtopt_ in $boost_rtopt '' -d; do
   for boost_full_suffix in \
     $boost_last_suffix \
