diff -r 438632ac61e5 MatrixElement/Matchbox/External/MadGraph/mg2herwig.py.in
--- a/MatrixElement/Matchbox/External/MadGraph/mg2herwig.py.in	Wed Oct 04 15:43:06 2017 +0100
+++ b/MatrixElement/Matchbox/External/MadGraph/mg2herwig.py.in	Wed Oct 11 15:26:38 2017 +0200
@@ -354,11 +354,32 @@
 file=open("makefile","w")
 file.write("include MG5/Source/make_opts  ")
 if Virtlist!=[]:
-    file.write("\nLIBDIR = MG5/lib\nLINKLIBS =  -L$(LIBDIR) -lcts  -liregi -L$(LIBDIR)/golem95_lib -lgolem") 
-    file.write("\nLIBS = $(LIBDIR)libcts.$(libext) $(LIBDIR)libgolem.$(libext) $(LIBDIR)libiregi.$(libext)")
+
+    collierlib=options.madgraph
+    collierlib=collierlib.replace("bin","HEPTools/collier")
+    onelooplib=options.madgraph
+    onelooplib=onelooplib.replace("bin","HEPTools/oneloop")
+
+    file.write("\nLIBDIR = MG5/lib")
+    file.write("\nLINKLIBS =-L$(LIBDIR) -lcts -liregi ")
+    if (os.path.exists("MG5/lib/ninja_lib")):
+	file.write("\nLINKLIBS += -L$(LIBDIR)/ninja_lib -lninja   ")
+    if (os.path.exists(collierlib)):
+        file.write("\nLINKLIBS += -L%s -lcollier "%(collierlib))
+    if (os.path.exists(onelooplib)):
+        file.write("\nLINKLIBS += -L%s -lavh_olo "%(onelooplib))
+    if (os.path.exists("MG5/lib/golem95_lib")):
+        file.write("\nLINKLIBS += -L$(LIBDIR)/golem95_lib -lgolem") 
+     
+    
 file.write("\nPROCESS= InterfaceMadGraph.f "+make+"\n\nall:  \n\t @FC@  @FFLAGS@ -w -fbounds-check -ffixed-line-length-132 -fPIC -fno-f2c -shared -s -o  InterfaceMadGraph.so -IMG5/SubProcesses/" )
 if Virtlist!=[]:
   file.write(" -IMG5/lib/golem95_include ")
+  file.write(" -IMG5/lib/ninja_include ")
+
+  incldir=options.madgraph.replace("bin","HEPTools/include")
+  
+  file.write(" -I%s -I./ "%incldir)
   # Find all .mod files also in /usr/include if golem was build there.
   # There can be an error message in the MadGraph output to add the golem include path to the makefiles.
   # Usually MadGraph finds the path if its Golem was build in an separate dictionary.
