diff --git a/MatrixElement/FxFx/FxFx.h b/MatrixElement/FxFx/FxFx.h
index a2133bb..23eba40 100644
--- a/MatrixElement/FxFx/FxFx.h
+++ b/MatrixElement/FxFx/FxFx.h
@@ -161,7 +161,7 @@ public:
     ISTUP.resize(NUP);
     MOTHUP.resize(NUP);
     ICOLUP.resize(NUP);
-    PUP.resize(NUP, std::vector<double>(5));
+    PUP.resize(NUP);
     VTIMUP.resize(NUP);
     SPINUP.resize(NUP);
   }
@@ -231,7 +231,7 @@ public:
    * Lab frame momentum (Px, Py, Pz, E and M in GeV) for the particle
    * entries in this event.
    */
-  vector< vector<double> > PUP;
+  vector< array<double,5> > PUP;
 
   /**
    * Invariant lifetime (c*tau, distance from production to decay im
diff --git a/MatrixElement/FxFx/FxFxReader.cc b/MatrixElement/FxFx/FxFxReader.cc
index 3dafe2c..d2340cd 100644
--- a/MatrixElement/FxFx/FxFxReader.cc
+++ b/MatrixElement/FxFx/FxFxReader.cc
@@ -1219,7 +1219,7 @@ bool FxFxReader::uncacheEvent() {
   pos = mread(pos, hepeup.MOTHUP[0], hepeup.NUP);
   hepeup.ICOLUP.resize(hepeup.NUP);
   pos = mread(pos, hepeup.ICOLUP[0], hepeup.NUP);
-  hepeup.PUP.resize(hepeup.NUP, vector<double>(5));
+  hepeup.PUP.resize(hepeup.NUP);
   for ( int i = 0; i < hepeup.NUP; ++i ) 
     pos = mread(pos, hepeup.PUP[i][0], 5);
   hepeup.VTIMUP.resize(hepeup.NUP);
