--- a/MatrixElement/Matchbox/External/MadGraph/mg2herwig.py.in	2017-09-22 15:56:17.000000001 +0200
+++ b/MatrixElement/Matchbox/External/MadGraph/mg2herwig.py.in	2017-09-22 15:52:59.000000001 +0200
@@ -354,11 +354,24 @@
 file=open("makefile","w")
 file.write("include MG5/Source/make_opts  ")
 if Virtlist!=[]:
+    ninjalib=options.madgraph
+    ninjalib=ninjalib.replace("bin","HEPTools/ninja/lib")
+    collierlib=options.madgraph
+    collierlib=collierlib.replace("bin","HEPTools/collier")
+    onelooplib=options.madgraph
+    onelooplib=onelooplib.replace("bin","HEPTools/oneloop")
+
+
+    file.write("\nLINKLIBS =-LMG5/lib -lcts -liregi  -L%s -lninja  -L%s -lcollier -L%s -lavh_olo "%(ninjalib,collierlib,onelooplib))
     file.write("\nLIBDIR = MG5/lib\nLINKLIBS =  -L$(LIBDIR) -lcts  -liregi -L$(LIBDIR)/golem95_lib -lgolem") 
-    file.write("\nLIBS = $(LIBDIR)libcts.$(libext) $(LIBDIR)libgolem.$(libext) $(LIBDIR)libiregi.$(libext)")
+    file.write("\nLIBS = $(LIBDIR)libcts.$(libext) $(LIBDIR)libgolem.$(libext) $(LIBDIR)libiregi.$(libext)") 
+    
 file.write("\nPROCESS= InterfaceMadGraph.f "+make+"\n\nall:  \n\t @FC@  @FFLAGS@ -w -fbounds-check -ffixed-line-length-132 -fPIC -fno-f2c -shared -s -o  InterfaceMadGraph.so -IMG5/SubProcesses/" )
 if Virtlist!=[]:
   file.write(" -IMG5/lib/golem95_include ")
+  incldir=options.madgraph.replace("bin","HEPTools/include")
+  
+  file.write(" -I%s -I./ "%incldir)
   # Find all .mod files also in /usr/include if golem was build there.
   # There can be an error message in the MadGraph output to add the golem include path to the makefiles.
   # Usually MadGraph finds the path if its Golem was build in an separate dictionary.
