--- Herwig-7.0.1/src/Makefile.am	2016-01-11 11:57:48.000000000 +0100
+++ Makefile-src.am	2016-02-28 11:08:02.354686202 +0100
@@ -94,53 +94,35 @@
 check_BSM_Full=
 check_BSM=
 
-if WANT_BSM
 
-check_BSM += check-LHC-RPV check-LHC-RS
+check_BSM_Full_valgrind =
+check_BSM_valgrind = 
 
-check_BSM_Full += \
-check-LHC-RPV check-LHC-MSSM check-ILC-MSSM \
-check-LHC-NMSSM \
-check-LHC-MUED check-ILC-MUED \
-check-LHC-RS check-ILC-RS check-LHC-ADD \
-check-LHC-LH check-LHC-LHTP \
-check-LHC-TRP \
-check-LHC-TTBA check-TVT-TTBA \
-check-LHC-ZP \
-check-LHC-LQ \
-check-LHC-Sextet
+check-local: 
 
-endif
-check_BSM_Full_valgrind = $(subst check,check-valgrind,$(check_BSM_Full))
-check_BSM_valgrind = $(subst check,check-valgrind,$(check_BSM))
+check-valgrind-local: 
 
-check-local: check-LEP check-LHC $(check_BSM) check-DIS check-ILC check-GammaGamma check-LHC-Powheg
+check-Powheg: 
 
-check-valgrind-local: check-valgrind-LEP check-valgrind-LHC $(check_BSM_valgrind) check-valgrind-DIS check-valgrind-ILC check-valgrind-GammaGamma check-valgrind-LHC-Powheg
+check-valgrind-Powheg: 
 
-check-Powheg: check-LHC-Powheg check-TVT-Powheg
+check-BSM: 
 
-check-valgrind-Powheg: check-valgrind-LHC-Powheg check-valgrind-TVT-Powheg
+check-valgrind-BSM: 
 
-check-BSM: $(check_BSM_Full)
+check_Matchbox= 
 
-check-valgrind-BSM: $(check_BSM_Full_valgrind)
+check-Matchbox: 
 
-check_Matchbox= \
-check-LEP-Matchbox check-DIS-Matchbox check-LHC-Matchbox
-check_Matchbox_valgrind = $(subst check,check-valgrind,$(check_Matchbox))
+check-valgrind-Matchbox: 
 
-check-Matchbox: $(check_Matchbox)
+check-extra: 
 
-check-valgrind-Matchbox: $(check_Matchbox_valgrind)
+check-valgrind-extra: 
 
-check-extra: check-LHC-diffractive check-LHC-GammaGamma check-LHC-MB check-TVT
+check-all: 
 
-check-valgrind-extra: check-valgrind-LHC-diffractive check-valgrind-LHC-GammaGamma check-valgrind-LHC-MB check-valgrind-TVT
-
-check-all: check-local check-Powheg check-BSM check-Matchbox check-extra
-
-check-valgrind-all: check-valgrind-local check-valgrind-Powheg check-valgrind-BSM check-valgrind-Matchbox check-valgrind-extra
+check-valgrind-all: 
 
 link-helper-files:
 	@for i in $(HELPERFILES); do \
@@ -151,27 +133,16 @@
 
 VALGRIND=valgrind --leak-check=full --num-callers=25 --freelist-vol=100000000 --leak-resolution=med --trace-children=yes
 
-check-valgrind-%: $(srcdir)/%.in HerwigDefaults.rpo link-helper-files
-	$(VALGRIND) ./Herwig read -d1 -D $< &> valgrind-$(notdir $(subst .in,,$<))-read.log
-	$(VALGRIND) ./Herwig run $(notdir $(subst .in,.run,$<)) -N 500 -d1 -D &> valgrind-$(notdir $(subst .in,,$<))-run.log
-
-valgrind: valgrind-init valgrind-read valgrind-run
+check-valgrind-%: 
+valgrind: 
 
 valgrind-init:
-	$(VALGRIND) ./Herwig init -d1 -D -L$(top_builddir)/lib defaults/HerwigDefaults.in \
-&> /tmp/valgrind-init.log
 valgrind-read:
-	$(VALGRIND) ./Herwig read -d1 -D LHC.in &> /tmp/valgrind-read.log
 valgrind-run:
-	$(VALGRIND) ./Herwig run -d1 -D -N5 LHC.run &> /tmp/valgrind-run.log
 
-CHECKCOMMAND = ./Herwig run  $(notdir $(subst .in,.run,$<)) -N500 -d1 -D
+CHECKCOMMAND = 
 
-check-%: $(srcdir)/%.in HerwigDefaults.rpo link-helper-files
-	./Herwig read -i . $< -D
-	@echo $(CHECKCOMMAND)
-	@$(CHECKCOMMAND) && echo "# $@ OK #" \
-			 || (echo "###### $@ BAD ######"; false)
+check-%: 
 SETUPTHEPEG=$(THEPEGPATH)/bin/setupThePEG
 THEPEGREPO=$(THEPEGLIBPATH)/ThePEGDefaults.rpo
 
