diff --git a/src/buffer.cc b/src/buffer.cc
index f12dcfd..40c74d1 100644
--- a/src/buffer.cc
+++ b/src/buffer.cc
@@ -62,12 +62,11 @@ IgProfBuffer::allocateRaw(size_t size)
 		    MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
   if (data != MAP_FAILED)
     return data;
-  else
-  {
-    igprof_debug("failed to allocate memory for profile buffer: %s (%d)\n",
-                 strerror(errno), errno);
-    igprof_abort();
-  }
+  
+  igprof_debug("failed to allocate memory for profile buffer: %s (%d)\n",
+                strerror(errno), errno);
+  igprof_abort();
+  __builtin_unreachable();
 }
 
 void
diff --git a/src/profile.cc b/src/profile.cc
index 33f069c..35bca5a 100644
--- a/src/profile.cc
+++ b/src/profile.cc
@@ -29,7 +29,7 @@
 #endif
 
 // Global variables initialised once here.
-HIDDEN void             (*igprof_abort)(void) __attribute__((noreturn)) = &abort;
+HIDDEN void             (*igprof_abort)(void) throw () = &abort;
 HIDDEN char *           (*igprof_getenv)(const char *) = &getenv;
 HIDDEN int              (*igprof_unsetenv)(const char *) = &unsetenv;
 HIDDEN bool             s_igprof_activated = false;
diff --git a/src/profile.h b/src/profile.h
index 2a980d2..933f2f4 100644
--- a/src/profile.h
+++ b/src/profile.h
@@ -21,7 +21,7 @@ extern IgProfAtomic     s_igprof_enabled;
 extern pthread_key_t    s_igprof_bufkey;
 extern pthread_key_t    s_igprof_flagkey;
 extern int              s_igprof_stderrOpen;
-extern void             (*igprof_abort) (void) __attribute__((noreturn));
+extern void             (*igprof_abort) (void) throw ();
 extern char *           (*igprof_getenv) (const char *);
 extern int              (*igprof_unsetenv) (const char *);
 
# bla bla
