diff -upw -r igprof-5.9.3.orig/src/igprof-book-build igprof-5.9.3/src/igprof-book-build
--- igprof-5.9.3.orig/src/igprof-book-build	2012-03-25 05:25:32.000000000 +0200
+++ igprof-5.9.3/src/igprof-book-build	2012-03-25 05:26:55.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 #
 # A script which merges all the relevant resources of a igprof-book into one
 # single file.
diff -upw -r igprof-5.9.3.orig/src/igpython-analyse igprof-5.9.3/src/igpython-analyse
--- igprof-5.9.3.orig/src/igpython-analyse	2012-03-25 05:25:32.000000000 +0200
+++ igprof-5.9.3/src/igpython-analyse	2012-03-25 05:26:52.000000000 +0200
@@ -1,4 +1,5 @@
-#!/usr/bin/python
+#!/usr/bin/env python
+
 from pstats import Stats
 from optparse import OptionParser
 from pprint import pprint
