diff -arupN igprof-5.9.3.orig/src/analyse.h igprof-5.9.3/src/analyse.h
--- igprof-5.9.3.orig/src/analyse.h	2012-03-07 10:14:51.000000000 +0100
+++ igprof-5.9.3/src/analyse.h	2012-03-07 10:16:40.000000000 +0100
@@ -15,6 +15,7 @@
 #include <inttypes.h>
 #include <cstdio>
 #include <cassert>
+#include <unistd.h>
 
 /** This class is the payload for a node in the stacktrace
     and holds all the information about the counter that we looking at.
