diff --git a/4.2/configure b/4.2/configure
index fa2cbb5..1f82871 100755
--- a/4.2/configure
+++ b/4.2/configure
@@ -82,7 +82,7 @@ check_arguments $*
 #fi
 #Platform & opt/dbg - independent flags and variables:
 
-export FFLAGS="${FFLAGS} -Wno-globals"
+export FFLAGS="${FFLAGS}"
 
 FFLAGSSHARED=-fPIC
 CFLAGSSHARED=-fPIC
@@ -92,17 +92,17 @@ CFLAGSSHARED=-fPIC
 export LDFLAGS=" -pthread -fPIC"
 
 ARCH=`uname`
-if [ ${ARCH} = "Linux" ]; then
-	export F77=g77
-	export FLIBS=" -lfrtbegin -lg2c "
-elif [ ${ARCH} = "Linux-gcc4" ]; then   # Linux platform with gcc4: new Fortran90 compiler.
-	export F77=gfortran
-	export FLIBS=" -lgfortran -lgfortranbegin "
-else
-	echo "Warning: unknown ARCH = $ARCH. Setting F77=f77, FLIBS=\"\"" 
-	export F77=f77
-	export FLIBS=""
-fi
+case ${ARCH} in
+  Linux)
+    export F77=gfortran
+    export FLIBS=" -lgfortran -lgfortranbegin "
+  ;;
+  *)
+    echo "Warning: unknown ARCH = $ARCH. Setting F77=f77, FLIBS=\"\""
+    export F77=f77
+    export FLIBS=""
+  ;;
+esac
 
 echo "Assuming the fortran compiler to be ... "$F77
 
diff --git a/4.2/src/Makefile b/4.2/src/Makefile
index 5d52ef0..0e9d2d2 100644
--- a/4.2/src/Makefile
+++ b/4.2/src/Makefile
@@ -19,7 +19,13 @@ lib_archive: $(LIB_ARCHIVE)
 $(LIB_SHARED):  $(objects)
 	mkdir -p $(LIBDIR_SHARED)
 	@echo creating the shared library $@
-	$(F77) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@)
+	case `uname` in \
+	   Linux*) \
+		$(F77) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) ;; \
+	   Darwin*) \
+		$(F77) $(LDFLAGSSHARED) $(objects) -L$(HERWIG_ROOT)/lib -lherwig -o $@ -shared -dynamic -Wl,-commons -Wl,use_dylibs ;; \
+	esac
+
 
 $(LIB_ARCHIVE) : $(objects)
 	mkdir -p $(LIBDIR_ARCHIVE)
# bla bla
