--- orig.configure	2007-10-23 19:38:20.000000000 +0200
+++ configure	2007-10-23 19:40:05.000000000 +0200
@@ -82,7 +82,7 @@
 #fi
 #Platform & opt/dbg - independent flags and variables:
 
-export FFLAGS="${FFLAGS} -Wno-globals"
+export FFLAGS="${FFLAGS}"
 
 FFLAGSSHARED=-fPIC
 CFLAGSSHARED=-fPIC
@@ -91,7 +91,7 @@
 
 export LDFLAGS=" -pthread -fPIC"
 
-ARCH=`uname`
+ARCH="Linux-gcc4" # Force it to be this for gcc4 builds
 if [ ${ARCH} = "Linux" ]; then
 	export F77=g77
 	export FLIBS=" -lfrtbegin -lg2c "
