diff -Naur orig.KtJet-1.06/src/KtEvent.cc KtJet-1.06/src/KtEvent.cc
--- orig.KtJet-1.06/src/KtEvent.cc	2006-06-21 19:16:56.000000000 +0200
+++ KtJet-1.06/src/KtEvent.cc	2007-05-29 22:03:17.000000000 +0200
@@ -370,31 +370,12 @@
     type.resize(16,' ');
     angle.resize(16,' ');
     recom.resize(16,' ');
-    std::cout << "******************************************\n";
-    std::cout << "* KtEvent constructor called: " << mode << " *\n";
-    std::cout << "* Collision type:       " << type  << " *\n";
-    std::cout << "* Kt scheme:            " << angle << " *\n";
-    std::cout << "* Recombination scheme: " << recom << " *\n";
-#ifdef KTDOUBLEPRECISION
-    std::cout << "* Compiled to use double precision.      *\n";
-#else
-    std::cout << "* Compiled to use single precision.      *\n";
-#endif
-    std::cout << "******************************************\n";
   }
 
   void KtEvent::listAuthors() const {
     static bool first = true;
     if (first) {
       first = false;
-      std::cout << "***********************************************\n";
-      std::cout << "* Package KtJet written by:                   *\n";
-      std::cout << "*   Jon Butterworth                           *\n";
-      std::cout << "*   Jon Couchman                              *\n";
-      std::cout << "*   Brian Cox                                 *\n";
-      std::cout << "*   Ben Waugh                                 *\n";
-      std::cout << "* See documentation at <http://www.ktjet.org> *\n";
-      std::cout << "***********************************************\n";
     }
   }
 
# bla bla
