diff -u a/configure b/configure
--- a/configure
+++ b/configure
@@ -2677,10 +2677,6 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "$CLHEPPATH"; then
   eval "$as_ac_File=yes"
 else
diff -u a/configure b/configure
--- a/configure
+++ b/configure
@@ -2699,10 +2699,6 @@
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
 if test -r "$CLHEPPATH/include/CLHEP/config/CLHEP.h"; then
   eval "$as_ac_File=yes"
 else
