diff --git a/bin/genhtml b/bin/genhtml
index e82fef5..73f157e 100755
--- a/bin/genhtml
+++ b/bin/genhtml
@@ -3165,6 +3165,28 @@ sub write_source($$$$$)
 		%count_data = %{$_[2]};
 	}
 
+       my $source_dirname = dirname($source_filename);
+       my $source_basename = basename($source_filename);
+       if ( -e $source_filename)
+       {
+       }
+       elsif ( -e join("/", $source_dirname, "src", $source_basename))
+       {
+               $source_filename = join("/", $source_dirname, "src", $source_basename);
+       }
+       elsif ( -e join("/", $source_dirname, "interface", $source_basename))
+       {
+               $source_filename = join("/", $source_dirname, "interface", $source_basename);
+       }
+       elsif ( -e join("/", $source_dirname, "plugins", $source_basename))
+       {
+               $source_filename = join("/", $source_dirname, "plugins", $source_basename);
+       }
+       elsif ( -e join("/", $source_dirname, "bin", $source_basename))
+       {
+               $source_filename = join("/", $source_dirname, "bin", $source_basename);
+       }
+
 	open(SOURCE_HANDLE, "<".$source_filename)
 		or die("ERROR: cannot open $source_filename for reading!\n");
 	
# bla bla
