--- a/genhtml	2022-07-05 08:55:12.599513155 +0200
+++ b/genhtml	2022-07-05 08:55:01.817596714 +0200
@@ -5279,6 +5279,28 @@
 		%count_data = %{$_[2]};
 	}
 
+  my $source_dirname = dirname($source_filename);
+  my $source_basename = basename($source_filename);
+  if ( -e $source_filename)
+  {
+  }
+	elsif ( -e join("/", $source_dirname, "src", $source_basename))
+  {
+	   $source_filename = join("/", $source_dirname, "src", $source_basename);
+  }
+  elsif ( -e join("/", $source_dirname, "interface", $source_basename))
+  {
+	   $source_filename = join("/", $source_dirname, "interface", $source_basename);
+  }
+  elsif ( -e join("/", $source_dirname, "plugins", $source_basename))
+  {
+	   $source_filename = join("/", $source_dirname, "plugins", $source_basename);
+  }
+  elsif ( -e join("/", $source_dirname, "bin", $source_basename))
+  {
+	   $source_filename = join("/", $source_dirname, "bin", $source_basename);
+  }
+
 	if (!open(SOURCE_HANDLE, "<", $source_filename)) {
 		my @lines;
 		my $last_line = 0;
