--- genhtml.orig	2020-08-12 16:17:04.000000000 +0200
+++ genhtml	2021-09-22 13:39:34.212361700 +0200
@@ -5138,6 +5138,28 @@ sub write_source($$$$$$$)
 		%count_data = %{$_[2]};
 	}
 
+    my $source_dirname = dirname($source_filename);
+    my $source_basename = basename($source_filename);
+    if ( -e $source_filename)
+    {
+    }
+    elsif ( -e join("/", $source_dirname, "src", $source_basename))
+    {
+            $source_filename = join("/", $source_dirname, "src", $source_basename);
+    }
+    elsif ( -e join("/", $source_dirname, "interface", $source_basename))
+    {
+            $source_filename = join("/", $source_dirname, "interface", $source_basename);
+    }
+    elsif ( -e join("/", $source_dirname, "plugins", $source_basename))
+    {
+            $source_filename = join("/", $source_dirname, "plugins", $source_basename);
+    }
+    elsif ( -e join("/", $source_dirname, "bin", $source_basename))
+    {
+            $source_filename = join("/", $source_dirname, "bin", $source_basename);
+    }
+
 	if (!open(SOURCE_HANDLE, "<", $source_filename)) {
 		my @lines;
 		my $last_line = 0;
