--- orig.lhapdf/5.6.0/Makefile.in	2008-10-22 11:47:13.000000000 +0200
+++ lhapdf/5.6.0/Makefile.in	2009-01-22 16:08:37.000000000 +0100
@@ -249,7 +249,7 @@
 	esac;
 
 $(top_builddir)/config.status: $(top_srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
+	#$(SHELL) ./config.status --recheck
 
 $(top_srcdir)/configure:  $(am__configure_deps)
 	cd $(srcdir) && $(AUTOCONF)
