--- orig.lhapdf/5.8.5/share/lhapdf/PDFsets/cteq65c.LHgrid	2010-08-12 23:31:06.000000000 +0200
+++ lhapdf/5.8.5/share/lhapdf/PDFsets/cteq65c.LHgrid	2011-02-22 17:21:52.000000000 +0100
@@ -2,7 +2,7 @@
 'Description:'
 'CTEQ6.5c'
 'Reference:'
-' J. Pumplin, H.L. Lai and W.K. Tung,
+' J. Pumplin, H.L. Lai and W.K. Tung, '
 ' "The charm parton content of the nucleon" '
 ' hep-ph/0701220, to be published in Phys. Rev. D. '
 ' This set has 7 member PDFs.'
