diff --git a/5.8.5/octave/Makefile.am b/5.8.5/octave/Makefile.am
index 7ef2eac..079c5d5 100644
--- a/5.8.5/octave/Makefile.am
+++ b/5.8.5/octave/Makefile.am
@@ -6,4 +6,4 @@ lhapdf.oct: wrapoctave.cc
 	$(MKOCTFILE) -I$(top_srcdir)/include -v -g -o lhapdf.oct wrapoctave.cc -L$(top_builddir)/lib -L$(top_builddir)/lib/.libs -lLHAPDF
 
 TESTS_ENVIRONMENT = LHAPATH=$(top_srcdir)/PDFsets $(OCTAVE)
-TESTS = $(srcdir)/lhapdf-octave-example1.m
+TESTS = lhapdf-octave-example1.m
