diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index dc35a33..cd2ced7 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -1531,6 +1531,8 @@ Linux::Linux(const HostInfo &Host, const llvm::Triple &Triple)
   Paths.push_back(Base + "/../../..");
   if (Arch == getArch() && IsUbuntu(Distro))
     Paths.push_back("/usr/lib/" + GccTriple);
+  // Hack to get llvm find the correct gcc runtime. Only works on 64bit slc.
+  Paths.push_back(getenv("COMPILER_RUNTIME_OBJECTS"));
 }
 
 bool Linux::HasNativeLLVMSupport() const {
