diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 7d70cd5..b6bbd7c 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -963,6 +963,8 @@ bool Generic_GCC::GCCVersion::operator<(const GCCVersion &RHS) const {
 }
 
 static StringRef getGCCToolchainDir(const ArgList &Args) {
+  if (const char *S = ::getenv("COMPILER_RUNTIME_OBJECTS"))
+    return StringRef(S);
   const Arg *A = Args.getLastArg(options::OPT_gcc_toolchain);
   if (A)
     return A->getValue();
