diff --git a/tools/scan-view/Reporter.py b/tools/scan-view/Reporter.py
index 9560fc1..d6aab9e 100644
--- a/tools/scan-view/Reporter.py
+++ b/tools/scan-view/Reporter.py
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 """Methods for reporting bugs."""
 
 import subprocess, sys, os
diff --git a/tools/scan-view/startfile.py b/tools/scan-view/startfile.py
index e8fbe2d..94d5f14 100644
--- a/tools/scan-view/startfile.py
+++ b/tools/scan-view/startfile.py
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 """Utility for opening a file using the default application in a cross-platform
 manner. Modified from http://code.activestate.com/recipes/511443/.
 """
