diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 81657d8..e6d8a7e 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -2100,6 +2100,8 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
   }
   addPathIfExists(SysRoot + "/lib", Paths);
   addPathIfExists(SysRoot + "/usr/lib", Paths);
+  // Hack to get llvm find the correct gcc runtime. Only works on 64bit slc.
+  Paths.push_back(getenv("COMPILER_RUNTIME_OBJECTS"));
 }
 
 bool Linux::HasNativeLLVMSupport() const {
