diff --git a/include/lwtnn/generic/Graph.tcc b/include/lwtnn/generic/Graph.tcc
index f0cc614..3af95ce 100644
--- a/include/lwtnn/generic/Graph.tcc
+++ b/include/lwtnn/generic/Graph.tcc
@@ -7,6 +7,7 @@
 
 #include <set>
 #include <memory>
+#include <cassert>
 
 namespace lwt {
 namespace generic {
diff --git a/include/lwtnn/generic/Stack.tcc b/include/lwtnn/generic/Stack.tcc
index d54680d..85acd45 100644
--- a/include/lwtnn/generic/Stack.tcc
+++ b/include/lwtnn/generic/Stack.tcc
@@ -4,6 +4,7 @@
 #include "lwtnn/generic/Stack.hh"
 
 #include <set>
+#include <cassert>
 
 
 namespace lwt {
diff --git a/src/lwtnn-test-arbitrary-net.cxx b/src/lwtnn-test-arbitrary-net.cxx
index 7678ac1..7de3018 100644
--- a/src/lwtnn-test-arbitrary-net.cxx
+++ b/src/lwtnn-test-arbitrary-net.cxx
@@ -10,6 +10,7 @@
 #include <sstream>
 #include <string>
 #include <fstream>
+#include <cassert>
 
 namespace {
   int run_on_files(const lwt::JSONConfig& config,
diff --git a/cmake/FindEigen3.cmake b/cmake/FindEigen3.cmake
--- a/cmake/FindEigen3.cmake
+++ b/cmake/FindEigen3.cmake
@@ -30,7 +30,11 @@ if(NOT Eigen3_FIND_VERSION)
 endif(NOT Eigen3_FIND_VERSION)
 
 macro(_eigen3_check_version)
-  file(READ "${EIGEN3_INCLUDE_DIR}/Eigen/src/Core/util/Macros.h" _eigen3_version_header)
+  if(EXISTS "${EIGEN3_INCLUDE_DIR}/Eigen/Version")
+    file(READ "${EIGEN3_INCLUDE_DIR}/Eigen/Version" _eigen3_version_header)
+  else()
+    file(READ "${EIGEN3_INCLUDE_DIR}/Eigen/src/Core/util/Macros.h" _eigen3_version_header)
+  endif()
 
   string(REGEX MATCH "define[ \t]+EIGEN_WORLD_VERSION[ \t]+([0-9]+)" _eigen3_world_version_match "${_eigen3_version_header}")
   set(EIGEN3_WORLD_VERSION "${CMAKE_MATCH_1}")
