--- orig.mcdb-api-1.0.2/src/parser/xerces/parser_xs.cpp	2007-10-23 17:26:04.000000000 +0200
+++ mcdb-api-1.0.2/src/parser/xerces/parser_xs.cpp	2010-04-09 22:20:30.000000000 +0200
@@ -41,7 +41,7 @@
     XMLCh* fUnicodeForm;
 };
 
-#define X(str) XStr::XStr(str).unicodeForm()
+#define X(str) XStr(str).unicodeForm()
 
 int init_parser_xs(DOMBuilder*& parser, bool reportErrors=false, bool processNs=false);
 void finish_parser_xs(DOMBuilder*& parser);
