diff --git a/1.25.0a/HEPEvent/Makefile b/1.25.0a/HEPEvent/Makefile
index 668da29..1984b92 100755
--- a/1.25.0a/HEPEvent/Makefile
+++ b/1.25.0a/HEPEvent/Makefile
@@ -64,14 +64,11 @@ all:     $(LIB) $(LIB_A)
 %.o: %.cxx %.h
 	$(CXX) $(CXXFLAGS) -c $<
 
-#$(DICT_SRC) $(DICT_HDR): $(LIB_HDR) 
-#	$(ROOTSYS)/bin/rootcint -f $(DICT_SRC) -c -p -DDebug ${INC} $(LIB_HDR)
-
 $(DICT_SRC) $(DICT_HDR): $(LIB_HDR)
-	$(ROOTBIN)/rootcint -f $(DICT_SRC) -c -p -D_USE_ROOT_ $(LIB_HDR)
+	$(ROOTBIN)/rootcling -f $(DICT_SRC) -D_USE_ROOT_ $(LIB_HDR)
 
 $(LIB): $(LIB_OBJ)
-	$(LD) $(LDFLAGS) $(SOFLAGS) -o $(LIB) $(LIB_OBJ) -L$(ROOTLIBPATH) -lCore -lCint
+	$(LD) $(LDFLAGS) $(SOFLAGS) -o $(LIB) $(LIB_OBJ) -L$(ROOTLIBPATH) -lCore -lCling
 
 $(LIB_A): $(LIB_OBJ)
 	ar cr $(LIB_A) $(LIB_OBJ)
diff --git a/1.25.0a/HepMCEvent/Makefile b/1.25.0a/HepMCEvent/Makefile
index e7be17c..2320568 100644
--- a/1.25.0a/HepMCEvent/Makefile
+++ b/1.25.0a/HepMCEvent/Makefile
@@ -30,7 +30,7 @@ endif
 HEPEVENT_INC=-I../include/
 HEPMC_INC=-I$(HEPMCLOCATION)/include
 
-ROOT_CINT=$(ROOTBIN)/rootcint
+ROOT_CLING=$(ROOTBIN)/rootcling
 
 ##############   What has to be built ###############################
 
@@ -44,11 +44,11 @@ all:  HEPMC_HEADERS $(LIB) $(LIB_A)
 	$(CXX) $(CXXFLAGS) $(HEPMC_INC) $(CLHEP_INC) $(HEPEVENT_INC) -c $<
 
 $(DICT_SRC) $(DICT_HDR): $(LIB_HDR) 
-	$(ROOT_CINT) -f $(DICT_SRC) -c -p -D_USE_ROOT_ ${INC} \
+	$(ROOT_CLING) -f $(DICT_SRC) -D_USE_ROOT_ ${INC} \
 	$(HEPMC_INC) $(CLHEP_INC) $(HEPEVENT_INC) $(LIB_HDR)
 
 $(LIB): $(LIB_OBJ)
-	$(LD) $(LDFLAGS) $(SOFLAGS) -o $(LIB) $(LIB_OBJ) -L$(ROOTLIBPATH) -lCore -lCint -L$(CURDIR)/../lib -lHEPEvent
+	$(LD) $(LDFLAGS) $(SOFLAGS) -o $(LIB) $(LIB_OBJ) -L$(ROOTLIBPATH) -lCore -lCling -L$(CURDIR)/../lib -lHEPEvent
 
 $(LIB_A): $(LIB_OBJ)
 	ar cr $(LIB_A) $(LIB_OBJ)
diff --git a/1.25.0a/platform/LCGCONFIG/HEPEvent/Makefile.am b/1.25.0a/platform/LCGCONFIG/HEPEvent/Makefile.am
index 74dd6e3..a0dad2b 100644
--- a/1.25.0a/platform/LCGCONFIG/HEPEvent/Makefile.am
+++ b/1.25.0a/platform/LCGCONFIG/HEPEvent/Makefile.am
@@ -45,7 +45,7 @@ $(lib_LTLIBRARIES): HEPEventDict.cxx
 HEPEventDict.cxx HEPEventDict.h: $(HDRS)
 	LD_LIBRARY_PATH=$(ROOT_LIBDIR):$(LD_LIBRARY_PATH)   \
 	DYLD_LIBRARY_PATH=$(ROOT_LIBDIR):$(DYLD_LIBRARY_PATH) \
-	$(ROOT_CINT) -f HEPEventDict.cxx -c -p $(AM_CPPFLAGS) $(HDRS)
+	$(ROOT_CLING) -f HEPEventDict.cxx $(AM_CPPFLAGS) $(HDRS)
 
 clean:
 	-rm -f HEPEventDict.cxx HEPEventDict.h
diff --git a/1.25.0a/platform/LCGCONFIG/HepMCEvent/Makefile.am b/1.25.0a/platform/LCGCONFIG/HepMCEvent/Makefile.am
index d7ba6c8..df4e2d2 100644
--- a/1.25.0a/platform/LCGCONFIG/HepMCEvent/Makefile.am
+++ b/1.25.0a/platform/LCGCONFIG/HepMCEvent/Makefile.am
@@ -26,7 +26,7 @@ $(lib_LTLIBRARIES): HepMCEventDict.cxx
 HepMCEventDict.cxx HepMCEventDict.h: $(HDRS)
 	LD_LIBRARY_PATH=$(ROOT_LIBDIR):$(LD_LIBRARY_PATH)   \
 	DYLD_LIBRARY_PATH=$(ROOT_LIBDIR):$(DYLD_LIBRARY_PATH) \
-	$(ROOT_CINT) -f HepMCEventDict.cxx -c -p $(AM_CPPFLAGS) $(INCLUDES) $(HDRS)
+	$(ROOT_CLING) -f HepMCEventDict.cxx $(AM_CPPFLAGS) $(INCLUDES) $(HDRS)
 
 clean:
 	-rm -f HepMCEventDict.cxx HepMCEventDict.h
diff --git a/1.25.0a/platform/LCGCONFIG/configure.ac b/1.25.0a/platform/LCGCONFIG/configure.ac
index f53e7f3..9765c00 100644
--- a/1.25.0a/platform/LCGCONFIG/configure.ac
+++ b/1.25.0a/platform/LCGCONFIG/configure.ac
@@ -100,14 +100,14 @@ if test "x$with_root" != "xno" ; then
 	echo ${_root_lib_}
 	AC_SUBST(ROOT_LIBDIR,[$_root_lib_])
 
-	echo "Checking for rootcint ... "
-	if test -x ${_root_bin_}/rootcint ; then
+	echo "Checking for rootcling ... "
+	if test -x ${_root_bin_}/rootcling ; then
 		echo found.
 	else
 		echo " NOT found in ${_root_bin_}. Check your ROOT installation">&2
 		exit 1
 	fi
-	AC_SUBST(ROOT_CINT,[${_root_bin_}/rootcint])
+	AC_SUBST(ROOT_CLING,[${_root_bin_}/rootcling])
 
 	echo "Checking for ROOT lib's ..."
 
@@ -173,7 +173,7 @@ rm -f examples-*/paths.sh
 	 echo "ROOTSYS=\"$ROOTSYS\" ;"
 	 echo "ROOT_BIN=\"$ROOT_BIN\" ;"
 	 echo "ROOT_CFLAGS=\"$ROOT_CFLAGS\" ;"
-	 echo "ROOT_CINT=\"$ROOT_CINT\" ;"
+	 echo "ROOT_CLING=\"$ROOT_CLING\" ;"
 	 echo "ROOT_LIBDIR=\"$ROOT_LIBDIR\" ;"
 	 echo "ROOT_LIBS=\"$ROOT_LIBS\" ;"
  fi
diff --git a/1.25.0a/platform/LCGCONFIG/src/Makefile.am b/1.25.0a/platform/LCGCONFIG/src/Makefile.am
index 1ef4966..929ed76 100644
--- a/1.25.0a/platform/LCGCONFIG/src/Makefile.am
+++ b/1.25.0a/platform/LCGCONFIG/src/Makefile.am
@@ -49,7 +49,7 @@ $(lib_LTLIBRARIES): MCTesterDict.cxx
 MCTesterDict.cxx MCTesterDict.h: $(HDRS)
 	LD_LIBRARY_PATH=$(ROOT_LIBDIR):$(LD_LIBRARY_PATH)   \
 	DYLD_LIBRARY_PATH=$(ROOT_LIBDIR):$(DYLD_LIBRARY_PATH) \
-	$(ROOT_CINT) -f MCTesterDict.cxx -c -I$(top_srcdir)/HEPEvent $(HDRS)
+	$(ROOT_CLING) -f MCTesterDict.cxx -I$(top_srcdir)/HEPEvent $(HDRS)
 
 clean:
 	-rm -f MCTesterDict.cxx MCTesterDict.h
diff --git a/1.25.0a/src/Makefile b/1.25.0a/src/Makefile
index b1b9556..27e67b3 100755
--- a/1.25.0a/src/Makefile
+++ b/1.25.0a/src/Makefile
@@ -48,11 +48,11 @@ all: $(LIB) $(LIB_A)
 
 
 $(DICT_SRC) $(DICT_HDR): $(LIB_HDR) 
-	$(ROOTBIN)/rootcint  -f $(DICT_SRC) -c -I../HEPEvent $(LIB_HDR) 
+	$(ROOTBIN)/rootcling -f $(DICT_SRC) -I../HEPEvent $(LIB_HDR) 
 
 
 $(LIB): $(LIB_OBJ)
-	$(LD) $(LDFLAGS) $(SOFLAGS) -o $(LIB) $(LIB_OBJ) -L$(ROOTLIBPATH) -lCore -lCint -lHist -lGpad -lGraf -lRIO -L$(CURDIR)/../lib -lHEPEvent
+	$(LD) $(LDFLAGS) $(SOFLAGS) -o $(LIB) $(LIB_OBJ) -L$(ROOTLIBPATH) -lCore -lCling -lHist -lGpad -lGraf -lRIO -L$(CURDIR)/../lib -lHEPEvent
 
 $(LIB_A): $(LIB_OBJ)
 	ar cr $(LIB_A) $(LIB_OBJ) 
# bla bla
