diff -u -r ../meschach-1.2-orig/machine.h ./machine.h
--- ../meschach-1.2-orig/machine.h	2003-10-06 20:49:33.000000000 +0200
+++ ./machine.h	2007-01-18 17:54:10.000000000 +0100
@@ -188,6 +188,6 @@
 
 
 #ifdef ANSI_C
-extern	int	isatty(int);
+//extern	int	isatty(int);
 #endif
 
diff -u -r ../meschach-1.2-orig/matrix.h ./matrix.h
--- ../meschach-1.2-orig/matrix.h	2003-10-28 22:25:38.000000000 +0100
+++ ./matrix.h	2007-01-18 17:53:57.000000000 +0100
@@ -200,7 +200,7 @@
 /* free (de-allocate) (band) matrices, vectors, permutations and 
    integer vectors */
 extern  int iv_free(IVEC *);
-extern	m_free(MAT *),v_free(VEC *),px_free(PERM *);
+extern	int m_free(MAT *),v_free(VEC *),px_free(PERM *);
 extern   int bd_free(BAND *);
 
 #endif /* ANSI_C */
# bla bla
