diff -Naur versWebEndMay2007plain/dynal.inc myTests2008_08/dynal.inc
--- versWebEndMay2007plain/dynal.inc	2006-09-29 20:09:30.000000000 +0200
+++ myTests2008_08/dynal.inc	2008-08-16 21:47:25.000000000 +0200
@@ -1,5 +1,8 @@
 
-      PARAMETER       (MEGA=100 000 000) ! 100 mio words
+       PARAMETER       (MEGA=NUMBER_OF_WORDS) ! make preprocessor decide
+*      PARAMETER       (MEGA=100 000 000) ! 100 mio words: 400 MB
+*         Following needs 64 bit and compiler option -mcmodel=medium:
+*      PARAMETER       (MEGA=1 000 000 000) ! 1000 mio words: 4 GB 
 *                           -----------
       COMMON/COMEGA/MQ(MEGA)
       REAL          QM(MEGA)
diff -Naur versWebEndMay2007plain/Makefile myTests2008_08/Makefile
--- versWebEndMay2007plain/Makefile	2008-07-23 22:35:45.000000000 +0200
+++ myTests2008_08/Makefile	2008-08-18 22:47:18.000000000 +0200
@@ -1,16 +1,25 @@
 # #################################################################
 # Makefile for MillePede II with possible input from C
-# (works on 32-bit SLC3 and and 64-bit SLC4)
+# (works on 32-bit SLC4)
 # #################################################################
 #
 # compiler options
 #
-# To make it link with a BIG static array in dynal.inc,
-# but does not work on 32 bit machines:
-#LARGE_MEMORY_OPT=-mcmodel=medium
+# To make it link with a BIG (>2 GB) static array in dynal.inc,
+# we need '-mcmodel=medium', but does not work on 32 bit machines,
+# so here we switch off and restrict to smaller binaries
 LARGE_MEMORY_OPT=
 # All but 'yes' disables support of reading C-binaries:
 SUPPORT_READ_C = yes
+# If yes (and if SUPPORT_READ_C is yes), uses rfio, i.e. shift library, for IO,
+# requires shift library to be installed
+# (valid only for default executables, pede_*rfio*-ones will always get it):
+SUPPORT_C_RFIO =
+# yes
+#
+# Define the number of words for matrix/vector storage for the default pede program:
+NUMBER_OF_WORDS = 100000000 # =400 MB
+#
 #
 FCOMP = gcc
 F_FLAGS = -Wall -fno-automatic -fno-backslash -O3 ${LARGE_MEMORY_OPT}
@@ -19,6 +28,7 @@
 CCOMP = gcc 
 C_FLAGS = -Wall -O3 -Df2cFortran ${LARGE_MEMORY_OPT}
 C_INCLUDEDIRS =  # e.g. -I .
+C_LIBS = -lg2c -lfrtbegin
 DEBUG =          # e.g. -g
 #
 LOADER = gcc
@@ -26,41 +36,91 @@
 #
 # objects for this project
 #
-USER_OBJ_PEDE = pede.o mptest.o mille.o mpnum.o mptext.o mphistab.o \
-	dynal.o minresblas.o minres.o vertpr.o linesrch.o
+USER_OBJ_PEDE_STATIC = mptest.o mille.o mpnum.o mptext.o mphistab.o \
+	minresblas.o minres.o vertpr.o linesrch.o
+USER_OBJ_PEDE = ${USER_OBJ_PEDE_STATIC} pede.o dynal.o
 #
 # Chose flags/object files for C-binary support:
 #
 ifeq ($(SUPPORT_READ_C),yes)
   F_FLAGS += -DREAD_C_FILES
   USER_OBJ_PEDE += readc.o
+  ifeq ($(SUPPORT_C_RFIO),yes)
+    C_FLAGS += -DUSE_SHIFT_RFIO
+    C_LIBS += -lshift
+  endif
 endif
 #  
 #
 # Make the executables
-#
+# The specific ones (part of hack, see below) + the single:
+EXECUTABLES = pede pede_1GB pede_1GB_rfio pede_2GB pede_2GB_rfio 
+
+all:	$(EXECUTABLES)
+
+# The single special one:
 pede : 	${USER_OBJ_PEDE} Makefile
-	$(LOADER) $(L_FLAGS) -lg2c -lfrtbegin \
+	$(LOADER) $(L_FLAGS) $(C_LIBS) \
 		-o $@ ${USER_OBJ_PEDE} 
-#
+#  
 #
 #
 clean:
 	rm -f *.o *~
 #
+clobber: clean 
+	rm -f $(EXECUTABLES)
+
+install: $(EXECUTABLES) #clean
+	mkdir -p bin
+	mv $(EXECUTABLES) bin
+
 # Make the object files - depend on source and include file 
 #
 %.o : %.F Makefile
 	${FCOMP} ${F_FLAGS} -c $< -o $@ 
-#
-# these two depend on the included memory:
-dynal.o : dynal.F dynal.inc 
-pede.o : pede.F dynal.inc 
-#
+# These two depend on the memory defined by NUMBER_OF_WORDS, 
+# the second also on variable definitions:
+dynal.o : dynal.F dynal.inc Makefile
+	${FCOMP} ${F_FLAGS} -DNUMBER_OF_WORDS=${NUMBER_OF_WORDS} -c $< -o $@
+pede.o : pede.F dynal.inc mpinds.inc Makefile
+	${FCOMP} ${F_FLAGS} -DNUMBER_OF_WORDS=${NUMBER_OF_WORDS} -c $< -o $@
 #
 %.o: %.c Makefile
 	$(CCOMP) -c $(C_FLAGS) $(DEFINES) $(C_INCLUDEDIRS) $(DEBUG) -o $@ $<
 #
+####################################################################
+# Here we start the hack for the various executables...
+
+%_1GB.o: %.F dynal.inc Makefile
+	${FCOMP} ${F_FLAGS} -DNUMBER_OF_WORDS=250000000 -c $< -o $@
+%_2GB.o: %.F dynal.inc Makefile
+	${FCOMP} ${F_FLAGS} -DNUMBER_OF_WORDS=500000000 -c $< -o $@
+%_rfio.o: %.c Makefile
+	$(CCOMP) -c $(C_FLAGS) -DUSE_SHIFT_RFIO $(DEFINES) $(C_INCLUDEDIRS) \
+		$(DEBUG) -o $@ $<
+#
+#
+#
+pede_1GB: ${USER_OBJ_PEDE_STATIC} pede_1GB.o dynal_1GB.o readc.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_1GB.o dynal_1GB.o readc.o  
+#
+pede_2GB: ${USER_OBJ_PEDE_STATIC} pede_2GB.o dynal_2GB.o readc.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_2GB.o dynal_2GB.o readc.o  
+#
+pede_1GB_rfio: ${USER_OBJ_PEDE_STATIC} pede_1GB.o dynal_1GB.o readc_rfio.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) -lshift \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_1GB.o dynal_1GB.o readc_rfio.o  
+#
+pede_2GB_rfio: ${USER_OBJ_PEDE_STATIC} pede_2GB.o dynal_2GB.o readc_rfio.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) -lshift \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_2GB.o dynal_2GB.o readc_rfio.o  
+#
+# End hack for the various executables...
+####################################################################
+#
 # ##################################################################
 # END
 # ##################################################################
diff -Naur versWebEndMay2007plain/mpinds.inc myTests2008_08/mpinds.inc
--- versWebEndMay2007plain/mpinds.inc	2007-04-10 11:46:52.000000000 +0200
+++ myTests2008_08/mpinds.inc	2008-08-16 21:47:24.000000000 +0200
@@ -21,18 +21,21 @@
       PARAMETER (NDIMB=5000)
       COMMON/MEMI/NST,GLDER(NDIMB),INDER(NDIMB)
 
-      PARAMETER (MFILES=100)
+*GF      PARAMETER (MFILES=100)
+      PARAMETER (MFILES=500)
       INTEGER MFD(MFILES)  ! MFD(.)=1 binary   MFD(.)=2 text
       INTEGER LFD(MFILES)  ! LFD(.)=  number of characters
       INTEGER NFD(MFILES)  ! NFD(.)=  index in file  
-      CHARACTER*240 TFD(MFILES)        ! file names
+*GF      CHARACTER*240 TFD(MFILES)        ! file names
+      CHARACTER*500 TFD(MFILES)        ! file names
       COMMON/MPFILE/IFILE,NFILES,NFILB,NFILC,MFD,LFD,TFD,NFD
 
       LOGICAL RECPRI,NEWITE
       COMMON/MPSTER/NTGB,NVGB,NAGB,NCGB,NAGBN,NALCN,
      +              METSOL,MATSTO,MPRINT,MDEBUG,MREQEN,INTRAC,
      +              MITERA,MBANDW,ICTEST,NAEQN,NREC,NLOOPN,
-     +              LUNKNO,LHUBER,CHICUT,CHIREM,DFLIM,NREJEC(3),
+*GF     +              LUNKNO,LHUBER,CHICUT,CHIREM,DFLIM,NREJEC(3),
+     +              LUNKNO,LHUBER,CHICUT,CHIREM,CHHUGE,DFLIM,NREJEC(3),
      +              ITERAT,NTERAT,TIMES(0:6),GBM,UBM,STEPL,
      +              NRECPR,NRECP2,NREC1,NREC2,VALUE1,VALUE2,
      +              RECPRI,NEWITE,TA(2),DWCUT,ISUBIT,NHISTP,
diff -Naur versWebEndMay2007plain/mptext.F myTests2008_08/mptext.F
--- versWebEndMay2007plain/mptext.F	2006-12-05 12:21:47.000000000 +0100
+++ myTests2008_08/mptext.F	2008-08-16 21:47:25.000000000 +0200
@@ -210,8 +210,11 @@
 *     returns MATCH=3, NPAT=4, NTEXT=8       
 *     ------------------------------------------------------------------
       CHARACTER*(*) PAT,TEXT
-      INTEGER ID(0:100,2)
-
+*GF
+*      INTEGER ID(0:100,2) 
+      PARAMETER (NPATMA=512)
+      INTEGER ID(0:NPATMA,2) 
+* end GF
       LOGICAL START                        ! for case conversion
       CHARACTER*26 CHU,CHL
       INTEGER NJ(0:255)
@@ -253,8 +256,12 @@
        IF(PAT(I:I).NE.' ') IPB=I
       END DO
       NPAT=IPB-IPA+1
-      IF(NPAT.GT.100) STOP 'MATINT: string PAT too long!   '
-
+*GF      IF(NPAT.GT.100) STOP 'MATINT: string PAT too long!   '
+      IF(NPAT.GT.NPATMA) THEN
+         WRITE(*,*) 'too long PAT (', PAT,'):', NPAT, ' >', NPATMA
+         STOP 'MATINT: string PAT too long!   '
+      END IF
+*GF end
       ID(0,1)=0 
       DO I=0,NPAT
        ID(I,2)=I
diff -Naur versWebEndMay2007plain/pede.F myTests2008_08/pede.F
--- versWebEndMay2007plain/pede.F	2007-05-25 12:37:09.000000000 +0200
+++ myTests2008_08/pede.F	2008-08-16 21:47:24.000000000 +0200
@@ -998,7 +998,22 @@
       END IF
       DIAG=DQ(IGMAT/2+JK)
       IF(GMATI*DIAG.GT.0.0) THEN   ! global correlation
-         GCOR=SQRT(ABS(1.0-1.0/(GMATI*DIAG)))
+* GF                GCOR=SQRT(ABS(1.0-1.0/(GMATI*DIAG)))
+* GF                 why abs??? cf. http://rkb.home.cern.ch/rkb/AN16pp/node40.html#39
+* GF start...
+         GCOR=(1.0-1.0/(GMATI*DIAG))
+         IF(GCOR.LT.0.0) THEN
+            WRITE(*,*) 'ERROR: global correlation^2 = ',GCOR,' set -2.'
+            GCOR=-2.0
+         ELSE 
+            GCOR=SQRT(GCOR)
+            IF(GCOR.GT.1.0) THEN
+               WRITE(*,*) 'ERROR: global correlation = ',
+     +              GCOR,', set -1.'
+               GCOR=-1.0
+            END IF
+         END IF
+* GF end...
       END IF
       WRITE(*,102) ITGBL,PAR,QM(IND1+ITGBI),DPA,ERR,GCOR
  101  FORMAT(1X,'    label    parameter  presigma     differ',
@@ -1508,7 +1523,8 @@
 *     open all binary files
 
 
-      CHARACTER*256 TEXT,FILNAM 
+* GF      CHARACTER*256 TEXT,FILNAM 
+      CHARACTER*512 TEXT,FILNAM 
 #include "dynal.inc"
       CHARACTER*14 BITE(3)
       CHARACTER*32 KEYSTX
@@ -1532,6 +1548,9 @@
       LHUBER=0  ! Huber down-weighting flag 
       CHICUT=0  ! cut in terms of 3-sigma cut, first iteration
       CHIREM=0  ! cut in terms of 3-sigma cut, other iterations
+* GF add
+      CHHUGE=50.0 ! cut in terms of 3-sigma for unreasonable data, all iterations
+* GF add end
       NRECPR=0  ! record number with printout
       NRECP2=0  ! record number with printout
       NREC1 =0  ! record number with largest residual
@@ -1642,18 +1661,18 @@
       END IF
 
       KEYSTX='fortranfiles'
-      MAT=MATINT(TEXT(IA:IB),KEYSTX,NPAT,NTEXT)
-      IF(MAT.GE.NTEXT-NTEXT/10) THEN
-c      IF (TEXT(IA:IB).EQ.KEYSTX) THEN ! exact comparison by GF
+cGF      MAT=MATINT(TEXT(IA:IB),KEYSTX,NPAT,NTEXT)
+cGF      IF(MAT.GE.NTEXT-NTEXT/10) THEN
+      IF (TEXT(IA:IB).EQ.KEYSTX) THEN ! exact comparison by GF
          NUF=3
 c         WRITE(*,*) 'Fortran files'
          GOTO 10   
       END IF
 
       KEYSTX='Cfiles'
-      MAT=MATINT(TEXT(IA:IB),KEYSTX,NPAT,NTEXT)
-      IF(MAT.GE.NTEXT-NTEXT/10) THEN 
-c      IF (TEXT(IA:IB).EQ.KEYSTX) THEN ! exact comparison by GF
+cGF      MAT=MATINT(TEXT(IA:IB),KEYSTX,NPAT,NTEXT)
+cGF      IF(MAT.GE.NTEXT-NTEXT/10) THEN 
+      IF (TEXT(IA:IB).EQ.KEYSTX) THEN ! exact comparison by GF
          NUF=1
 c         WRITE(*,*) 'Cfiles'
          GOTO 10
@@ -1723,13 +1742,17 @@
           WRITE(*,*) 'Opening C file ',NFILC+1, ': ',TFD(I)(1:LFD(I)) ! by GF
           CALL OPENC(TFD(I)(1:LFD(I)),IOS)
           IF(IOS.NE.0) THEN
-             WRITE(*,*) 'Open error for file ',TFD(I)(1:LFD(I))
+*GF             WRITE(*,*) 'Open error for file ',TFD(I)(1:LFD(I))
+             WRITE(*,*) 'Open error ',IOS,' for file ',TFD(I)(1:LFD(I))
              IOSUM=IOSUM+1 ! typo fixed by GF
           ELSE 
              NFILC=NFILC+1
           END IF 
 #else
           WRITE(*,*) 'Opening of C-files not supported.'
+* GF add
+          IOSUM=IOSUM+1
+* GF add end
 #endif
        END IF
       END DO
@@ -2000,9 +2023,19 @@
       SAVE
 *     ...
       NUFILE=0
-      INQUIRE(FILE=FILNAM,IOSTAT=IOS,EXIST=EX)
-      IF(IOS.NE.0) NUFILE=-ABS(IOS)
-      IF(IOS.NE.0) RETURN
+* GF avoid INQUIRE on rfio files (will fail) and assume them to be binary
+      IF(FILNAM(1:5).EQ.'rfio:') THEN
+         LL=LEN(FILNAM) 
+         FILNAM=FILNAM(6:LL)
+         EX=.TRUE.
+         NUFILE=1
+         RETURN
+      ELSE
+* GF end avoid...
+         INQUIRE(FILE=FILNAM,IOSTAT=IOS,EXIST=EX)
+         IF(IOS.NE.0) NUFILE=-ABS(IOS)
+         IF(IOS.NE.0) RETURN
+      END IF ! GF
       IF(EX) THEN
          NUFILE=1                                   ! binary
          LL=LEN(FILNAM) 
@@ -2084,7 +2117,8 @@
          IF(MAT.GE.(NPAT-NPAT/5)) THEN          
 c         IF(MAT.GE.(NTEXT+NTEXT+3)/3) THEN
             MDEBUG=1
-            IF(NUMS.GT.0) MPRINT=DNUM(1)
+* GF            IF(NUMS.GT.0) MPRINT=DNUM(1)
+            IF(NUMS.GT.0) MDEBUG=DNUM(1)
             RETURN
          END IF
 
@@ -2125,6 +2159,16 @@
             RETURN
          END IF
 
+* GF added:
+         KEYSTX='hugecut'
+         MAT=MATINT(TEXT(KEYA:KEYB),KEYSTX,NPAT,NTEXT) ! comparison
+         IF(MAT.GE.(NPAT-NPAT/5)) THEN          
+            IF(NUMS.GT.0) CHHUGE=DNUM(1)
+            IF(CHHUGE.LT.1.0) CHHUGE=1.0 ! at least (!!) 3-sigma
+            RETURN
+         END IF
+* GF added end
+
          KEYSTX='bandwidth'
          MAT=MATINT(TEXT(KEYA:KEYB),KEYSTX,NPAT,NTEXT) ! comparison
 c         IF(MAT.GE.(NTEXT+NTEXT+3)/3) THEN
@@ -3833,7 +3877,8 @@
 c            RETURN
 c         END IF
       CHICHI=CHINDL(3,NDF)*FLOAT(NDF)
-      IF(SUMM.GT.50.0*CHICHI) THEN ! huge 
+* GF      IF(SUMM.GT.50.0*CHICHI) THEN ! huge 
+      IF(SUMM.GT.CHHUGE*CHICHI) THEN ! huge 
           NREJEC(2)=NREJEC(2)+1    ! count cases with huge chi^2
           IF(RECPRI) THEN
                WRITE(1,*) '   ---> rejected!'
@@ -4076,8 +4121,24 @@
              GCOR=0.0
              DIAG=DQ(IDUX1/2+IVGBI)
              IF(GMATI*DIAG.GT.0.0) THEN   ! global correlation
-                GCOR=SQRT(ABS(1.0-1.0/(GMATI*DIAG)))
-                IF(GCOR.LT.0.0.OR.GCOR.GT.1.0) GCOR=-1.0
+* GF                GCOR=SQRT(ABS(1.0-1.0/(GMATI*DIAG)))
+* GF                 why abs??? cf. http://rkb.home.cern.ch/rkb/AN16pp/node40.html#39
+* GF                IF(GCOR.LT.0.0.OR.GCOR.GT.1.0) GCOR=-1.0
+* GF start...
+                GCOR=(1.0-1.0/(GMATI*DIAG))
+                IF(GCOR.LT.0.0) THEN
+                   WRITE(*,*) 'ERROR: global correlation^2 = ',
+     +                  GCOR, ' set -2.'
+                   GCOR=-2.0
+                ELSE 
+                   GCOR=SQRT(GCOR)
+                   IF(GCOR.GT.1.0) THEN
+                      WRITE(*,*) 'ERROR: global correlation = ',
+     +                     GCOR,', set -1.'
+                      GCOR=-1.0
+                   END IF
+                END IF
+* GF end...
              END IF
           END IF
        END IF
diff -Naur versWebEndMay2007plain/readc.c myTests2008_08/readc.c
--- versWebEndMay2007plain/readc.c	2007-05-04 09:35:49.000000000 +0200
+++ myTests2008_08/readc.c	2008-08-16 21:47:25.000000000 +0200
@@ -10,16 +10,27 @@
    goes through all files as if it were only one, in contrast to the 
    fortran READ used in routine PEREAD of pede.F.
 
-   written by Gero Flucke (gero.flucke@cern.ch),
-   last update on March 1st, 2007
+   if compiled with preprocessor macro USE_SHIFT_RFIO, uses libRFIO,
+   i.e. includes shift.h instead of stdio.h
+
+   written by Gero Flucke (gero.flucke@cern.ch) in 2006/7,
+   last update on July 14th, 2008
 */
 
+#ifdef USE_SHIFT_RFIO
+#include <shift.h>
+// or this??
+// // needed?#define _LARGEFILE64_SOURCE
+//#include <sys/types.h>
+//#include "rfio_api.h"
+#else
 #include <stdio.h>
+#endif
 #include "cfortran.h"
 
 /* ________ global variables used for file handling __________ */
 
-#define MAXNUMFILES 90
+#define MAXNUMFILES 490        /* should roughly match MFILES in mpinds.inc */
 FILE *files[MAXNUMFILES];      /* pointers to opened binary files */
 int fileReadOnce[MAXNUMFILES]; /* flag to printout record number once */
 unsigned int numAllFiles;      /* number of opened files */
@@ -116,7 +127,9 @@
    size_t nCheckR = fread(&recordLength, sizeof(recordLength), 1,
  			 files[fileIndex]);
    while (feof(files[fileIndex])) {
-     rewind(files[fileIndex]);
+     /* rewind(files[fileIndex]);  Does not work with rfio, so call: */
+     fseek(files[fileIndex], 0L, SEEK_SET);
+     clearerr(files[fileIndex]); /* These two should be the same as rewind... */
      if (!fileReadOnce[fileIndex]) {
        printf("readC: %d. file read the first time, found %d records.\n",
 	      fileIndex+1, nRec);
