Index: Makefile
===================================================================
--- rfio/Makefile	(revision 78)
+++ rfio/Makefile	(working copy)
@@ -81,14 +81,15 @@
 #
 # Make the executables
 # The specific ones (part of hack, see below) + the single:
-EXECUTABLES = pede pede_1GB pede_2GB 
+#EXECUTABLES = pede pede_1GB pede_2GB 
 # Or also those supporting rfio-reading:
 #EXECUTABLES = pede pede_1GB pede_2GB pede_1GB_rfio pede_2GB_rfio 
 # If you need also the executables with more memory, you need a
 # 64-bit environment and the '-mcmodel=medium' option (see above).
 #	pede_4GB pede_4GB_rfio pede_8GB pede_8GB_rfio
-#EXECUTABLES = pede pede_1GB pede_2GB pede_1GB_rfio pede_2GB_rfio \
-#	pede_4GB pede_4GB_rfio pede_8GB pede_8GB_rfio
+EXECUTABLES = pede pede_1GB pede_2GB pede_1GB_rfio pede_2GB_rfio \
+	pede_8GB pede_8GB_rfio pede_16GB pede_16GB_rfio \
+	pede_24GB pede_24GB_rfio pede_32GB pede_32GB_rfio
 
 all:	$(EXECUTABLES)
 
@@ -186,12 +187,25 @@
 	$(LOADER) $(L_FLAGS) $(C_LIBS) \
 		-o $@ ${USER_OBJ_PEDE_STATIC} pede_16GB.o dynal_16GB.o readc.o
 #
+pede_16GB_rfio: ${USER_OBJ_PEDE_STATIC} pede_16GB.o dynal_16GB.o readc_rfio.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_16GB.o dynal_16GB.o readc_rfio.o
+#
 pede_24GB: ${USER_OBJ_PEDE_STATIC} pede_24GB.o dynal_24GB.o readc.o Makefile
 	$(LOADER) $(L_FLAGS) $(C_LIBS) \
 		-o $@ ${USER_OBJ_PEDE_STATIC} pede_24GB.o dynal_24GB.o readc.o
+#
+pede_24GB_rfio: ${USER_OBJ_PEDE_STATIC} pede_24GB.o dynal_24GB.o readc_rfio.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_24GB.o dynal_24GB.o readc_rfio.o
+#
 pede_32GB: ${USER_OBJ_PEDE_STATIC} pede_32GB.o dynal_32GB.o readc.o Makefile
 	$(LOADER) $(L_FLAGS) $(C_LIBS) \
 		-o $@ ${USER_OBJ_PEDE_STATIC} pede_32GB.o dynal_32GB.o readc.o
+#
+pede_32GB_rfio: ${USER_OBJ_PEDE_STATIC} pede_32GB.o dynal_32GB.o readc_rfio.o Makefile
+	$(LOADER) $(L_FLAGS) $(C_LIBS) \
+		-o $@ ${USER_OBJ_PEDE_STATIC} pede_32GB.o dynal_32GB.o readc_rfio.o
 pede_48GB: ${USER_OBJ_PEDE_STATIC} pede_48GB.o dynal_48GB.o readc.o Makefile
 	$(LOADER) $(L_FLAGS) $(C_LIBS) \
 		-o $@ ${USER_OBJ_PEDE_STATIC} pede_48GB.o dynal_48GB.o readc.o
