diff --git a/mimetic/tokenizer.h b/mimetic/tokenizer.h
index 247b181..b968778 100644
--- a/mimetic/tokenizer.h
+++ b/mimetic/tokenizer.h
@@ -156,7 +156,7 @@ struct ContTokenizer: public ItTokenizer<typename Container::const_iterator,type
     ContTokenizer(const Container* cont, const DelimCont& delims)
     : ItTokenizer<const_iterator,value_type>(cont->begin(), cont->end())
     {
-        setDelimList(delims);
+        this->setDelimList(delims);
     }
     void setSource(const Container* cont)
     {
