diff -Naur orig.mimetic-0.9.6/mimetic/rfc822/messageid.h mimetic-0.9.6/mimetic/rfc822/messageid.h
--- orig.mimetic-0.9.6/mimetic/rfc822/messageid.h	2009-05-08 15:47:58.000000000 +0200
+++ mimetic-0.9.6/mimetic/rfc822/messageid.h	2009-12-26 05:21:38.000000000 +0100
@@ -26,6 +26,7 @@
 #include <mimetic/utils.h>
 #include <mimetic/os/utils.h>
 #include <mimetic/rfc822/fieldvalue.h>
+typedef unsigned int            uint32_t;
 
 namespace mimetic
 {
