diff -aur mimetic-0.9.5/mimetic/codec/other_codecs.h mimetic-0.9.5/mimetic/codec/other_codecs.h
--- mimetic-0.9.5/mimetic/codec/other_codecs.h	2008-10-07 13:06:26.000000000 +0200
+++ mimetic-0.9.5/mimetic/codec/other_codecs.h	2009-09-09 15:50:36.000000000 +0200
@@ -118,7 +118,7 @@
     : m_max(0), m_written(0)
     {
     }
-    MaxLineLen(uint m)
+    MaxLineLen(unsigned int m)
     : m_max(m), m_written(0)
     {
     }
