--- orig.mimetic-0.8.9/mimetic/utils.h	2005-03-07 16:35:08.000000000 +0100
+++ mimetic-0.8.9/mimetic/utils.h	2007-06-28 13:24:22.000000000 +0200
@@ -49,20 +49,6 @@
 /// returns a string hexadecimal representation of \p n
 std::string int2hex(unsigned int n);
 
-// boyer-moore find 
-/**
- * find the first occurrence of \p word in (\p bit, \p eit]
- *
- * returns an Iterator pointing at the first character of the found pattern
- * or \p eit if the search fails
- */
-template<typename Iterator>
-Iterator find_bm(Iterator bit, Iterator eit, const std::string& word)
-{
-    return find_bm(bit, eit, word, 
-        typename std::iterator_traits<Iterator>::iterator_category());
-}
-
 // find_bm specialization for random access iterators
 template<typename Iterator>
 Iterator find_bm(Iterator bit, Iterator eit, const std::string& word, const std::random_access_iterator_tag&)
@@ -94,6 +80,20 @@
     return bit + i + 1;
 }
 
+// boyer-moore find 
+/**
+ * find the first occurrence of \p word in (\p bit, \p eit]
+ *
+ * returns an Iterator pointing at the first character of the found pattern
+ * or \p eit if the search fails
+ */
+template<typename Iterator>
+Iterator find_bm(Iterator bit, Iterator eit, const std::string& word)
+{
+    return find_bm(bit, eit, word, 
+        typename std::iterator_traits<Iterator>::iterator_category());
+}
+
 
 
 } // ns utils
