--- WrapXS/Apache2/Connection/Connection.xs.orig	2014-10-13 21:51:24.000000000 +0200
+++ WrapXS/Apache2/Connection/Connection.xs	2014-10-13 21:56:55.000000000 +0200
@@ -114,7 +114,7 @@
 
 
     CODE:
-    RETVAL = (APR__SockAddr) obj->remote_addr;
+    RETVAL = (APR__SockAddr) obj->client_addr;
 
     OUTPUT:
     RETVAL
@@ -131,11 +131,11 @@
 
 
     CODE:
-    RETVAL = (char *) obj->remote_ip;
+    RETVAL = (char *) obj->client_ip;
 
     if (items > 1) {
          
-         obj->remote_ip = (char *) (SvOK(ST(1)) ?
+         obj->client_ip = (char *) (SvOK(ST(1)) ?
                     apr_pstrndup(mpxs_Apache2__Connection_pool(obj), val, val_len) : NULL);
     }
 
