--- SConstruct.orig	2012-09-06 16:41:37.000000000 +0200
+++ SConstruct	2012-09-06 16:42:22.000000000 +0200
@@ -498,9 +498,6 @@
        env.Append( EXTRALIBPATH=["/usr/64/lib"] )
        if installDir == DEFAULT_INSTALL_DIR:
            installDir = "/usr/64/"
-    else:
-       env.Append( EXTRACPPPATH=filterExists(["/sw/include" , "/opt/local/include"]) )
-       env.Append( EXTRALIBPATH=filterExists(["/sw/lib/", "/opt/local/lib"]) )
 
 elif os.sys.platform.startswith("linux"):
     linux = True
