--- SConstruct	2011-07-09 02:33:31.000000000 +0200
+++ SConstruct	2011-07-09 02:36:24.000000000 +0200
@@ -459,9 +459,6 @@ if "darwin" == os.sys.platform:
         env.Append( LIBPATH=["/usr/64/lib"] )
         if installDir == DEFAULT_INSTALL_DIR and not distBuild:
             installDir = "/usr/64/"
-    else:
-        env.Append( CPPPATH=filterExists(["/sw/include" , "/opt/local/include"]) )
-        env.Append( LIBPATH=filterExists(["/sw/lib/", "/opt/local/lib"]) )
 
 elif "linux2" == os.sys.platform:
     linux = True
@@ -1176,8 +1173,6 @@ elif not onlyServer:
         shellEnv["LINKFLAGS"].remove("-m64")
         shellEnv["CPPPATH"].remove( "/usr/64/include" )
         shellEnv["LIBPATH"].remove( "/usr/64/lib" )
-        shellEnv.Append( CPPPATH=filterExists(["/sw/include" , "/opt/local/include"]) )
-        shellEnv.Append( LIBPATH=filterExists(["/sw/lib/", "/opt/local/lib" , "/usr/lib", "/usr/local/lib" ]) )
 
     l = shellEnv["LIBS"]
 
