diff --git a/mozilla/security/nss/cmd/platlibs.mk b/mozilla/security/nss/cmd/platlibs.mk
index 332681d..f2bd502 100644
--- a/mozilla/security/nss/cmd/platlibs.mk
+++ b/mozilla/security/nss/cmd/platlibs.mk
@@ -123,6 +123,7 @@ endif
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+	-L$(SQLITE_LIBS_DIR) \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
diff --git a/mozilla/security/nss/lib/softoken/config.mk b/mozilla/security/nss/lib/softoken/config.mk
index 908d1d9..52426ce 100644
--- a/mozilla/security/nss/lib/softoken/config.mk
+++ b/mozilla/security/nss/lib/softoken/config.mk
@@ -48,6 +48,7 @@ else
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+	-L$(SQLITE_LIBS_DIR) \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
