diff --git a/nss/cmd/platlibs.mk b/nss/cmd/platlibs.mk
index 833952a..516089c 100644
--- a/nss/cmd/platlibs.mk
+++ b/nss/cmd/platlibs.mk
@@ -136,6 +136,7 @@ endif
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+	-L$(SQLITE_LIBS_DIR) \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
diff --git a/nss/lib/softoken/config.mk b/nss/lib/softoken/config.mk
index 6058e71..ac6d7fe 100644
--- a/nss/lib/softoken/config.mk
+++ b/nss/lib/softoken/config.mk
@@ -48,6 +48,7 @@ else
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+	-L$(SQLITE_LIBS_DIR) \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
