diff -rupN nss-3.12.9.orig/mozilla/security/nss/cmd/platlibs.mk nss-3.12.9/mozilla/security/nss/cmd/platlibs.mk
--- nss-3.12.9.orig/mozilla/security/nss/cmd/platlibs.mk	2012-01-28 21:16:37.000000000 +0100
+++ nss-3.12.9/mozilla/security/nss/cmd/platlibs.mk	2012-01-28 21:18:45.000000000 +0100
@@ -177,6 +177,7 @@ endif
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+  $(SQLITE_LIBS_DIR) \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
diff -rupN nss-3.12.9.orig/mozilla/security/nss/lib/softoken/config.mk nss-3.12.9/mozilla/security/nss/lib/softoken/config.mk
--- nss-3.12.9.orig/mozilla/security/nss/lib/softoken/config.mk	2012-01-28 21:16:38.000000000 +0100
+++ nss-3.12.9/mozilla/security/nss/lib/softoken/config.mk	2012-01-28 21:19:27.000000000 +0100
@@ -80,6 +80,7 @@ else
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+  $(SQLITE_LIBS_DIR) \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
