diff --git a/nss/cmd/modutil/Makefile b/nss/cmd/modutil/Makefile
index 1a64b47..c8808d8 100644
--- a/nss/cmd/modutil/Makefile
+++ b/nss/cmd/modutil/Makefile
@@ -25,8 +25,12 @@ include $(CORE_DEPTH)/coreconf/config.mk
 #######################################################################
 include ../platlibs.mk
 
+ifdef ZLIB_INCLUDE_DIR
+INCLUDES += -I$(ZLIB_INCLUDE_DIR)
+endif
+
 ifdef USE_SYSTEM_ZLIB
-OS_LIBS += $(ZLIB_LIBS)
+OS_LIBS += -L$(ZLIB_LIBS_DIR) $(ZLIB_LIBS)
 else
 EXTRA_LIBS += $(ZLIB_LIBS)
 endif
diff --git a/nss/cmd/signtool/Makefile b/nss/cmd/signtool/Makefile
index 20c45d6..1e5d8e7 100644
--- a/nss/cmd/signtool/Makefile
+++ b/nss/cmd/signtool/Makefile
@@ -26,8 +26,12 @@ include $(CORE_DEPTH)/coreconf/config.mk
 
 include ../platlibs.mk
 
+ifdef ZLIB_INCLUDE_DIR
+INCLUDES += -I$(ZLIB_INCLUDE_DIR)
+endif
+
 ifdef USE_SYSTEM_ZLIB
-OS_LIBS += $(ZLIB_LIBS)
+OS_LIBS += -L$(ZLIB_LIBS_DIR) $(ZLIB_LIBS)
 else
 EXTRA_LIBS += $(ZLIB_LIBS)
 endif
diff --git a/nss/lib/ssl/config.mk b/nss/lib/ssl/config.mk
index 40b1c30..37f7152 100644
--- a/nss/lib/ssl/config.mk
+++ b/nss/lib/ssl/config.mk
@@ -80,11 +80,15 @@ ifdef NSS_ENABLE_ZLIB
 
 DEFINES += -DNSS_ENABLE_ZLIB
 
+ifdef ZLIB_INCLUDE_DIR
+INCLUDES += -I$(ZLIB_INCLUDE_DIR)
+endif
+
 # If a platform has a system zlib, set USE_SYSTEM_ZLIB to 1 and
 # ZLIB_LIBS to the linker command-line arguments for the system zlib
 # (for example, -lz) in the platform's config file in coreconf.
 ifdef USE_SYSTEM_ZLIB
-OS_LIBS += $(ZLIB_LIBS)
+OS_LIBS += -L$(ZLIB_LIBS_DIR) $(ZLIB_LIBS)
 else
 ZLIB_LIBS = $(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX)
 EXTRA_LIBS += $(ZLIB_LIBS)
