diff -rupN nss-3.12.9.orig/mozilla/security/nss/cmd/modutil/Makefile nss-3.12.9/mozilla/security/nss/cmd/modutil/Makefile
--- nss-3.12.9.orig/mozilla/security/nss/cmd/modutil/Makefile	2012-01-28 20:15:06.000000000 +0100
+++ nss-3.12.9/mozilla/security/nss/cmd/modutil/Makefile	2012-01-28 20:23:32.000000000 +0100
@@ -58,7 +58,7 @@ include $(CORE_DEPTH)/coreconf/config.mk
 include ../platlibs.mk
 
 ifdef USE_SYSTEM_ZLIB
-OS_LIBS += $(ZLIB_LIBS)
+OS_LIBS += $(ZLIB_LIBS_DIR) $(ZLIB_LIBS)
 else
 EXTRA_LIBS += $(ZLIB_LIBS)
 endif
diff -rupN nss-3.12.9.orig/mozilla/security/nss/cmd/signtool/Makefile nss-3.12.9/mozilla/security/nss/cmd/signtool/Makefile
--- nss-3.12.9.orig/mozilla/security/nss/cmd/signtool/Makefile	2012-01-28 20:15:06.000000000 +0100
+++ nss-3.12.9/mozilla/security/nss/cmd/signtool/Makefile	2012-01-28 20:22:51.000000000 +0100
@@ -59,7 +59,7 @@ include $(CORE_DEPTH)/coreconf/config.mk
 include ../platlibs.mk
 
 ifdef USE_SYSTEM_ZLIB
-OS_LIBS += $(ZLIB_LIBS)
+OS_LIBS += $(ZLIB_LIBS_DIR) $(ZLIB_LIBS)
 else
 EXTRA_LIBS += $(ZLIB_LIBS)
 endif
diff -rupN nss-3.12.9.orig/mozilla/security/nss/lib/ssl/config.mk nss-3.12.9/mozilla/security/nss/lib/ssl/config.mk
--- nss-3.12.9.orig/mozilla/security/nss/lib/ssl/config.mk	2012-01-28 20:15:08.000000000 +0100
+++ nss-3.12.9/mozilla/security/nss/lib/ssl/config.mk	2012-01-28 20:20:34.000000000 +0100
@@ -111,7 +111,7 @@ DEFINES += -DNSS_ENABLE_ZLIB
 # ZLIB_LIBS to the linker command-line arguments for the system zlib
 # (for example, -lz) in the platform's config file in coreconf.
 ifdef USE_SYSTEM_ZLIB
-OS_LIBS += $(ZLIB_LIBS)
+OS_LIBS += $(ZLIB_LIBS_DIR) $(ZLIB_LIBS)
 else
 ZLIB_LIBS = $(DIST)/lib/$(LIB_PREFIX)zlib.$(LIB_SUFFIX)
 EXTRA_LIBS += $(ZLIB_LIBS)
