diff --git a/numpy/f2py/__init__.py b/numpy/f2py/__init__.py
index e583250f70..750ab35c5c 100644
--- a/numpy/f2py/__init__.py
+++ b/numpy/f2py/__init__.py
@@ -11,6 +11,12 @@
 __all__ = ['run_main', 'compile', 'get_include']
 
 import sys
+while True:
+  if sys.path[0].endswith(".zip"):
+    break
+  temp_path = sys.path.pop(0)
+  sys.path.append(temp_path)
+
 import subprocess
 import os
 import warnings
