diff --git a/pyol/build/cpp.scons b/pyol/build/cpp.scons
index 6636833..a47390c 100644
--- a/pyol/build/cpp.scons
+++ b/pyol/build/cpp.scons
@@ -33,7 +33,7 @@ sys.path.insert(0, os.path.abspath(os.path.join('pyol', 'tools')))
 import OLBaseConfig
 config = OLBaseConfig.get_config()
 
-cpp_flags = ['--traditional-cpp', '-P']
+cpp_flags = ['--traditional-cpp', '-P', '-undef']
 
 def split_and_clean(ls):
     """Create a list from a comma separated string and remove empty elements."""
# bla bla
