--- a/modules/imgproc/test/test_houghlines.cpp	2023-07-18 08:46:05.532555671 +0200
+++ b/modules/imgproc/test/test_houghlines.cpp	2023-07-18 08:46:39.922490881 +0200
@@ -53,7 +53,7 @@
     T value;
     float theta_eps;
     float rho_eps;
-    SimilarWith<T>(T val, float e, float r_e): value(val), theta_eps(e), rho_eps(r_e) { };
+    SimilarWith(T val, float e, float r_e): value(val), theta_eps(e), rho_eps(r_e) { };
     bool operator()(const T& other);
 };
 
