*** Child.pm	2009-06-02 16:10:53.120660674 +0200
--- Child.pm.patched	2009-05-28 16:38:08.687685140 +0200
***************
*** 265,279 ****
  
  	# clean up
  
! 	delete $self->{$PKG}{wheels}{$id};
! 	delete $self->{$PKG}{pids}{$id};
  
  	# all expiring children should issue a "done" except when
  	# the return code is non-zero which indicates a failure
  	# if the caller asked we quit, fire a "done" regardless
  	# of the child's return code value (we might have hard killed)
  
! 	my $event = ($self->{$PKG}{wheels}{$id}{quit} || $rc == 0)
  		? "done" : "died"
  		;
  	$self->callback($event, { wheel => $id, rc => $rc });
--- 265,281 ----
  
  	# clean up
  
! 	delete $self->{$PKG}{pids}{$self->{$PKG}{wheels}{$id}{ref}->PID};
!         my $wheel = delete $self->{$PKG}{wheels}{$id};
! 	delete $self->{$PKG}{SIGCHLD}{$id};
! 	delete $self->{$PKG}{CLOSED}{$id};
  
  	# all expiring children should issue a "done" except when
  	# the return code is non-zero which indicates a failure
  	# if the caller asked we quit, fire a "done" regardless
  	# of the child's return code value (we might have hard killed)
  
! 	my $event = ($wheel->{quit} || $rc == 0)
  		? "done" : "died"
  		;
  	$self->callback($event, { wheel => $id, rc => $rc });
***************
*** 306,311 ****
--- 308,314 ----
  sub wheel {
  	my $self = shift;
  	my $id = shift || $self->{$PKG}{wheels}{current};
+ 	return undef unless exists $self->{$PKG}{wheels}{$id};
  	$self->{$PKG}{wheels}{$id}{ref};
  	}
  
