*** Child.pm	2009-05-08 18:37:39.000000000 +0200
--- Child.pm.patched	2009-05-08 18:37:32.000000000 +0200
***************
*** 266,271 ****
  	# clean up
  
  	delete $self->{$PKG}{wheels}{$id};
! 	delete $self->{$PKG}{pids}{$id};
  
  	# all expiring children should issue a "done" except when
--- 266,273 ----
  	# clean up
  
+ 	delete $self->{$PKG}{pids}{$self->{$PKG}{wheels}{$id}{ref}->PID};
  	delete $self->{$PKG}{wheels}{$id};
! 	delete $self->{$PKG}{SIGCHLD}{$id};
! 	delete $self->{$PKG}{CLOSED}{$id};
  
  	# all expiring children should issue a "done" except when
