diff --git a/photos/215.5/configure b/photos/215.5/configure
index c8382cc..9adb5eb 100755
--- a/photos/215.5/configure
+++ b/photos/215.5/configure
@@ -115,7 +115,8 @@ if [ $ARCH = Linux ]; then
   fi
 fi
 # Linux platform with gcc4: new Fortran90 compiler.
-if [ $ARCH = Linux-gcc4 ]; then
+case $ARCH in 
+	*-gcc4)
   export FC=gfortran
   export FFLAGS="${FFLAGS_OPT}"
   export CFLAGS="${CFLAGS_OPT}"
@@ -131,7 +132,8 @@ if [ $ARCH = Linux-gcc4 ]; then
     export CFLAGS="${CFLAGS_DBG}"
     export CXXFLAGS="${CXXFLAGS_DBG}"
   fi
-fi
+  ;;
+esac
 
 #Platform & opt/dbg - independent flags and variables:
 
diff --git a/photos/215.5/Makeshared.subdir b/photos/215.5/Makeshared.subdir
index f167145..025e173 100644
--- a/photos/215.5/Makeshared.subdir
+++ b/photos/215.5/Makeshared.subdir
@@ -42,7 +42,14 @@ objects += $(patsubst $(DIRNAME)/%.c,$(TMPDIR)/%.o,$(wildcard $(DIRNAME)/*.c))
 $(LIBNAME): $(objects)
 	@mkdir -p $(LIBDIR)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@)
+	case ${ARCH} in \
+	  Linux-*) \
+	    $(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) \
+	  ;;\
+	  Darwin-*) \
+	    $(CXX) $(LDFLAGSSHARED) $(objects) -Wl,-commons,use_dylibs -o $@ -shared -L$(dir $(shell gfortran --print-file-name=libgfortran.a)) $(FLIBS) -dynamic \
+	  ;; \
+	esac
 
 deps := $(patsubst $(DIRNAME)/%.F,$(TMPDIR)/%.d,$(wildcard $(DIRNAME)/*.F))
 
