--- setup.py	2011-07-09 01:32:21.000000000 +0200
+++ setup.py	2011-07-09 01:32:50.000000000 +0200
@@ -54,14 +54,15 @@ simply send your email to yajl@librelist
 if USE_SETUPTOOLS:
     setup_kwargs.update({'test_suite' : 'tests.unit'})
 
-if not os.listdir('yajl'):
+if False:
+  if not os.listdir('yajl'):
     # Submodule hasn't been created, let's inform the user
     print('>>> It looks like the `yajl` submodule hasn\'t been initialized')
     print('>>> I\'ll try to do that, but if I fail, you can run:')
     print('>>>      `git submodule update --init`')
     subprocess.call(['git', 'submodule', 'update', '--init'])
 
-subprocess.call(['git', 'submodule', 'update',])
+  subprocess.call(['git', 'submodule', 'update',])
 
 if not os.path.exists('includes'):
     # Our symlink into the yajl directory isn't there, let's fixulate that
