diff --git a/oamap/compiler.py b/oamap/compiler.py
index f45e42a..04073b4 100644
--- a/oamap/compiler.py
+++ b/oamap/compiler.py
@@ -294,7 +294,7 @@ else:
                     else:
                         raise IndexError("item {0} out of range for type {1}".format(idx, tpe.schema))
 
-    @numba.extending.lower_builtin("static_getitem", SchemaType, numba.types.Const)
+    @numba.extending.lower_builtin("static_getitem", SchemaType, numba.types.Literal)
     def schema_static_getitem(context, builder, sig, args):
         typ, _ = sig.args
         return numba.cgutils.create_struct_proxy(typ)(context, builder)._getvalue()
@@ -1415,7 +1415,7 @@ else:
                     else:
                         raise IndexError("item {0} out of range for type {1}".format(idx, tpe.generator.schema))
 
-    @numba.extending.lower_builtin("static_getitem", TupleProxyNumbaType, numba.types.Const)
+    @numba.extending.lower_builtin("static_getitem", TupleProxyNumbaType, numba.types.Literal)
     def tupleproxy_static_getitem(context, builder, sig, args):
         tupletpe, _ = sig.args
         tupleval, idx = args
