diff --git a/scipy/sparse/sparsetools/SConscript b/scipy/sparse/sparsetools/SConscript
index 1aa5557..85d9e05 100644
--- a/scipy/sparse/sparsetools/SConscript
+++ b/scipy/sparse/sparsetools/SConscript
@@ -3,6 +3,7 @@
 from numscons import GetNumpyEnvironment
 
 env = GetNumpyEnvironment(ARGUMENTS)
+env.PrependUnique(CPPDEFINES = '__STDC_FORMAT_MACROS')
 
 for fmt in ['csr','csc','coo','bsr','dia']:
     sources = [ fmt + '_wrap.cxx' ]
diff --git a/scipy/sparse/sparsetools/setup.py b/scipy/sparse/sparsetools/setup.py
index f4027a7..c1024ee 100755
--- a/scipy/sparse/sparsetools/setup.py
+++ b/scipy/sparse/sparsetools/setup.py
@@ -9,7 +9,9 @@ def configuration(parent_package='',top_path=None):
     for fmt in ['csr','csc','coo','bsr','dia']:
         sources = [ fmt + '_wrap.cxx' ]
         depends = [ fmt + '.h' ]
-        config.add_extension('_' + fmt, sources=sources, depends=depends)
+        config.add_extension('_' + fmt, sources=sources, 
+            define_macros=[('__STDC_FORMAT_MACROS', 1)], 
+            depends=depends) 
 
     return config
 
