diff -Naur orig/ffi/Makefile.linux llvmlite-0.15.0/ffi/Makefile.linux
--- orig/ffi/Makefile.linux	2017-04-21 18:12:05.000000000 +0200
+++ llvmlite-0.15.0/ffi/Makefile.linux	2017-04-21 18:12:34.000000000 +0200
@@ -18,8 +18,8 @@
 
 $(OUTPUT): $(SRC) $(INCLUDE)
 	# static-libstdc++ avoids runtime dependencies on a
-	# particular libstdc++ version.
-	$(CXX) -static-libstdc++ -shared $(CXXFLAGS) $(SRC) -o $(OUTPUT) $(LDFLAGS) $(LIBS)
+	# particular libstdc++ version. (but doesnt work for us)
+	$(CXX) -shared $(CXXFLAGS) $(SRC) -o $(OUTPUT) $(LDFLAGS) $(LIBS)
 
 clean:
 	rm -rf test $(OUTPUT)
