--- a/numpy/distutils/system_info.py	2011-09-14 15:28:38.000000000 +0200
+++ b/numpy/distutils/system_info.py	2011-09-14 15:36:41.000000000 +0200
@@ -1283,23 +1283,13 @@ class lapack_opt_info(system_info):
 
         if sys.platform=='darwin' and not os.environ.get('ATLAS',None):
             args = []
-            link_args = []
-            if get_platform()[-4:] == 'i386':
+	    link_args = [os.environ["LAPACK"], os.environ["BLAS"]]
+            if get_platform()[-6:] == 'x86_64':
                 intel = 1
+                args.extend(['-msse3'])
             else:
                 intel = 0
-            if os.path.exists('/System/Library/Frameworks/Accelerate.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                link_args.extend(['-Wl,-framework','-Wl,Accelerate'])
-            elif os.path.exists('/System/Library/Frameworks/vecLib.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                link_args.extend(['-Wl,-framework','-Wl,vecLib'])
+                args.extend(['-faltivec'])
             if args:
                 self.set_info(extra_compile_args=args,
                               extra_link_args=link_args,
@@ -1369,28 +1359,14 @@ class blas_opt_info(system_info):
     def calc_info(self):
 
         if sys.platform=='darwin' and not os.environ.get('ATLAS',None):
-            args = []
-            link_args = []
-            if get_platform()[-4:] == 'i386':
+            args = ['-I' + os.environ.get("LAPACK_ROOT", ".")]
+	    link_args = [os.environ["BLAS"]]
+            if get_platform()[-6:] == 'x86_64':
                 intel = 1
+                args.extend(['-msse3'])
             else:
                 intel = 0
-            if os.path.exists('/System/Library/Frameworks/Accelerate.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                args.extend([
-                    '-I/System/Library/Frameworks/vecLib.framework/Headers'])
-                link_args.extend(['-Wl,-framework','-Wl,Accelerate'])
-            elif os.path.exists('/System/Library/Frameworks/vecLib.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                args.extend([
-                    '-I/System/Library/Frameworks/vecLib.framework/Headers'])
-                link_args.extend(['-Wl,-framework','-Wl,vecLib'])
+                args.extend(['-faltivec'])
             if args:
                 self.set_info(extra_compile_args=args,
                               extra_link_args=link_args,
