diff --git a/xgboost/src/common/hist_util.cc b/xgboost/src/common/hist_util.cc
index 8e13403..9a68e88 100644
--- a/xgboost/src/common/hist_util.cc
+++ b/xgboost/src/common/hist_util.cc
@@ -12,14 +12,7 @@
 #include "./hist_util.h"
 #include "./quantile.h"
 
-#if defined(XGBOOST_MM_PREFETCH_PRESENT)
-  #include <xmmintrin.h>
-  #define PREFETCH_READ_T0(addr) _mm_prefetch(reinterpret_cast<const char*>(addr), _MM_HINT_T0)
-#elif defined(XGBOOST_BUILTIN_PREFETCH_PRESENT)
-  #define PREFETCH_READ_T0(addr) __builtin_prefetch(reinterpret_cast<const char*>(addr), 0, 3)
-#else  // no SW pre-fetching available; PREFETCH_READ_T0 is no-op
-  #define PREFETCH_READ_T0(addr) do {} while (0)
-#endif  // defined(XGBOOST_MM_PREFETCH_PRESENT)
+#define PREFETCH_READ_T0(addr) do {} while (0)
 
 namespace xgboost {
 namespace common {
