--- a/setup.py	2017-06-19 15:04:52.000000000 +0200
+++ b/setup.py	2017-06-19 15:05:52.000000000 +0200
@@ -32,13 +32,8 @@
 from setuptools.command.build_ext import build_ext
 
 # For guessing the capabilities of the CPU for C-Blosc
-try:
-    import cpuinfo
-    cpu_info = cpuinfo.get_cpu_info()
-    cpu_flags = cpu_info['flags']
-except Exception as e:
-    print('cpuinfo failed, assuming no CPU features:', e)
-    cpu_flags = []
+# avoid guessing for CMS
+cpu_flags = []
 
 # The name for the pkg-config utility
 PKG_CONFIG = 'pkg-config'
# bla bla
