diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index e0fd213..e944fd1 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -1254,18 +1254,16 @@ class lapack_opt_info(system_info):
                 intel = 1
             else:
                 intel = 0
-            if os.path.exists('/System/Library/Frameworks/Accelerate.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                link_args.extend(['-Wl,-framework','-Wl,Accelerate'])
-            elif os.path.exists('/System/Library/Frameworks/vecLib.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                link_args.extend(['-Wl,-framework','-Wl,vecLib'])
+            if intel:
+                args.extend(['-msse3'])
+            else:
+                args.extend(['-faltivec'])
+            link_args.extend(['-llapack'])
+            if intel:
+                args.extend(['-msse3'])
+            else:
+                args.extend(['-faltivec'])
+            link_args.extend(['-lblas'])
             if args:
                 self.set_info(extra_compile_args=args,
                               extra_link_args=link_args,
diff --git a/numpy/distutils/system_info.py b/numpy/distutils/system_info.py
index e944fd1..0c6fd27 100644
--- a/numpy/distutils/system_info.py
+++ b/numpy/distutils/system_info.py
@@ -1349,22 +1349,13 @@ class blas_opt_info(system_info):
                 intel = 1
             else:
                 intel = 0
-            if os.path.exists('/System/Library/Frameworks/Accelerate.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                args.extend([
-                    '-I/System/Library/Frameworks/vecLib.framework/Headers'])
-                link_args.extend(['-Wl,-framework','-Wl,Accelerate'])
-            elif os.path.exists('/System/Library/Frameworks/vecLib.framework/'):
-                if intel:
-                    args.extend(['-msse3'])
-                else:
-                    args.extend(['-faltivec'])
-                args.extend([
-                    '-I/System/Library/Frameworks/vecLib.framework/Headers'])
-                link_args.extend(['-Wl,-framework','-Wl,vecLib'])
+            if intel:
+                args.extend(['-msse3'])
+            else:
+                args.extend(['-faltivec'])
+            args.extend([
+                '-I' + os.environ.get("LAPACK_ROOT",".")])
+            link_args.extend(['-lblas'])
             if args:
                 self.set_info(extra_compile_args=args,
                               extra_link_args=link_args,
