diff -rupN xml.orig/xpath/ParsedAbbreviatedAbsoluteLocationPath.py xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py
--- xml.orig/xpath/ParsedAbbreviatedAbsoluteLocationPath.py	2011-12-14 14:01:09.000000000 +0100
+++ xml/xpath/ParsedAbbreviatedAbsoluteLocationPath.py	2011-12-14 14:01:39.000000000 +0100
@@ -24,8 +24,8 @@ class ParsedAbbreviatedAbsoluteLocationP
         self._rel = rel
         nt = ParsedNodeTest.ParsedNodeTest('node', '')
         ppl = ParsedPredicateList.ParsedPredicateList([])
-        as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
-        self._step = ParsedStep.ParsedStep(as, nt, ppl)
+        pas = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+        self._step = ParsedStep.ParsedStep(pas, nt, ppl)
         return
 
     def evaluate(self, context):
diff -rupN xml.orig/xpath/ParsedAbbreviatedRelativeLocationPath.py xml/xpath/ParsedAbbreviatedRelativeLocationPath.py
--- xml.orig/xpath/ParsedAbbreviatedRelativeLocationPath.py	2011-12-14 14:01:09.000000000 +0100
+++ xml/xpath/ParsedAbbreviatedRelativeLocationPath.py	2011-12-14 14:02:06.000000000 +0100
@@ -28,8 +28,8 @@ class ParsedAbbreviatedRelativeLocationP
         self._right = right
         nt = ParsedNodeTest.ParsedNodeTest('node','')
         ppl = ParsedPredicateList.ParsedPredicateList([])
-        as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
-        self._middle = ParsedStep.ParsedStep(as, nt, ppl)
+        pas = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+        self._middle = ParsedStep.ParsedStep(pas, nt, ppl)
 
     def evaluate(self, context):
         res = []
# bla bla
