diff -Naur a/ffi/build.py b/ffi/build.py
--- a/ffi/build.py	2020-01-10 14:51:38.000000000 +0100
+++ b/ffi/build.py	2020-01-10 14:45:27.059843040 +0100
@@ -110,7 +110,7 @@
     out = out.decode('latin1')
     print(out)
     if not (out.startswith('8.0.') or out.startswith('7.0.')
-            or out.startswith('7.1.')):
+            or out.startswith('7.1.') or out.startswith('9.0.')):
         msg = (
             "Building llvmlite requires LLVM 7.0.x, 7.1.x or 8.0.x Be sure to "
             "set LLVM_CONFIG to the right executable path.\n"
