diff --git a/ffi/build.py b/ffi/build.py
index 6408bf5..22f3b3c 100755
--- a/ffi/build.py
+++ b/ffi/build.py
@@ -163,7 +163,7 @@ def main_posix(kind, library_ext):
         print(warning + '\n')
     else:
 
-        if not out.startswith('11'):
+        if int(out.split(".")[0])<11:
             msg = ("Building llvmlite requires LLVM 11.x.x, got "
                    "{!r}. Be sure to set LLVM_CONFIG to the right executable "
                    "path.\nRead the documentation at "
