diff --git a/setup.py b/setup.py
index d9af186..edfdd3c 100644
--- a/setup.py
+++ b/setup.py
@@ -302,12 +302,8 @@ def get_ext_modules():
 
 packages = find_packages(include=["numba", "numba.*"])
 
-build_requires = [f'numpy >={min_numpy_build_version}']
-install_requires = [
-    f'llvmlite >={min_llvmlite_version},<={max_llvmlite_version}',
-    f'numpy >={min_numpy_run_version}',
-    'setuptools',
-]
+build_requires = ['numpy >=1.11']
+install_requires = ['llvmlite>=0.31.0dev0,<=0.33.0.dev0', 'numpy', 'setuptools']
 
 metadata = dict(
     name='numba',
@@ -346,7 +342,7 @@ metadata = dict(
     packages=packages,
     setup_requires=build_requires,
     install_requires=install_requires,
-    python_requires=f">={min_python_version}",
+    python_requires=">=3.6",
     license="BSD",
     cmdclass=cmdclass,
 )
