--- configure.py.orig	2011-12-14 12:13:54.000000000 +0100
+++ configure.py	2011-12-14 12:22:41.000000000 +0100
@@ -33,6 +33,7 @@
 import glob
 import optparse
 import shutil
+import string
 
 import sipconfig
 
@@ -1984,6 +1985,12 @@
     remove_file(exe_file)
     run_command("%s -f %s%s" % (make, make_file, make_target))
 
+    # Fix Mac OS X qtdirs.app (it must use qt.conf)
+    # Relocate qt.conf to qtdirs.app/Contents/Resources/qt.conf
+    # More details: http://doc.qt.nokia.com/stable/qt-conf.html
+    if string.lower(os.uname()[0]) == "darwin":
+        shutil.copyfile("qt.conf", "qtdirs.app/Contents/Resources/qt.conf")
+
     if not os.access(exe_file, os.X_OK):
         sipconfig.error("Failed to determine the layout of your Qt installation. Try again using the --verbose flag to see more detail about the problem.")
 
