diff -urw orig/pythia8201/src/BeamParticle.cc new/pythia8201/src/BeamParticle.cc
--- orig/pythia8201/src/BeamParticle.cc 2014-10-13 20:22:12.000000000 +0200
+++ new/pythia8201/src/BeamParticle.cc  2014-11-17 20:21:40.678702937 +0100
@@ -113,6 +113,9 @@
   pBeam             = Vec4( 0., 0., pzIn, eIn);
   mBeam             = mIn;
 
+  // Remove remnants? 
+  clear();
+
 }
 
 //--------------------------------------------------------------------------

