Index: include/Pythia8/SigmaTotal.h
===================================================================
--- include/Pythia8/SigmaTotal.h	(revision 536)
+++ include/Pythia8/SigmaTotal.h	(working copy)
@@ -84,8 +84,8 @@
   // Constants: could only be changed in the code itself.
   static const int    IHADATABLE[], IHADBTABLE[], ISDTABLE[], IDDTABLE[];
   static const double MMIN, EPSILON, ETA, X[], Y[], BETA0[], BHAD[],
-                      ALPHAPRIME, CONVERTEL, CONVERTSD, CONVERTDD, MMIN0,
-                      CRES, MRES0, CSD[10][8], CDD[10][9], SPROTON;
+                      ALPHAPRIME, CONVERTEL, CONVERTSD, CONVERTDD,
+                      CSD[10][8], CDD[10][9], SPROTON;
 
   // Integration of MBR cross sections and form factor approximation.
   static const int    NINTEG, NINTEG2;
@@ -96,6 +96,7 @@
   double sigAXB2TeV, sigTotOwn, sigElOwn, sigXBOwn, sigAXOwn, sigXXOwn,
          sigAXBOwn, maxXBOwn, maxAXOwn, maxXXOwn, maxAXBOwn, bSlope, rho,
          lambda, tAbsMin, alphaEM0, sigmaPomP, mPomP, pPomP, fracSigmaNDgamma;
+  double MMIN0, CRES, MRES0;
 
   // Parameters of MBR model.
   double MBReps, MBRalpha, MBRbeta0, MBRsigma0, m2min, dyminSDflux,
Index: src/SigmaTotal.cc
===================================================================
--- src/SigmaTotal.cc	(revision 536)
+++ src/SigmaTotal.cc	(working copy)
@@ -70,9 +70,9 @@
 
 // Diffractive mass spectrum starts at m + MMIN0 and has a low-mass
 // enhancement, factor cRes, up to around m + mRes0.
-const double SigmaTotal::MMIN0 = 0.28;
-const double SigmaTotal::CRES  = 2.0;
-const double SigmaTotal::MRES0 = 1.062;
+//const double SigmaTotal::MMIN0 = 0.28;
+//const double SigmaTotal::CRES  = 2.0;
+//const double SigmaTotal::MRES0 = 1.062;
 
 // Parameters and coefficients for single diffractive scattering.
 const int SigmaTotal::ISDTABLE[] = { 0, 0, 1, 1, 1, 2, 3, 4, 5,
@@ -176,6 +176,12 @@
   dyminSigDD  = settings.parm("Diffraction:MBRdyminSigDD");
   dyminSigCD  = settings.parm("Diffraction:MBRdyminSigCD");
 
+// Diffractive mass spectrum starts at m + MMIN0 and has a low-mass
+// enhancement, factor cRes, up to around m + mRes0.
+  MMIN0       = settings.parm("SigmaDiffractive:mMin");
+  CRES        = settings.parm("SigmaDiffractive:lowMEnhance");
+  MRES0       = settings.parm("SigmaDiffractive:mResMax");  
+  
   // Parameters for gamma+gamma total cross section.
   fracSigmaNDgamma = settings.parm("Photon:sigmaNDfrac");
 }
Index: share/Pythia8/xmldoc/TotalCrossSections.xml
===================================================================
--- share/Pythia8/xmldoc/TotalCrossSections.xml	(revision 536)
+++ share/Pythia8/xmldoc/TotalCrossSections.xml	(working copy)
@@ -113,7 +113,26 @@
 is what makes up the minimum-bias event class, and plays a major role 
 in the description of multiparton interactions, it is important that a 
 consistent set is used. 
- 
+
+<h3>Modify diffractive cross sections</h3> 
+
+   The default description of diffractive interactions was
+   parameterized and fit in <ref>Sch94, Sch97</ref>.   The following parameters
+   allow for a modification of that fit.    Note, these parameters
+   have no effect on the MBR model.
+
+  <parm name = "SigmaDiffractive:mMin" default="0.28" min="0.0">
+  Diffractive cross section calculation begins a mHadron + mMin.
+  </parm>
+  <parm name  = "SigmaDiffractive:lowMEnhance" default="2.0" min="0.0">
+    Normalization factor for the contribution of low-mass resonances
+    to the diffractive cross section.
+  </parm>
+  <parm name = "SigmaDiffractive:mResMax" default="1.062" min="0.0">
+    The contribution of low-mass resonances is allowed up ot
+    mHadron+mResMax.  To make sense, we should have mResMax>mMin.
+  </parm>
+
 <h3>Dampen diffractive cross sections</h3> 
  
 As already noted, unitarization effects may dampen the rise of diffractive 
