diff --git a/200pre2/configure b/200pre2/configure
index adac4a5..5234fef 100755
--- a/200pre2/configure
+++ b/200pre2/configure
@@ -192,7 +192,8 @@ if [ -z "$CXX_SHARED" ]; then
     if [ "$ARCH" = "LINUX" ];  then CXX_SHARED="-shared"; fi
     if [ "$ARCH" = "DARWIN" ]; then 
 	CXX_SHARED="-dynamiclib -flat_namespace"
-	CXX_SHARED="$CXX_SHARED -undefined suppress -Qunused-arguments"; fi; fi
+	CXX_SHARED="$CXX_SHARED -undefined suppress"
+	if [[ "$(basename $CXX)" = *clang* ]]; then CXX_SHARED="$CXX_SHARED -Qunused-arguments"; fi; fi; fi
 if [ -z "$CXX_SONAME" ]; then 
     if [ "$ARCH" = "LINUX" ];  then CXX_SONAME="-Wl,-soname"; fi
     if [ "$ARCH" = "DARWIN" ]; then CXX_SONAME="-Wl,-dylib_install_name"; fi; fi
