diff -ur orig/pythia8200/configure new/pythia8200/configure
--- orig/pythia8200/configure	2014-10-10 21:46:33.000000000 +0200
+++ new/pythia8200/configure	2014-10-13 01:25:23.472416381 +0200
@@ -77,7 +77,7 @@
 --lib-suffix=SUF     : Shared library name suffix: [.so] for Linux and [.dylib]
                        for OS X.
 BLOCKTEXT
-OPTIONS="-h --h -help --help --enable-debug --enable-64bit --enable-64bit --lcg"
+OPTIONS="-h --h -help --help --enable-debug --enable-64bit --enable-64bit --lcg --enable-shared"
 OPTIONS+=" --prefix --prefix-bin --prefix-lib --prefix-include --prefix-share"
 for PKG in "fastjet3" "hepmc2" "hepmc3" "lhapdf5" "lhapdf6" "root"\
     "boost" "gzip"; do
@@ -206,7 +206,8 @@
     if [ "$ARCH" = "LINUX" ];  then CXX_SHARED="-shared"; fi
     if [ "$ARCH" = "DARWIN" ]; then 
 	CXX_SHARED="-dynamiclib -flat_namespace"
-	CXX_SHARED="$CXX_SHARED -undefined suppress -Qunused-arguments"; fi; fi
+	CXX_SHARED="$CXX_SHARED -undefined suppress"
+	if [[ "$(basename $CXX)" = *clang* ]]; then CXX_SHARED="$CXX_SHARED -Qunused-arguments"; fi; fi; fi
 if [ -z "$CXX_SONAME" ]; then 
     if [ "$ARCH" = "LINUX" ];  then CXX_SONAME="-Wl,-soname"; fi
     if [ "$ARCH" = "DARWIN" ]; then CXX_SONAME="-Wl,-dylib_install_name"; fi; fi
