--- a/Lib/distutils/util.py	2012-03-10 16:40:10.000000000 +0100
+++ b/Lib/distutils/util.py	2012-03-10 17:38:44.000000000 +0100
@@ -162,11 +162,14 @@ def get_platform ():
                     raise ValueError(
                        "Don't know machine value for archs=%r"%(archs,))
 
-
             elif machine in ('PowerPC', 'Power_Macintosh'):
                 # Pick a sane name for the PPC architecture.
                 machine = 'ppc'
 
+            elif machine == 'i386' and sys.maxsize > 2**32:
+	        # If built 64-bit but running on 32-bit kernel, report 64-bit
+                machine = 'x86_64'
+
     return "%s-%s-%s" % (osname, release, machine)
 
 # get_platform ()
