diff --git a/cmake/External/aotriton.cmake b/cmake/External/aotriton.cmake
index 2678cfde3c4..bb601775d2a 100644
--- a/cmake/External/aotriton.cmake
+++ b/cmake/External/aotriton.cmake
@@ -33,7 +33,7 @@ if(NOT __AOTRITON_INCLUDED)
       "rocm6.3"
       "rocm6.4"
       )
-  set(__AOTRITON_CI_COMMIT "b388d223d8c7213545603e00f6f3148c54d1f525")
+  set(__AOTRITON_CI_COMMIT "e328a7776da72be798725109747cdf350c615c3e")
   set(__AOTRITON_SHA256_LIST
       "08d84f96f4c984179f80f517c0431c7511ee26bb0ce9bd05a827573ddd78cc79"  # rocm6.2
       "9094d59717e7e6eace9126ca100dd0e86510f07fc6c3a349569fc4e2d9056604"  # rocm6.3
@@ -53,7 +53,7 @@ if(NOT __AOTRITON_INCLUDED)
     set(target_gpus "")
     get_target_gpus_from_pytorch(target_gpus)
     ExternalProject_Add(aotriton_external
-      GIT_REPOSITORY https://github.com/ROCm/aotriton.git
+      GIT_REPOSITORY https://github.com/cms-externals/aotriton.git
       GIT_TAG ${__AOTRITON_CI_COMMIT}
       PREFIX ${__AOTRITON_EXTERN_PREFIX}
       INSTALL_DIR ${__AOTRITON_INSTALL_DIR}
