diff -arupN qt-everywhere-opensource-src-4.6.3.orig/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h qt-everywhere-opensource-src-4.6.3/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h
--- qt-everywhere-opensource-src-4.6.3.orig/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-07 16:27:21.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.3/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-08 00:04:35.000000000 +0100
@@ -55,9 +55,7 @@ namespace JSC {
 
     class JSGlobalObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
-
-        struct JSGlobalObjectData : public JSVariableObjectData {
+        struct JSGlobalObjectData : public JSVariableObject::JSVariableObjectData {
             JSGlobalObjectData()
                 : JSVariableObjectData(&symbolTable, 0)
                 , registerArraySize(0)
diff -arupN qt-everywhere-opensource-src-4.6.3.orig/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h qt-everywhere-opensource-src-4.6.3/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h
--- qt-everywhere-opensource-src-4.6.3.orig/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h	2012-03-07 16:27:21.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.3/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h	2012-03-07 16:29:34.000000000 +0100
@@ -32,8 +32,7 @@ namespace JSC{
     
     class JSStaticScopeObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
-        struct JSStaticScopeObjectData : public JSVariableObjectData {
+        struct JSStaticScopeObjectData : public JSVariableObject::JSVariableObjectData {
             JSStaticScopeObjectData()
                 : JSVariableObjectData(&symbolTable, &registerStore + 1)
             {
diff -arupN qt-everywhere-opensource-src-4.6.3.orig/src/3rdparty/javascriptcore/JavaScriptCore/wtf/RefPtr.h qt-everywhere-opensource-src-4.6.3/src/3rdparty/javascriptcore/JavaScriptCore/wtf/RefPtr.h
--- qt-everywhere-opensource-src-4.6.3.orig/src/3rdparty/javascriptcore/JavaScriptCore/wtf/RefPtr.h	2012-03-07 16:27:21.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.3/src/3rdparty/javascriptcore/JavaScriptCore/wtf/RefPtr.h	2012-03-08 09:31:17.000000000 +0100
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include "AlwaysInline.h"
 #include "FastAllocBase.h"
+#include "PassRefPtr.h"
 
 namespace WTF {
 
diff -arupN qt-everywhere-opensource-src-4.6.3.orig/tools/porting/src/codemodel.h qt-everywhere-opensource-src-4.6.3/tools/porting/src/codemodel.h
--- qt-everywhere-opensource-src-4.6.3.orig/tools/porting/src/codemodel.h	2012-03-07 16:27:23.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.3/tools/porting/src/codemodel.h	2012-03-08 10:03:06.000000000 +0100
@@ -96,7 +96,7 @@ class Collection: public QMultiHash<QByt
 {
 public:
      void add(CollectedType *collectedItem)
-    { insert(collectedItem->name(), collectedItem); }
+    { this->insert(collectedItem->name(), collectedItem); }
 };
 
 typedef Collection<Scope> ScopeCollection;
# bla bla
