? reflex/src/reflex.Typedef.diff
Index: reflex/src/Typedef.h
===================================================================
RCS file: /user/cvs/root/reflex/src/Typedef.h,v
retrieving revision 1.15
diff -u -r1.15 Typedef.h
--- reflex/src/Typedef.h	5 Sep 2006 17:13:15 -0000	1.15
+++ reflex/src/Typedef.h	24 Nov 2006 15:12:24 -0000
@@ -743,7 +743,10 @@
 //-------------------------------------------------------------------------------
 inline std::string ROOT::Reflex::Typedef::Name( unsigned int mod ) const {
 //-------------------------------------------------------------------------------
-   if ( 0 != ( mod & ( FINAL | F ))) return FinalType().Name( mod );
+   if ( 0 != ( mod & ( FINAL | F ))) {
+      if (IsFundamental()) return fTypedefType.Name( mod );
+      else return FinalType().Name( mod );
+   }
    else                              return TypeBase::Name( mod );
 }
 
