diff --git a/2.5.2/include/Rivet/Tools/RivetSTL.hh b/2.5.2/include/Rivet/Tools/RivetSTL.hh
index 7b38c1f..ae6c2e6 100644
--- a/2.5.2/include/Rivet/Tools/RivetSTL.hh
+++ b/2.5.2/include/Rivet/Tools/RivetSTL.hh
@@ -19,7 +19,7 @@
 #include <iomanip>
 #include <cmath>
 #include <limits>
-
+#include <functional>
 
 #ifndef foreach
 /// @decl A foreach macro for backward compatibility with BOOST_FOREACH
