diff --git a/include/LWH/Axis.h b/include/LWH/Axis.h
index 8c5ca9f..c3ce5bc 100644
--- a/include/LWH/Axis.h
+++ b/include/LWH/Axis.h
@@ -115,7 +115,7 @@ public:
    *
    */
   int coordToIndex(double coord) const {
-    assert( ! isnan(coord) );
+    assert( ! std::isnan(coord) );
     if ( coord >= upper ) return OVERFLOW_BIN;
     else if ( coord < lower ) return UNDERFLOW_BIN;
     else return int((coord - lower)/binWidth(0));
diff --git a/src/Analyses/LHCB_2011_I917009.cc b/src/Analyses/LHCB_2011_I917009.cc
index 5e8cc60..f0d2fc8 100644
--- a/src/Analyses/LHCB_2011_I917009.cc
+++ b/src/Analyses/LHCB_2011_I917009.cc
@@ -97,7 +97,7 @@ namespace Rivet {
         const FourMomentum& qmom = p.momentum();
         y = log((qmom.E() + qmom.pz())/(qmom.E() - qmom.pz()))/2.;
         // skip this particle if it has too high or too low rapidity (extremely rare cases when E = +- pz)
-        if ( isnan(y) || isinf(y) ) continue;
+        if ( std::isnan(y) || std::isinf(y) ) continue;
         y = fabs(y);
         if ( (y < rap_min) || (y > rap_max) ) continue;
         pT = sqrt((qmom.px() * qmom.px()) + (qmom.py() * qmom.py()));
