diff -r 607da7b1c45b include/Rivet/AnalysisHandler.hh
--- a/include/Rivet/AnalysisHandler.hh	Thu Jul 18 15:56:06 2019 +0100
+++ b/include/Rivet/AnalysisHandler.hh	Tue Jul 23 16:08:40 2019 +0200
@@ -71,6 +71,8 @@
 
     /// Set the weight names from a GenEvent
     void setWeightNames(const GenEvent& ge);
+    /// Set the weight names from vector<string>
+    void setWeightNames(vector<string> weightNames) { _weightNames = weightNames; }
 
     /// Get the index of the nominal weight-stream
     size_t defaultWeightIndex() const { return _defaultWeightIdx; }
@@ -200,7 +202,7 @@
     //@{
 
     /// Initialize a run, with the run beams taken from the example event.
-    void init(const GenEvent& event);
+    void init(const GenEvent& event, const vector<string>& externalWeightNames = vector<string>());
 
     /// @brief Analyze the given \a event by reference.
     ///
diff -r 607da7b1c45b src/Core/AnalysisHandler.cc
--- a/src/Core/AnalysisHandler.cc	Thu Jul 18 15:56:06 2019 +0100
+++ b/src/Core/AnalysisHandler.cc	Tue Jul 23 16:08:40 2019 +0200
@@ -62,7 +62,7 @@
   }
 
 
-  void AnalysisHandler::init(const GenEvent& ge) {
+  void AnalysisHandler::init(const GenEvent& ge, const vector<string>& externalWeightNames) {
     if (_initialised)
       throw UserError("AnalysisHandler::init has already been called: cannot re-initialize!");
 
@@ -72,7 +72,12 @@
     MSG_DEBUG("Initialising the analysis handler");
     _eventNumber = ge.event_number();
 
-    setWeightNames(ge);
+    if (externalWeightNames.empty()) {
+      setWeightNames(ge);
+    }
+    else {
+      setWeightNames(externalWeightNames);
+    }
     if (_skipWeights)
         MSG_INFO("Only using nominal weight. Variation weights will be ignored.");
     else if (haveNamedWeights())
