diff -Naur src/Analyses/ATLAS_2014_I1307243.cc srcNEW/Analyses/ATLAS_2014_I1307243.cc
--- src/Analyses/ATLAS_2014_I1307243.cc	2015-03-19 14:25:42.000000000 +0100
+++ srcNEW/Analyses/ATLAS_2014_I1307243.cc	2015-04-27 11:13:43.321554321 +0200
@@ -23,12 +23,16 @@
       , _dy_max(8)
       , _nEventsInAcceptance(0)
       , _sumOfAcceptedWeights(0.)
+      , _fiducialRegions{2010, 2011}
+      , _vetoScale{20.*GeV, 30.*GeV}
+      , _yFiducial{4.4, 2.4}
+      , _gapCategories{"inclusive","gap"}
     {
       // Cannot do this in initialiser list without C++11
-      _fiducialRegions = boost::assign::list_of<unsigned int>(2010)(2011);
-      _vetoScale = boost::assign::list_of<double>(20.*GeV)(30.*GeV);
-      _yFiducial = boost::assign::list_of<double>(4.4)(2.4);
-      _gapCategories = boost::assign::list_of<std::string>("inclusive")("gap");
+//      _fiducialRegions = boost::assign::list_of<unsigned int>(2010)(2011);
+//      _vetoScale = boost::assign::list_of<double>(20.*GeV)(30.*GeV);
+//      _yFiducial = boost::assign::list_of<double>(4.4)(2.4);
+//      _gapCategories = boost::assign::list_of<std::string>("inclusive")("gap");
     }
 
     /// Book histograms and initialise projections before the run
diff -Naur src/Analyses/PDG_TAUS.cc srcNEW/Analyses/PDG_TAUS.cc
--- src/Analyses/PDG_TAUS.cc	2015-03-20 12:29:35.000000000 +0100
+++ srcNEW/Analyses/PDG_TAUS.cc	2015-04-27 11:14:02.209747219 +0200
@@ -121,16 +121,16 @@
 
     // Set up a lookup table for decays
     void populateDecayMap() {
-      decay_pids["muids"]     = boost::assign::list_of(13)(14)(16);
-      decay_pids["elids"]     = boost::assign::list_of(11)(12)(16);
-      decay_pids["pinu"]      = boost::assign::list_of(211)(16);
-      decay_pids["Kpnu"]      = boost::assign::list_of(321)(16);
-      decay_pids["pipinu"]    = boost::assign::list_of(111)(211)(16);
-      decay_pids["Kppinu"]    = boost::assign::list_of(111)(321)(16);
-      decay_pids["pipipinu"]  = boost::assign::list_of(111)(111)(211)(16);
-      decay_pids["KSpinu"]    = boost::assign::list_of(211)(310)(16);
-      decay_pids["KLpinu"]    = boost::assign::list_of(211)(130)(16);
-      decay_pids["3pipipinu"] = boost::assign::list_of(211)(211)(211)(16);
+      //decay_pids["muids"]     = boost::assign::list_of(13)(14)(16);
+      //decay_pids["elids"]     = boost::assign::list_of(11)(12)(16);
+      //decay_pids["pinu"]      = boost::assign::list_of(211)(16);
+      //decay_pids["Kpnu"]      = boost::assign::list_of(321)(16);
+      //decay_pids["pipinu"]    = boost::assign::list_of(111)(211)(16);
+      //decay_pids["Kppinu"]    = boost::assign::list_of(111)(321)(16);
+      //decay_pids["pipipinu"]  = boost::assign::list_of(111)(111)(211)(16);
+      //decay_pids["KSpinu"]    = boost::assign::list_of(211)(310)(16);
+      //decay_pids["KLpinu"]    = boost::assign::list_of(211)(130)(16);
+      //decay_pids["3pipipinu"] = boost::assign::list_of(211)(211)(211)(16);
     }
 
 
@@ -206,7 +206,18 @@
     //@}
 
     double _weights_had, _weights_mu, _weights_el;
-    map<string, vector<int> > decay_pids;
+    map<string, vector<int> > decay_pids {
+        {"muids", {13, 14, 16}},
+        {"elids", {11,12,16}},
+        {"pinu", {211, 16}},
+        {"Kpnu", {321, 16}},
+        {"pipinu", {111, 211, 16}},
+        {"Kppinu", {111, 321, 16}},
+        {"pipipinu", {111, 111, 211, 16}},
+        {"KSpinu", {211, 310, 16}},
+        {"KLpinu", {211, 130, 16}},
+        {"3pipipinu", {211, 211, 211, 16}}
+        };
 
   };
 
# bla bla
