--- src/Core/AnalysisLoader.cc
+++ src/Core/AnalysisLoader.cc
@@ -90,6 +90,7 @@ namespace Rivet {
     // Find plugin module library files
     const string libsuffix = ".so";
     vector<string> pluginfiles;
+    vector<string> pluginfileNames;
     for (const string& d : dirs) {
       if (d.empty()) continue;
       oslink::directory dir(d);
@@ -103,8 +104,13 @@ namespace Rivet {
         /// @todo Sys-dependent path separator instead of "/"
         const string path = d + "/" + filename;
         // Ensure no duplicate paths
-        if (find(pluginfiles.begin(), pluginfiles.end(), path) == pluginfiles.end()) {
+        // CMS PATCH: ensure also no duplicate library names. Ensures that only the first plugin library in the RIVET_ANALYIS_PATH is taken.
+        // This ensures that, when the same library is both in the developer area and in the release, only the first one is taken. 
+        // The developer area preceeds the release in the RIVET_ANALYSIS_PATH
+        if (find(pluginfiles.begin(), pluginfiles.end(), path) == pluginfiles.end() &&
+            find(pluginfileNames.begin(), pluginfileNames.end(), filename) == pluginfileNames.end()){
           pluginfiles += path;
+          pluginfileNames += filename;
         }
       }
     }

