--- src/Core/AnalysisLoader.cc
+++ src/Core/AnalysisLoader.cc
@@ -90,6 +90,7 @@ namespace Rivet {
     // Find plugin module library files
     const string libsuffix = ".so";
     vector<string> pluginfiles;
+    vector<string> pluginfileNames;
     for (const string& d : dirs) {
       if (d.empty()) continue;
       oslink::directory dir(d);
@@ -103,8 +104,13 @@ namespace Rivet {
         /// @todo Sys-dependent path separator instead of "/"
         const string path = d + "/" + filename;
         // Ensure no duplicate paths
-        if (find(pluginfiles.begin(), pluginfiles.end(), path) == pluginfiles.end()) {
+        // CMS PATCH: ensure also no duplicate library names. Ensures that only the first plugin library in the RIVET_ANALYIS_PATH is taken.
+        // This ensures that, when the same library is both in the developer area and in the release, only the first one is taken. 
+        // The developer area preceeds the release in the RIVET_ANALYSIS_PATH
+        if (find(pluginfiles.begin(), pluginfiles.end(), path) == pluginfiles.end() &&
+            find(pluginfileNames.begin(), pluginfileNames.end(), filename) == pluginfileNames.end()){
           pluginfiles += path;
+          pluginfileNames += filename;
         }
       }
     }
diff --git analyses/pluginCESR/CLEOII_1997_I446031.cc analyses/pluginCESR/CLEOII_1997_I446031.cc
index 85fc07e..ab1a2be 100644
--- analyses/pluginCESR/CLEOII_1997_I446031.cc
+++ analyses/pluginCESR/CLEOII_1997_I446031.cc
@@ -66,7 +66,7 @@ namespace Rivet {
     void analyze(const Event& event) {
       // find scattered leptons and calc Q2
       const Beam & beams = apply<Beam>(event, "Beams");
-      double Q12,Q22;
+      double Q12 = 0.,Q22 = 0.;
       if(!findScattered(beams.beams().first ,Q12)) vetoEvent;
       if(!findScattered(beams.beams().second,Q22)) vetoEvent;
       double scale = max(Q12,Q22);

