diff -r 0d419d5ca2fa -r ea1ef11f4da8 analyses/pluginCMS/CMS_2017_I1499471.info
--- a/analyses/pluginCMS/CMS_2017_I1499471.info
+++ b/analyses/pluginCMS/CMS_2017_I1499471.info
@@ -3,7 +3,7 @@
 Summary: Measurements of the associated production of a Z boson and b jets in pp collisions at $\sqrt{s} = 8$ TeV
 Experiment: CMS
 Collider: LHC
-InspireID: I1499471
+InspireID: 1499471
 Status: VALIDATED
 Authors:
  - Fabio Cossutti (fabio.cossutti@ts.infn.it)
diff -r 0d419d5ca2fa -r ea1ef11f4da8 analyses/pluginCMS/CMS_2018_I1690148.cc
--- a/analyses/pluginCMS/CMS_2018_I1690148.cc
+++ b/analyses/pluginCMS/CMS_2018_I1690148.cc
@@ -9,7 +9,6 @@
 #include "Rivet/Projections/InvMassFinalState.hh"
 #include "Rivet/Projections/MissingMomentum.hh"
 #include "Rivet/Math/MatrixN.hh"
-#include "Rivet/Math/MatrixDiag.hh"
 #include "Rivet/Tools/fjcontrib/Nsubjettiness.hh"
 #include "Rivet/Tools/fjcontrib/EnergyCorrelator.hh"
 
@@ -73,7 +72,7 @@
         for (int o = 0; o < 33; ++o) { // observable
           d += 1;
           for (int f = 0; f < 4; ++f) { // flavor
-            char buffer [11];
+            char buffer [20];
             sprintf(buffer, "d%02d-x01-y%02d", d, f+1);
             _h[r][o][f] = bookHisto1D(buffer);
           }
@@ -361,8 +360,23 @@
         M += MPart * p.e();
       }
       // Calculate eccentricity from eigenvalues
-      const EigenSystem<2> eigen = diagonalize(M);
-      return 1. - eigen.getEigenValues()[1]/eigen.getEigenValues()[0];
+      // Check that the matrix is symmetric.
+      const bool isSymm = M.isSymm();
+      if (!isSymm) {
+        MSG_ERROR("Error: energy tensor not symmetric:");
+        MSG_ERROR("[0,1] vs. [1,0]: " << M.get(0,1) << ", " << M.get(1,0));
+      }
+      // If not symmetric, something's wrong (we made sure the error msg appeared first).
+
+      assert(isSymm);
+      const double a = M.get(0,0);
+      const double b = M.get(1,1);
+      const double c = M.get(1,0);
+
+      const double l1 = 0.5*(a+b+sqrt( (a-b)*(a-b) + 4 *c*c));
+      const double l2 = 0.5*(a+b-sqrt( (a-b)*(a-b) + 4 *c*c));
+
+      return 1. - l2/l1;
     }
 
 
