diff --git a/analyses/pluginATLAS/ATLAS_2023_I2663725.cc b/analyses/pluginATLAS/ATLAS_2023_I2663725.cc
index 41d39daad7645a563654a55f423ebe4a71a23ef1..e0c1e538ce9d4ddd793ed202db53d7294d3b8144 100644
--- a/analyses/pluginATLAS/ATLAS_2023_I2663725.cc
+++ b/analyses/pluginATLAS/ATLAS_2023_I2663725.cc
@@ -5,6 +5,7 @@
 #include "Rivet/Projections/DressedLeptons.hh"
 #include "Rivet/Projections/PromptFinalState.hh"
 #include "Rivet/Projections/InvisibleFinalState.hh"
+#include "Rivet/Projections/VetoedFinalState.hh"
 
 namespace Rivet {
 
