diff --git a/include/Rivet/Analysis.hh b/include/Rivet/Analysis.hh
index 4734e4e..784d6da 100644
--- a/include/Rivet/Analysis.hh
+++ b/include/Rivet/Analysis.hh
@@ -1187,7 +1187,7 @@ namespace Rivet {
       if (re != "") {
         std::smatch match;
         const bool needsDP = std::regex_search(path, match, std::regex(re));
-        if (needsDP)  yao.template setAnnotation("WriterDoublePrecision", "1");
+        if (needsDP)  yao.template setAnnotation<>("WriterDoublePrecision", "1");
       }
     }
 
