diff --git a/include/Rivet/AnalysisHandler.hh b/include/Rivet/AnalysisHandler.hh
index 4add600a2..d5c6cb633 100644
--- a/include/Rivet/AnalysisHandler.hh
+++ b/include/Rivet/AnalysisHandler.hh
@@ -68,6 +68,8 @@ namespace Rivet {
 
     /// Set the weight names from a GenEvent
     void setWeightNames(const GenEvent& ge);
+    /// Set the weight names from vector<string>
+    void setWeightNames(vector<string> weightNames) { _weightNames = weightNames; }
 
     /// Get the index of the nominal weight-stream
     size_t defaultWeightIndex() const { return _rivetDefaultWeightIdx; }
@@ -221,7 +223,7 @@ namespace Rivet {
     /// @{
 
     /// Initialize a run, with the run beams taken from the example event.
-    void init(const GenEvent& event);
+    void init(const GenEvent& event, const vector<string>& externalWeightNames = vector<string>());
 
     /// @brief Analyze the given \a event by reference.
     ///
diff --git a/src/Core/AnalysisHandler.cc b/src/Core/AnalysisHandler.cc
index f8769a4ad..3c908fd2f 100644
--- a/src/Core/AnalysisHandler.cc
+++ b/src/Core/AnalysisHandler.cc
@@ -71,7 +71,7 @@ namespace Rivet {
   }
 
 
-  void AnalysisHandler::init(const GenEvent& ge) {
+  void AnalysisHandler::init(const GenEvent& ge, const vector<string>& externalWeightNames) {
     if (_initialised)
       throw UserError("AnalysisHandler::init has already been called: cannot re-initialize!");
 
@@ -81,7 +81,12 @@ namespace Rivet {
     MSG_DEBUG("Initialising the analysis handler");
     _eventNumber = ge.event_number();
 
-    setWeightNames(ge);
+    if (externalWeightNames.empty()) {
+      setWeightNames(ge);
+    }
+    else {
+      setWeightNames(externalWeightNames);
+    }
     if (_skipWeights)
         MSG_INFO("Only using nominal weight. Variation weights will be ignored.");
     else if (haveNamedWeights())

