--- Makefile.am.orig	2022-10-31 16:05:58.132999546 +0100
+++ Makefile.am	2022-10-31 16:06:15.565836240 +0100
@@ -19,11 +19,7 @@
 	@rm -rf a.out
 
 install-data-local:
-	cd doc && $(MAKE) dat json
-	$(mkdir_p) $(DESTDIR)$(pkgdatadir)
-	$(install_sh_DATA) doc/analists/analyses.dat $(DESTDIR)$(pkgdatadir)/
-	$(install_sh_DATA) doc/analists/analyses.json $(DESTDIR)$(pkgdatadir)/
-
+	@true
 
 ## Upload tarballs to HepForge
 DEST = login.hepforge.org:rivet/downloads/
