diff --git a/include/Rivet/AnalysisHandler.hh b/include/Rivet/AnalysisHandler.hh
index 38c93cb5e..0d9ca9d1d 100644
--- a/include/Rivet/AnalysisHandler.hh
+++ b/include/Rivet/AnalysisHandler.hh
@@ -76,6 +76,8 @@ namespace Rivet {
 
     /// Set the weight names from a GenEvent
     void setWeightNames(const GenEvent& ge);
+    /// Set the weight names from vector<string>
+    void setWeightNames(vector<string> weightNames) { _weightNames = weightNames; }
 
     /// Get the index of the nominal weight-stream
     size_t defaultWeightIndex() const { return _rivetDefaultWeightIdx; }
@@ -227,7 +229,7 @@ namespace Rivet {
     //@{
 
     /// Initialize a run, with the run beams taken from the example event.
-    void init(const GenEvent& event);
+    void init(const GenEvent& event, const vector<string>& externalWeightNames = vector<string>());
 
     /// @brief Analyze the given \a event by reference.
     ///
diff --git a/src/Core/AnalysisHandler.cc b/src/Core/AnalysisHandler.cc
index cc0fa0ddb..a34c20023 100644
--- a/src/Core/AnalysisHandler.cc
+++ b/src/Core/AnalysisHandler.cc
@@ -69,7 +69,7 @@ namespace Rivet {
   }
 
 
-  void AnalysisHandler::init(const GenEvent& ge) {
+  void AnalysisHandler::init(const GenEvent& ge, const vector<string>& externalWeightNames) {
     if (_initialised)
       throw UserError("AnalysisHandler::init has already been called: cannot re-initialize!");
 
@@ -79,7 +79,12 @@ namespace Rivet {
     MSG_DEBUG("Initialising the analysis handler");
     _eventNumber = ge.event_number();
 
-    setWeightNames(ge);
+    if (externalWeightNames.empty()) {
+      setWeightNames(ge);
+    }
+    else {
+      setWeightNames(externalWeightNames);
+    }
     if (_skipWeights)
         MSG_INFO("Only using nominal weight. Variation weights will be ignored.");
     else if (haveNamedWeights())
