Index: roofit/roofitcore/src/RooSimSplitGenContext.cxx
===================================================================
--- roofit/roofitcore/src/RooSimSplitGenContext.cxx	(revision 44631)
+++ roofit/roofitcore/src/RooSimSplitGenContext.cxx	(revision 44632)
@@ -205,18 +205,18 @@
 
 
 //_____________________________________________________________________________
-RooDataSet* RooSimSplitGenContext::generate(Int_t nEvents, Bool_t skipInit, Bool_t extendedMode)
+RooDataSet* RooSimSplitGenContext::generate(Double_t nEvents, Bool_t skipInit, Bool_t extendedMode)
 {
   if(!isValid()) {
     coutE(Generation) << ClassName() << "::" << GetName() << ": context is not valid" << endl;
     return 0;
   }
-  
+
+
   // Calculate the expected number of events if necessary
   if(nEvents <= 0) {
     nEvents= _expectedEvents;
   }
-  
   coutI(Generation) << ClassName() << "::" << GetName() << ":generate: will generate "
 		    << nEvents << " events" << endl;
     
@@ -229,15 +229,15 @@
   }
 
   // Generate lookup table from expected event counts
-  vector<Int_t> nGen(_numPdf) ;
-  if (extendedMode) {
-
+  vector<Double_t> nGen(_numPdf) ;
+  if (extendedMode ) {
     _proxyIter->Reset() ;
     RooRealProxy* proxy ;
     Int_t i(0) ;
     while((proxy=(RooRealProxy*)_proxyIter->Next())) {
       RooAbsPdf* pdf=(RooAbsPdf*)proxy->absArg() ;
-      nGen[i] = Int_t(pdf->expectedEvents(&_allVarsPdf)+0.5) ;
+      //nGen[i] = Int_t(pdf->expectedEvents(&_allVarsPdf)+0.5) ;
+      nGen[i] = pdf->expectedEvents(&_allVarsPdf) ;
       i++ ;
     }     
     
@@ -269,6 +269,7 @@
       }
     }
   }
+
     
 
   // Now loop over states
Index: roofit/roofitcore/src/RooAbsGenContext.cxx
===================================================================
--- roofit/roofitcore/src/RooAbsGenContext.cxx	(revision 44631)
+++ roofit/roofitcore/src/RooAbsGenContext.cxx	(revision 44632)
@@ -135,7 +135,7 @@
 
 
 //_____________________________________________________________________________
-RooDataSet *RooAbsGenContext::generate(Int_t nEvents, Bool_t skipInit, Bool_t extendedMode) 
+RooDataSet *RooAbsGenContext::generate(Double_t nEvents, Bool_t skipInit, Bool_t extendedMode) 
 {
   // Generate the specified number of events with nEvents>0 and
   // and return a dataset containing the generated events. With nEvents<=0,
Index: roofit/roofitcore/src/RooBinnedGenContext.cxx
===================================================================
--- roofit/roofitcore/src/RooBinnedGenContext.cxx	(revision 44631)
+++ roofit/roofitcore/src/RooBinnedGenContext.cxx	(revision 44632)
@@ -131,9 +131,10 @@
 
 
 //_____________________________________________________________________________
-RooDataSet *RooBinnedGenContext::generate(Int_t nEvt, Bool_t /*skipInit*/, Bool_t extended)
+RooDataSet *RooBinnedGenContext::generate(Double_t nEvt, Bool_t /*skipInit*/, Bool_t extended)
 {
   
+
   // Scale to number of events and introduce Poisson fluctuations
   _hist->reset() ;
 
@@ -146,7 +147,7 @@
       return 0 ;
     } else {
       // Don't round in expectedData mode
-      if (_expectedData) {
+      if (_expectedData || extended) {
 	nEvents = _pdf->expectedEvents(_vars) ;
       } else {
 	nEvents = Int_t(_pdf->expectedEvents(_vars)+0.5) ;
Index: roofit/roofitcore/src/RooAbsPdf.cxx
===================================================================
--- roofit/roofitcore/src/RooAbsPdf.cxx	(revision 44631)
+++ roofit/roofitcore/src/RooAbsPdf.cxx	(revision 44632)
@@ -1684,7 +1684,7 @@
   if (protoData) {
     data = generate(whatVars,*protoData,nEvents,verbose,randProto,resampleProto) ;
   } else {
-    data = generate(whatVars,nEvents,verbose,autoBinned,binnedTag,expectedData) ;
+     data = generate(whatVars,nEvents,verbose,autoBinned,binnedTag,expectedData, extended) ;
   }
 
   // Rename dataset to given name if supplied
@@ -1811,7 +1811,7 @@
 
 
 //_____________________________________________________________________________
-RooDataSet *RooAbsPdf::generate(const RooArgSet &whatVars, Int_t nEvents, Bool_t verbose, Bool_t autoBinned, const char* binnedTag, Bool_t expectedData) const 
+RooDataSet *RooAbsPdf::generate(const RooArgSet &whatVars, Int_t nEvents, Bool_t verbose, Bool_t autoBinned, const char* binnedTag, Bool_t expectedData, Bool_t extended) const 
 {
   // Generate a new dataset containing the specified variables with
   // events sampled from our distribution. Generate the specified
@@ -1833,7 +1833,7 @@
   
   RooDataSet *generated = 0;
   if(0 != context && context->isValid()) {
-    generated= context->generate(nEvents);
+     generated= context->generate(nEvents, kFALSE, extended);
   }
   else {
     coutE(Generation)  << "RooAbsPdf::generate(" << GetName() << ") cannot create a valid context" << endl;
@@ -2087,7 +2087,8 @@
   const char* dsetName = pc.getString("dsetName") ;
 
   if (extended) {
-    nEvents = (nEvents==0?Int_t(expectedEvents(&whatVars)+0.5):nEvents) ;
+     //nEvents = (nEvents==0?Int_t(expectedEvents(&whatVars)+0.5):nEvents) ;
+    nEvents = (nEvents==0 ? expectedEvents(&whatVars) :nEvents) ;
     cxcoutI(Generation) << " Extended mode active, number of events generated (" << nEvents << ") is Poisson fluctuation on " 
 			<< GetName() << "::expectedEvents() = " << nEvents << endl ;
     // If Poisson fluctuation results in zero events, stop here
@@ -2138,8 +2139,9 @@
       delete hist ;
       return 0 ;
     } else {
-      // Don't round in expectedData mode
-      if (expectedData) {
+
+      // Don't round in expectedData or extended mode
+      if (expectedData || extended) {
 	nEvents = expectedEvents(&whatVars) ;
       } else {
 	nEvents = Int_t(expectedEvents(&whatVars)+0.5) ;

Property changes on: roofit/roofitcore/inc/LinkDef3.h
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/roofit/roofitcore/inc/LinkDef3.h:r42432,43113,44630

Index: roofit/roofitcore/inc/RooAbsPdf.h
===================================================================
--- roofit/roofitcore/inc/RooAbsPdf.h	(revision 44631)
+++ roofit/roofitcore/inc/RooAbsPdf.h	(revision 44632)
@@ -57,7 +57,7 @@
                        const RooCmdArg& arg3=RooCmdArg::none(),const RooCmdArg& arg4=RooCmdArg::none(),
                        const RooCmdArg& arg5=RooCmdArg::none(),const RooCmdArg& arg6=RooCmdArg::none()) ;
   RooDataSet *generate(const RooArgSet &whatVars, Int_t nEvents = 0, Bool_t verbose=kFALSE, Bool_t autoBinned=kTRUE, 
-		       const char* binnedTag="", Bool_t expectedData=kFALSE) const;
+		       const char* binnedTag="", Bool_t expectedData=kFALSE, Bool_t extended = kFALSE) const;
   RooDataSet *generate(const RooArgSet &whatVars, const RooDataSet &prototype, Int_t nEvents= 0,
 		       Bool_t verbose=kFALSE, Bool_t randProtoOrder=kFALSE, Bool_t resampleProto=kFALSE) const;
 
Index: roofit/roofitcore/inc/RooBinnedGenContext.h
===================================================================
--- roofit/roofitcore/inc/RooBinnedGenContext.h	(revision 44631)
+++ roofit/roofitcore/inc/RooBinnedGenContext.h	(revision 44632)
@@ -33,7 +33,7 @@
                    const RooArgSet* auxProto=0, Bool_t _verbose= kFALSE);
   virtual ~RooBinnedGenContext();
   
-  RooDataSet* generate(Int_t nEvents=0, Bool_t skipInit=kFALSE, Bool_t extendedMode=kFALSE) ;
+  RooDataSet* generate(Double_t nEvents=0, Bool_t skipInit=kFALSE, Bool_t extendedMode=kFALSE) ;
 
   virtual void setProtoDataOrder(Int_t*)  {}
 
Index: roofit/roofitcore/inc/RooAbsGenContext.h
===================================================================
--- roofit/roofitcore/inc/RooAbsGenContext.h	(revision 44631)
+++ roofit/roofitcore/inc/RooAbsGenContext.h	(revision 44632)
@@ -29,7 +29,7 @@
 		   Bool_t _verbose= kFALSE) ;
   virtual ~RooAbsGenContext();
   
-  virtual RooDataSet *generate(Int_t nEvents= 0, Bool_t skipInit=kFALSE, Bool_t extendedMode=kFALSE);
+  virtual RooDataSet *generate(Double_t nEvents= 0, Bool_t skipInit=kFALSE, Bool_t extendedMode=kFALSE);
 
   Bool_t isValid() const { 
     // If true generator context is in a valid state
Index: roofit/roofitcore/inc/RooSimSplitGenContext.h
===================================================================
--- roofit/roofitcore/inc/RooSimSplitGenContext.h	(revision 44631)
+++ roofit/roofitcore/inc/RooSimSplitGenContext.h	(revision 44632)
@@ -35,7 +35,7 @@
 
   virtual void printMultiline(ostream &os, Int_t content, Bool_t verbose=kFALSE, TString indent="") const ;
 
-  virtual RooDataSet *generate(Int_t nEvents= 0, Bool_t skipInit=kFALSE, Bool_t extendedMode=kFALSE);
+  virtual RooDataSet *generate(Double_t nEvents= 0, Bool_t skipInit=kFALSE, Bool_t extendedMode=kFALSE);
 
   virtual void setExpectedData(Bool_t) ;
 

Property changes on: roofit/roofitcore
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/roofit/roofitcore:r42432,43113,44630

