diff --git a/roofit/roostats/src/HypoTestInverterResult.cxx b/roofit/roostats/src/HypoTestInverterResult.cxx
index 2859430..1d34862 100644
--- a/roofit/roostats/src/HypoTestInverterResult.cxx
+++ b/roofit/roostats/src/HypoTestInverterResult.cxx
@@ -332,7 +332,7 @@ double HypoTestInverterResult::GetGraphX(const TGraph & graph, double y0, bool l
    bool ret = brf.Solve(100, 1.E-6, 1.E-6);
    if (!ret) { 
       ooccoutE(this,Eval) << "HypoTestInverterResult - interpolation failed - return inf" << std::endl;
-         return TMath::Infinity();
+         return std::numeric_limits<Double_t>::infinity();
    }
    double limit =  brf.Root();
 
