--- orig.root/core/meta/src/TClass.cxx	2009-02-12 19:03:18.000000000 +0100
+++ root/core/meta/src/TClass.cxx	2009-03-04 17:29:20.000000000 +0100
@@ -1536,30 +1536,40 @@
 //______________________________________________________________________________
 TObject *TClass::Clone(const char *new_name) const
 {
+   // Create a Clone of this TClass object, in particular using the same 'dictionary'.
+   // This effectively creates an alias for the class name
+
    if (new_name == 0 || new_name[0]=='\0' || fName == new_name) {
       Error("Clone","The name of the class must be changed when cloning a TClass object.");
       return 0;
    }
+   // Temporarily remove the original from the list of classes.
+   TClass::RemoveClass(const_cast<TClass*>(this));
+ 
    TClass *copy;
    if (fTypeInfo) {
-      copy = new TClass(new_name,
-                             fClassVersion,
-                             *fTypeInfo,
-                             new TIsAProxy(*fTypeInfo),
-                             fShowMembers,
-                             GetDeclFileName(),
-                             GetImplFileName(),
-                             GetDeclFileLine(),
-                             GetImplFileLine());
+      copy = new TClass(GetName(),
+                        fClassVersion,
+                        *fTypeInfo,
+                        new TIsAProxy(*fTypeInfo),
+                        fShowMembers,
+                        GetDeclFileName(),
+                        GetImplFileName(),
+                        GetDeclFileLine(),
+                        GetImplFileLine());
    } else {
-      copy = new TClass(new_name,
+      copy = new TClass(GetName(),
                         fClassVersion,
                         GetDeclFileName(),
                         GetImplFileName(),
                         GetDeclFileLine(),
                         GetImplFileLine());
       copy->fShowMembers = fShowMembers;
-   }      
+   }
+   // Remove the copy before renaming it
+   TClass::RemoveClass(copy);
+   copy->SetName(new_name);
+   TClass::AddClass(copy);
 
    copy->SetNew(fNew);
    copy->SetNewArray(fNewArray);
@@ -1579,6 +1589,7 @@
    if (fRefProxy) {
       copy->AdoptReferenceProxy( fRefProxy->Clone() );
    }
+   TClass::AddClass(const_cast<TClass*>(this));
    return copy;
 }   
 
