diff --git a/interpreter/llvm/src/include/llvm/ADT/IntrusiveRefCntPtr.h b/interpreter/llvm/src/include/llvm/ADT/IntrusiveRefCntPtr.h
index f9df378..c859c98 100644
--- a/interpreter/llvm/src/include/llvm/ADT/IntrusiveRefCntPtr.h
+++ b/interpreter/llvm/src/include/llvm/ADT/IntrusiveRefCntPtr.h
@@ -197,6 +197,9 @@ public:
   private:
     void retain() { if (Obj) IntrusiveRefCntPtrInfo<T>::retain(Obj); }
     void release() { if (Obj) IntrusiveRefCntPtrInfo<T>::release(Obj); }
+
+    template <typename X>
+    friend class IntrusiveRefCntPtr;
   };
 
   template<class T, class U>
