diff -Naur orig.root/cint/cint/src/struct.cxx root/cint/cint/src/struct.cxx
--- orig.root/cint/cint/src/struct.cxx	2010-11-05 15:46:29.000000000 +0100
+++ root/cint/cint/src/struct.cxx	2010-12-13 23:50:54.000000000 +0100
@@ -1959,7 +1959,8 @@
       if (G__struct.type[i] != 'a'
           && G__struct.type[i] != 0
           && G__UserSpecificUpdateClassInfo) {
-         (*G__UserSpecificUpdateClassInfo)(G__struct.name[i],i);
+         G__FastAllocString fullname(G__fulltagname(i,0));
+         (*G__UserSpecificUpdateClassInfo)(fullname,i);
       }
    }
    else if (G__struct.type[i]==0 || (G__struct.type[i] == 'a')) {
@@ -1968,7 +1969,8 @@
          if (G__struct.type[i] != 'a'
              && G__struct.type[i] != 0
              && G__UserSpecificUpdateClassInfo) {
-            (*G__UserSpecificUpdateClassInfo)(G__struct.name[i],i);
+            G__FastAllocString fullname(G__fulltagname(i,0));
+            (*G__UserSpecificUpdateClassInfo)(fullname,i);
          }
       }
       ++G__struct.nactives;
