Index: root/configure
===================================================================
--- root/configure	(revision 41601)
+++ root/configure	(revision 41602)
@@ -4321,6 +4321,7 @@
    fi
 fi
 bjrcomm=
+hasxrdutils="no"
 if test "x$enable_xrootd" = "xyes" && test "x$externalxrd" = "xyes"; then
 
    if test "x$enable_xrootd" = "xyes" ; then
@@ -4330,8 +4331,13 @@
       decreftwo=20070723
       decrefthree=20071004
       decreffour=300000002
+      decreffive=300010000
       if [ $decver -gt $decrefone ] ; then
-         if [ $decver -gt $decreffour ] ; then
+         if [ $decver -ge $decreffive ] ; then
+            vercomm=" (>=$decreffive: setting -DHAVEXRDSYSDNS)"
+            extraxrdflags="-DHAVEXRDSYSDNS $extraxrdflags"
+            hasxrdutils="yes"
+         elif [ $decver -gt $decreffour ] ; then
             xrdcomm="$xrdcomm"
             extraxrdflags="$extraxrdflags"
          elif [ $decver -gt $decrefthree ] ; then
@@ -6542,6 +6548,7 @@
     -e "s|@xrdlibdir@|$xrdlibdir|"              \
     -e "s|@xrdincdir@|$xrdincdir|"              \
     -e "s|@xrdversion@|$decver|"                \
+    -e "s|@hasxrdutils@|$hasxrdutils|"          \
     -e "s|@cfitsioincdir@|$cfitsioincdir|"      \
     -e "s|@cfitsiolib@|$cfitsiolib|"            \
     -e "s|@cfitsiolibdir@|$cfitsiolibdir|"      \
Index: root/proof/proofd/src/XrdProofConn.cxx
===================================================================
--- root/proof/proofd/src/XrdProofConn.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofConn.cxx	(revision 41602)
@@ -41,7 +41,7 @@
 #include "XrdClient/XrdClientPhyConnection.hh"
 #include "XrdClient/XrdClientMessage.hh"
 #include "XrdClient/XrdClientUrlInfo.hh"
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XrdOuc/XrdOucErrInfo.hh"
 #include "XrdOuc/XrdOucString.hh"
 #include "XrdSec/XrdSecInterface.hh"
@@ -342,7 +342,7 @@
 
    // Resolve the DNS information
    char *haddr[10] = {0}, *hname[10] = {0};
-   int naddr = XrdNetDNS::getAddrName(fUrl.Host.c_str(), 10, haddr, hname);
+   int naddr = XrdSysDNS::getAddrName(fUrl.Host.c_str(), 10, haddr, hname);
 
    int i = 0;
    for (; i < naddr; i++ ) {
@@ -1240,7 +1240,7 @@
    // for the authentication.
    struct sockaddr_in netaddr;
    char **hosterrmsg = 0;
-   if (XrdNetDNS::getHostAddr((char *)fUrl.HostAddr.c_str(),
+   if (XrdSysDNS::getHostAddr((char *)fUrl.HostAddr.c_str(),
                                 (struct sockaddr &)netaddr, hosterrmsg) <= 0) {
       TRACE(XERR, "getHostAddr: "<< *hosterrmsg);
       return protocol;
Index: root/proof/proofd/src/XrdProofdProofServMgr.cxx
===================================================================
--- root/proof/proofd/src/XrdProofdProofServMgr.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofdProofServMgr.cxx	(revision 41602)
@@ -32,7 +32,7 @@
 #include "Xrd/XrdPoll.hh"
 #include "Xrd/XrdScheduler.hh"
 #include "XrdNet/XrdNet.hh"
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XrdNet/XrdNetPeer.hh"
 #include "XrdOuc/XrdOucRash.hh"
 #include "XrdOuc/XrdOucStream.hh"
@@ -2820,7 +2820,7 @@
 
    // Make sure we have the full host name
    if (peerpsrv.InetName) free(peerpsrv.InetName);
-   peerpsrv.InetName = XrdNetDNS::getHostName("localhost");
+   peerpsrv.InetName = XrdSysDNS::getHostName("localhost");
 
    // Allocate a new network object
    if (!(linkpsrv = XrdLink::Alloc(peerpsrv, lnkopts))) {
Index: root/proof/proofd/src/XrdProofdNetMgr.cxx
===================================================================
--- root/proof/proofd/src/XrdProofdNetMgr.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofdNetMgr.cxx	(revision 41602)
@@ -27,7 +27,7 @@
 #include "XrdClient/XrdClientEnv.hh"
 #include "XrdClient/XrdClientMessage.hh"
 #include "XrdClient/XrdClientUrlInfo.hh"
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XrdOuc/XrdOucStream.hh"
 #include "XrdSys/XrdSysPlatform.hh"
 
@@ -77,7 +77,7 @@
    fRequestTO = 30;
    fBonjourEnabled = false;
 #if defined(BUILD_BONJOUR)
-   char *host = XrdNetDNS::getHostName();
+   char *host = XrdSysDNS::getHostName();
    fBonjourName = host ? host : "";
    SafeFree(host);
    fBonjourCores = XrdProofdAux::GetNumCPUs();
@@ -1042,7 +1042,7 @@
       XrdClientUrlInfo uu(host);
       if (uu.Port <= 0) uu.Port = 1093;
       // Fully qualified name
-      char *fqn = XrdNetDNS::getHostName(uu.Host.c_str());
+      char *fqn = XrdSysDNS::getHostName(uu.Host.c_str());
       if (fqn && (strstr(fqn, "localhost") || !strcmp(fqn, "127.0.0.1") ||
                   !strcmp(fMgr->Host(), fqn))) {
          if (!checkport || (uu.Port == fMgr->Port()))
Index: root/proof/proofd/src/XrdProofdProtocol.cxx
===================================================================
--- root/proof/proofd/src/XrdProofdProtocol.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofdProtocol.cxx	(revision 41602)
@@ -33,7 +33,7 @@
 
 #include "XrdVersion.hh"
 #include "Xrd/XrdBuffer.hh"
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 
 #include "XrdProofdClient.h"
 #include "XrdProofdClientMgr.h"
Index: root/proof/proofd/src/XrdProofWorker.cxx
===================================================================
--- root/proof/proofd/src/XrdProofWorker.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofWorker.cxx	(revision 41602)
@@ -27,7 +27,7 @@
 #include "XrdProofWorker.h"
 #include "XrdProofdProofServ.h"
 #include "XrdClient/XrdClientUrlInfo.hh"
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XProofProtocol.h"
 
 // Tracing utilities
@@ -99,7 +99,7 @@
    // Take the user name, if specified
    fUser = ui.User;
    char *err;
-   char *fullHostName = XrdNetDNS::getHostName((char *)ui.Host.c_str(), &err);
+   char *fullHostName = XrdSysDNS::getHostName((char *)ui.Host.c_str(), &err);
    if (!fullHostName || !strcmp(fullHostName, "0.0.0.0")) {
       TRACE(XERR, "DNS could not resolve '" << ui.Host << "'");
       return;
Index: root/proof/proofd/src/XrdProofdConfig.cxx
===================================================================
--- root/proof/proofd/src/XrdProofdConfig.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofdConfig.cxx	(revision 41602)
@@ -28,7 +28,7 @@
 #  include "XrdSys/XrdSysError.hh"
 #  include "XrdSys/XrdSysLogger.hh"
 #endif
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XrdOuc/XrdOucEnv.hh"
 #include "XrdOuc/XrdOucStream.hh"
 #include "XrdOuc/XrdOucString.hh"
@@ -115,7 +115,7 @@
 
    // Local FQDN
    if (fgHost.length() <= 0) {
-      char *host = XrdNetDNS::getHostName();
+      char *host = XrdSysDNS::getHostName();
       fgHost = host ? host : "";
       SafeFree(host);
    }
Index: root/proof/proofd/src/XrdProofdManager.cxx
===================================================================
--- root/proof/proofd/src/XrdProofdManager.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofdManager.cxx	(revision 41602)
@@ -32,7 +32,7 @@
 #  include "XrdSys/XrdSysPlugin.hh"
 #  include "XrdSys/XrdSysTimer.hh"
 #endif
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XrdOuc/XrdOucEnv.hh"
 #include "XrdOuc/XrdOucStream.hh"
 #include "XrdSys/XrdSysPriv.hh"
@@ -668,7 +668,7 @@
       }
 
       // Local FQDN
-      char *host = XrdNetDNS::getHostName();
+      char *host = XrdSysDNS::getHostName();
       fHost = host ? host : "";
       SafeFree(host);
 
Index: root/proof/proofd/src/XrdProofPhyConn.cxx
===================================================================
--- root/proof/proofd/src/XrdProofPhyConn.cxx	(revision 41601)
+++ root/proof/proofd/src/XrdProofPhyConn.cxx	(revision 41602)
@@ -27,7 +27,7 @@
 #include "XrdClient/XrdClientConst.hh"
 #include "XrdClient/XrdClientLogConnection.hh"
 #include "XrdClient/XrdClientMessage.hh"
-#include "XrdNet/XrdNetDNS.hh"
+#include "XpdSysDNS.h"
 #include "XrdSec/XrdSecInterface.hh"
 
 #ifndef WIN32
@@ -91,7 +91,7 @@
 
    // Host and Port
    if (!fTcp) {
-      fHost = XrdNetDNS::getHostName(((fUrl.Host.length() > 0) ?
+      fHost = XrdSysDNS::getHostName(((fUrl.Host.length() > 0) ?
                                        fUrl.Host.c_str() : "localhost"));
       fPort = -1;
       fUrl.Host = "";
Index: root/proof/proofd/inc/XpdSysDNS.h
===================================================================
--- root/proof/proofd/inc/XpdSysDNS.h	(revision 0)
+++ root/proof/proofd/inc/XpdSysDNS.h	(revision 41602)
@@ -0,0 +1,13 @@
+
+//
+// These are needed to handle backward compatibility
+//
+
+#ifndef HAVEXRDSYSDNS
+#define XrdSysDNS           XrdNetDNS
+#include "XrdNet/XrdNetDNS.hh"
+#else
+#include "XrdSys/XrdSysDNS.hh"
+#endif
+
+

Property changes on: root/proof/proofd/inc/XpdSysDNS.h
___________________________________________________________________
Added: svn:eol-style
   + LF
Added: svn:keywords
   + Id

Index: root/proof/proofd/Module.mk
===================================================================
--- root/proof/proofd/Module.mk	(revision 41601)
+++ root/proof/proofd/Module.mk	(revision 41602)
@@ -123,10 +123,12 @@
 # Extra include paths and libs
 XPROOFDEXELIBS :=
 XPROOFDEXESYSLIBS := $(DNSSDLIB)
-XPROOFDEXE     :=
+XPROOFDEXE     := bin/xproofd
 ifeq ($(HASXRD),yes)
 XPDINCEXTRA    := $(XROOTDDIRI:%=-I%)
 XPDINCEXTRA    += $(PROOFDDIRI:%=-I%)
+ifeq ($(HASXRDUTILS),no)
+
 XPDLIBEXTRA    += -L$(XROOTDDIRL) -lXrdOuc -lXrdNet -lXrdSys \
                   -lXrdClient -lXrdSut $(DNSSDLIB)
 XPROOFDEXELIBS := $(XROOTDDIRL)/libXrd.a $(XROOTDDIRL)/libXrdClient.a \
@@ -145,12 +147,23 @@
 XPDLIBEXTRA    += -L$(XROOTDDIRL) -lXrdNetUtil
 XPROOFDEXELIBS += $(XROOTDDIRL)/libXrdNetUtil.a
 endif
+
+else
+
+XPDLIBEXTRA    := -L$(XROOTDDIRL) -lXrdClient -lXrdUtils
+ifeq ($(PLATFORM),macosx)
+XPROOFDEXELIBS := $(XROOTDDIRL)/libXrdMain.dylib $(XROOTDDIRL)/libXrdClient.dylib  $(XROOTDDIRL)/libXrdUtils.dylib
+else
+XPROOFDEXELIBS := $(XROOTDDIRL)/libXrdMain.$(SOEXT) $(XROOTDDIRL)/libXrdClient.$(SOEXT)  $(XROOTDDIRL)/libXrdUtils.$(SOEXT)
+endif
+
+endif
 XPDLIBEXTRA    +=  $(DNSSDLIB)
+XPROOFDEXELIBS +=  $(DNSSDLIB)
 
 ifeq ($(PLATFORM),solaris)
 XPROOFDEXESYSLIBS := -lsendfile
 endif
-XPROOFDEXE     := bin/xproofd
 endif
 
 # used in the main Makefile
@@ -181,7 +194,7 @@
 $(XPROOFDEXE):  $(XPDO) $(XPROOFDEXELIBS) $(XRDPROOFXD) $(RPDCONNO)
 		$(LD) $(LDFLAGS) -o $@ $(XPDO) $(RPDCONNO) $(XPROOFDEXELIBS) $(SYSLIBS) $(XPROOFDEXESYSLIBS)
 
-$(XPDLIB):      $(XPDO) $(XPDH) $(ORDER_) $(MAINLIBS) $(XRDPROOFXD) $(RPDCONNO)
+$(XPDLIB):      $(XPDO) $(XPDH) $(ORDER_) $(XPROOFDEXELIBS) $(MAINLIBS) $(XRDPROOFXD) $(RPDCONNO)
 		@$(MAKELIB) $(PLATFORM) $(LD) "$(LDFLAGS)" \
 		   "$(SOFLAGS)" libXrdProofd.$(SOEXT) $@ "$(XPDO) $(RPDCONNO)" \
 		   "$(XPDLIBEXTRA)"
@@ -224,6 +237,16 @@
 
 endif
 
+ifeq ($(PLATFORM),macosx)
+ifeq ($(HASXRDUTILS),no)
+$(XPDLIB): SOFLAGS := -undefined dynamic_lookup $(SOFLAGS)
+endif
+endif
+ifeq ($(PLATFORM),linux)
+comma := ,
+$(XPDLIB): LDFLAGS := $(subst -Wl$(comma)--no-undefined,,$(LDFLAGS))
+endif
+
 $(PROOFEXECVO): $(RPDCONNO) $(RPDPRIVO)
 
 endif
Index: root/proof/proofx/Module.mk
===================================================================
--- root/proof/proofx/Module.mk	(revision 41601)
+++ root/proof/proofx/Module.mk	(revision 41602)
@@ -66,6 +66,7 @@
 ifeq ($(PLATFORM),win32)
 PROOFXLIBEXTRA += $(XROOTDDIRL)/libXrdClient.lib
 else
+ifeq ($(HASXRDUTILS),no)
 PROOFXLIBEXTRA += -L$(XROOTDDIRL) -lXrdOuc -lXrdSys -lXrdNet -lXrdClient
 # Starting from Jul 2010 XrdNet has been split in two libs: XrdNet and XrdNetUtil;
 # both are needed
@@ -78,7 +79,10 @@
 ifeq ($(XRDNETUTIL),yes)
 PROOFXLIBEXTRA += -lXrdNetUtil
 endif
+else
+PROOFXLIBEXTRA += -L$(XROOTDDIRL) -lXrdUtils -lXrdClient
 endif
+endif
 
 ##### local rules #####
 .PHONY:         all-$(MODNAME) clean-$(MODNAME) distclean-$(MODNAME)
Index: root/net/netx/Module.mk
===================================================================
--- root/net/netx/Module.mk	(revision 41601)
+++ root/net/netx/Module.mk	(revision 41602)
@@ -58,9 +58,13 @@
 ifeq ($(PLATFORM),win32)
 NETXLIBEXTRA += $(XROOTDDIRL)/libXrdClient.lib
 else
+ifeq ($(HASXRDUTILS),no)
 NETXLIBEXTRA += -L$(XROOTDDIRL) -lXrdOuc -lXrdSys \
                 -lXrdClient
+else
+NETXLIBEXTRA += -L$(XROOTDDIRL) -lXrdUtils -lXrdClient
 endif
+endif
 
 ##### local rules #####
 .PHONY:         all-$(MODNAME) clean-$(MODNAME) distclean-$(MODNAME)
Index: root/config/Makefile.in
===================================================================
--- root/config/Makefile.in	(revision 41601)
+++ root/config/Makefile.in	(revision 41602)
@@ -252,6 +252,7 @@
 XRDADDOPTS     := @xrdaddopts@
 EXTRA_XRDFLAGS := @extraxrdflags@
 XRDVERSION     := @xrdversion@
+HASXRDUTILS    := @hasxrdutils@
 
 SRPLIBDIR      := @srplibdir@
 SRPLIB         := @srplib@
