diff --git a/configure b/configure
index 769ee8a..032ac0b 100755
--- a/configure
+++ b/configure
@@ -2634,7 +2634,7 @@ result "$enable_builtin_pcre"
 #
 if test "x$enable_builtin_zlib" = "xno" ; then
     check_header "zlib.h" "" \
-        $ZLIB ${ZLIB:+$ZLIB/include} \
+        $ZLIB_ROOT/include $ZLIB ${ZLIB:+$ZLIB/include} \
         ${finkdir:+$finkdir/include} \
         /usr/local/include /usr/include/zlib /usr/local/include/zlib \
         /opt/zlib/include /usr/include
@@ -2646,7 +2646,7 @@ if test "x$enable_builtin_zlib" = "xno" ; then
     fi
 
     check_library "libz" "$enable_shared" "" \
-        $ZLIB ${ZLIB:+$ZLIB/lib} \
+        $ZLIB_ROOT/lib $ZLIB ${ZLIB:+$ZLIB/lib} \
         /usr/local/zlib/lib /usr/local/lib \
         /usr/lib/zlib /usr/local/lib/zlib /usr/zlib/lib /usr/lib \
         /usr/zlib /usr/local/zlib /opt/zlib /opt/zlib/lib
@@ -4084,12 +4084,12 @@ if test ! "x$enable_asimage" = "xno" ; then
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" \
-          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+          $LIBJPG_ROOT/include $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
           ${finkdir:+$finkdir/include} \
           /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" \
-          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+          $LIBPNG_ROOT/include $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
           ${ASPNG:+$ASPNG/include} \
           ${finkdir:+$finkdir/include} \
           /usr/local/include /usr/X11/include /usr/include \
@@ -4097,7 +4097,7 @@ if test ! "x$enable_asimage" = "xno" ; then
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" \
-               $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+               $LIBTIFF_ROOT/include $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
                ${finkdir:+$finkdir/include} \
                /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
@@ -4113,11 +4113,14 @@ if test ! "x$enable_asimage" = "xno" ; then
        # add libpng which needs libz
        if test ! "x$enable_astiff" = "xno" ; then
            aslibs="libjpeg libtiff libz libpng"
+           extra_libtiff_path=$LIBTIFF_ROOT/lib
        else
            aslibs="libjpeg libz libpng"
+           extra_libtiff_path=
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
+               $extra_libtiff_path $LIBJPG_ROOT/lib $LIBPNG_ROOT/lib $ZLIB_ROOT/lib \
                $ASIMAGE ${ASIMAGE:+$ASIMAGE/lib} ${ASPNG:+$ASPNG/lib} \
                ${finkdir:+$finkdir/lib} \
                /usr/local/lib /usr/X11/lib /usr/lib /opt/lib
