--- ./math/mathcore/inc/TMath.h 2011-10-05 11:01:30.000000000 +0200
+++ ./math/mathcore/inc/TMath.h	2011-10-05 11:04:29.000000000 +0200
@@ -356,7 +356,7 @@
 #endif
 #if defined(R__AIX) || defined(R__SOLARIS_CC50) || \
     defined(R__HPUX11) || defined(R__GLIBC) || \
-    (defined(R__MACOSX) && (defined(__INTEL_COMPILER) || defined(__arm__)))
+    defined(R__MACOSX)
 // math functions are defined inline so we have to include them here
 #   include <math.h>
 #   ifdef R__SOLARIS_CC50
@@ -479,7 +479,7 @@
 inline Int_t TMath::Finite(Double_t x)
 #if defined(R__HPUX11)
    { return isfinite(x); }
-#elif defined(R__MACOSX) && defined(__arm__)
+#elif defined(R__MACOSX)
 #ifdef isfinite
    // from math.h
    { return isfinite(x); }
@@ -492,7 +492,7 @@
 #endif
 
 inline Int_t TMath::IsNaN(Double_t x)
-#if defined(R__MACOSX) && defined(__arm__)
+#if defined(R__MACOSX)
 #ifdef isnan
    // from math.h
    { return isnan(x); }
