--- branches/v5-30-00-patches/core/meta/src/TClass.cxx	2011/09/07 16:58:28	40830
+++ branches/v5-30-00-patches/core/meta/src/TClass.cxx	2011/09/07 20:18:17	40831
@@ -1543,6 +1543,10 @@
       return;
    }
 
+   if (fClassVersion == 0) {
+      isTransient = kTRUE;
+   }
+
    // When called via TMapFile (e.g. Update()) make sure that the dictionary
    // gets allocated on the heap and not in the mapped file.
    TMmallocDescTemp setreset;
