Index: root/core/unix/src/TUnixSystem.cxx
===================================================================
--- root/core/unix/src/TUnixSystem.cxx       (revision 39195)
+++ root/core/unix/src/TUnixSystem.cxx       (revision 39196)
@@ -4343,6 +4343,9 @@ static const char *DynamicPath(const char *newpath = 0, Bool_t reset = kFALSE)
       if (!ldpath.IsNull())
          ldpath += ":";
       ldpath += gSystem->Getenv("LD_LIBRARY_PATH");
+      if (!ldpath.IsNull())
+         ldpath += ":";
+      ldpath += gSystem->Getenv("DYLD_FALLBACK_LIBRARY_PATH");
 #else
       ldpath = gSystem->Getenv("LD_LIBRARY_PATH");
 #endif
@@ -4440,11 +4440,12 @@
    char *name;
 
    int ext = 0, len = strlen(lib);
-   if (len > 3 && (!strcmp(lib+len-3, ".sl") ||
-                   !strcmp(lib+len-3, ".dl") ||
-                   !strcmp(lib+len-4, ".dll")||
-                   !strcmp(lib+len-4, ".DLL")||
-                   !strcmp(lib+len-3, ".so") ||
+   if (len > 3 && (!strcmp(lib+len-3, ".so")    ||
+                   !strcmp(lib+len-3, ".dl")    ||
+                   !strcmp(lib+len-4, ".dll")   ||
+                   !strcmp(lib+len-4, ".DLL")   ||
+                   !strcmp(lib+len-6, ".dylib") ||
+                   !strcmp(lib+len-3, ".sl")    ||
                    !strcmp(lib+len-2, ".a"))) {
       name = gSystem->Which(GetDynamicPath(), lib, kReadPermission);
       ext  = 1;
@@ -4456,14 +4457,18 @@
          fname.Form("%s.dll", lib);
          name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
          if (!name) {
-            fname.Form("%s.sl", lib);
+            fname.Form("%s.dylib", lib);
             name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
             if (!name) {
-               fname.Form("%s.dl", lib);
+               fname.Form("%s.sl", lib);
                name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
                if (!name) {
-                  fname.Form("%s.a", lib);
+                  fname.Form("%s.dl", lib);
                   name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
+                  if (!name) {
+                     fname.Form("%s.a", lib);
+                     name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
+                  }
                }
             }
          }
@@ -4476,7 +4481,7 @@
                "%s does not exist in %s", lib, GetDynamicPath());
       else
          Error("DynamicPathName",
-               "%s[.so | .sl | .dl | .a | .dll] does not exist in %s", lib, GetDynamicPath());
+               "%s[.so | .dll | .dylib | .sl | .dl | .a] does not exist in %s", lib, GetDynamicPath());
    }
 
    return name;
Index: root/core/meta/src/TCint.cxx
===================================================================
--- root/core/meta/src/TCint.cxx	(revision 39195)
+++ root/core/meta/src/TCint.cxx	(revision 39196)
@@ -2116,11 +2116,12 @@
             needToSkip = (!strncmp(basename, excludelist[i], excludelen[i]));
       }
       if (!needToSkip &&
-           ((len>3 && strcmp(end-2,".a") == 0)    ||
-            (len>4 && (strcmp(end-3,".sl") == 0   ||
+           ((len>2 && strcmp(end-2,".a") == 0)    ||
+            (len>3 && (strcmp(end-3,".sl") == 0   ||
                        strcmp(end-3,".dl") == 0   ||
                        strcmp(end-3,".so") == 0)) ||
-            (len>5 && (strcasecmp(end-4,".dll") == 0)))) {
+            (len>4 && (strcasecmp(end-4,".dll") == 0)) ||
+            (len>6 && (strcasecmp(end-6,".dylib") == 0)))) {
          if (!fSharedLibs.IsNull())
             fSharedLibs.Append(" ");
          fSharedLibs.Append(filename);
Index: root/cint/cint/src/shl.cxx
===================================================================
--- root/cint/cint/src/shl.cxx	(revision 39195)
+++ root/cint/cint/src/shl.cxx	(revision 39196)
@@ -183,6 +183,7 @@
                strcmp(filename+len-3,".so")==0)) ||
      (len>4&& (strcmp(filename+len-4,".dll")==0 ||
                strcmp(filename+len-4,".DLL")==0)) ||
+     (len>6&& (strcmp(filename+len-6,".dylib")==0)) ||
 #if defined(R__FBSD) || defined(R__OBSD)
      (len>strlen(soext) && strcmp(filename+len-strlen(soext), soext)==0) ||
 #endif
Index: root/cint/cint/src/loadfile.cxx
===================================================================
--- root/cint/cint/src/loadfile.cxx	(revision 39195)
+++ root/cint/cint/src/loadfile.cxx	(revision 39196)
@@ -1218,6 +1218,10 @@
             filename[len - 4] = 0;
             filename += G__getmakeinfo1("DLLPOST");
          }
+         else if((len>6&& (strcmp(filename+len-6,".dylib")==0))) {
+            filename[len - 6] = 0;
+            filename += G__getmakeinfo1("DLLPOST");
+         }
          else if((len>2&& (strcmp(filename+len-2,".a")==0 ||
                            strcmp(filename+len-2,".A")==0))) {
             filename[len - 2] = 0;
@@ -1408,7 +1412,7 @@
 /******************************************************************
 * G__loadfile(filename)
 *
-*  0) If .sl .dl .so .dll .DLL call G__shl_load()
+*  0) If .sl .dl .so .dylib .dll .DLL call G__shl_load()
 *  1) check G__MAXFILE                       return -2 if fail(fatal)
 *  2) check if file is already loaded        return 1 if already loaded
 *  3) Open filename
@@ -1695,6 +1699,10 @@
            filename[len - 4] = 0;
            filename += G__getmakeinfo1("DLLPOST");
         }
+        else if((len>6&& (strcmp(filename+len-6,".dylib")==0))) {
+           filename[len - 6] = 0;
+           filename += G__getmakeinfo1("DLLPOST");
+        }
         else if((len>2&& (strcmp(filename+len-2,".a")==0 ||
                           strcmp(filename+len-2,".A")==0))) {
            filename[len - 2] = 0;
@@ -2198,6 +2206,7 @@
                strcmp(filename+len-3,".so")==0)) ||
      (len>4&& (strcmp(filename+len-4,".dll")==0 ||
                strcmp(filename+len-4,".DLL")==0)) ||
+     (len>6&& (strcmp(filename+len-6,".dylib")==0)) ||
 #if defined(R__FBSD) || defined(R__OBSD)
      (len>strlen(soext) && strcmp(filename+len-strlen(soext), soext)==0) ||
 #endif
