--- a/interpreter/cling/include/cling/boost.modulemap	2020-07-07 15:31:53.120798326 +0200
+++ b/interpreter/cling/include/cling/boost.modulemap		      2020-07-15 10:45:19.408188217 +0200
@@ -1,5 +1,6 @@
 module boost_accumulators [system] {
   export *
+
   module "accumulators__accumulators_fwd" { header "accumulators/accumulators_fwd.hpp" export * }
   module "accumulators__accumulators" { header "accumulators/accumulators.hpp" export * }
   module "accumulators__framework__accumulator_base" { header "accumulators/framework/accumulator_base.hpp" export * }
@@ -4684,6 +4685,14 @@
   module "pool__simple_segregated_storage" { header "pool/simple_segregated_storage.hpp" export * }
   module "pool__singleton_pool" { header "pool/singleton_pool.hpp" export * }
 }
+
+module boost_serialization {
+ export *
+ umbrella "serialization"
+ module * { export * }
+}
+
+
 /*
 module boost_preprocessor {
   module "preprocessor__arithmetic__add" { header "preprocessor/arithmetic/add.hpp" export * }
@@ -6774,9 +6783,6 @@
     module "spirit__home__support__detail__lexer__size_t" { header "spirit/home/support/detail/lexer/size_t.hpp" export * }
     module "spirit__home__support__detail__make_cons" { header "spirit/home/support/detail/make_cons.hpp" export * }
     module "spirit__home__support__detail__make_vector" { header "spirit/home/support/detail/make_vector.hpp" export * }
-    module "spirit__home__support__detail__math__detail__fp_traits" { header "spirit/home/support/detail/math/detail/fp_traits.hpp" export * }
-    module "spirit__home__support__detail__math__fpclassify" { header "spirit/home/support/detail/math/fpclassify.hpp" export * }
-    module "spirit__home__support__detail__math__signbit" { header "spirit/home/support/detail/math/signbit.hpp" export * }
 /*
  * While building module 'boost_modules_wrapper' imported from /home/teemperor/llvm/boost-compile/cms.cpp:2:
 In file included from <module-includes>:1023:
@@ -6911,7 +6917,6 @@
     module "spirit__home__x3__support__traits__has_attribute" { header "spirit/home/x3/support/traits/has_attribute.hpp" export * }
     module "spirit__home__x3__support__traits__is_parser" { header "spirit/home/x3/support/traits/is_parser.hpp" export * }
     module "spirit__home__x3__support__traits__is_variant" { header "spirit/home/x3/support/traits/is_variant.hpp" export * }
-    module "spirit__home__x3__support__traits__make_attribute" { header "spirit/home/x3/support/traits/make_attribute.hpp" export * }
     module "spirit__home__x3__support__traits__move_to" { header "spirit/home/x3/support/traits/move_to.hpp" export * }
     module "spirit__home__x3__support__traits__numeric_traits" { header "spirit/home/x3/support/traits/numeric_traits.hpp" export * }
     module "spirit__home__x3__support__traits__optional_traits" { header "spirit/home/x3/support/traits/optional_traits.hpp" export * }
@@ -7427,7 +7432,6 @@
     module "variant__detail__enable_recursive_fwd" { header "variant/detail/enable_recursive_fwd.hpp" export * }
     module "variant__detail__enable_recursive" { header "variant/detail/enable_recursive.hpp" export * }
     module "variant__detail__forced_return" { header "variant/detail/forced_return.hpp" export * }
-    module "variant__detail__generic_result_type" { header "variant/detail/generic_result_type.hpp" export * }
     module "variant__detail__hash_variant" { header "variant/detail/hash_variant.hpp" export * }
     module "variant__detail__has_result_type" { header "variant/detail/has_result_type.hpp" export * }
     module "variant__detail__initializer" { header "variant/detail/initializer.hpp" export * }
@@ -7449,85 +7453,6 @@
     module "variant__variant_fwd" { header "variant/variant_fwd.hpp" export * }
     module "variant__variant" { header "variant/variant.hpp" export * }
   }
-  module boost_serialization {
-    module "serialization__access" { header "serialization/access.hpp" export * }
-    module "serialization__archive_input_unordered_map" { header "serialization/archive_input_unordered_map.hpp" export * }
-    module "serialization__archive_input_unordered_set" { header "serialization/archive_input_unordered_set.hpp" export * }
-    module "serialization__array" { header "serialization/array.hpp" export * }
-    module "serialization__array_optimization" { header "serialization/array_optimization.hpp" export * }
-    module "serialization__array_wrapper" { header "serialization/array_wrapper.hpp" export * }
-    module "serialization__assume_abstract" { header "serialization/assume_abstract.hpp" export * }
-    module "serialization__base_object" { header "serialization/base_object.hpp" export * }
-    module "serialization__binary_object" { header "serialization/binary_object.hpp" export * }
-    module "serialization__bitset" { header "serialization/bitset.hpp" export * }
-    module "serialization__boost_array" { header "serialization/boost_array.hpp" export * }
-    module "serialization__boost_unordered_map" { header "serialization/boost_unordered_map.hpp" export * }
-    module "serialization__boost_unordered_set" { header "serialization/boost_unordered_set.hpp" export * }
-    module "serialization__collection_size_type" { header "serialization/collection_size_type.hpp" export * }
-    module "serialization__collections_load_imp" { header "serialization/collections_load_imp.hpp" export * }
-    module "serialization__collections_save_imp" { header "serialization/collections_save_imp.hpp" export * }
-    module "serialization__collection_traits" { header "serialization/collection_traits.hpp" export * }
-    module "serialization__complex" { header "serialization/complex.hpp" export * }
-    module "serialization__config" { header "serialization/config.hpp" export * }
-    module "serialization__deque" { header "serialization/deque.hpp" export * }
-    module "serialization__detail__is_default_constructible" { header "serialization/detail/is_default_constructible.hpp" export * }
-    module "serialization__detail__stack_constructor" { header "serialization/detail/stack_constructor.hpp" export * }
-    module "serialization__ephemeral" { header "serialization/ephemeral.hpp" export * }
-    module "serialization__export" { header "serialization/export.hpp" export * }
-    module "serialization__extended_type_info" { header "serialization/extended_type_info.hpp" export * }
-    module "serialization__extended_type_info_no_rtti" { header "serialization/extended_type_info_no_rtti.hpp" export * }
-    module "serialization__extended_type_info_typeid" { header "serialization/extended_type_info_typeid.hpp" export * }
-    module "serialization__factory" { header "serialization/factory.hpp" export * }
-    module "serialization__force_include" { header "serialization/force_include.hpp" export * }
-    module "serialization__forward_list" { header "serialization/forward_list.hpp" export * }
-    module "serialization__hash_collections_load_imp" { header "serialization/hash_collections_load_imp.hpp" export * }
-    module "serialization__hash_collections_save_imp" { header "serialization/hash_collections_save_imp.hpp" export * }
-    module "serialization__is_bitwise_serializable" { header "serialization/is_bitwise_serializable.hpp" export * }
-    module "serialization__item_version_type" { header "serialization/item_version_type.hpp" export * }
-    module "serialization__level_enum" { header "serialization/level_enum.hpp" export * }
-    module "serialization__level" { header "serialization/level.hpp" export * }
-    module "serialization__list" { header "serialization/list.hpp" export * }
-    module "serialization__map" { header "serialization/map.hpp" export * }
-    module "serialization__nvp" { header "serialization/nvp.hpp" export * }
-    module "serialization__optional" { header "serialization/optional.hpp" export * }
-    module "serialization__priority_queue" { header "serialization/priority_queue.hpp" export * }
-    module "serialization__queue" { header "serialization/queue.hpp" export * }
-    module "serialization__scoped_ptr" { header "serialization/scoped_ptr.hpp" export * }
-    module "serialization__serialization" { header "serialization/serialization.hpp" export * }
-    module "serialization__set" { header "serialization/set.hpp" export * }
-    module "serialization__shared_ptr_helper" { header "serialization/shared_ptr_helper.hpp" export * }
-    module "serialization__shared_ptr" { header "serialization/shared_ptr.hpp" export * }
-    module "serialization__singleton" { header "serialization/singleton.hpp" export * }
-    module "serialization__slist" { header "serialization/slist.hpp" export * }
-    module "serialization__smart_cast" { header "serialization/smart_cast.hpp" export * }
-    module "serialization__split_free" { header "serialization/split_free.hpp" export * }
-    module "serialization__split_member" { header "serialization/split_member.hpp" export * }
-    module "serialization__stack" { header "serialization/stack.hpp" export * }
-    module "serialization__state_saver" { header "serialization/state_saver.hpp" export * }
-    module "serialization__static_warning" { header "serialization/static_warning.hpp" export * }
-    module "serialization__string" { header "serialization/string.hpp" export * }
-    module "serialization__strong_typedef" { header "serialization/strong_typedef.hpp" export * }
-    module "serialization__throw_exception" { header "serialization/throw_exception.hpp" export * }
-    module "serialization__tracking_enum" { header "serialization/tracking_enum.hpp" export * }
-    module "serialization__tracking" { header "serialization/tracking.hpp" export * }
-    module "serialization__traits" { header "serialization/traits.hpp" export * }
-    module "serialization__type_info_implementation" { header "serialization/type_info_implementation.hpp" export * }
-    module "serialization__unique_ptr" { header "serialization/unique_ptr.hpp" export * }
-    module "serialization__unordered_collections_load_imp" { header "serialization/unordered_collections_load_imp.hpp" export * }
-    module "serialization__unordered_collections_save_imp" { header "serialization/unordered_collections_save_imp.hpp" export * }
-    module "serialization__unordered_map" { header "serialization/unordered_map.hpp" export * }
-    module "serialization__unordered_set" { header "serialization/unordered_set.hpp" export * }
-    module "serialization__utility" { header "serialization/utility.hpp" export * }
-    module "serialization__valarray" { header "serialization/valarray.hpp" export * }
-    module "serialization__variant" { header "serialization/variant.hpp" export * }
-    module "serialization__vector_135" { header "serialization/vector_135.hpp" export * }
-    module "serialization__vector" { header "serialization/vector.hpp" export * }
-    module "serialization__version" { header "serialization/version.hpp" export * }
-    module "serialization__void_cast_fwd" { header "serialization/void_cast_fwd.hpp" export * }
-    module "serialization__void_cast" { header "serialization/void_cast.hpp" export * }
-    module "serialization__weak_ptr" { header "serialization/weak_ptr.hpp" export * }
-    module "serialization__wrapper" { header "serialization/wrapper.hpp" export * }
-  }
   module boost_archive {
     module "archive__archive_exception" { header "archive/archive_exception.hpp" export * }
     module "archive__basic_archive" { header "archive/basic_archive.hpp" export * }
@@ -8021,7 +7946,6 @@
   module "regex__config__cwchar" { header "regex/config/cwchar.hpp" export * }
   module "regex__config" { header "regex/config.hpp" export * }
   module "regex" { header "regex.hpp" export * }
-  module "regex__icu" { header "regex/icu.hpp" export * }
   module "regex__pattern_except" { header "regex/pattern_except.hpp" export * }
   module "regex__pending__object_cache" { header "regex/pending/object_cache.hpp" export * }
   module "regex__pending__static_mutex" { header "regex/pending/static_mutex.hpp" export * }
