--- orig.root/configure	2010-09-29 11:29:35.000000000 +0200
+++ root/configure	2010-11-02 18:41:16.000000000 +0100
@@ -2233,7 +2233,7 @@
 ### echo %%% Use included ZLib or use systems
 #
 if test "x$enable_builtin_zlib" = "xno" ; then
-    check_header "zlib.h" ""  $ZLIB $ZLIB/include \
+    check_header "zlib.h" ""  $ZLIB_ROOT/include $ZLIB $ZLIB/include \
         /usr/local/include /usr/include/zlib /usr/local/include/zlib \
         /opt/zlib/include /usr/include
     if test "x$found_dir" = "x" ; then
@@ -2244,7 +2244,7 @@
     fi
 
     check_library "libz" "$enable_shared" "" \
-        $ZLIB $ZLIB/lib /usr/local/zlib/lib /usr/local/lib \
+        $ZLIB_ROOT/lib $ZLIB $ZLIB/lib /usr/local/zlib/lib /usr/local/lib \
         /usr/lib/zlib /usr/local/lib/zlib /usr/zlib/lib /usr/lib \
         /usr/zlib /usr/local/zlib /opt/zlib /opt/zlib/lib
     if test "x$found_lib" = "x" ; then
@@ -3523,21 +3523,21 @@
     # If we didn't find the system libAfterImage, or the user didn't ask
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
-       check_header "jpeglib.h" "" $ASIMAGE $ASIMAGE/include \
+       check_header "jpeglib.h" "" $LIBJPG_ROOT/include $ASIMAGE $ASIMAGE/include \
           /usr/local/include /usr/include /opt/include $finkdir/include
        asjpegincdir=$found_dir
-       check_header "png.h" "" $ASIMAGE $ASIMAGE/include $ASPNG/include \
+       check_header "png.h" "" $LIBPNG_ROOT/include $ASIMAGE $ASIMAGE/include $ASPNG/include \
           /usr/local/include /usr/include /usr/local/include/libpng \
           /opt/include $finkdir/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
-           check_header "tiffio.h" "" $ASIMAGE $ASIMAGE/include \
+           check_header "tiffio.h" "" $LIBTIFF_ROOT/include $ASIMAGE $ASIMAGE/include \
                /usr/local/include /usr/include /opt/include $finkdir/include
            astiffincdir=$found_dir
        else
            astiffincdir="--with-tiff=no"
        fi
-       check_header "gif_lib.h" "" $ASIMAGE $ASIMAGE/include \
+       check_header "gif_lib.h" "" $LIBUNGIF_ROOT/include $ASIMAGE $ASIMAGE/include \
           /usr/local/include /usr/include /opt/include $finkdir/include
        asgifincdir=$found_dir
 
@@ -3551,11 +3551,14 @@
        # libAfterImage contains embedded libungif source if needed
        if test ! "x$enable_astiff" = "xno" ; then
            aslibs="libjpeg libtiff libungif libz libpng"
+           extra_libtiff_path=$LIBTIFF_ROOT/lib 
        else
            aslibs="libjpeg libungif libz libpng"
+           extra_libtiff_path=
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
+              $extra_libtiff_path $LIBJPG_ROOT/lib $LIBPNG_ROOT/lib $LIBUNGIF_ROOT/lib $ZLIB_ROOT/lib \
              $ASIMAGE $ASIMAGE/lib $ASPNG/lib /usr/local/lib /usr/lib /opt/lib \
 			 $finkdir/lib
            if test "x$k" = "xlibz" ; then
