--- branches/v5-30-00-patches/core/metautils/inc/TSchemaRuleProcessor.h	2011/08/22 20:01:47	40663
+++ branches/v5-30-00-patches/core/metautils/inc/TSchemaRuleProcessor.h	2011/08/22 22:59:25	40664
@@ -88,9 +88,10 @@
                  
                   // Split between the typename and the membername
                   // Take in consideration template names.
-                  for(std::string::size_type i=0; i<elem.size(); ++i) {
-                     if (elem[i]=='<') { ++level; }
-                     else if (elem[i]=='>') { --level; }
+                  for(std::string::size_type j=elem.size(); j>0; --j) {
+                     std::string::size_type i = j-1;
+                     if (elem[i]=='<') { --level; }
+                     else if (elem[i]=='>') { ++level; }
                      else if (level == 0 && isspace(elem[i])) {
                         type = elem.substr( 0, i );
                         // At the first iteration we know we have a space.
