diff -Naur orig.root/core/base/src/TBuffer.cxx root/core/base/src/TBuffer.cxx
--- orig.root/core/base/src/TBuffer.cxx	2010-11-05 15:46:19.000000000 +0100
+++ root/core/base/src/TBuffer.cxx	2010-11-29 14:17:51.000000000 +0100
@@ -185,8 +185,13 @@
    // Expand the I/O buffer to newsize bytes.
 
    Int_t l  = Length();
-   fBuffer  = fReAllocFunc(fBuffer, newsize+kExtraSpace,
-                           fBufSize+kExtraSpace);
+   if ( (fMode&kWrite)!=0 ) {
+      fBuffer  = fReAllocFunc(fBuffer, newsize+kExtraSpace,
+                              fBufSize+kExtraSpace);
+   } else {
+      fBuffer  = fReAllocFunc(fBuffer, newsize,
+                              fBufSize);
+   }
    if (fBuffer == 0) {
       if (fReAllocFunc == TStorage::ReAllocChar) {
          Fatal("Expand","Failed to expand the data buffer using TStorage::ReAllocChar.");
