--- a/interpreter/cling/include/cling/boost.modulemap
+++ b/interpreter/cling/include/cling/boost.modulemap
@@ -309,7 +309,7 @@
   module "asio__yield" { header "asio/yield.hpp" export * }
 }
 module boost_assert {
-  module "assert" { header "assert.hpp" export * }
+  module "assert" { textual header "assert.hpp" export * }
 }
 module boost_assign {
   module "assign__assignment_exception" { header "assign/assignment_exception.hpp" export * }
@@ -5339,12 +5339,14 @@
   exclude header "date_time/gregorian/gregorian_io.hpp"
   // Likely fixed by the the same commit:
   exclude header "date_time/posix_time/time_formatters.hpp"
+  exclude header "date_time/posix_time/time_formatters_limited.hpp"
   //error: too many template arguments for class template 'date_formatter'
   //  std::basic_string<charT> s(date_time::date_formatter<date,date_time::iso_format<charT>,charT>::date_to_string(d.begin()));
   exclude header "date_time/posix_time/posix_time_types.hpp"
   exclude header "date_time/dst_rules.hpp"
   exclude header "date_time/gregorian/formatters.hpp"
   exclude header "date_time/gregorian/gregorian.hpp"
+  exclude header "date_time/gregorian/formatters_limited.hpp"
   exclude header "date_time/date_formatting.hpp"
   exclude header "date_time/date_formatting_limited.hpp"
   // Disentangle from boost_serialization. Fixed by boostorg/date_time@356c5c2
@@ -5659,6 +5661,9 @@
 
 module boost_thread {
   module "thread" { header "thread.hpp" export * }
+  exclude header "thread/thread_time.hpp"
+  // Broken and seems never included.
+  exclude header "thread/detail/function_wrapper.hpp"
   umbrella "thread" module * { export * }
 }
