diff --git a/core/unix/src/TUnixSystem.cxx b/core/unix/src/TUnixSystem.cxx
index 0110839..008fa66 100644
--- a/core/unix/src/TUnixSystem.cxx
+++ b/core/unix/src/TUnixSystem.cxx
@@ -4343,6 +4343,9 @@ static const char *DynamicPath(const char *newpath = 0, Bool_t reset = kFALSE)
       if (!ldpath.IsNull())
          ldpath += ":";
       ldpath += gSystem->Getenv("LD_LIBRARY_PATH");
+      if (!ldpath.IsNull())
+         ldpath += ":";
+      ldpath += gSystem->Getenv("DYLD_FALLBACK_LIBRARY_PATH");
 #else
       ldpath = gSystem->Getenv("LD_LIBRARY_PATH");
 #endif
