diff -aur root.orig/configure root/configure
--- root.orig/configure	2011-05-25 10:02:20.000000000 +0200
+++ root/configure	2011-06-01 12:55:07.000000000 +0200
@@ -2371,7 +2371,7 @@
 #
 if test "x$enable_builtin_zlib" = "xno" ; then
     check_header "zlib.h" "" \
-        $ZLIB ${ZLIB:+$ZLIB/include} \
+        $ZLIB_ROOT/include $ZLIB ${ZLIB:+$ZLIB/include} \
         ${finkdir:+$finkdir/include} \
         /usr/local/include /usr/include/zlib /usr/local/include/zlib \
         /opt/zlib/include /usr/include
@@ -2383,7 +2383,7 @@
     fi
 
     check_library "libz" "$enable_shared" "" \
-        $ZLIB ${ZLIB:+$ZLIB/lib} \
+        $ZLIB_ROOT/lib $ZLIB ${ZLIB:+$ZLIB/lib} \
         /usr/local/zlib/lib /usr/local/lib \
         /usr/lib/zlib /usr/local/lib/zlib /usr/zlib/lib /usr/lib \
         /usr/zlib /usr/local/zlib /opt/zlib /opt/zlib/lib
@@ -3794,12 +3794,12 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" \
-          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+          $LIBJPG_ROOT/include $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
           ${finkdir:+$finkdir/include} \
           /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" \
-          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+          $LIBPNG_ROOT/include $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
           ${ASPNG:+$ASPNG/include} \
           ${finkdir:+$finkdir/include} \
           /usr/local/include /usr/include /usr/local/include/libpng \
@@ -3807,7 +3807,7 @@
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" \
-               $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+               $LIBTIFF_ROOT/include $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
                ${finkdir:+$finkdir/include} \
                /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
@@ -3815,7 +3815,7 @@
            astiffincdir="--with-tiff=no"
        fi
        check_header "gif_lib.h" "" \
-          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
+          $LIBUNGIF_ROOT/include $$ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
           ${finkdir:+$finkdir/include} \
           /usr/local/include /usr/include /opt/include
        asgifincdir=$found_dir
@@ -3830,11 +3830,14 @@
        # libAfterImage contains embedded libungif source if needed
        if test ! "x$enable_astiff" = "xno" ; then
            aslibs="libjpeg libtiff libungif libz libpng"
+           extra_libtiff_path=$LIBTIFF_ROOT/lib
        else
            aslibs="libjpeg libungif libz libpng"
+           extra_libtiff_path=
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
+               $extra_libtiff_path $LIBJPG_ROOT/lib $LIBPNG_ROOT/lib $LIBUNGIF_ROOT/lib $ZLIB_ROOT/lib \
                $ASIMAGE ${ASIMAGE:+$ASIMAGE/lib} ${ASPNG:+$ASPNG/lib} \
                ${finkdir:+$finkdir/lib} \
                /usr/local/lib /usr/lib /opt/lib
