diff -aur root.orig/tree/tree/src/TBranchElement.cxx root/tree/tree/src/TBranchElement.cxx
--- root.orig/tree/tree/src/TBranchElement.cxx	2011-05-25 10:02:21.000000000 +0200
+++ root/tree/tree/src/TBranchElement.cxx	2011-06-01 12:02:05.000000000 +0200
@@ -1906,6 +1906,12 @@
       SetAddress(fAddress);
    }
    SetupAddresses();
+   if (fObject == 0) 
+   {
+      // We have nowhere to copy the data (probably because the data member was
+      // 'dropped' from the current schema) so let's no copy it in a random place.
+      return 0;
+   }
 
    Int_t nbranches = fBranches.GetEntriesFast();
    if (nbranches) {
