From 9c30f57aabdec5a6ed845e00f89c3173f883fcd5 Mon Sep 17 00:00:00 2001
From: Axel Naumann <Axel.Naumann@cern.ch>
Date: Sat, 30 Apr 2016 01:21:51 +0200
Subject: [PATCH 1/1] Likely fix for rootcling unhappiness / missing
 Transaction.

---
 core/metautils/src/TMetaUtils.cxx | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/metautils/src/TMetaUtils.cxx b/core/metautils/src/TMetaUtils.cxx
index 0ac60ff..07149db 100644
--- a/core/metautils/src/TMetaUtils.cxx
+++ b/core/metautils/src/TMetaUtils.cxx
@@ -972,6 +972,10 @@ ROOT::TMetaUtils::EIOCtorCategory ROOT::TMetaUtils::CheckConstructor(const clang
    if (ioctortype.GetType() ==0 && (arg == 0 || arg[0] == '\0')) {
       // We are looking for a constructor with zero non-default arguments.
       clang::CXXRecordDecl* ncCl = const_cast<clang::CXXRecordDecl*>(cl);
+
+      // We may induce template instantiation
+      cling::Interpreter::PushTransactionRAII clingRAII(const_cast<cling::Interpreter*>(&interpreter));
+      
       if (auto* Ctor = interpreter.getCI()->getSema().LookupDefaultConstructor(ncCl)) {
          if (Ctor->getAccess() == clang::AS_public) {
             return EIOCtorCategory::kDefault;
-- 
2.7.4

