diff -aur root.orig/config/Makefile.linux root/config/Makefile.linux
--- root.orig/config/Makefile.linux	2011-05-25 10:02:21.000000000 +0200
+++ root/config/Makefile.linux	2011-06-01 11:53:53.000000000 +0200
@@ -33,7 +33,7 @@
 
 # Linker:
 LD            = g++
-LDFLAGS       = -m32 $(OPT) $(EXTRA_LDFLAGS)
+LDFLAGS       = -m32 -Wl,--hash-style=gnu $(OPT) $(EXTRA_LDFLAGS)
 SOFLAGS       = -shared -Wl,-soname,
 SOEXT         = so
 
diff -aur root.orig/config/Makefile.linuxx8664gcc root/config/Makefile.linuxx8664gcc
--- root.orig/config/Makefile.linuxx8664gcc	2011-05-25 10:02:21.000000000 +0200
+++ root/config/Makefile.linuxx8664gcc	2011-06-01 11:53:53.000000000 +0200
@@ -36,7 +36,7 @@
 
 # Linker:
 LD            = g++
-LDFLAGS       = -m64 $(OPT) $(EXTRA_LDFLAGS)
+LDFLAGS       = -m64 -Wl,--hash-style=gnu $(OPT) $(EXTRA_LDFLAGS)
 SOFLAGS       = -shared -Wl,-soname,
 SOEXT         = so
 
