diff --git a/core/unix/src/TUnixSystem.cxx b/core/unix/src/TUnixSystem.cxx
index 0110839..008fa66 100644
--- a/core/unix/src/TUnixSystem.cxx
+++ b/core/unix/src/TUnixSystem.cxx
@@ -4343,6 +4343,9 @@ static const char *DynamicPath(const char *newpath = 0, Bool_t reset = kFALSE)
       if (!ldpath.IsNull())
          ldpath += ":";
       ldpath += gSystem->Getenv("LD_LIBRARY_PATH");
+      if (!ldpath.IsNull())
+         ldpath += ":";
+      ldpath += gSystem->Getenv("DYLD_FALLBACK_LIBRARY_PATH");
 #else
       ldpath = gSystem->Getenv("LD_LIBRARY_PATH");
 #endif
@@ -4411,11 +4411,12 @@
    char *name;
 
    int ext = 0, len = strlen(lib);
-   if (len > 3 && (!strcmp(lib+len-3, ".sl") ||
-                   !strcmp(lib+len-3, ".dl") ||
-                   !strcmp(lib+len-4, ".dll")||
-                   !strcmp(lib+len-4, ".DLL")||
-                   !strcmp(lib+len-3, ".so") ||
+   if (len > 3 && (!strcmp(lib+len-3, ".so")    ||
+                   !strcmp(lib+len-3, ".dl")    ||
+                   !strcmp(lib+len-4, ".dll")   ||
+                   !strcmp(lib+len-4, ".DLL")   ||
+                   !strcmp(lib+len-6, ".dylib") ||
+                   !strcmp(lib+len-3, ".sl")    ||
                    !strcmp(lib+len-2, ".a"))) {
       name = gSystem->Which(GetDynamicPath(), lib, kReadPermission);
       ext  = 1;
@@ -4427,14 +4428,18 @@
          fname.Form("%s.dll", lib);
          name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
          if (!name) {
-            fname.Form("%s.sl", lib);
+            fname.Form("%s.dylib", lib);
             name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
             if (!name) {
-               fname.Form("%s.dl", lib);
+               fname.Form("%s.sl", lib);
                name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
                if (!name) {
-                  fname.Form("%s.a", lib);
+                  fname.Form("%s.dl", lib);
                   name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
+                  if (!name) {
+                     fname.Form("%s.a", lib);
+                     name = gSystem->Which(GetDynamicPath(), fname, kReadPermission);
+                  }
                }
             }
          }
@@ -4447,7 +4452,7 @@
                "%s does not exist in %s", lib, GetDynamicPath());
       else
          Error("DynamicPathName",
-               "%s[.so | .sl | .dl | .a | .dll] does not exist in %s", lib, GetDynamicPath());
+               "%s[.so | .dll | .dylib | .sl | .dl | .a] does not exist in %s", lib, GetDynamicPath());
    }
 
    return name;
