diff --git a/config/Makefile.linux b/config/Makefile.linux
index 53b3246..3d9faff 100644
--- a/config/Makefile.linux
+++ b/config/Makefile.linux
@@ -36,7 +36,7 @@ COMPILER      = gnu
 
 # Linker:
 LD            = g++
-LDFLAGS       = -m32 $(OPT) $(EXTRA_LDFLAGS) -Wl,--no-undefined -Wl,--as-needed
+LDFLAGS       = -m32 -Wl,--hash-style=gnu $(OPT) $(EXTRA_LDFLAGS)
 SOFLAGS       = -shared -Wl,-soname,
 SOEXT         = so
 
diff --git a/config/Makefile.linuxx8664gcc b/config/Makefile.linuxx8664gcc
index f2f483c..04d4ec4 100644
--- a/config/Makefile.linuxx8664gcc
+++ b/config/Makefile.linuxx8664gcc
@@ -39,7 +39,7 @@ COMPILER      = gnu
 
 # Linker:
 LD            = g++
-LDFLAGS       = -m64 $(OPT) $(EXTRA_LDFLAGS) -Wl,--no-undefined -Wl,--as-needed
+LDFLAGS       = -m64 -Wl,--hash-style=gnu $(OPT) $(EXTRA_LDFLAGS)
 SOFLAGS       = -shared -Wl,-soname,
 SOEXT         = so
 
