--- /dev/null
+++ b/math/mathcore/src/SMatrix.cxx
@@ -0,0 +1,14 @@
+#include "Math/MatrixRepresentationsStatic.h"
+
+void ROOT::Math::RowOffsetsBase::init(int *v, int *offsets, unsigned int D)
+{
+   v[0]=0;
+   for (unsigned int i=1; i<D; ++i)
+      v[i]=v[i-1]+i;
+      for (unsigned int i=0; i<D; ++i) {
+         for (unsigned int j=0; j<=i; ++j)
+            offsets[i*D+j] = v[i]+j;
+         for (unsigned int j=i+1; j<D; ++j)
+            offsets[i*D+j] = v[j]+i ;
+  }
+}

--- a/math/smatrix/inc/Math/MatrixRepresentationsStatic.h
+++ b/math/smatrix/inc/Math/MatrixRepresentationsStatic.h
@@ -122,25 +122,34 @@ namespace Math {
       symmetric matrix
    */
 
+   struct RowOffsetsBase 
+   {
+     protected:
+       static void init(int *v, int *offsets, unsigned int D);
+   };
+
    template<unsigned int D>
-   struct RowOffsets {
+   struct RowOffsets : RowOffsetsBase {
       RowOffsets() {
          int v[D];
-         v[0]=0;
-         for (unsigned int i=1; i<D; ++i)
-            v[i]=v[i-1]+i;
-         for (unsigned int i=0; i<D; ++i) { 
-            for (unsigned int j=0; j<=i; ++j)
-               fOff[i*D+j] = v[i]+j; 
-            for (unsigned int j=i+1; j<D; ++j)
-               fOff[i*D+j] = v[j]+i ;
-         }
+         this->init(v, fOff, D);
       }
       int operator()(unsigned int i, unsigned int j) const { return fOff[i*D+j]; }
       int apply(unsigned int i) const { return fOff[i]; }
       int fOff[D*D];
    };
 
+   template <unsigned int D> struct SymMatrixOffsets
+   {
+    protected:
+      static RowOffsets<D> offsets;
+   };
+
+   template <unsigned int D>
+   RowOffsets<D>
+   SymMatrixOffsets<D>::offsets;
+
+
 //_________________________________________________________________________________
    /**
       MatRepSym
@@ -161,11 +170,11 @@ namespace Math {
       @ingroup MatRep 
    */
    template <class T, unsigned int D>
-   class MatRepSym {
+   class MatRepSym : SymMatrixOffsets<D> {
 
    public: 
 
-      MatRepSym() :fOff(0) { CreateOffsets(); } 
+      MatRepSym() :fOff(&SymMatrixOffsets<D>::offsets) { } 
 
       typedef T  value_type;
 

