diff --git a/interpreter/llvm-project/llvm/include/llvm/ADT/FoldingSet.h b/interpreter/llvm-project/llvm/include/llvm/ADT/FoldingSet.h
index f82eabd5..df6fcdc9 100644
--- a/interpreter/llvm-project/llvm/include/llvm/ADT/FoldingSet.h
+++ b/interpreter/llvm-project/llvm/include/llvm/ADT/FoldingSet.h
@@ -628,7 +628,7 @@ public:
 /// to provide the interface of FoldingSet but with deterministic iteration
 /// order based on the insertion order. T must be a subclass of FoldingSetNode
 /// and implement a Profile function.
-template <class T, class VectorT = SmallVector<T*, 8>>
+template <class T, class VectorT = SmallVector<T*, 32>>
 class FoldingSetVector {
   FoldingSet<T> Set;
   VectorT Vector;
