--- trunk/tree/tree/src/TBranchElement.cxx	2010/12/01 14:11:45	37149
+++ trunk/tree/tree/src/TBranchElement.cxx	2010/12/02 22:56:23	37210
@@ -4726,6 +4726,12 @@
    for (Int_t elemID = 0; elemID < ndata; ++elemID) {
       // -- Loop over all the streamer elements and create sub-branches as needed.
       TStreamerElement* elem = (TStreamerElement*) elems[elemID];
+      if (elem->IsA() == TStreamerArtificial::Class()) {
+         continue;
+      }
+      if (elem->TestBit(TStreamerElement::kRepeat)) {
+         continue;
+      }
       Int_t offset = elem->GetOffset();
       // FIXME: An STL container as a base class gets TStreamerSTL as its class, so this test is not enough.
       // See InitializeOffsets() for the proper test.
