diff --git a/interpreter/cling/include/cling/std.modulemap b/interpreter/cling/include/cling/std.modulemap
index 6b9acf1..81700c8 100644
--- a/interpreter/cling/include/cling/std.modulemap
+++ b/interpreter/cling/include/cling/std.modulemap
@@ -429,4 +429,8 @@ module "std" [system] {
     export *
     header "bits/stl_tree.h"
   }
+  module "ext/alloc_traits" {
+    export *
+    header "ext/alloc_traits.h"
+  }
 }
