diff --git a/cmake/modules/SearchInstalledSoftware.cmake b/cmake/modules/SearchInstalledSoftware.cmake
index 84001e8..658f491 100644
--- a/cmake/modules/SearchInstalledSoftware.cmake
+++ b/cmake/modules/SearchInstalledSoftware.cmake
@@ -287,7 +287,7 @@ if(opengl)
   else()
     find_package(OpenGL)
   endif()
-  if(NOT OPENGL_FOUND OR NOT OPENGL_GLU_FOUND)
+  if(NOT OPENGL_LIBRARIES)
     if(fail-on-missing)
       message(FATAL_ERROR "OpenGL package (with GLU) not found and opengl option required")
     else()
