Index: hist/hist/src/TH1.cxx
===================================================================
--- hist/hist/src/TH1.cxx	(revision 44641)
+++ hist/hist/src/TH1.cxx	(revision 44642)
@@ -5081,20 +5081,34 @@
    Bool_t allHaveLimits = kTRUE;
    Bool_t allSameLimits = kTRUE;
    Bool_t foundLabelHist = kFALSE;
+   Bool_t firstNonEmptyHist = kTRUE;
 
 
    TIter next(&inlist);
    // start looping with this histogram 
    TH1 * h = this; 
+
    do  {
       // skip empty histograms
       if (h->fTsumw == 0 && h->GetEntries() == 0) continue;
 
+
       Bool_t hasLimits = h->GetXaxis()->GetXmin() < h->GetXaxis()->GetXmax();
       allHaveLimits = allHaveLimits && hasLimits;
 
       if (hasLimits) {
          h->BufferEmpty();
+
+         // this is done in case the first histograms are empty and 
+         // the histogram have different limits
+         if (firstNonEmptyHist ) { 
+            // set axis limits in the case the first histogram was empty
+            if (h != this && !SameLimitsAndNBins( fXaxis, *h->GetXaxis()) ) { 
+               fXaxis.Set(h->GetXaxis()->GetNbins(), h->GetXaxis()->GetXmin(),h->GetXaxis()->GetXmax());
+            }
+            firstNonEmptyHist = kFALSE;     
+         }
+
          // this is executed the first time an histogram with limits is found
          // to set some initial values on the new axis
          if (!initialLimitsFound) {
Index: hist/hist/src/TH2.cxx
===================================================================
--- hist/hist/src/TH2.cxx	(revision 44641)
+++ hist/hist/src/TH2.cxx	(revision 44642)
@@ -1446,6 +1446,7 @@
    Bool_t initialLimitsFound = kFALSE;
    Bool_t allSameLimits = kTRUE;
    Bool_t allHaveLimits = kTRUE;
+   Bool_t firstNonEmptyHist = kTRUE;
 
    TIter next(&inlist);
    TH2 * h = this;
@@ -1458,6 +1459,20 @@
 
       if (hasLimits) {
          h->BufferEmpty();
+
+         // this is done in case the first histograms are empty and 
+         // the histogram have different limits
+         if (firstNonEmptyHist ) { 
+            // set axis limits in the case the first histogram was empty
+            if (h != this ) { 
+               if (!SameLimitsAndNBins(fXaxis, *(h->GetXaxis())) ) 
+                  fXaxis.Set(h->GetXaxis()->GetNbins(), h->GetXaxis()->GetXmin(),h->GetXaxis()->GetXmax());
+               if (!SameLimitsAndNBins(fYaxis, *(h->GetYaxis())) ) 
+                  fYaxis.Set(h->GetYaxis()->GetNbins(), h->GetYaxis()->GetXmin(),h->GetYaxis()->GetXmax());
+            }
+            firstNonEmptyHist = kFALSE;     
+         }
+
          if (!initialLimitsFound) {
             // this is executed the first time an histogram with limits is found
             // to set some initial values on the new axes
Index: hist/hist/src/TH3.cxx
===================================================================
--- hist/hist/src/TH3.cxx	(revision 44641)
+++ hist/hist/src/TH3.cxx	(revision 44642)
@@ -1480,6 +1480,7 @@
    Bool_t initialLimitsFound = kFALSE;
    Bool_t allSameLimits = kTRUE;
    Bool_t allHaveLimits = kTRUE;
+   Bool_t firstNonEmptyHist = kTRUE;
 
    TIter next(&inlist);
    TH3* h = this;
@@ -1492,6 +1493,22 @@
 
       if (hasLimits) {
          h->BufferEmpty();
+
+         // this is done in case the first histograms are empty and 
+         // the histogram have different limits
+         if (firstNonEmptyHist ) { 
+            // set axis limits in the case the first histogram was empty
+            if (h != this ) { 
+               if (!SameLimitsAndNBins(fXaxis, *(h->GetXaxis())) ) 
+                  fXaxis.Set(h->GetXaxis()->GetNbins(), h->GetXaxis()->GetXmin(),h->GetXaxis()->GetXmax());
+               if (!SameLimitsAndNBins(fYaxis, *(h->GetYaxis())) ) 
+                  fYaxis.Set(h->GetYaxis()->GetNbins(), h->GetYaxis()->GetXmin(),h->GetYaxis()->GetXmax());
+               if (!SameLimitsAndNBins(fZaxis, *(h->GetZaxis())) ) 
+                  fZaxis.Set(h->GetZaxis()->GetNbins(), h->GetZaxis()->GetXmin(),h->GetZaxis()->GetXmax());
+            }
+            firstNonEmptyHist = kFALSE;     
+         }
+
          if (!initialLimitsFound) {
             // this is executed the first time an histogram with limits is found
             // to set some initial values on the new axes

Property changes on: hist/hist/src/TFormula.cxx
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/hist/hist/src/TFormula.cxx:r44639

Index: hist/hist/src/TProfileHelper.h
===================================================================
--- hist/hist/src/TProfileHelper.h	(revision 44641)
+++ hist/hist/src/TProfileHelper.h	(revision 44642)
@@ -167,6 +167,7 @@
    Bool_t initialLimitsFound = kFALSE;
    Bool_t allSameLimits = kTRUE;
    Bool_t allHaveLimits = kTRUE;
+   Bool_t firstNonEmptyHist = kTRUE;
 
    TIter next(&inlist);
    T* h = p;
@@ -181,6 +182,24 @@
 
       if (hasLimits) {
          h->BufferEmpty();
+
+         // this is done in case the first histograms are empty and 
+         // the histogram have different limits
+         if (firstNonEmptyHist ) { 
+            // set axis limits in the case the first histogram was empty
+            if (h != p ) { 
+               if (!p->SameLimitsAndNBins(p->fXaxis, *(h->GetXaxis())) ) 
+                  p->fXaxis.Set(h->GetXaxis()->GetNbins(), h->GetXaxis()->GetXmin(),h->GetXaxis()->GetXmax());
+               if (!p->SameLimitsAndNBins(p->fYaxis, *(h->GetYaxis())) ) 
+                  p->fYaxis.Set(h->GetYaxis()->GetNbins(), h->GetYaxis()->GetXmin(),h->GetYaxis()->GetXmax());
+               if (!p->SameLimitsAndNBins(p->fZaxis, *(h->GetZaxis())) ) 
+                  p->fZaxis.Set(h->GetZaxis()->GetNbins(), h->GetZaxis()->GetXmin(),h->GetZaxis()->GetXmax());
+            }
+            firstNonEmptyHist = kFALSE;     
+         }
+
+         // this is executed the first time an histogram with limits is found
+         // to set some initial values on the new axis
          if (!initialLimitsFound) {
             initialLimitsFound = kTRUE;
             newXAxis.Set(h->GetXaxis()->GetNbins(), h->GetXaxis()->GetXmin(),

Property changes on: hist/hist/src
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/hist/hist/src:r44639


Property changes on: hist/hist/inc
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/hist/hist/inc:r44639


Property changes on: hist/hist
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/hist/hist:r44639

