--- cint/cint7/src/tmplt.cxx	(revision 28414)
+++ cint/cint7/src/tmplt.cxx	(revision 28415)
@@ -298,6 +298,10 @@
    if (fileNum < 0) {
       return -1;
    }
+   if (G__srcfile[fileNum].filename[0] == '{') {
+      // ignore "{CINTEX dictionary translator}"
+      return -1;
+   }
    fileNum = G__getIndex(fileNum, -1, headers);
    if (fileNum == -1) {
       return -1;
@@ -315,6 +319,10 @@
          if (index < 0) {
             return -1;
          }
+         if (G__srcfile[index].filename[0] == '{') {
+            // ignore "{CINTEX dictionary translator}"
+            return -1;
+         }
          index = G__getIndex(index, tagnum, headers); // Note: Do real work, headers is modified here.
          if (index == -1) {
             return -1;
--- cint/cint/src/tmplt.cxx	(revision 28414)
+++ cint/cint/src/tmplt.cxx	(revision 28415)
@@ -2114,6 +2114,9 @@
   //getting 'vector' header file
   int fileNum = deftmpclass->filenum;
   if( fileNum < 0 ) return -1;
+  if (G__srcfile[fileNum].filename[0] == '{')
+     // ignore "{CINTEX dictionary translator}"
+     return -4;
   fileNum = G__getIndex(fileNum,-1, headers);
   if( fileNum < 0 ) return fileNum;
   //headers.push_back( G__srcfile[ fileNum ].filename );
@@ -2125,6 +2128,9 @@
       int index = gValue.tagnum;
       index = G__struct.filenum[index];
       if( index < 0 ) return -3;
+      if (G__srcfile[index].filename[0] == '{')
+         // ignore "{CINTEX dictionary translator}"
+         return -4;
       index = G__getIndex(index, gValue.tagnum, headers);
       if( index < 0 ) return index;
       //it = find( headers.begin(), headers.end(),G__srcfile[ index ].filename  );
