diff -Naur orig.root/cint/reflex/python/genreflex/selclass.py root/cint/reflex/python/genreflex/selclass.py
--- orig.root/cint/reflex/python/genreflex/selclass.py	2010-02-11 14:20:31.000000000 +0100
+++ root/cint/reflex/python/genreflex/selclass.py	2010-04-13 14:07:50.000000000 +0200
@@ -181,7 +181,8 @@
 
       for chk in lst:
         try:
-          i = int( chk )
+          if ( chk != "*" ):
+             i = int( chk )
         except:
           print warning, chk, 'is not a valid value of checksum parameter - an integer expected'
           return False
@@ -202,17 +203,18 @@
         return False
 
       for v in lst:
-        matchObj = self.ver_re.match( v )
-        if not matchObj:
-          print warning, '-', v, 'is not a valid value of version parameter'
-          return False
-        else:
-          rng = matchObj.groups()
-          if rng[0] and rng[1]:
-            b, e = int(rng[0]), int(rng[1])
-            if b >= e:
-              print warning, '-', v, 'is not a valid version range'
-              return False
+        if ( v != "*" ):
+          matchObj = self.ver_re.match( v )
+          if not matchObj:
+             print warning, '-', v, 'is not a valid value of version parameter'
+             return False
+          else:
+            rng = matchObj.groups()
+            if rng[0] and rng[1]:
+              b, e = int(rng[0]), int(rng[1])
+              if b >= e:
+                print warning, '-', v, 'is not a valid version range'
+                return False
 
     #------------------------------------------------------------------------------
     # Check if we deal with renameing rule
