diff -Naur orig.root/etc/gdb-backtrace.sh root/etc/gdb-backtrace.sh
--- orig.root/etc/gdb-backtrace.sh	2010-11-05 15:46:52.000000000 +0100
+++ root/etc/gdb-backtrace.sh	2010-11-29 12:04:41.000000000 +0100
@@ -39,7 +39,7 @@
    GDB=${GDB:-gdb}
 
    # Run GDB, strip out unwanted noise.
-   $GDB -q -batch -n -x $TMPFILE $1 $2 2>&1 |
+   $GDB -q -batch -n -x $TMPFILE $1 $2 2>&1  < /dev/null |
    /usr/bin/sed -n \
     -e 's/^(gdb) //' \
     -e '/^#/p' \
@@ -85,7 +85,7 @@
    # Run GDB, strip out unwanted noise.
    have_eval_command=`gdb --help 2>&1 |grep eval-command`
    if ! test "x$have_eval_command" = "x"; then
-      $GDB --batch --eval-command="$backtrace" /proc/$1/exe $1 2>&1 |
+      $GDB --batch --eval-command="$backtrace" /proc/$1/exe $1 2>&1 < /dev/null |
       /bin/sed -n \
          -e 's/^(gdb) //' \
          -e '/^#/p' \
