--- trunk/cint/reflex/python/genreflex/gendict.py	2011/03/08 20:07:10	38324
+++ trunk/cint/reflex/python/genreflex/gendict.py	2011/03/08 20:07:34	38325
@@ -2651,7 +2651,7 @@
       if clname.find('<') != -1:
         template = 'template<> '
         # specialization of A::B::f() needs namespace A { template<> B<...>::f(){} }
-        specclname = attrs['name']
+        specclname = None
         enclattrs = attrs
         while 'context' in enclattrs:
           if self.xref[enclattrs['id']]['elem'] == 'Namespace' :
@@ -2664,7 +2664,11 @@
             returnValue += 'namespace ' + namespname.replace('::', ' { namespace ')
             returnValue += ' { \n'
             break
-          specclname = enclattrs['name'] + '::' + specclname
+          if specclname:
+            specclname = enclattrs['name'] + '::' + specclname
+          else:
+            #this is the first time through so we want the class name
+            specclname = enclattrs['name']
           enclattrs = self.xref[enclattrs['context']]['attrs']
       else :
         specclname = clname
