diff -Naur --exclude=.svn -I '@(#)root' -I '// Author' -I '\$Id' root/config/Makefile.in root-noungif/config/Makefile.in
--- root/config/Makefile.in	2012-02-12 20:59:23.062918496 -0800
+++ root-noungif/config/Makefile.in	2012-02-29 18:47:16.611126427 -0800
@@ -198,7 +198,6 @@
 ASJPEGINCDIR   := $(filter-out /usr/include, @asjpegincdir@)
 ASPNGINCDIR    := $(filter-out /usr/include, @aspngincdir@)
 ASTIFFINCDIR   := $(filter-out /usr/include, @astiffincdir@)
-ASGIFINCDIR    := $(filter-out /usr/include, @asgifincdir@)
 ASINCDIR       := $(filter-out /usr/include, @asimageincdir@)
 ASLIB          := @asimagelib@
 ASLIBDIR       := @asimagelibdir@
diff -Naur --exclude=.svn -I '@(#)root' -I '// Author' -I '\$Id' root/configure root-noungif/configure
--- root/configure	2012-02-29 18:34:17.316160474 -0800
+++ root-noungif/configure	2012-02-29 18:48:48.370122418 -0800
@@ -3988,11 +3988,6 @@
        else
            astiffincdir="--with-tiff=no"
        fi
-       check_header "gif_lib.h" "" \
-          $LIBUNGIF_ROOT/include $$ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
-          ${finkdir:+$finkdir/include} \
-          /usr/local/include /usr/include /opt/include
-       asgifincdir=$found_dir
 
        asextralib=""
        asextralibdir=""
@@ -4000,13 +3995,11 @@
        found_dirz=""
        # libz must be before libpng, if libz is not found don't
        # add libpng which needs libz
-       # note that failure to find system libungif is not fatal since
-       # libAfterImage contains embedded libungif source if needed
        if test ! "x$enable_astiff" = "xno" ; then
-           aslibs="libjpeg libtiff libungif libz libpng"
+           aslibs="libjpeg libtiff libz libpng"
            extra_libtiff_path=$LIBTIFF_ROOT/lib
        else
-           aslibs="libjpeg libungif libz libpng"
+           aslibs="libjpeg libz libpng"
            extra_libtiff_path=
        fi
        for k in $aslibs ; do
@@ -4023,9 +4016,6 @@
            if test "x$k" = "xlibpng" && test "x$found_libz" = "x" ; then
                break;
            fi
-           if test "x$k" = "xlibungif" && test "x$found_lib" = "x" ; then
-               asgifincdir=""
-           fi
            if test ! "x$found_lib" = "x" ; then
                asextralib="$asextralib $found_lib"
                asextralibdir="$asextralibdir $found_dir"
@@ -4037,7 +4027,6 @@
            asjpegincdir=""
            aspngincdir=""
            astiffincdir=""
-           asgifincdir=""
        else
            asextralib="$asextralib $found_libz"
            asextralibdir="$asextralibdir $found_dirz"
@@ -6418,7 +6407,6 @@
     -e "s|@asjpegincdir@|$asjpegincdir|"        \
     -e "s|@aspngincdir@|$aspngincdir|"          \
     -e "s|@astiffincdir@|$astiffincdir|"        \
-    -e "s|@asgifincdir@|$asgifincdir|"          \
     -e "s|@asimageincdir@|$asimageincdir|"      \
     -e "s|@asimagelib@|$asimagelib|"            \
     -e "s|@asimagelibdir@|$asimagelibdir|"      \
diff -Naur --exclude=.svn -I '@(#)root' -I '// Author' -I '\$Id' root/graf2d/asimage/Module.mk root-noungif/graf2d/asimage/Module.mk
--- root/graf2d/asimage/Module.mk	2012-02-12 20:59:20.841918286 -0800
+++ root-noungif/graf2d/asimage/Module.mk	2012-02-29 18:47:14.380126525 -0800
@@ -169,9 +169,6 @@
 		elif [ "$(ASTIFFINCDIR)" != "" ]; then \
 			TIFFINCDIR="--with-tiff-includes=$(ASTIFFINCDIR)"; \
 		fi; \
-		if [ "$(ASGIFINCDIR)" != "" ]; then \
-			GIFINCDIR="--with-gif-includes=$(ASGIFINCDIR)"; \
-		fi; \
 		if [ "$(FREETYPEDIRI)" != "" ]; then \
 			TTFINCDIR="--with-ttf-includes=-I../../../../$(FREETYPEDIRI)"; \
 		fi; \
@@ -184,7 +181,6 @@
 		$$MMX \
 		$(ASTEPDBG) \
 		--with-builtin-ungif \
-		$$GIFINCDIR \
 		--with-jpeg \
 		$$JPEGINCDIR \
 		--with-png \
