

diff --git a/configure b/configure
index 599346b..94e0765 100755
--- a/configure
+++ b/configure
@@ -2846,7 +2846,7 @@ if test ! "x$enable_asimage" = "xno" ; then
           /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" $ASIMAGE $ASIMAGE/include $ASPNG/include \
-          /usr/local/include /usr/include /usr/local/include/libpng \
+          $LIBPNG_ROOT/include /usr/local/include /usr/include /usr/local/include/libpng \
           /opt/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
@@ -2875,7 +2875,7 @@ if test ! "x$enable_asimage" = "xno" ; then
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
-             $ASIMAGE $ASIMAGE/lib $ASPNG/lib /usr/local/lib /usr/lib /opt/lib
+             $ASIMAGE $ASIMAGE/lib $ASPNG/lib $LIBPNG_ROOT/lib /usr/local/lib /usr/lib /opt/lib
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
