Index: branches/v5-28-00-patches/tree/tree/src/TTreeCloner.cxx
===================================================================
--- branches/v5-28-00-patches/tree/tree/src/TTreeCloner.cxx	(revision 39758)
+++ branches/v5-28-00-patches/tree/tree/src/TTreeCloner.cxx	(revision 39759)
@@ -267,6 +267,10 @@
    }
 
    fFromBranches.AddLast(from);
+   if (!from->TestBit(TBranch::kDoNotUseBufferMap)) {
+      // Make sure that we reset the Buffer's map if needed.
+      to->ResetBit(TBranch::kDoNotUseBufferMap);
+   }
    fToBranches.AddLast(to);
 
    numBaskets += from->GetWriteBasket();

Property changes on: branches/v5-28-00-patches/tree/tree/src/TTreeCloner.cxx
___________________________________________________________________
Added: svn:mergeinfo
   Merged /trunk/io/io/src/tree/tree/src/TTreeCloner.cxx:r38156
   Merged /branches/v5-28-00-patches/io/tree/src/TTreeCloner.cxx:r38059
   Merged /trunk/tree/tree/src/TTreeCloner.cxx:r37618,37641,37836,37906,37920-37921,37947,37950,37955,37965,37985,38059,38061,38080,38202,38205,38393,38554,38614,38743,38854,38869,39270,39755

Index: branches/v5-28-00-patches/tree/tree/inc/TBranch.h
===================================================================
--- branches/v5-28-00-patches/tree/tree/inc/TBranch.h	(revision 39758)
+++ branches/v5-28-00-patches/tree/tree/inc/TBranch.h	(revision 39759)
@@ -47,6 +47,7 @@
 class TDirectory;
 class TFile;
 class TClonesArray;
+class TTreeCloner;
 
    const Int_t kDoNotProcess = BIT(10); // Active bit for branches
    const Int_t kIsClone      = BIT(11); // to indicate a TBranchClones
@@ -57,6 +58,7 @@
 class TBranch : public TNamed , public TAttFill {
 
 protected:
+   friend class TTreeCloner;
    // TBranch status bits
    enum EStatusBits {
       kAutoDelete = BIT(15),

Property changes on: branches/v5-28-00-patches/tree/tree/inc/TBranch.h
___________________________________________________________________
Added: svn:mergeinfo
   Merged /trunk/tree/tree/inc/TBranch.h:r37618,37641,37836,37906,37920-37921,37955,37965,37985,38055,38059,38061,38403-38404,38462,38801,38803,38854,38869,39551,39642,39644,39757
   Merged /trunk/io/io/src/tree/tree/inc/TBranch.h:r38156
   Merged /branches/v5-28-00-patches/io/tree/inc/TBranch.h:r38059

