--- configure.old	2011-09-26 15:22:56.000000000 +0200
+++ configure	2011-09-26 15:23:48.000000000 +0200
@@ -356,14 +356,14 @@
 
     case $arch in
     macosx)
-        logmsg " file -L $chklibcompat | grep 'i386'"
-        if file -L $chklibcompat | grep 'i386' > /dev/null 2>& 1 ; then
+        logmsg " otool -fhv $chklibcompat | grep -i 'i386'"
+        if otool -fhv $chklibcompat | grep -i 'i386' > /dev/null 2>& 1 ; then
             ret=1
         fi
         ;;
     macosx64)
-        logmsg " file -L $chklibcompat | grep 'x86_64'"
-        if file -L $chklibcompat | grep 'x86_64' > /dev/null 2>& 1 ; then
+        logmsg " otool -fhv $chklibcompat | grep -i 'x86_64'"
+        if otool -fhv $chklibcompat | grep -i 'x86_64' > /dev/null 2>& 1 ; then
             ret=1
         fi
         ;;
