Index: cint/reflex/src/Class.cxx
===================================================================
--- cint/reflex/src/Class.cxx	(revision 28049)
+++ cint/reflex/src/Class.cxx	(revision 28086)
@@ -199,14 +199,10 @@
       // we found a destructor -> Invoke it
       Object dummy = Object(Type(), instance);
       fDestructor.Invoke(dummy, (Object*)0);
-      // if deallocation of memory wanted
-      if (dealloc) {
-         Deallocate(instance);
-      }
    }
-   else {
-      // this class has no destructor defined we call the operator delete on it
-      ::operator delete(instance);
+   // if deallocation of memory wanted
+   if (dealloc) {
+      Deallocate(instance);
    }
 }
 
