Index: graf3d/gl/src/CsgOps.cxx
===================================================================
--- graf3d/gl/src/CsgOps.cxx	(revision 46519)
+++ graf3d/gl/src/CsgOps.cxx	(revision 46520)
@@ -1892,7 +1892,7 @@
       PLIST& polyList = Polys();
       UInt_t i;
       for (i=0;i < polyList.size(); i++) {
-         TGBinder binder(fMesh,i);
+         TGBinder binder(*this, i);
          polyList[i].SetPlane(compute_plane(binder));
       }
    }
