Index: core/meta/src/TClass.cxx
===================================================================
--- core/meta/src/TClass.cxx	(revision 46238)
+++ core/meta/src/TClass.cxx	(revision 46239)
@@ -3206,9 +3206,6 @@
          fRealData->Delete();
       delete fRealData;  fRealData=0;
       
-      if (fStreamerInfo)
-         fStreamerInfo->Delete();
-      delete fStreamerInfo; fStreamerInfo=0;
    }
 }
 

Property changes on: core/meta/src/TClass.cxx
___________________________________________________________________
Modified: svn:mergeinfo
   Merged /trunk/core/meta/src/TClass.cxx:r46238

