diff --git a/cint/ROOT/cintdlls.mk b/cint/ROOT/cintdlls.mk
index 7c2872b..abc26c8 100644
--- a/cint/ROOT/cintdlls.mk
+++ b/cint/ROOT/cintdlls.mk
@@ -147,6 +147,9 @@ ifeq ($(subst $(MACOSX_MINOR),,456789),456789)
 # MACOSX_MINOR < 4
   CINTDLLSOEXTCMD += ;mv $(@:.dll=.so) $@
 else
+  # On macosx one should change the install_name as well.
+  # FIXME: not tested on 10.4, should be the same also there?
+  CINTDLLSOEXTCMD += ;install_name_tool -id `otool -D $@ | tail -1 | sed -e's|:$$||;s|[.]so$$|.dll|'` $@
   CINTDLLSOEXTCMD += ;rm -f $(@:.dll=.so)
 endif
 endif # macosx
