Index: graf2d/asimage/Module.mk
===================================================================
--- graf2d/asimage/Module.mk	(revision 45902)
+++ graf2d/asimage/Module.mk	(revision 45903)
@@ -303,7 +303,7 @@
 
 ##### extra rules ######
 $(ASIMAGEO): $(ASTEPDEP) $(FREETYPEDEP)
-$(ASIMAGEO): CXXFLAGS += $(FREETYPEINC) $(ASTEPDIRI)
+$(ASIMAGEO): CXXFLAGS += $(FREETYPEINC) $(ASTEPDIRI) $(X11INCDIR:%=-I%)
 
 $(ASIMAGEGUIO) $(ASIMAGEGUIDO) $(ASIMAGEDO): $(ASTEPDEP)
 $(ASIMAGEGUIO) $(ASIMAGEGUIDO) $(ASIMAGEDO): CXXFLAGS += $(ASTEPDIRI)

Property changes on: graf2d/asimage
___________________________________________________________________
Added: svn:mergeinfo
   Merged /trunk/graf2d/asimage:r44180-44181,45318,45568,45597,45610,45613,45630,45643,45902

