diff -arupN root-5.32.00.orig/bindings/pyroot/src/Converters.cxx root-5.32.00/bindings/pyroot/src/Converters.cxx
--- root-5.32.00.orig/bindings/pyroot/src/Converters.cxx	2012-04-05 11:00:15.000000000 +0200
+++ root-5.32.00/bindings/pyroot/src/Converters.cxx	2012-04-05 11:19:05.000000000 +0200
@@ -87,7 +87,7 @@ Bool_t PyROOT::T##name##Converter::SetAr
             func->SetArg( para.fl );                                          \
       } else {                                                                \
          PyErr_Format( PyExc_TypeError,                                       \
-            #type" expected, got string of size "PY_SSIZE_T_FORMAT, PyROOT_PyUnicode_GET_SIZE( pyobject ) );\
+            #type" expected, got string of size " PY_SSIZE_T_FORMAT, PyROOT_PyUnicode_GET_SIZE( pyobject ) );\
          return kFALSE;                                                       \
       }                                                                       \
    } else {                                                                   \
diff -arupN root-5.32.00.orig/bindings/pyroot/src/PyBufferFactory.cxx root-5.32.00/bindings/pyroot/src/PyBufferFactory.cxx
--- root-5.32.00.orig/bindings/pyroot/src/PyBufferFactory.cxx	2012-04-05 11:00:15.000000000 +0200
+++ root-5.32.00/bindings/pyroot/src/PyBufferFactory.cxx	2012-04-05 11:18:19.000000000 +0200
@@ -111,7 +111,7 @@ namespace {
    PyObject* name##_buffer_str( PyObject* self )                             \
    {                                                                         \
       Py_ssize_t l = buffer_length( self );                                  \
-      return PyROOT_PyUnicode_FromFormat( "<"#type" buffer, size "PY_SSIZE_T_FORMAT">", l );\
+      return PyROOT_PyUnicode_FromFormat( "<"#type" buffer, size " PY_SSIZE_T_FORMAT ">", l );\
    }                                                                         \
                                                                              \
    PyObject* name##_buffer_item( PyObject* self, Py_ssize_t idx ) {          \
diff -arupN root-5.32.00.orig/bindings/pyroot/src/RootModule.cxx root-5.32.00/bindings/pyroot/src/RootModule.cxx
--- root-5.32.00.orig/bindings/pyroot/src/RootModule.cxx	2012-04-05 11:00:15.000000000 +0200
+++ root-5.32.00/bindings/pyroot/src/RootModule.cxx	2012-04-05 11:13:48.000000000 +0200
@@ -275,7 +275,7 @@ namespace {
       Py_ssize_t argc = PyTuple_GET_SIZE( args );
       if ( argc != 2 ) {
          PyErr_Format( PyExc_TypeError,
-           "BindObject takes exactly 2 argumenst ("PY_SSIZE_T_FORMAT" given)", argc );
+           "BindObject takes exactly 2 argumenst (" PY_SSIZE_T_FORMAT " given)", argc );
          return 0;
       }
 
@@ -306,7 +306,7 @@ namespace {
       Py_ssize_t argc = PyTuple_GET_SIZE( args );
       if ( argc != 0 && argc != 1 ) {
          PyErr_Format( PyExc_TypeError,
-            "MakeNullPointer takes at most 1 argument ("PY_SSIZE_T_FORMAT" given)", argc );
+            "MakeNullPointer takes at most 1 argument (" PY_SSIZE_T_FORMAT " given)", argc );
          return 0;
       }
 
diff -arupN root-5.32.00.orig/io/sql/src/TSQLFile.cxx root-5.32.00/io/sql/src/TSQLFile.cxx
--- root-5.32.00.orig/io/sql/src/TSQLFile.cxx	2012-04-05 11:00:16.000000000 +0200
+++ root-5.32.00/io/sql/src/TSQLFile.cxx	2012-04-05 11:01:40.000000000 +0200
@@ -1152,7 +1152,7 @@ void TSQLFile::CreateBasicTables()
 
    #define WrintCfg(name, type, value)                              \
    {                                                                \
-      sqlcmd.Form("INSERT INTO %s%s%s VALUES (%s%s%s, %s"type"%s)", \
+      sqlcmd.Form("INSERT INTO %s%s%s VALUES (%s%s%s, %s" type "%s)", \
                   quote, sqlio::ConfigTable, quote,                 \
                   vquote, name, vquote,                             \
                   vquote, value, vquote);                           \
