diff --git a/interpreter/CMakeLists.txt b/interpreter/CMakeLists.txt
index de5056d..8f993f7 100644
--- a/interpreter/CMakeLists.txt
+++ b/interpreter/CMakeLists.txt
@@ -229,6 +229,11 @@ if(builtin_llvm)
 
   set(LLVM_ENABLE_PROJECTS "clang" CACHE STRING "")
 
+  # Always build LLVM with C++17. It is not necessary to compile with the same
+  # C++ standard as the rest of ROOT and sometimes it doesn't even work.
+  set(_cxx_standard ${CMAKE_CXX_STANDARD})
+  set(CMAKE_CXX_STANDARD 17)
+
   #---Add the sub-directory excluding all the targets from all-----------------------------------------
   if(CMAKE_GENERATOR MATCHES "Xcode")
     add_subdirectory(llvm-project/llvm)
@@ -236,6 +241,8 @@ if(builtin_llvm)
     add_subdirectory(llvm-project/llvm EXCLUDE_FROM_ALL)
   endif()
 
+   set(CMAKE_CXX_STANDARD ${_cxx_standard})
+
   set(LLVM_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/interpreter/llvm-project/llvm/include
     ${CMAKE_BINARY_DIR}/interpreter/llvm-project/llvm/include
     CACHE STRING "LLVM include directories."
@@ -446,7 +453,14 @@ else()
     # Disable linking against shared LLVM
     set(LLVM_LINK_LLVM_DYLIB OFF)
 
+    # Always build LLVM with C++17. It is not necessary to compile with the same
+    # C++ standard as the rest of ROOT and sometimes it doesn't even work.
+    set(_cxx_standard ${CMAKE_CXX_STANDARD})
+    set(CMAKE_CXX_STANDARD 17)
+
     add_subdirectory(llvm-project/clang EXCLUDE_FROM_ALL)
+
+    set(CMAKE_CXX_STANDARD ${_cxx_standard})
   endif(builtin_clang)
 
   set( CLANG_BUILT_STANDALONE 1 )
