--- trunk/cint/reflex/inc/Reflex/Builder/CollectionProxy.h	2010/10/08 13:27:53	36196
+++ trunk/cint/reflex/inc/Reflex/Builder/CollectionProxy.h	2010/11/16 16:40:22	36698
@@ -110,6 +110,11 @@
    : public Address<typename T::const_reference>
 #endif
 {
+#ifdef _KCC  // KAI compiler
+   typedef Address<typename T::value_type&> Address_t;
+#else
+   typedef Address<typename T::const_reference> Address_t;
+#endif
    typedef T Cont_t;
    typedef typename T::iterator Iter_t;
    typedef typename T::value_type Value_t;
@@ -153,7 +158,7 @@
 #else
       typename T::const_reference ref = *(e->iter());
 #endif
-      return e->fStart = address(ref);
+      return e->fStart = Address_t::address(ref);
    }
 
 
@@ -171,7 +176,7 @@
 #else
       typename T::const_reference ref = *(e->iter());
 #endif
-      return address(ref);
+      return Address_t::address(ref);
    }
 
 
