--- orig.root/tree/tree/src/TTreeCloner.cxx	2008-12-18 09:56:42.000000000 +0100
+++ root/tree/tree/src/TTreeCloner.cxx	2009-01-26 22:50:51.000000000 +0100
@@ -241,15 +241,15 @@
       Int_t firstfi = fi;
       while (strcmp(fb->GetName(), tb->GetName())) {
          ++fi;
+         if (fi >= fnb) {
+            // continue at the beginning
+            fi = 0;
+         }
          if (fi==firstfi) {
             // We tried all the branches and there is not match.
             fb = 0;
             break;
          }
-         if (fi >= fnb) {
-            // continue at the beginning
-            fi = 0;
-         }
          fb = (TBranch*) from->UncheckedAt(fi);
       }
       if (fb) {
