diff --git a/math/mathcore/src/mixmax.h b/math/mathcore/src/mixmax.h
index 16decf1..176ed7e 100755
--- a/math/mathcore/src/mixmax.h
+++ b/math/mathcore/src/mixmax.h
@@ -33,7 +33,7 @@
 
 #include <stdio.h>
 #include <stdint.h>
-
+#include <math.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -134,7 +134,7 @@ void branch_inplace( rng_state_t* Xin, myID_t* ID ); // almost the same as apply
 #define MOD_REM(k) ((k) % MERSBASE )  // latest Intel CPU is supposed to do this in one CPU cycle, but on my machines it seems to be 20% slower than the best tricks
 #define MOD_MERSENNE(k) MOD_PAYNE(k)
 
-#define INV_MERSBASE (0x1p-61)
+#define INV_MERSBASE pow(2, -61)
 
 
 // the charpoly is irreducible for the combinations of N and SPECIAL and has maximal period for N=508, 256, half period for 1260, and 1/12 period for 3150
