diff -aur root.orig/tree/tree/src/TTreeCache.cxx root/tree/tree/src/TTreeCache.cxx
--- root.orig/tree/tree/src/TTreeCache.cxx	2011-05-25 10:02:21.000000000 +0200
+++ root/tree/tree/src/TTreeCache.cxx	2011-06-01 12:10:22.000000000 +0200
@@ -684,12 +684,7 @@
 
    if (needLearningStart) {
       // Restart learning
-      fIsLearning = kTRUE;
-      fIsManual = kFALSE;
-      fNbranches  = 0;
-      fZipBytes   = 0;
-      if (fBrNames) fBrNames->Delete();
-      fEntryCurrent = -1;
+      StartLearningPhase();
    }
 }
 
