diff --git a/hist/hist/src/TH1.cxx b/hist/hist/src/TH1.cxx
index b6b4859..b7dd5a7 100644
--- a/hist/hist/src/TH1.cxx
+++ b/hist/hist/src/TH1.cxx
@@ -2601,6 +2601,8 @@ Bool_t TH1::Divide(const TH1 *h1)
       Warning("Divide","Attempt to divide histograms with different axis limits");
    } catch(DifferentBinLimits&) {
       Warning("Divide","Attempt to divide histograms with different bin limits");
+   } catch(DifferentLabels&) {
+      Warning("Divide","Attempt to divide histograms with different labels");
    }
 
    if (fDimension < 2) nbinsy = -1;
@@ -2695,6 +2697,8 @@ Bool_t TH1::Divide(const TH1 *h1, const TH1 *h2, Double_t c1, Double_t c2, Optio
       Warning("Divide","Attempt to divide histograms with different axis limits");
    } catch(DifferentBinLimits&) {
       Warning("Divide","Attempt to divide histograms with different bin limits");
+   }  catch(DifferentLabels&) {
+      Warning("Divide","Attempt to divide histograms with different labels");
    }
 
    if (!c2) {
@@ -5542,6 +5546,8 @@ Bool_t TH1::Multiply(const TH1 *h1)
       Warning("Multiply","Attempt to multiply histograms with different axis limits");
    } catch(DifferentBinLimits&) {
       Warning("Multiply","Attempt to multiply histograms with different bin limits");
+   } catch(DifferentLabels&) {
+      Warning("Multiply","Attempt to multiply histograms with different labels");
    }
 
    if (fDimension < 2) nbinsy = -1;
@@ -5626,6 +5632,8 @@ Bool_t TH1::Multiply(const TH1 *h1, const TH1 *h2, Double_t c1, Double_t c2, Opt
       Warning("Multiply","Attempt to multiply histograms with different axis limits");
    } catch(DifferentBinLimits&) {
       Warning("Multiply","Attempt to multiply histograms with different bin limits");
+   } catch(DifferentLabels&) {
+      Warning("Multiply","Attempt to multiply histograms with different labels");
    }
 
    if (fDimension < 2) nbinsy = -1;
