From 048d5497a2a79188cb0d173ec725ebb31360b4cc Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <davidlt@cern.ch>
Date: Sat, 19 Oct 2013 17:11:44 +0200
Subject: [PATCH 3/8] Increase macro buffer size (10x)

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>
---
 rpmio/macro.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpmio/macro.c b/rpmio/macro.c
index 09cb563..ee8dc08 100644
--- a/rpmio/macro.c
+++ b/rpmio/macro.c
@@ -18,7 +18,7 @@ extern int optind;
 
 #define	STREQ(_t, _f, _fn)	((_fn) == (sizeof(_t)-1) && rstreqn((_t), (_f), (_fn)))
 
-#define MACROBUFSIZ (BUFSIZ * 2)
+#define MACROBUFSIZ (BUFSIZ * 20)
 
 #include <rpm/rpmio.h>
 #include <rpm/rpmstring.h>
-- 
1.8.0

