diff --git a/rpmio/fts.h b/rpmio/fts.h
index eaa4a8b..da5f646 100644
--- a/rpmio/fts.h
+++ b/rpmio/fts.h
@@ -56,7 +56,7 @@
 #endif
 
 #if defined(__APPLE__)
-# define _D_EXACT_NAMLEN(d) ((d)->d_reclen)
+# define _D_EXACT_NAMLEN(d) (strlen((d)->d_name))
 #endif
 
 #endif
diff --git a/find-provides b/find-provides
index 5064414..b068912 100755
--- a/autodeps/darwin.prov
+++ b/autodeps/darwin.prov
@@ -5,7 +5,7 @@
 # This script reads filenames from STDIN and outputs any relevant provides
 # information that needs to be included in the package.
 
-filelist=$(sed -n -e '/\.dylib/p' -e '/\.so/p' -e '/\.bundle/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*ppc" | cut -d: -f1)
+filelist=$(sed -n -e '/\.dylib/p' -e '/\.so/p' -e '/\.bundle/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*\(ppc\|i386\)" | cut -d: -f1)
 
 for f in $filelist; do
     libname=$(basename $f | sed -e 's;\..*;;')
