diff --git a/autodeps/darwin.req b/autodeps/darwin.req
index 9bb08b7..05dccf8 100644
--- a/autodeps/darwin.req
+++ b/autodeps/darwin.req
@@ -5,8 +5,8 @@
 ulimit -c 0
 
 filelist=`sed "s/['\"]/\\\&/g"`
-exelist=`echo $filelist | xargs file | fgrep Mach-O | cut -d: -f1 `
-scriptlist=`echo $filelist | xargs file | egrep ":.* (commands|script) " | cut -d: -f1 `
+exelist=`echo $filelist | xargs -n 100 file | fgrep Mach-O | cut -d: -f1 `
+scriptlist=`echo $filelist | xargs -n 100 file | egrep ":.* (commands|script) " | cut -d: -f1 `
 
 for f in $exelist; do
     if [ -x $f ]; then
