commit d40c31900f58514fa5425ef91d3456b74df5d49d
Author: Giulio Eulisse <giulio.eulisse@cern.ch>
Date:   Thu Apr 22 13:35:21 2010 +0200

    Workaround empty buildroot message

diff --git a/build/parsePreamble.c b/build/parsePreamble.c
index b5f1780..1ba4014 100644
--- a/build/parsePreamble.c
+++ b/build/parsePreamble.c
@@ -892,10 +892,6 @@ int parsePreamble(rpmSpec spec, int initialPackage)
      */
     if (initialPackage) {
 	char *buildRoot = rpmGetPath(spec->buildRoot, NULL);
-	if (*buildRoot == '\0') {
-	    rpmlog(RPMLOG_ERR, _("%%{buildroot} couldn't be empty\n"));
-	    goto exit;
-	}
 	if (rstreq(buildRoot, "/")) {
 	    rpmlog(RPMLOG_ERR, _("%%{buildroot} can not be \"/\"\n"));
 	    goto exit;
diff --git a/macros.in b/macros.in
index ae46fcb..c5555d1 100644
--- a/macros.in
+++ b/macros.in
@@ -224,7 +224,7 @@ package or when debugging this package.\
 %_buildrootdir		%{_topdir}/BUILDROOT
 
 #	Build root path, where %install installs the package during build.
-%buildroot		%{_buildrootdir}/%{name}-%{version}-%{release}.%{_arch}
+%buildroot
 
 #	Directory where temporaray files can be created.
 %_tmppath		%{_var}/tmp
