diff -arupN rpm-4.8.0.orig/macros.in rpm-4.8.0/macros.in
--- rpm-4.8.0.orig/macros.in	2012-02-27 10:57:02.000000000 +0100
+++ rpm-4.8.0/macros.in	2012-02-27 10:57:25.000000000 +0100
@@ -417,7 +417,7 @@ package or when debugging this package.\
 
 #
 # Use internal dependency generator rather than external helpers?
-%_use_internal_dependency_generator	1
+%_use_internal_dependency_generator	0
 
 #
 # Filter GLIBC_PRIVATE Provides: and Requires:
