commit b6afe3d6e0b4508c4fe1f525dde7027e2901a54e
Author: Giulio Eulisse <giulio.eulisse@cern.ch>
Date:   Thu Apr 22 11:32:16 2010 +0200

    Adds proper __fxstat64 in case of a mac  build.

diff --git a/rpmio/fts.c b/rpmio/fts.c
index 325e1a9..0169a02 100644
--- a/rpmio/fts.c
+++ b/rpmio/fts.c
@@ -59,6 +59,7 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #if defined(__APPLE__)
 #   define __errno_location()	(__error())
 #   define _STAT_VER		0
+#   define __fxstat64(_stat_ver, _fd, _sbp)    fstat64((_fd), (_sbp))
 #endif
 #include "system.h"
 #include "rpmio/fts.h"
