diff --git a/autodeps/darwin.prov b/autodeps/darwin.prov
index b068912..3169378 100644
--- a/autodeps/darwin.prov
+++ b/autodeps/darwin.prov
@@ -5,7 +5,7 @@
 # This script reads filenames from STDIN and outputs any relevant provides
 # information that needs to be included in the package.
 
-filelist=$(sed -n -e '/\.dylib/p' -e '/\.so/p' -e '/\.bundle/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*\(ppc\|i386\)" | cut -d: -f1)
+filelist=$(sed -n -e '/\.dylib/p' -e '/\.so/p' -e '/\.bundle/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*\(ppc\|i386\|x86_64\)" | cut -d: -f1)
 
 for f in $filelist; do
     libname=$(basename $f | sed -e 's;\..*;;')
