 diff -Naur autodeps/darwin.prov   autodeps/darwin.prov.new
--- autodeps/darwin.prov	2015-05-17 03:13:19.000000000 -0500
+++ autodeps/darwin.prov.new	2016-08-18 13:01:51.000000000 -0500
@@ -8,14 +8,5 @@
 filelist=$(sed -n -e '/\.dll/p' -e '/\.dylib/p' -e '/\.iglet/p' -e '/\.so/p' -e '/\.bundle/p' | sort | uniq | xargs file -L 2>/dev/null | grep "Mach-O.*\(ppc\|i386\|x86_64\|64-bit\)" | cut -d: -f1)
 
 for f in $filelist; do
-    libname=$(basename $f | sed -e 's;\..*;;')
-    soname=$(otool -l $f | grep $libname | awk '/ name/ {print $2}')
-
-    if [ "$soname" != "" ]; then
-	if [ ! -L $f ]; then
-	    basename $soname
-	fi
-    else
 	echo ${f##*/}
-    fi
 done | sort -u

