diff --git a/compiler/rustc_llvm/build.rs b/compiler/rustc_llvm/build.rs
index ac758c1..11abfac 100644
--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -347,18 +347,18 @@ fn main() {
 
     // C++ runtime library
     if !target.contains("msvc") {
-        if let Some(s) = llvm_static_stdcpp {
-            assert!(!cxxflags.contains("stdlib=libc++"));
-            let path = PathBuf::from(s);
-            println!("cargo:rustc-link-search=native={}", path.parent().unwrap().display());
-            if target.contains("windows") {
-                println!("cargo:rustc-link-lib=static:-bundle={}", stdcppname);
-            } else {
-                println!("cargo:rustc-link-lib=static={}", stdcppname);
-            }
-        } else if cxxflags.contains("stdlib=libc++") {
-            println!("cargo:rustc-link-lib=c++");
-        } else {
+        if let Some(_s) = llvm_static_stdcpp {
+        //    assert!(!cxxflags.contains("stdlib=libc++"));
+        //    let path = PathBuf::from(s);
+        //    println!("cargo:rustc-link-search=native={}", path.parent().unwrap().display());
+        //    if target.contains("windows") {
+        //        println!("cargo:rustc-link-lib=static:-bundle={}", stdcppname);
+        //    } else {
+        //        println!("cargo:rustc-link-lib=static={}", stdcppname);
+        //    }
+        //} else if cxxflags.contains("stdlib=libc++") {
+        //    println!("cargo:rustc-link-lib=c++");
+        //} else {
             println!("cargo:rustc-link-lib={}", stdcppname);
         }
     }
