--- src/Scripting/PyLCGDict2/python/setup.py	2005-10-04 03:33:16.000000000 +0200
+++ src/Scripting/PyLCGDict2/python/setup.py.new	2006-02-22 14:01:40.000000000 +0100
@@ -148,7 +148,7 @@
 ###
 regenerate(testdictCPP,
            [testdictH],
-           'lcgdict %s -I%s -o %s -s %s' % (testdictH, python_include, testdictCPP, testdictSelect),
+           lcgdict + ' %s -I%s -o %s -s %s' % (testdictH, python_include, testdictCPP, testdictSelect),
            )
 
 regenerate(seconddictCPP,
