diff -rupN SHERPA-MC-1.4.0.orig/PDF/LHAPDF/Makefile.am SHERPA-MC-1.4.0/PDF/LHAPDF/Makefile.am
--- PDF/LHAPDF/Makefile.am	2012-03-18 00:03:08.000000000 +0100
+++ PDF/LHAPDF/Makefile.am	2012-05-11 10:55:18.000000000 +0200
@@ -15,5 +15,6 @@ LHAPDFEXTRADIST = SVN_Info.C LHAPDF_Fort
 endif
 
 libLHAPDFSherpa_la_CPPFLAGS = $(AM_CPPFLAGS) @CONDITIONAL_LHAPDFINCS@
+libLHAPDFSherpa_la_LDFLAGS = -lz -L@CONDITIONAL_LHAPDFDIR@/lib -Bstatic -lLHAPDF
 
 EXTRA_DIST    = @SVNINCLUDE@ $(LHAPDFEXTRADIST)
diff -rupN SHERPA-MC-1.4.0.orig/SHERPA/Initialization/Initialization_Handler.C SHERPA-MC-1.4.0/SHERPA/Initialization/Initialization_Handler.C
--- SHERPA/Initialization/Initialization_Handler.C	2012-03-18 00:02:58.000000000 +0100
+++ SHERPA/Initialization/Initialization_Handler.C	2012-05-11 10:55:48.000000000 +0200
@@ -524,8 +524,8 @@ bool Initialization_Handler::InitializeT
     if (*pdflib=="None") continue;
     if (*pdflib=="LHAPDFSherpa") {
 #ifdef USING__LHAPDF
-      s_loader->AddPath(std::string(LHAPDF_PATH)+"/lib");
-      s_loader->LoadLibrary("LHAPDF");
+//      s_loader->AddPath(std::string(LHAPDF_PATH)+"/lib");
+//      s_loader->LoadLibrary("LHAPDF");
 #else
       THROW(fatal_error, "Sherpa not compiled with LHAPDF support.");
 #endif
