diff -Naur SHERPA-MC-2.1.0-orig/PDF/LHAPDF/Makefile.am SHERPA-MC-2.1.0/PDF/LHAPDF/Makefile.am
--- SHERPA-MC-2.1.0-orig/PDF/LHAPDF/Makefile.am	2014-04-11 12:03:48.000000000 +0200
+++ SHERPA-MC-2.1.0/PDF/LHAPDF/Makefile.am	2014-04-11 12:08:02.000000000 +0200
@@ -27,5 +27,6 @@
 libLHAPDFSherpa_la_LIBADD = @CONDITIONAL_LHAPDFLIBS@
 
 libLHAPDFSherpa_la_CPPFLAGS = $(AM_CPPFLAGS) @CONDITIONAL_LHAPDFINCS@
+libLHAPDFSherpa_la_LDFLAGS = -lz -L@CONDITIONAL_LHAPDFDIR@/lib -Bstatic -lLHAPDF
 
 EXTRA_DIST    = $(LHAPDFEXTRADIST)
diff -Naur SHERPA-MC-2.1.0-orig/SHERPA/Initialization/Initialization_Handler.C SHERPA-MC-2.1.0/SHERPA/Initialization/Initialization_Handler.C
--- SHERPA-MC-2.1.0-orig/SHERPA/Initialization/Initialization_Handler.C	2014-04-11 12:04:33.000000000 +0200
+++ SHERPA-MC-2.1.0/SHERPA/Initialization/Initialization_Handler.C	2014-04-11 12:08:52.000000000 +0200
@@ -527,8 +527,8 @@
     if (*pdflib=="None") continue;
     if (*pdflib=="LHAPDFSherpa") {
 #ifdef USING__LHAPDF
-      s_loader->AddPath(std::string(LHAPDF_PATH)+"/lib");
-      s_loader->LoadLibrary("LHAPDF");
+    //  s_loader->AddPath(std::string(LHAPDF_PATH)+"/lib");
+    //  s_loader->LoadLibrary("LHAPDF");
 #else
       THROW(fatal_error, "Sherpa not compiled with LHAPDF support.");
 #endif
# bla bla
