diff --git a/MODEL/Main/SM_U1_B.C b/MODEL/Main/SM_U1_B.C
index 1c6d011..97d1cbb 100644
--- a/MODEL/Main/SM_U1_B.C
+++ b/MODEL/Main/SM_U1_B.C
@@ -175,10 +175,10 @@ void SM_U1_B::FixMix() {
       for (int j=0;j<3;++j) {
 	Complex entry=Complex(0.,0.);
 	for (int k=0;k<3;++k) entry += Mix[i][k]*Mixconj[k][j];
-	if (ATOOLS::dabs(entry.real())<1.e-12) entry.real() = 0.;
-	if (ATOOLS::dabs(entry.imag())<1.e-12) entry.imag() = 0.;
-	if (ATOOLS::dabs(1.-entry.real())<1.e-12) entry.real() = 1.;
-	if (ATOOLS::dabs(1.-entry.imag())<1.e-12) entry.imag() = 1.;
+	if (ATOOLS::dabs(entry.real())<1.e-12) entry.real(0.);
+	if (ATOOLS::dabs(entry.imag())<1.e-12) entry.imag(0.);
+	if (ATOOLS::dabs(1.-entry.real())<1.e-12) entry.real(1.);
+	if (ATOOLS::dabs(1.-entry.imag())<1.e-12) entry.imag(1.);
 	msg_Out()<<std::setw(os)<<entry;
       }
       msg_Out()<<"\n";
