diff --git a/AMEGIC++/Main/Single_Process.C b/AMEGIC++/Main/Single_Process.C
index b14ddcf..7dc6630 100644
--- a/AMEGIC++/Main/Single_Process.C
+++ b/AMEGIC++/Main/Single_Process.C
@@ -14,6 +14,8 @@
 #include "ATOOLS/Org/MyStrStream.H"
 #include "ATOOLS/Org/Data_Reader.H"
 
+#include <unistd.h>
+
 using namespace AMEGIC;
 using namespace MODEL;
 using namespace PHASIC;
diff --git a/AMEGIC++/Main/Single_Process_Combined.C b/AMEGIC++/Main/Single_Process_Combined.C
index 5bb02f9..3fcb636 100644
--- a/AMEGIC++/Main/Single_Process_Combined.C
+++ b/AMEGIC++/Main/Single_Process_Combined.C
@@ -14,6 +14,8 @@
 #include "ATOOLS/Org/MyStrStream.H"
 #include "ATOOLS/Org/Data_Reader.H"
 
+#include <unistd.h>
+
 using namespace AMEGIC;
 using namespace MODEL;
 using namespace PHASIC;
diff --git a/ATOOLS/Org/Shell_Tools.C b/ATOOLS/Org/Shell_Tools.C
index 18055c5..0e6abbe 100644
--- a/ATOOLS/Org/Shell_Tools.C
+++ b/ATOOLS/Org/Shell_Tools.C
@@ -15,6 +15,8 @@
 #include <iostream>
 #endif
 
+#include <unistd.h>
+
 using namespace ATOOLS;
 
 bool ATOOLS::MakeDir(std::string path,const bool create_tree,
diff --git a/MODEL/Main/SM_U1_B.C b/MODEL/Main/SM_U1_B.C
index 1c6d011..97d1cbb 100644
--- a/MODEL/Main/SM_U1_B.C
+++ b/MODEL/Main/SM_U1_B.C
@@ -175,10 +175,10 @@ void SM_U1_B::FixMix() {
       for (int j=0;j<3;++j) {
 	Complex entry=Complex(0.,0.);
 	for (int k=0;k<3;++k) entry += Mix[i][k]*Mixconj[k][j];
-	if (ATOOLS::dabs(entry.real())<1.e-12) entry.real() = 0.;
-	if (ATOOLS::dabs(entry.imag())<1.e-12) entry.imag() = 0.;
-	if (ATOOLS::dabs(1.-entry.real())<1.e-12) entry.real() = 1.;
-	if (ATOOLS::dabs(1.-entry.imag())<1.e-12) entry.imag() = 1.;
+	if (ATOOLS::dabs(entry.real())<1.e-12) entry.real(0.);
+	if (ATOOLS::dabs(entry.imag())<1.e-12) entry.imag(0.);
+	if (ATOOLS::dabs(1.-entry.real())<1.e-12) entry.real(1.);
+	if (ATOOLS::dabs(1.-entry.imag())<1.e-12) entry.imag(1.);
 	msg_Out()<<std::setw(os)<<entry;
       }
       msg_Out()<<"\n";
