diff -Naur SHERPA-MC-2.0.beta2.orig/PDF/LHAPDF/Makefile.am SHERPA-MC-2.0.beta2/PDF/LHAPDF/Makefile.am
--- PDF/LHAPDF/Makefile.am	2013-03-18 18:04:56.000000000 +0100
+++ PDF/LHAPDF/Makefile.am	2013-03-18 18:06:31.000000000 +0100
@@ -18,5 +18,6 @@
 libLHAPDFSherpa_la_LIBADD = @CONDITIONAL_LHAPDFLIBS@
 
 libLHAPDFSherpa_la_CPPFLAGS = $(AM_CPPFLAGS) @CONDITIONAL_LHAPDFINCS@
+libLHAPDFSherpa_la_LDFLAGS = -lz -L@CONDITIONAL_LHAPDFDIR@/lib -Bstatic -lLHAPDF
 
 EXTRA_DIST    = @SVNINCLUDE@ $(LHAPDFEXTRADIST)
diff -Naur SHERPA-MC-2.0.beta2.orig/SHERPA/Initialization/Initialization_Handler.C SHERPA-MC-2.0.beta2/SHERPA/Initialization/Initialization_Handler.C
--- SHERPA/Initialization/Initialization_Handler.C	2013-03-18 18:04:53.000000000 +0100
+++ SHERPA/Initialization/Initialization_Handler.C	2013-03-18 18:07:21.000000000 +0100
@@ -549,8 +549,8 @@
     if (*pdflib=="None") continue;
     if (*pdflib=="LHAPDFSherpa") {
 #ifdef USING__LHAPDF
-      s_loader->AddPath(std::string(LHAPDF_PATH)+"/lib");
-      s_loader->LoadLibrary("LHAPDF");
+//      s_loader->AddPath(std::string(LHAPDF_PATH)+"/lib");
+//      s_loader->LoadLibrary("LHAPDF");
 #else
       THROW(fatal_error, "Sherpa not compiled with LHAPDF support.");
 #endif
