diff --git a/Makefile.am b/Makefile.am
index c85ab94..9a8fb49 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -18,9 +18,7 @@ SUBDIRS       = \
 		PHOTONS++ \
 		SHRiMPS \
 		SHERPA \
-		AddOns \
-		Manual \
-		Examples
+		AddOns
 
 EXTRA_DIST = GUIDELINES
 
diff --git a/Makefile.in b/Makefile.in
index 90d6901..2beff38 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -409,9 +409,7 @@ SUBDIRS = \
 		PHOTONS++ \
 		SHRiMPS \
 		SHERPA \
-		AddOns \
-		Manual \
-		Examples
+		AddOns
 
 EXTRA_DIST = GUIDELINES
 all: config.h
# bla bla
