diff --git a/AMEGIC++/Amplitude/Vertex.C b/AMEGIC++/Amplitude/Vertex.C
index 53da1de..56a7986 100644
--- a/AMEGIC++/Amplitude/Vertex.C
+++ b/AMEGIC++/Amplitude/Vertex.C
@@ -428,9 +428,9 @@ void Vertex::ColorExchange(MODEL::Color_Function* colfunc,int new0,int new1,int
 {
   //T[0,1,2] -> T[new0,new1,new2]
   int  partarg[3]  ={-1,-1,-1};
-  char strarg[3]   ={-1,-1,-1};
+  signed char strarg[3]   ={-1,-1,-1};
   int  partargn[3] ={-1,-1,-1};
-  char strargn[3]  ={-1,-1,-1};
+  signed char strargn[3]  ={-1,-1,-1};
  
   for (short int i=0;i<3;i++) {
     if (colfunc->Type()==MODEL::cf::D && i==2) break;
# bla bla
