diff --git a/PDF/LHAPDF/LHAPDF_Fortran_Interface.C b/PDF/LHAPDF/LHAPDF_Fortran_Interface.C
index 1edde9e..0f69dcf 100644
--- a/PDF/LHAPDF/LHAPDF_Fortran_Interface.C
+++ b/PDF/LHAPDF/LHAPDF_Fortran_Interface.C
@@ -11,11 +11,7 @@
 #include <dirent.h>
 #include <cstring>
 
-#ifdef ARCH_LINUX
 #define DIRENT_TYPE const dirent
-#else
-#define DIRENT_TYPE dirent
-#endif
 
 #ifndef _D_EXACT_NAMLEN
 #define _D_EXACT_NAMLEN(ENTRY) ENTRY->d_namlen
