diff --git a/AMEGIC++/Main/makelibs.in b/AMEGIC++/Main/makelibs.in
index 728f3a245..0b77f63ec 100755
--- a/AMEGIC++/Main/makelibs.in
+++ b/AMEGIC++/Main/makelibs.in
@@ -1,5 +1,5 @@
 #!/bin/bash
-# 
+#
 # To successfully employ this script, make sure that your bash is in
 # the bin - directory or adjust accordingly.
 #
@@ -37,7 +37,7 @@ do
   \?)
     shift `expr $OPTIND - 1`
     if [ "$1" = "--help" ]; then print_help && exit 0;
-    else 
+    else
       echo -n "makelibs: error: unrecognized option "
       if [ $OPTARG != "-" ]; then echo "'-$OPTARG'. try '-h'"
       else echo "'$1'. try '-h'"
@@ -49,13 +49,13 @@ do
   esac
 done
 
-if [ "$ONE" = "TRUE" ] ; then 
+if [ "$ONE" = "TRUE" ] ; then
   echo "makelibs: select module ( q) to quit )"
   select DIRS in $DIRS ; do
-    if [ "$DIRS" != "" ] ; then 
+    if [ "$DIRS" != "" ] ; then
       SINGLE="TRUE"
       break
-    else 
+    else
       if [ "$REPLY" = "q" ] || [ "$REPLY" = "Q" ]  ; then exit ; fi
     fi
   done
@@ -68,7 +68,7 @@ if test -z "$CFLAGS"; then export CFLAGS="@CFLAGS@"; fi
 if test -z "$FC"; then export FC="@FC@"; fi
 if test -z "$FCFLAGS"; then export FCFLAGS="@FCFLAGS@"; fi
 if test -z "$CMAKE"; then export CMAKE="@CMAKE_COMMAND@"; fi
-if test -z "$SHERPA_CMAKE_PATH"; then export SHERPA_CMAKE_PATH="@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_DATADIR@/SHERPA-MC"; fi
+if test -z "$SHERPA_CMAKE_PATH"; then export SHERPA_CMAKE_PATH="${SHERPA3_CMAKE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_DATADIR@/SHERPA-MC@VERSIONING@}"; fi
 if [[ ! -f $CMAKE ]]; then
   echo "makelibs: the original cmake is not present in the system. Falling back to 'cmake3' or 'cmake'"
   if [[ `type -p cmake3` != "" ]]; then
@@ -77,7 +77,7 @@ if [[ ! -f $CMAKE ]]; then
     if [[ `type -p cmake` != "" ]]; then
       export CMAKE=cmake
     else
-      echo "makelibs: cannot find cmake in the system" 
+      echo "makelibs: cannot find cmake in the system"
       exit 1
     fi
   fi
@@ -89,7 +89,7 @@ for J in $DIRS ; do
   echo "======================"
   echo "$J";
   echo "======================"
-  cd $J 
+  cd $J
   test -f CMakeLists.txt.orig || cp CMakeLists.txt CMakeLists.txt.orig
   if test $LIBMODE -eq 1; then
     cp CMakeLists.txt.orig CMakeLists.txt
diff --git a/ATOOLS/Org/Run_Parameter.C b/ATOOLS/Org/Run_Parameter.C
index 675eb5358..651c580a3 100644
--- a/ATOOLS/Org/Run_Parameter.C
+++ b/ATOOLS/Org/Run_Parameter.C
@@ -164,15 +164,15 @@ void Run_Parameter::AnalyseEnvironment()
   }
 
   gen.m_variables["SHERPA_SHARE_PATH"]=
-    (var=getenv("SHERPA_SHARE_PATH"))==NULL?sharepath:var;
+    (var=getenv("SHERPA3_SHARE_PATH"))==NULL?sharepath:var;
 
   // set include path
   gen.m_variables["SHERPA_INC_PATH"]=
-    (var=getenv("SHERPA_INCLUDE_PATH"))==NULL?includepath:var;
+    (var=getenv("SHERPA3_INCLUDE_PATH"))==NULL?includepath:var;
 
   // set library path
   gen.m_variables["SHERPA_LIBRARY_PATH"]=
-    (var=getenv("SHERPA_LIBRARY_PATH"))==NULL?librarypath:var;
+    (var=getenv("SHERPA3_LIBRARY_PATH"))==NULL?librarypath:var;
 
 }
 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index d421173a2..25c1f47bc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,7 @@ else()
 endif()
 message(STATUS "SHERPA: CMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}")
 if (SHERPA_ENABLE_VERSIONING)
-  set (VERSIONING "-${EXTERNAL_VERSION_STRING}")
+  set (VERSIONING "${SHERPA_VERSION_MAJOR}")
 else()
   set (VERSIONING "")
 endif()
@@ -338,7 +338,7 @@ macro(sherpa_handle_shared_library  mylib mylibname)
     $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC${VERSIONING}>)
   install(TARGETS ${mylib} EXPORT SHERPATargets DESTINATION ${CMAKE_INSTALL_LIBDIR}/SHERPA-MC${VERSIONING} COMPONENT libs)
-  set_target_properties(${mylib}        PROPERTIES POSITION_INDEPENDENT_CODE ON  SOVERSION 0.0.0)
+  set_target_properties(${mylib}        PROPERTIES POSITION_INDEPENDENT_CODE ON  SOVERSION ${SHERPA_VERSION_MAJOR})
   set_target_properties(${mylib}        PROPERTIES DEFINE_SYMBOL "")
   set_target_properties(${mylib}        PROPERTIES OUTPUT_NAME ${mylibname})
   target_compile_features(${mylib} PRIVATE cxx_std_11)
@@ -956,7 +956,7 @@ $<IF:$<BOOL:${SHERPA_ENABLE_EWSUD}>,#define USING__EWSud \"1\",/* #undef USING__
 file(GENERATE OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/ATOOLS/Org/CXXFLAGS_PACKAGES.H CONTENT "${CXXFLAGS_PACKAGES_H_CONTENT}")
 ########################################################################
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/AMEGIC++/Main/makelibs.in ${CMAKE_CURRENT_BINARY_DIR}/AMEGIC++/Main/makelibs @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/AMEGIC++/Main/makelibs    DESTINATION  ${CMAKE_INSTALL_DATADIR}/SHERPA-MC COMPONENT doc PERMISSIONS  OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE )
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/AMEGIC++/Main/makelibs    DESTINATION  ${CMAKE_INSTALL_DATADIR}/SHERPA-MC${VERSIONING} COMPONENT doc PERMISSIONS  OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE )
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/SHERPA/Run/Sherpa-config.in ${CMAKE_CURRENT_BINARY_DIR}/SHERPA/Run/Sherpa${VERSIONING}-config @ONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/EXTAMP/Example/madgraph/CMakeLists.txt.in ${CMAKE_CURRENT_BINARY_DIR}/EXTAMP/Example/madgraph/CMakeLists.txt @ONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/EXTAMP/Example/subtraction/CMakeLists.txt.in ${CMAKE_CURRENT_BINARY_DIR}/EXTAMP/Example/subtraction/CMakeLists.txt @ONLY)
@@ -964,9 +964,9 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Examples/V_plus_Bs/LHC_Wbb/CMakeLists
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Examples/V_plus_Jets/LHC_WJets/CMakeLists.txt.in ${CMAKE_CURRENT_BINARY_DIR}/Examples/V_plus_Jets/LHC_WJets/CMakeLists.txt @ONLY)
 
 if (SHERPA_ENABLE_UFO)
-SET(SHERPA_CMAKE_CONFIG_DIR ${CMAKE_INSTALL_PREFIX}/share/SHERPA-MC/)
+SET(SHERPA_CMAKE_CONFIG_DIR ${CMAKE_INSTALL_PREFIX}/share/SHERPA-MC${VERSIONING}/)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/MODEL/UFO/src/ufo_interface/parser.py.in ${CMAKE_CURRENT_BINARY_DIR}/MODEL/UFO/src/ufo_interface/parser.py @ONLY)
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/MODEL/UFO/bin/Sherpa-generate-model.in ${CMAKE_CURRENT_BINARY_DIR}/MODEL/UFO/bin/Sherpa-generate-model @ONLY)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/MODEL/UFO/bin/Sherpa-generate-model.in ${CMAKE_CURRENT_BINARY_DIR}/MODEL/UFO/bin/Sherpa${VERSIONING}-generate-model @ONLY)
 if (NOT SHERPA_ENABLE_PYTHON AND SHERPA_ENABLE_TESTING)
     message(FATAL_ERROR "SHERPA: UFO model generation tests requires Python Addons!")
 endif()
@@ -984,17 +984,17 @@ add_custom_command(TARGET unZipTestProcesses PRE_BUILD
      VERBATIM)
 endif()
 endif()
-install(EXPORT SHERPATargets DESTINATION ${CMAKE_INSTALL_DATADIR}/SHERPA-MC)
+install(EXPORT SHERPATargets DESTINATION ${CMAKE_INSTALL_DATADIR}/SHERPA-MC${VERSIONING})
 
 ########################################################################
 include(CMakePackageConfigHelpers)
 configure_package_config_file(cmake/Templates/SHERPA-MCConfig.cmake.in ${PROJECT_BINARY_DIR}/outputs/share/SHERPA-MC/SHERPA-MCConfig.cmake
-                              INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/SHERPA-MC)
+                              INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/SHERPA-MC${VERSIONING})
 write_basic_package_version_file(${PROJECT_BINARY_DIR}/outputs/share/SHERPA-MC/SHERPA-MCConfig-version.cmake
                                  COMPATIBILITY SameMajorVersion VERSION ${PROJECT_VERSION})
 install(FILES ${PROJECT_BINARY_DIR}/outputs/share/SHERPA-MC/SHERPA-MCConfig.cmake
               ${PROJECT_BINARY_DIR}/outputs/share/SHERPA-MC/SHERPA-MCConfig-version.cmake
-              DESTINATION ${CMAKE_INSTALL_DATADIR}/SHERPA-MC COMPONENT devel)
+              DESTINATION ${CMAKE_INSTALL_DATADIR}/SHERPA-MC${VERSIONING} COMPONENT devel)
 export(EXPORT SHERPATargets  FILE ${PROJECT_BINARY_DIR}/outputs/share/SHERPA-MC/SHERPATargets.cmake)
 ########################################################################
 if (CMAKE_SYSTEM_NAME MATCHES Darwin)
@@ -1031,17 +1031,17 @@ $<IF:$<BOOL:${SHERPA_ENABLE_BINRELOC}>,#define ENABLE_BINRELOC \"1\",/* #undef E
 /* Sherpa installation prefix */\n\
 #define SHERPA_PREFIX \"${CMAKE_INSTALL_PREFIX}\"\n\n\
 /* Sherpa include directory */\n\
-#define SHERPA_INCLUDE_PATH \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC\"\n\n\
+#define SHERPA_INCLUDE_PATH \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC${VERSIONING}\"\n\n\
 /* Sherpa include subdirectory */\n\
-#define SHERPA_INCLUDE_SUBDIR \"${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC\"\n\n\
+#define SHERPA_INCLUDE_SUBDIR \"${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC${VERSIONING}\"\n\n\
 /* Sherpa library directory */\n\
-#define SHERPA_LIBRARY_PATH \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/SHERPA-MC/\"\n\n\
+#define SHERPA_LIBRARY_PATH \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/SHERPA-MC${VERSIONING}/\"\n\n\
 /* Sherpa library subdirectory */\n\
-#define SHERPA_LIBRARY_SUBDIR \"${CMAKE_INSTALL_LIBDIR}/SHERPA-MC/\"\n\n\
+#define SHERPA_LIBRARY_SUBDIR \"${CMAKE_INSTALL_LIBDIR}/SHERPA-MC${VERSIONING}/\"\n\n\
 /* Sherpa data directory */\n\
-#define SHERPA_SHARE_PATH \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/SHERPA-MC/\"\n\n\
+#define SHERPA_SHARE_PATH \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATADIR}/SHERPA-MC${VERSIONING}/\"\n\n\
 /* Sherpa subversion */\n\
-#define SHERPA_SHARE_SUBDIR \"${CMAKE_INSTALL_DATADIR}/SHERPA-MC/\"\n\n\
+#define SHERPA_SHARE_SUBDIR \"${CMAKE_INSTALL_DATADIR}/SHERPA-MC${VERSIONING}/\"\n\n\
 /* Sherpa subversion */\n\
 #define SHERPA_SUBVERSION \"${SHERPA_VERSION_MINOR}.${SHERPA_VERSION_PATCH}\"\n\n\
 /* Sherpa version */\n\
@@ -1071,7 +1071,7 @@ file(GENERATE OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/ATOOLS/Org/CXXFLAGS.H CONTENT "
 install(FILES
       ${CMAKE_CURRENT_BINARY_DIR}/ATOOLS/Org/CXXFLAGS.H
       ${CMAKE_CURRENT_BINARY_DIR}/ATOOLS/Org/CXXFLAGS_PACKAGES.H DESTINATION
-      ${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC/ATOOLS/Org/  COMPONENT devel )
+      ${CMAKE_INSTALL_INCLUDEDIR}/SHERPA-MC${VERSIONING}/ATOOLS/Org/  COMPONENT devel )
 ########################################################################
 if (SHERPA_ENABLE_TESTING)
   message(STATUS "SHERPA: Will use the installed SHERPA for testing! Run ctest only after the installation!")
diff --git a/MODEL/UFO/CMakeLists.txt b/MODEL/UFO/CMakeLists.txt
index 0db255640..67bbea5c3 100644
--- a/MODEL/UFO/CMakeLists.txt
+++ b/MODEL/UFO/CMakeLists.txt
@@ -17,7 +17,7 @@ file(COPY src/ufo_interface DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/src/)
 file(COPY src/opt_einsum/opt_einsum DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/src/)
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/src/
         DESTINATION ${SHERPA_Python_SITEARCH}/)
-install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/bin/Sherpa-generate-model
+install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/bin/Sherpa${VERSIONING}-generate-model
         DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 if(SHERPA_ENABLE_TESTING)
diff --git a/SHERPA/Run/Sherpa-config.in b/SHERPA/Run/Sherpa-config.in
index 5689776c9..f51ddde53 100644
--- a/SHERPA/Run/Sherpa-config.in
+++ b/SHERPA/Run/Sherpa-config.in
@@ -6,37 +6,37 @@ case $1 in
   --prefix)
     echo @CMAKE_INSTALL_PREFIX@;;
   --libs)
-    echo -L${SHERPA_LIBRARY_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/SHERPA-MC} -lSherpaMain @CONDITIONAL_HEPMC3LIBS@ @RIVET_CONFIG_LIBS_STRING@ @ROOT_CONFIG_LIBS_STRING@ -L@LHAPDF_LIBRARY_DIR@ -lLHAPDF;;
+    echo -L${SHERPA_LIBRARY_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/SHERPA-MC@VERSIONING@} -lSherpaMain @CONDITIONAL_HEPMC3LIBS@ @RIVET_CONFIG_LIBS_STRING@ @ROOT_CONFIG_LIBS_STRING@ -L@LHAPDF_LIBRARY_DIR@ -lLHAPDF;;
   --python-libs)
     echo @SHERPA_Python_SITEARCH@;;
   --load-mpi4py)
     echo @LOADMPIFORPY@;;
   --incdir)
-    echo ${SHERPA_INCLUDE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/SHERPA-MC};;
+    echo ${SHERPA_INCLUDE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/SHERPA-MC@VERSIONING@};;
   --libdir)
-    echo ${SHERPA_LIBRARY_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/SHERPA-MC};;
+    echo ${SHERPA_LIBRARY_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/SHERPA-MC@VERSIONING@};;
   --datadir)
-    echo ${SHERPA_SHARE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_DATADIR@/SHERPA-MC};;
+    echo ${SHERPA_SHARE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_DATADIR@/SHERPA-MC@VERSIONING@};;
   --ldflags)
-    echo -L${SHERPA_LIBRARY_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/SHERPA-MC} -lSherpaMain @CONDITIONAL_HEPMC3LIBS@ @RIVET_CONFIG_LIBS_STRING@ @ROOT_CONFIG_LIBS_STRING@ -L@LHAPDF_LIBRARY_DIR@ -lLHAPDF;;
+    echo -L${SHERPA_LIBRARY_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/SHERPA-MC@VERSIONING@} -lSherpaMain @CONDITIONAL_HEPMC3LIBS@ @RIVET_CONFIG_LIBS_STRING@ @ROOT_CONFIG_LIBS_STRING@ -L@LHAPDF_LIBRARY_DIR@ -lLHAPDF;;
   --cppflags)
-    echo -I${SHERPA_INCLUDE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/SHERPA-MC} @CONDITIONAL_HEPMC3INCS@ @RIVET_CONFIG_CPPFLAGS_STRING@ @ROOT_CONFIG_CPPFLAGS_STRING@ @LHAPDF_CONFIG_CPPFLAGS_STRING@;;
+    echo -I${SHERPA_INCLUDE_PATH:-@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_INCLUDEDIR@/SHERPA-MC@VERSIONING@} @CONDITIONAL_HEPMC3INCS@ @RIVET_CONFIG_CPPFLAGS_STRING@ @ROOT_CONFIG_CPPFLAGS_STRING@ @LHAPDF_CONFIG_CPPFLAGS_STRING@;;
   --cxx)
     echo @CXX@;;
   --fc)
     echo @FC@;;
   *)
-    echo "Sherpa-config: configuration tool for the Monte Carlo";
+    echo "Sherpa3-config: configuration tool for the Monte Carlo";
     echo "               event generator Sherpa: http://sherpa.hepforge.org";
     echo "Options:";
-    echo "  Sherpa-config --version     : returns Sherpa version";
-    echo "  Sherpa-config --prefix      : returns the installation path";
-    echo "  Sherpa-config --incdir      : returns the path of Sherpa headers";
-    echo "  Sherpa-config --libs        : returns the linker line to link with Sherpa";
-    echo "  Sherpa-config --datadir     : returns the path of Sherpa data (like Decaydata, PDFs)";
-    echo "  Sherpa-config --ldflags     : returns the LDFLAGS used for Sherpa";
-    echo "  Sherpa-config --cppflags    : returns the preprocessor FLAGS needed for Sherpa headers";
-    echo "  Sherpa-config --cxx         : returns the C++ compiler used";
-    echo "  Sherpa-config --fc          : returns the FORTRAN compiler used";
-    echo "  Sherpa-config --python-libs : returns the installation path of python libraries";;
+    echo "  Sherpa3-config --version     : returns Sherpa version";
+    echo "  Sherpa3-config --prefix      : returns the installation path";
+    echo "  Sherpa3-config --incdir      : returns the path of Sherpa headers";
+    echo "  Sherpa3-config --libs        : returns the linker line to link with Sherpa";
+    echo "  Sherpa3-config --datadir     : returns the path of Sherpa data (like Decaydata, PDFs)";
+    echo "  Sherpa3-config --ldflags     : returns the LDFLAGS used for Sherpa";
+    echo "  Sherpa3-config --cppflags    : returns the preprocessor FLAGS needed for Sherpa headers";
+    echo "  Sherpa3-config --cxx         : returns the C++ compiler used";
+    echo "  Sherpa3-config --fc          : returns the FORTRAN compiler used";
+    echo "  Sherpa3-config --python-libs : returns the installation path of python libraries";;
 esac
