diff -Naur orig.libsigc++-2.2.3/sigc++/signal.h libsigc++-2.2.3/sigc++/signal.h
--- orig.libsigc++-2.2.3/sigc++/signal.h	2008-02-28 17:47:42.000000000 +0100
+++ libsigc++-2.2.3/sigc++/signal.h	2011-03-05 19:22:39.000000000 +0100
@@ -10,6 +10,7 @@
 #include <sigc++/trackable.h>
 #include <sigc++/functors/slot.h>
 #include <sigc++/functors/mem_fun.h>
+#include <stddef.h>
 
 //SIGC_TYPEDEF_REDEFINE_ALLOWED:
 // TODO: This should have its own test, but I can not create one that gives the error instead of just a warning. murrayc.
diff -Naur orig.libsigc++-2.2.3/sigc++/signal_base.h libsigc++-2.2.3/sigc++/signal_base.h
--- orig.libsigc++-2.2.3/sigc++/signal_base.h	2008-02-26 20:44:34.000000000 +0100
+++ libsigc++-2.2.3/sigc++/signal_base.h	2011-03-05 19:22:31.000000000 +0100
@@ -27,6 +27,7 @@
 #include <sigc++/trackable.h>
 #include <sigc++/functors/slot.h>
 #include <sigc++/functors/mem_fun.h>
+#include <stddef.h>
 
 namespace sigc
 {
# bla bla
