--- orig.tauola++/1.0.2a/Makefile	2010-09-29 11:20:06.000000000 +0200
+++ tauola++/1.0.2a/Makefile	2011-03-24 14:35:37.000000000 +0100
@@ -3,9 +3,9 @@
 LIB_VER = 1.0.0
 
 #Name of libraries to create
-LIB_TAUOLA_CXX_INT_SO = libTauolaCxxInterface.so
+LIB_TAUOLA_CXX_INT_SO = libTauolaCxxInterface
 LIB_TAUOLA_CXX_INT_A  = libTauolaCxxInterface.a
-LIB_TAUOLA_FORTRAN_SO = libTauolaFortran.so
+LIB_TAUOLA_FORTRAN_SO = libTauolaFortran
 LIB_TAUOLA_FORTRAN_A  = libTauolaFortran.a
 
 TAUOLA_CXX_INT_OBJECTS = src/$(FORTRAN_TAUOLA_INTERFACE_DIR)/*.o \
@@ -32,10 +32,10 @@
 src: $(EVENT_RECORD_INTERFACE_DIR) $(FORTRAN_TAUOLA_INTERFACE_DIR) $(C_TAUOLA_INTERFACE_DIR) $(UTILITIES_DIR)
 	ar cr lib/$(LIB_TAUOLA_CXX_INT_A) $(TAUOLA_CXX_INT_OBJECTS)
 	ar cr lib/$(LIB_TAUOLA_FORTRAN_A) $(TAUOLA_FORTRAN_OBJECTS)
-	$(LD) $(LDFLAGS) $(SOFLAGS) -o lib/$(LIB_TAUOLA_CXX_INT_SO).$(LIB_VER) $(TAUOLA_CXX_INT_OBJECTS)
-	$(LD) $(LDFLAGS) $(SOFLAGS) -o lib/$(LIB_TAUOLA_FORTRAN_SO).$(LIB_VER) $(TAUOLA_FORTRAN_OBJECTS)
-	ln -sf $(LIB_TAUOLA_CXX_INT_SO).$(LIB_VER) lib/$(LIB_TAUOLA_CXX_INT_SO)
-	ln -sf $(LIB_TAUOLA_FORTRAN_SO).$(LIB_VER) lib/$(LIB_TAUOLA_FORTRAN_SO)
+	$(LD) $(LDFLAGS) $(SOFLAGS) -install_name $(LIB_TAUOLA_CXX_INT_SO).$(LIB_VER).dylib -o lib/$(LIB_TAUOLA_CXX_INT_SO).$(LIB_VER).dylib $(TAUOLA_CXX_INT_OBJECTS)
+	$(LD) $(LDFLAGS) $(SOFLAGS) -install_name $(LIB_TAUOLA_FORTRAN_SO).$(LIB_VER).dylib -o lib/$(LIB_TAUOLA_FORTRAN_SO).$(LIB_VER).dylib $(TAUOLA_FORTRAN_OBJECTS)
+	ln -sf $(LIB_TAUOLA_CXX_INT_SO).$(LIB_VER).dylib lib/$(LIB_TAUOLA_CXX_INT_SO).dylib
+	ln -sf $(LIB_TAUOLA_FORTRAN_SO).$(LIB_VER).dylib lib/$(LIB_TAUOLA_FORTRAN_SO).dylib
 	@echo "##################################################################"	
 	@echo " Tauola C++ Interface library created and moved to lib/ directory "
 	@echo "##################################################################"
# bla bla
