--- orig.tauola/27.121.5/pretauola/tauola_srs.F	2007-05-14 18:38:41.000000000 +0200
+++ tauola/27.121.5/pretauola/tauola_srs.F	2010-11-25 11:19:06.000000000 +0100
@@ -95,9 +95,16 @@
 *  srs
 * 
 *------------
+      integer         MSTU,               MSTJ
+      double precision          PARU,               PARJ     
+      COMMON /PYDAT1/ MSTU(200),PARU(200),MSTJ(200),PARJ(200)
+
       IF(MODE.EQ.-1) THEN
 C     ***********************
 
+C     SWITCH OFF TAU-PHOTON RADIATION IN PY6
+         MSTJ(39) = 15
+
          JAK1  =  PJAK1      ! decay mode first tau 
          JAK2  =  PJAK2      ! decay mode second tau
 *         JAK1  =  0     ! decay mode first tau 
@@ -243,6 +250,9 @@
             ELSEIF ((IMOTH.EQ.IM0.OR.IMOTH.EQ.IM).AND.ISEL.EQ.0) THEN
                ISON(2)=I
             ELSEIF ((IMOTH.NE.IM0.AND.IMOTH.NE.IM).AND.ISEL.EQ.0) THEN
+               IF( IDHEP(I).EQ.22. AND. JMOHEP(1,I).GT.0 ) THEN
+                  IF(ABS(IDHEP(JMOHEP(1,I))).NE.KFTAU) GOTO 76
+               ENDIF
                ISEL=1
                GOTO 77
             ENDIF
