diff --git a/27.121.5/src/taueta.F b/27.121.5/src/taueta.F
index 15cd088..a82149d 100644
--- a/27.121.5/src/taueta.F
+++ b/27.121.5/src/taueta.F
@@ -43,7 +43,7 @@ c
       REAL  PHOT1(4),PHOT2(4),PHOT3(4)
       REAL*8    X(4),    Y(4),    Z(4)
       REAL                                YM1,YM2,YM3
-      REAL*8  R,RU,PETA(4),XM1,XM2,XM3,XM,XLAM
+      REAL*8  R,RU,PETA(4),XM1,XM2,XM3,XM,AM2,XLAM
       REAL*8 a,b,c
       REAL*8 AM2
       XLAM(a,b,c)=SQRT(ABS((a-b-c)**2-4.0*b*c))
diff --git a/27.121.5/src/taueta.F b/27.121.5/src/taueta.F
index a82149d..ced7f80 100644
--- a/27.121.5/src/taueta.F
+++ b/27.121.5/src/taueta.F
@@ -45,7 +45,6 @@ c
       REAL                                YM1,YM2,YM3
       REAL*8  R,RU,PETA(4),XM1,XM2,XM3,XM,AM2,XLAM
       REAL*8 a,b,c
-      REAL*8 AM2
       XLAM(a,b,c)=SQRT(ABS((a-b-c)**2-4.0*b*c))
 C position of decaying particle:
       IF((KTO.EQ. 1).OR.(KTO.EQ.11)) THEN
# bla bla
