diff --git a/tauola/27.121.5/Makefile b/tauola/27.121.5/Makefile
index acb9e77..3f89fda 100644
--- a/tauola/27.121.5/Makefile
+++ b/tauola/27.121.5/Makefile
@@ -50,20 +50,12 @@ all: libs
 libs:
 	@echo FFLAGS_tauola = $(FFLAGS_tauola)
 
-	@LIBNAME=$(LIBDIR)/libtauola.so ; DIRNAME=src ; FFLAGS="$(FFLAGS_tauola)" ; \
-	export LIBNAME ; export DIRNAME ; export FFLAGS; \
-	$(MAKE) -f Makeshared.subdir
-
 	@LIBNAME=$(LIBDIR)/archive/libtauola.a ; DIRNAME=src ; FFLAGS="$(FFLAGS_tauola)" ; \
 	export LIBNAME ; export DIRNAME ; export FFLAGS; \
 	$(MAKE) -f Makearchive.subdir
 
 	@echo FFLAGS_pretauola = $(FFLAGS_pretauola)
 
-	@LIBNAME=$(LIBDIR)/libpretauola.so ; DIRNAME=pretauola ; FFLAGS="$(FFLAGS_pretauola)" ; \
-	export LIBNAME ; export DIRNAME ; export FFLAGS; \
-	$(MAKE) -f Makeshared.subdir
-
 	@LIBNAME=$(LIBDIR)/archive/libpretauola.a ; DIRNAME=pretauola ; FFLAGS="$(FFLAGS_pretauola)" ; \
 	export LIBNAME ; export DIRNAME ; export FFLAGS; \
 	$(MAKE) -f Makearchive.subdir
