--- orig.tauola/27.121.5/src/taueta.F	2007-01-21 21:54:54.000000000 +0100
+++ tauola/27.121.5/src/taueta.F	2008-11-08 17:33:26.000000000 +0100
@@ -43,7 +43,7 @@
       REAL  PHOT1(4),PHOT2(4),PHOT3(4)
       REAL*8    X(4),    Y(4),    Z(4)
       REAL                                YM1,YM2,YM3
-      REAL*8  R,RU,PETA(4),XM1,XM2,XM3,XM,XLAM
+      REAL*8  R,RU,PETA(4),XM1,XM2,XM3,XM,AM2,XLAM
       REAL*8 a,b,c
       XLAM(a,b,c)=SQRT(ABS((a-b-c)**2-4.0*b*c))
 C position of decaying particle:
