diff --git a/tauola/27.121.5/Makeshared.subdir b/tauola/27.121.5/Makeshared.subdir
index cfb17bf..e4056c3 100644
--- a/tauola/27.121.5/Makeshared.subdir
+++ b/tauola/27.121.5/Makeshared.subdir
@@ -57,12 +57,24 @@ ifeq ($(PYTHIA6FORSHAREDLIBS),)
  $(LIBNAME): $(objects)
 	@mkdir -p $(LIBDIR)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@)
+	case $ARCH in \
+	   Linux*) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) ;; \
+	   Darwin*) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -dynamic ;; \
+	esac
 else
  $(LIBNAME): $(objects)
 	@mkdir -p $(LIBDIR)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) -L$(PYTHIA6FORSHAREDLIBS) -l$(PYTHIA6LIBNAMEFORSHAREDLIBS)
+	case ${ARCH} in \
+           Linux*) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) -L$(PYTHIA6FORSHAREDLIBS) -l$(PYTHIA6LIBNAMEFORSHAREDLIBS) \
+	   ;; \
+	   Darwin*) \
+	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -dynamic -Wl,-commons,use_dylibs $(shell echo $@ | grep pretauola >/dev/null && echo lib/libtauola.dylib) -L$(PYTHIA6FORSHAREDLIBS) -l$(PYTHIA6LIBNAMEFORSHAREDLIBS) -L$(PHOTOS_ROOT)/lib -lphotos $(shell gfortran --print-file-name=libgfortran.a) \
+	   ;; \
+	esac
 endif
 
 deps := $(patsubst $(DIRNAME)/%.F,$(TMPDIR)/%.d,$(wildcard $(DIRNAME)/*.F))
diff --git a/tauola/27.121.5/configure b/tauola/27.121.5/configure
index 81e13c4..cd03ba9 100755
--- a/tauola/27.121.5/configure
+++ b/tauola/27.121.5/configure
@@ -141,7 +141,8 @@ if [ $ARCH = Linux ]; then
   fi
 fi
 # Linux platform with gcc4: new Fortran90 compiler.
-if [ $ARCH = Linux-gcc4 ]; then
+case $ARCH in 
+  *-gcc4 )
   export FC=gfortran
   export FFLAGS="${FFLAGS_OPT}"
   export CFLAGS="${CFLAGS_OPT}"
@@ -157,7 +158,8 @@ if [ $ARCH = Linux-gcc4 ]; then
     export CFLAGS="${CFLAGS_DBG}"
     export CXXFLAGS="${CXXFLAGS_DBG}"
   fi
-fi
+  ;;
+esac
 
 #Platform & opt/dbg - independent flags and variables:
 
diff --git a/tauola/27.121.5/Makeshared.subdir b/tauola/27.121.5/Makeshared.subdir
index 215a200..96a1af6 100644
--- a/tauola/27.121.5/Makeshared.subdir
+++ b/tauola/27.121.5/Makeshared.subdir
@@ -42,9 +42,9 @@ $(TMPDIR)/%.o : $(DIRNAME)/%.c
 # Creating the dependency files *.d
 
 $(TMPDIR)/%.d : $(DIRNAME)/%.F
-	@echo Making dependency for file $<; \
+	echo Making dependency for file $<; \
 	mkdir -p $(TMPDIR); \
-	$(CC) -M -I$(INCDIR) $< | \
+	$(FC) -M -I$(INCDIR) $< | \
 	sed 's,\($*\)\.o[ :]*,\1.o $@ : ,g' | \
 	sed 's/$*.o/$(TMPDIR)\/$*.o/' > $@; \
 	[ -s $@ ] || rm -f $@
