--- tensorflow/xla_aot_runtime_src/CMakeLists.txt	2024-08-20 16:04:12.973175653 +0200
+++ tensorflow/xla_aot_runtime_src/CMakeLists.txt	2024-08-20 16:06:37.748479188 +0200
@@ -14,6 +14,13 @@
   -Wno-sign-compare
 )
 
-add_library(tf_xla_runtime STATIC
+find_package(absl REQUIRED)
+add_library(tf_xla_runtime SHARED
   $<TARGET_OBJECTS:tf_xla_runtime_objects>
 )
+set(LIBFFT_WRAPPER "" CACHE FILEPATH "Path to the libfft shared library")
+if(NOT LIBFFT_WRAPPER)
+  message(FATAL_ERROR "LIBFFT_WRAPPER path not provided. Use -DLIBFFT_WRAPPER=<path to libfft_wrapper.so> to specify the library location.")
+endif()
+
+target_link_libraries(tf_xla_runtime absl::strings absl::str_format_internal ${LIBFFT_WRAPPER})
