diff --git a/configure b/configure
index 8925b2a..d8b115a 100755
--- a/configure
+++ b/configure
@@ -15723,7 +15723,7 @@ else
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $with_hepmc" >&5
 $as_echo "$with_hepmc" >&6; }
 	HEPMCINCLUDE=-I$with_hepmc/include
-	HEPMCLIBS="-L$with_hepmc/lib -R$with_hepmc/lib -lHepMC"
+	HEPMCLIBS="-L$with_hepmc/lib -lHepMC"
 fi
 
 if test "x$with_hepmc" != "xno"; then
