diff -arupN thepeg.orig/1.7.0/Analysis/ProgressLog.cc thepeg/1.7.0/Analysis/ProgressLog.cc
--- thepeg.orig/1.7.0/Analysis/ProgressLog.cc	2012-03-08 10:55:59.000000000 +0100
+++ thepeg/1.7.0/Analysis/ProgressLog.cc	2012-03-08 11:19:00.000000000 +0100
@@ -16,6 +16,7 @@
 #include "ThePEG/Persistency/PersistentIStream.h"
 
 #include <sys/times.h>
+#include <unistd.h>
 
 using namespace ThePEG;
 
diff -arupN thepeg.orig/1.7.0/Interface/ParVector.tcc thepeg/1.7.0/Interface/ParVector.tcc
--- thepeg.orig/1.7.0/Interface/ParVector.tcc	2012-03-08 10:55:59.000000000 +0100
+++ thepeg/1.7.0/Interface/ParVector.tcc	2012-03-08 10:59:07.000000000 +0100
@@ -298,16 +298,16 @@ template <typename T, typename Type>
 void ParVector<T,Type>::doxygenDescription(ostream & os) const {
   ParVectorTBase<Type>::doxygenDescription(os);
   os << "<b>Default value:</b> ";
-  putUnit(os, theDef);
+  this->putUnit(os, theDef);
   if ( theDefFn ) os << " (May be changed by member function.)";
   if ( ParVectorBase::lowerLimit() ) {
     os << "<br>\n<b>Minimum value:</b> ";
-    putUnit(os, theMin);
+    this->putUnit(os, theMin);
     if ( theMinFn ) os << " (May be changed by member function.)";
   }
   if ( ParVectorBase::upperLimit() ) {
     os << "<br>\n<b>Maximum value:</b> ";
-    putUnit(os, theMax);
+    this->putUnit(os, theMax);
     if ( theMaxFn ) os << " (May be changed by member function.)";
   }
   os << "<br>\n";
