--- orig.thepeg/1.7.0/configure	2011-03-25 15:19:42.000000000 +0100
+++ thepeg/1.7.0/configure	2011-03-25 15:31:56.000000000 +0100
@@ -15242,7 +15242,7 @@
 
     ORIG_LIBS="$LIBS"
     for readline_lib in readline edit editline; do
-      for termcap_lib in "" termcap curses ncurses; do
+      for termcap_lib in "" xxxtermcap xxxcurses xxxxncurses; do
         if test -z "$termcap_lib"; then
           TRY_LIB="-l$readline_lib"
         else
