--- orig.thepeg/1.6.1/PDF/LHAPDF.cc	2009-11-13 13:49:29.000000000 +0100
+++ thepeg/1.6.1/PDF/LHAPDF.cc	2010-12-10 14:34:53.000000000 +0100
@@ -34,6 +34,8 @@
 #include "ThePEG/Persistency/PersistentOStream.h"
 #include "ThePEG/Persistency/PersistentIStream.h"
 
+#include <cstdlib>
+
 #ifdef ThePEG_HAS_FPU_CONTROL
 #include <fpu_control.h>
 #endif
@@ -214,7 +216,16 @@
 
 std::string LHAPDF::getIndexPath() {
   // macro is passed in from -D compile flag
-  return std::string(LHAPDF_PKGDATADIR) + "/PDFsets.index";
+  //  return std::string(LHAPDF_PKGDATADIR) + "/PDFsets.index";
+  const char * cp = std::getenv("LHAPATH");
+  std::string path = "";
+  if ( cp ) {
+    path = (std::string)cp + "/../PDFsets.index";
+  }
+  else {
+    path = std::string(LHAPDF_PKGDATADIR) + "/PDFsets.index";
+  }
+  return path;
 }
 
 bool LHAPDF::openLHAIndex(ifstream & is) {
