diff --git a/FecSoftwareV3_0/FecPciDeviceDriver/include/datatypes.h b/FecSoftwareV3_0/FecPciDeviceDriver/include/datatypes.h
index ad5f313..af58310 100755
--- a/FecSoftwareV3_0/FecPciDeviceDriver/include/datatypes.h
+++ b/FecSoftwareV3_0/FecPciDeviceDriver/include/datatypes.h
@@ -54,8 +54,15 @@ Include home-made values definition
 /*!
 Include system defines
 */
+#if !defined(__APPLE__)
 #include <linux/kd.h>
 #include <linux/types.h>
+#else
+#include <stdint.h>
+#define __u32 uint32_t
+#define __u16 uint16_t
+#define __u8  uint8_t
+#endif
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/FecSoftwareV3_0/generic/include/keyType.h b/FecSoftwareV3_0/generic/include/keyType.h
index 5d944b0..85b035a 100755
--- a/FecSoftwareV3_0/generic/include/keyType.h
+++ b/FecSoftwareV3_0/generic/include/keyType.h
@@ -21,7 +21,6 @@
 #define KEYTYPE_H
 
 #include <cstdio>
-#include <asm/types.h>
 
 /* This file define the key to define a unique device, channel, ...   */
 /* The bits allowed are:                                              */
@@ -34,7 +33,13 @@
 /* XXXX X  XXX X  XXX XXXX  XXXX XXXX      XXXX XXXXX                 */
 
 /* Type of the key */
+#if !defined(__APPLE__)
+#include <asm/types.h>
 typedef __u32 keyType ;
+#else
+#include <stdint.h>
+typedef uint32_t keyType; typedef uint16_t __u16;
+#endif
 
 /* Value for each part */
 #define MASKFECKEY     0x1F
diff --git a/TrackerOnline/Fed9U/Fed9USoftware/Makevars b/TrackerOnline/Fed9U/Fed9USoftware/Makevars
index 3314811..3abf2e7 100755
--- a/TrackerOnline/Fed9U/Fed9USoftware/Makevars
+++ b/TrackerOnline/Fed9U/Fed9USoftware/Makevars
@@ -120,6 +120,7 @@ LIBS += $(ENV_CMS_TK_FED9U_HAL_LIBDIR) \
 	-L../Fed9UVmeBase/ \
 	-L../Fed9UVmeDevice/ \
 	-lm \
+        $(ENV_CMS_TK_FED9U_ORACLE_LIBRARY) \
 	$(CUSTOM_LIBS)
 #	-L$(ENV_CMS_TK_FED9U_DAQ_ROOT)/TriDAS/daq/extern/log4cplus/linuxx86/lib/ \
 #	-L$(ENV_CMS_TK_FED9U_DAQ_ROOT)/TriDAS/lib/linux/x86/               \
diff --git a/FecSoftwareV3_0/generic/include/FecRingRegisters.h b/FecSoftwareV3_0/generic/include/FecRingRegisters.h
index 364cb3a..3af98c0 100755
--- a/FecSoftwareV3_0/generic/include/FecRingRegisters.h
+++ b/FecSoftwareV3_0/generic/include/FecRingRegisters.h
@@ -698,11 +698,11 @@ class FecRingRegisters {
     ccuerrstr += "\n" ;
 
     ushort channelBusy[18] = {0} ;
-    for (ushort i = 0 ; i < 18 ; i ++)
+    for (unsigned short i = 0 ; i < 18 ; i ++)
       if (ccureg & (0x1 << i)) channelBusy[i] = 1 ;
 
     ccuerrstr += "  i2c busy channel: " ;
-    for (ushort i = 0 ; i < 16 ; i ++) 
+    for (unsigned short i = 0 ; i < 16 ; i ++) 
       if (channelBusy[i]) {
 	sprintf(hexwrd,"%d",(unsigned int)i+1);
 	ccuerrstr += hexwrd ;
@@ -825,9 +825,9 @@ class FecRingRegisters {
     ccuerrstr += hexwrd;
     ccuerrstr += "\n" ;
 
-    ushort channelEnable[23] = {0} ;
+    unsigned short channelEnable[23] = {0} ;
     bool i2cDisplay = false, piaDisplay = false ;
-    for (ushort i = 0 ; i < 23 ; i ++) {
+    for (unsigned short i = 0 ; i < 23 ; i ++) {
       channelEnable[i] = ccureg & (0x1 << i) ;
       if (channelEnable[i]) {
 	if (i < 16) i2cDisplay = true ;
@@ -839,7 +839,7 @@ class FecRingRegisters {
     ccuerrstr += "  i2c enabled: " ;
     if (! i2cDisplay) ccuerrstr += "None" ;
     else {
-      for (ushort i = 0 ; i < 16 ; i ++) {
+      for (unsigned short i = 0 ; i < 16 ; i ++) {
 	if (channelEnable[i]) {
 	  sprintf(hexwrd,"%d ",(i+1));
 	  ccuerrstr += hexwrd ;
@@ -851,7 +851,7 @@ class FecRingRegisters {
     ccuerrstr += "  PIA enabled: " ;
     if (! piaDisplay) ccuerrstr += "None" ;
     else {
-      for (ushort i = 16; i < 20 ; i ++) 
+      for (unsigned short i = 16; i < 20 ; i ++) 
 	if (channelEnable[i]) {
 	  sprintf(hexwrd,"%d",(i+1));
 	  ccuerrstr += hexwrd ;
diff --git a/FecSoftwareV3_0/generic/src/common/TShare.cc b/FecSoftwareV3_0/generic/src/common/TShare.cc
index a5d6c8c..48653ed 100644
--- a/FecSoftwareV3_0/generic/src/common/TShare.cc
+++ b/FecSoftwareV3_0/generic/src/common/TShare.cc
@@ -11,7 +11,7 @@
 
 #include <iostream>
 
-#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)
+#if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) || defined(__APPLE__)
 /* union semun is defined by including <sys/sem.h> */
 #else
 /* according to X/OPEN we have to define it ourselves */
diff --git a/TrackerOnline/Fed9U/Fed9USoftware/Fed9UDeviceFactory/src/Fed9UXMLDescriptionFromDb.cc b/TrackerOnline/Fed9U/Fed9USoftware/Fed9UDeviceFactory/src/Fed9UXMLDescriptionFromDb.cc
index 21a0cb9..4a73c2d 100755
--- a/TrackerOnline/Fed9U/Fed9USoftware/Fed9UDeviceFactory/src/Fed9UXMLDescriptionFromDb.cc
+++ b/TrackerOnline/Fed9U/Fed9USoftware/Fed9UDeviceFactory/src/Fed9UXMLDescriptionFromDb.cc
@@ -35,8 +35,6 @@
 #include "xercesc/framework/Wrapper4InputSource.hpp"
 #include "xercesc/util/BinInputStream.hpp"
 
-
-
 namespace Fed9U {
   u32 xmlBufferLength=0;
 //   using xercesc::XMLPlatformUtils;
diff --git a/FecSoftwareV3_0/generic/include/FecRingRegisters.h b/FecSoftwareV3_0/generic/include/FecRingRegisters.h
index 87407b4..3af98c0 100755
--- a/FecSoftwareV3_0/generic/include/FecRingRegisters.h
+++ b/FecSoftwareV3_0/generic/include/FecRingRegisters.h
@@ -697,7 +697,7 @@ class FecRingRegisters {
     ccuerrstr += hexwrd;
     ccuerrstr += "\n" ;
 
-    ushort channelBusy[18] = {0} ;
+    unsigned short channelBusy[18] = {0} ;
     for (unsigned short i = 0 ; i < 18 ; i ++)
       if (ccureg & (0x1 << i)) channelBusy[i] = 1 ;
 
# bla bla
