diff --git a/FecSoftwareV3_0/generic/include/FecExceptionHandler.h b/FecSoftwareV3_0/generic/include/FecExceptionHandler.h
index d12892e..ce95d47 100755
--- a/FecSoftwareV3_0/generic/include/FecExceptionHandler.h
+++ b/FecSoftwareV3_0/generic/include/FecExceptionHandler.h
@@ -778,7 +778,7 @@ class FecExceptionHandler {
    */
   static void setFrame ( tscType8 *dst, tscType8 *src, bool dAck = false ) {
 
-    unsigned int realSize = src[2] & FEC_LENGTH_2BYTES ? ((src[2] & 0x7F) << 8 | src[3] + 1 +3) : src[2]+3 ;
+    unsigned int realSize = src[2] & FEC_LENGTH_2BYTES ? (((src[2] & 0x7F) << 8) | (src[3] + 1 +3)) : src[2]+3 ;
     //                                                                                    ^ size is two bytes
     //                                                                                       ^ dst src length
     //                                                                                         ^ if dAck is set
diff --git a/FecSoftwareV3_0/generic/include/FecRingRegisters.h b/FecSoftwareV3_0/generic/include/FecRingRegisters.h
index 85184fa..5274970 100755
--- a/FecSoftwareV3_0/generic/include/FecRingRegisters.h
+++ b/FecSoftwareV3_0/generic/include/FecRingRegisters.h
@@ -462,7 +462,7 @@ class FecRingRegisters {
    */
   static std::string decodeFrame ( tscType8 *frame ) {
 
-    int realSize = frame[2] & FEC_LENGTH_2BYTES ? ((frame[2] & 0x7F) << 8 | frame[3] + 1) : frame[2] ;
+    int realSize = frame[2] & FEC_LENGTH_2BYTES ? (((frame[2] & 0x7F) << 8) | (frame[3] + 1)) : frame[2] ;
 
     if (realSize != 0) {
 
diff --git a/FecSoftwareV3_0/generic/include/ParameterDescription.h b/FecSoftwareV3_0/generic/include/ParameterDescription.h
index 7472b92..1f6c0c5 100755
--- a/FecSoftwareV3_0/generic/include/ParameterDescription.h
+++ b/FecSoftwareV3_0/generic/include/ParameterDescription.h
@@ -196,7 +196,6 @@ class ParameterDescription {
     tscType16 *val16 ;
     tscType32 *val32 ;
     double *valDble ;
-    std::string *valString ;
 
     switch (type_) {
     case INTEGER8:
@@ -223,7 +222,6 @@ class ParameterDescription {
       break ;
     case STRING:
       valConvert_ = new std::string (value_) ;
-      valString = (std::string *)valConvert_ ;
       break ;
     }
   }
