diff --git a/toprex/4.23/Makeshared.subdir b/toprex/4.23/Makeshared.subdir
index 9a832ec..acac5aa 100644
--- a/toprex/4.23/Makeshared.subdir
+++ b/toprex/4.23/Makeshared.subdir
@@ -42,7 +42,12 @@ objects += $(patsubst $(DIRNAME)/%.c,$(TMPDIR)/%.o,$(wildcard $(DIRNAME)/*.c))
 $(LIBNAME): $(objects)
 	@mkdir -p $(LIBDIR)
 	@echo creating the shared library $@
-	$(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@)
+	case ${ARCH} in \
+	  Linux-*) \
+	    $(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -shared -Wl,-soname,$(notdir $@) ;; \
+	  Darwin-*) \
+	    $(CXX) $(LDFLAGSSHARED) $(objects) -o $@ -L$(PYTHIA6_ROOT)/lib -lpythia6 -shared -dynamic -Wl,-commons -Wl,use_dylibs $(shell gfortran --print-file-name=libgfortran.a);; \
+	esac
 
 deps := $(patsubst $(DIRNAME)/%.F,$(TMPDIR)/%.d,$(wildcard $(DIRNAME)/*.F))
 
diff --git a/toprex/4.23/configure b/toprex/4.23/configure
index 1f7007c..df051f3 100755
--- a/toprex/4.23/configure
+++ b/toprex/4.23/configure
@@ -107,23 +107,25 @@ if [ $ARCH = Linux ]; then
   fi
 fi
 # Linux platform with gcc4: new Fortran90 compiler.
-if [ $ARCH = Linux-gcc4 ]; then
-  export FC=gfortran
-  export FFLAGS="${FFLAGS_OPT}"
-  export CFLAGS="${CFLAGS_OPT}"
-  export CXXFLAGS="${CXXFLAGS_OPT}"
-  export FLIBS="-lgfortran -lgfortranbegin"
-  if [ ${COMPMODE} = OPT ]; then
+case $ARCH in
+  *-gcc4)
+    export FC=gfortran
     export FFLAGS="${FFLAGS_OPT}"
     export CFLAGS="${CFLAGS_OPT}"
     export CXXFLAGS="${CXXFLAGS_OPT}"
-  fi
-  if [ ${COMPMODE} = DBG ]; then
-    export FFLAGS="${FFLAGS_DBG}"
-    export CFLAGS="${CFLAGS_DBG}"
-    export CXXFLAGS="${CXXFLAGS_DBG}"
-  fi
-fi
+    export FLIBS="-lgfortran -lgfortranbegin"
+    if [ ${COMPMODE} = OPT ]; then
+      export FFLAGS="${FFLAGS_OPT}"
+      export CFLAGS="${CFLAGS_OPT}"
+      export CXXFLAGS="${CXXFLAGS_OPT}"
+    fi
+    if [ ${COMPMODE} = DBG ]; then
+      export FFLAGS="${FFLAGS_DBG}"
+      export CFLAGS="${CFLAGS_DBG}"
+      export CXXFLAGS="${CXXFLAGS_DBG}"
+    fi
+  ;;
+esac
 
 #Platform & opt/dbg - independent flags and variables:
 
# bla bla
