--- orig.configure	2007-10-23 19:12:14.000000000 +0200
+++ configure	2007-10-23 19:13:19.000000000 +0200
@@ -128,8 +128,7 @@
 #Platform & opt/dbg - independent flags and variables:
 
 export FFLAGS="${FFLAGS} -fno-automatic -Df2cFortran -fno-second-underscore"
-export FFLAGS="${FFLAGS} -fbounds-check -finit-local-zero -fugly-logint"
-export FFLAGS="${FFLAGS} -fugly-complex -fno-globals"
+export FFLAGS="${FFLAGS} -fbounds-check"
 
 rm -f config.mk
 
# bla bla
