diff --git a/toprex/4.23/Makefile b/toprex/4.23/Makefile
index 8305219..2667835 100644
--- a/toprex/4.23/Makefile
+++ b/toprex/4.23/Makefile
@@ -47,10 +47,6 @@ all: libs
 
 
 libs:
-	@LIBNAME=$(LIBDIR)/libtoprex.so ; DIRNAME=src ; \
-	export LIBNAME ; export DIRNAME ; \
-	$(MAKE) -f Makeshared.subdir
-
 	@LIBNAME=$(LIBDIR)/archive/libtoprex.a ; DIRNAME=src ; \
 	export LIBNAME ; export DIRNAME ; \
 	$(MAKE) -f Makearchive.subdir
# bla bla
