--- makefile.orig	2014-10-02 10:34:09.000000000 -0500
+++ makefile	2014-10-02 10:40:15.622458716 -0500
@@ -1,5 +1,5 @@
-CFLAGS	:= -Ire2/ $(shell pkg-config --cflags sqlite3) -Wall -fPIC -ansi
-LDFLAGS := -Lre2/obj -lre2 $(shell pkg-config --libs sqlite3) -shared
+CFLAGS	:= -Ire2/ -I$(SQLITE_ROOT)/include -Wall -fPIC -ansi
+LDFLAGS := -Lre2/obj -lre2 -L$(SQLITE_ROOT)/lib -lsqlite3 -shared
 SOURCES := kmp.c scanstr.c varint.c hash.c expr.c match.c regexp.cpp cursor.c vtable.c trilite.c
 OBJECTS := $(patsubst %.cpp,%.o,$(patsubst %.c,%.o,$(SOURCES))) 
 all: debug
@@ -9,7 +9,7 @@
 release: LDFLAGS += -O3
 release: re2/obj/libre2.a libtrilite.so
 re2:
-	hg clone https://re2.googlecode.com/hg re2
+	curl https://re2.googlecode.com/files/re2-20140304.tgz | tar xfz -
 re2/obj/libre2.a: re2
 	$(MAKE) -C re2 CXXFLAGS='-Wall -O3 -pthread -fPIC'
 %.o: %.cpp

