diff --git a/callgrind/global.h b/callgrind/global.h
index d764e16..b71d15c 100644
--- a/callgrind/global.h
+++ b/callgrind/global.h
@@ -113,7 +113,7 @@ struct _CommandLineOptions {
 
 /* Size of various buffers used for storing strings */
 #define FILENAME_LEN                    VKI_PATH_MAX
-#define FN_NAME_LEN                    4096 /* for C++ code :-) */
+#define FN_NAME_LEN                     32768 /* for C++ code :-) */
 #define OBJ_NAME_LEN                    256
 #define COSTS_LEN                       512 /* at least 17x 64bit values */
 #define BUF_LEN                         512
