Index: massif/ms_main.c
===================================================================
--- massif/ms_main.c	(revision 7353)
+++ massif/ms_main.c	(working copy)
@@ -774,6 +774,11 @@
 #define MAX_OVERESTIMATE   50
 #define MAX_IPS            (MAX_DEPTH + MAX_OVERESTIMATE)
 
+// This is used for various buffers which can hold function names/IP
+// description.  Some C++ names can get really long so 1024 isn't big
+// enough.
+#define BUF_LEN   2048
+
 // Get the stack trace for an XCon, filtering out uninteresting entries:
 // alloc-fns and entries above alloc-fns, and entries below main-or-below-main.
 //   Eg:       alloc-fn1 / alloc-fn2 / a / b / main / (below main) / c
@@ -783,7 +788,6 @@
 static
 Int get_IPs( ThreadId tid, Bool is_custom_alloc, Addr ips[])
 {
-   #define BUF_LEN   1024
    Char buf[BUF_LEN];
    Int n_ips, i, n_alloc_fns_removed;
    Int overestimate;
@@ -1863,16 +1867,27 @@
 // The output file name.  Controlled by --massif-out-file.
 static Char* massif_out_file = NULL;
 
-#define FP_BUF_SIZE     1024
-Char FP_buf[FP_BUF_SIZE];
+Char FP_buf[BUF_LEN];
 
 // XXX: implement f{,n}printf in m_libcprint.c eventually, and use it here.
 // Then change Cachegrind to use it too.
 #define FP(format, args...) ({ \
-   VG_(snprintf)(FP_buf, FP_BUF_SIZE, format, ##args); \
+   VG_(snprintf)(FP_buf, BUF_LEN, format, ##args); \
+   FP_buf[BUF_LEN-1] = '\0';  /* Make sure the string is terminated. */ \
    VG_(write)(fd, (void*)FP_buf, VG_(strlen)(FP_buf)); \
 })
 
+// Same as FP, but guarantees a '\n' at the end.
+#define FPn(format, args...) ({ \
+   VG_(snprintf)(FP_buf, BUF_LEN, format, ##args); \
+   FP_buf[BUF_LEN-5] = '.';   /* "..." at the end make the truncation */ \
+   FP_buf[BUF_LEN-4] = '.';   /*  more obvious */ \
+   FP_buf[BUF_LEN-3] = '.'; \
+   FP_buf[BUF_LEN-2] = '\n';  /* Make sure the last char is a newline. */ \
+   FP_buf[BUF_LEN-1] = '\0';  /* Make sure the string is terminated. */ \
+   VG_(write)(fd, (void*)FP_buf, VG_(strlen)(FP_buf)); \
+})
+
 // Nb: uses a static buffer, each call trashes the last string returned.
 static Char* make_perc(ULong x, ULong y)
 {
@@ -1892,7 +1907,6 @@
                             Int depth_str_len,
                             SizeT snapshot_heap_szB, SizeT snapshot_total_szB)
 {
-   #define BUF_LEN   1024
    Int   i, n_insig_children_sxpts;
    Char* perc;
    Char  ip_desc_array[BUF_LEN];
@@ -1925,7 +1939,10 @@
          ip_desc = VG_(describe_IP)(sxpt->Sig.ip-1, ip_desc, BUF_LEN);
       }
       perc = make_perc(sxpt->szB, snapshot_total_szB);
-      FP("%sn%d: %lu %s\n",
+      // Nb: we deliberately use 'FPn', not 'FP'.  So if the ip_desc is
+      // too long (eg. due to a long C++ function name), it'll get
+      // truncated, but the '\n' is still there so its a valid file.
+      FPn("%sn%d: %lu %s\n",     
          depth_str, sxpt->Sig.n_children, sxpt->szB, ip_desc);
 
       // Indent.
