--- callgrind/dump.c.old	2007-03-12 17:00:01.000000000 +0100
+++ callgrind/dump.c	2007-03-12 16:33:10.000000000 +0100
@@ -57,7 +57,7 @@ EventMapping* CLG_(dumpmap) = 0;
  *  print_fn_pos, fprint_apos, fprint_fcost, fprint_jcc,
  *  fprint_fcc_ln, dump_run_info, dump_state_info
  */
-static Char outbuf[FILENAME_LEN + FN_NAME_LEN + OBJ_NAME_LEN];
+static Char outbuf[FILENAME_LEN + 256 * FN_NAME_LEN + OBJ_NAME_LEN];
 
 Int CLG_(get_dump_counter)(void)
 {
