--- orig.valgrind-3.3.0/coregrind/m_aspacemgr/aspacemgr-linux.c	2007-12-11 00:18:46.000000000 +0100
+++ valgrind-3.3.0/coregrind/m_aspacemgr/aspacemgr-linux.c	2007-12-14 01:43:33.000000000 +0100
@@ -262,13 +262,13 @@
 /* ------ start of STATE for the address-space manager ------ */
 
 /* Max number of segments we can track. */
-#define VG_N_SEGMENTS 5000
+#define VG_N_SEGMENTS 20000
 
 /* Max number of segment file names we can track. */
-#define VG_N_SEGNAMES 1000
+#define VG_N_SEGNAMES 4000
 
 /* Max length of a segment file name. */
-#define VG_MAX_SEGNAMELEN 1000
+#define VG_MAX_SEGNAMELEN 4000
 
 
 typedef
