diff --git a/CMakeLists.txt b/CMakeLists.txt
index dddf5f1..7b79c9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -307,6 +307,8 @@ elseif(Intel)
       list(APPEND CMAKE_CXX_FLAGS -std=c++11)
     elseif(CMAKE_CXX_STANDARD STREQUAL 14)
       list(APPEND CMAKE_CXX_FLAGS -std=c++14)
+    elseif(CMAKE_CXX_STANDARD STREQUAL 17)
+      list(APPEND CMAKE_CXX_FLAGS -std=c++17)
     else()
       message(FATAL_ERROR "Unsupported C++ standard requested")
     endif()
