diff -aur xdaq.orig/config/mfSet.coretools xdaq/config/mfSet.coretools
--- xdaq.orig/config/mfSet.coretools	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/config/mfSet.coretools	2010-02-05 14:24:20.000000000 +0100
@@ -11,8 +11,6 @@
 #########################################################################
 
 Packages=\
-	../config \
-	../java \
 	config \
 	xcept \
 	log/udpappender \
diff -aur xdaq.orig/config/mfSet.extern_coretools xdaq/config/mfSet.extern_coretools
--- xdaq.orig/config/mfSet.extern_coretools	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/config/mfSet.extern_coretools	2010-02-05 14:26:07.000000000 +0100
@@ -13,9 +13,6 @@
 Packages=\
 	extern/asyncresolv \
 	extern/i2o \
-	extern/xerces \
 	extern/xalan \
 	extern/log4cplus \
-	extern/cgicc \
-	extern/mimetic \
-	extern/tinyproxy
+	extern/cgicc
diff -aur xdaq.orig/config/mfSet.extern_powerpack xdaq/config/mfSet.extern_powerpack
--- xdaq.orig/config/mfSet.extern_powerpack	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/config/mfSet.extern_powerpack	2010-02-05 14:27:17.000000000 +0100
@@ -12,8 +12,5 @@
 
 Packages=\
 	extern/slp \
-	extern/appweb \
-	extern/pstreams \
-	extern/ssh \
-        extern/sqlite
+	extern/pstreams
 
diff -aur xdaq.orig/config/mfSet.general_worksuite xdaq/config/mfSet.general_worksuite
--- xdaq.orig/config/mfSet.general_worksuite	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/config/mfSet.general_worksuite	2010-02-05 14:47:39.000000000 +0100
@@ -12,12 +12,5 @@
 
 Packages=\
 	xdaq2rc \
-	interface/evb \
-	interface/shared \
-	pt/atcp \
-	rubuilder \
-	state/dashboard \
-	state/panel \
-	jobcontrol \
-	evf
+	pt/atcp
 
diff -aur xdaq.orig/config/mfSet.powerpack xdaq/config/mfSet.powerpack
--- xdaq.orig/config/mfSet.powerpack	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/config/mfSet.powerpack	2010-02-05 14:29:14.000000000 +0100
@@ -14,7 +14,6 @@
 	b2in/utils \
 	b2in/eventing \
 	pt/tcp \
-	pt/appweb \
 	xplore/utils \
 	xslp \
 	xplore \
diff -aur xdaq.orig/daq/extern/cgicc/Makefile xdaq/daq/extern/cgicc/Makefile
--- xdaq.orig/daq/extern/cgicc/Makefile	2010-02-05 14:20:20.000000000 +0100
+++ xdaq/daq/extern/cgicc/Makefile	2010-02-05 14:30:48.000000000 +0100
@@ -67,14 +67,14 @@
 
 all: cgicc-3.2.3/configure
 	cd ./cgicc-3.2.3; \
-	./configure --prefix $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM); \
+	./configure --prefix $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
 	make all ; make install
 
 _installall: install
 
 install:
 	cd cgicc-3.2.3; \
-	./configure --prefix $(INSTALL_PREFIX)/$(XDAQ_PLATFORM); \
+	./configure --prefix $(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
 	make; make install
 
 _cleanall: clean
diff -aur xdaq.orig/daq/extern/log4cplus/Makefile xdaq/daq/extern/log4cplus/Makefile
--- xdaq.orig/daq/extern/log4cplus/Makefile	2010-02-05 14:20:20.000000000 +0100
+++ xdaq/daq/extern/log4cplus/Makefile	2010-02-05 14:32:17.000000000 +0100
@@ -67,13 +67,13 @@
 	
 all: log4cplus-1.0.2/configure
 	cd ./log4cplus-1.0.2; \
-	./configure --prefix=$(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --exec-prefix=$(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM); \
+	./configure --prefix=$(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --exec-prefix=$(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
 	make; make install
 _installall: install
 
 install: log4cplus-1.0.2/configure
 	cd ./log4cplus-1.0.2; \
-	./configure --prefix=$(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --exec-prefix=$(INSTALL_PREFIX)/$(XDAQ_PLATFORM); \
+	./configure --prefix=$(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --exec-prefix=$(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
 	make; make install
 _cleanall: clean
 
diff -aur xdaq.orig/daq/extern/slp/Makefile xdaq/daq/extern/slp/Makefile
--- xdaq.orig/daq/extern/slp/Makefile	2010-02-05 14:20:20.000000000 +0100
+++ xdaq/daq/extern/slp/Makefile	2010-02-05 14:33:15.000000000 +0100
@@ -63,7 +63,7 @@
 	
 all: openslp-2.0.0.svn20070726/configure
 	cd ./openslp-2.0.0.svn20070726; \
-	./configure --with-pic --prefix $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM); \
+	./configure --with-pic --prefix $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
 	make;  make install
 	mkdir -p $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM)/etc
 	cp slp.conf $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM)/etc
@@ -72,7 +72,7 @@
 
 install: openslp-2.0.0.svn20070726/configure
 	cd openslp-2.0.0.svn20070726; \
-	./configure --with-pic --prefix $(INSTALL_PREFIX)/$(XDAQ_PLATFORM); \
+	./configure --with-pic --prefix $(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
 	make; make install
 	mkdir -p $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM)/etc
 	cp slp.conf $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM)/etc
diff -aur xdaq.orig/daq/extern/xalan/gcc43.patch xdaq/daq/extern/xalan/gcc43.patch
--- xdaq.orig/daq/extern/xalan/gcc43.patch	2010-01-27 15:03:22.000000000 +0100
+++ xdaq/daq/extern/xalan/gcc43.patch	2010-01-27 15:49:59.000000000 +0100
@@ -46,3 +46,14 @@
  
  #include <xercesc/sax/AttributeList.hpp>
  
+diff -aur xml-xalan/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp xml-xalan/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp
+--- xml-xalan/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2005-05-05 02:32:04.000000000 +0200
++++ xml-xalan/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2009-09-11 10:15:46.000000000 +0200
+@@ -34,6 +34,7 @@
+ 
+ 
+ 
++#include <string.h>
+ #include <functional>
+ #include <iterator>
+ 
diff -aur xdaq.orig/daq/i2o/utils/Makefile xdaq/daq/i2o/utils/Makefile
--- xdaq.orig/daq/i2o/utils/Makefile	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/daq/i2o/utils/Makefile	2010-02-05 14:50:42.000000000 +0100
@@ -73,7 +73,7 @@
 #
 # Compile the source files and create a shared library
 #
-DynamicLibrary=
+DynamicLibrary=i2outils
 StaticLibrary=i2outils
 Executables=
 
diff -aur xdaq.orig/daq/xdaq/Makefile xdaq/daq/xdaq/Makefile
--- xdaq.orig/daq/xdaq/Makefile	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/daq/xdaq/Makefile	2010-02-05 14:39:38.000000000 +0100
@@ -95,7 +95,8 @@
 	$(PT_LIB_PREFIX) \
 	$(XDAQ_LIB_PREFIX) \
 	$(I2O_LIB_PREFIX) \
-	$(XI2O_LIB_PREFIX)
+	$(XI2O_LIB_PREFIX) \
+	$(UUID_LIB_PREFIX)
 
 UserSourcePath = 
 UserCFlags =
