diff -arupN xdaq.orig/daq/toolbox/include/toolbox/mem/MemoryPartition.h xdaq/daq/toolbox/include/toolbox/mem/MemoryPartition.h
--- xdaq.orig/daq/toolbox/include/toolbox/mem/MemoryPartition.h	2012-03-08 12:35:57.000000000 +0100
+++ xdaq/daq/toolbox/include/toolbox/mem/MemoryPartition.h	2012-03-08 13:06:24.000000000 +0100
@@ -14,6 +14,7 @@
 #define _toolbox_mem_MemoryPartition_h
 
 #include <map>
+#include <unistd.h>
 
 #include "toolbox/mem/exception/Corruption.h"
 #include "toolbox/mem/exception/InvalidAddress.h"
diff -arupN xdaq.orig/daq/toolbox/include/toolbox/utils.h xdaq/daq/toolbox/include/toolbox/utils.h
--- xdaq.orig/daq/toolbox/include/toolbox/utils.h	2012-03-08 12:35:57.000000000 +0100
+++ xdaq/daq/toolbox/include/toolbox/utils.h	2012-03-08 13:24:49.000000000 +0100
@@ -15,6 +15,7 @@
 
 #include <string>
 #include <time.h>
+#include <unistd.h>
 
 #include "toolbox/exception/Exception.h"
 
diff -arupN xdaq.orig/daq/toolbox/src/linux/common/Task.cc xdaq/daq/toolbox/src/linux/common/Task.cc
--- xdaq.orig/daq/toolbox/src/linux/common/Task.cc	2012-03-08 12:35:57.000000000 +0100
+++ xdaq/daq/toolbox/src/linux/common/Task.cc	2012-03-08 12:51:09.000000000 +0100
@@ -13,6 +13,8 @@
 #include <string>
 #include <stdarg.h>
 #include <signal.h>
+#include <unistd.h>
+
 #include "toolbox/Task.h"
 #include "toolbox/TaskAttributes.h"
 #include "toolbox/TaskGroup.h"
diff -arupN xdaq.orig/daq/xoap/include/xoap/HTTPURLConnection.h xdaq/daq/xoap/include/xoap/HTTPURLConnection.h
--- xdaq.orig/daq/xoap/include/xoap/HTTPURLConnection.h	2012-03-08 12:35:57.000000000 +0100
+++ xdaq/daq/xoap/include/xoap/HTTPURLConnection.h	2012-03-08 13:40:12.000000000 +0100
@@ -15,6 +15,7 @@
 
 #include "xoap/exception/HTTPException.h"
 #include <string>
+#include <unistd.h>
 
 namespace xoap
 {
