--- xdaq.orig/config/mfDefs.linux	2010-02-05 14:20:21.000000000 +0100
+++ xdaq/config/mfDefs.linux	2010-02-05 15:04:16.000000000 +0100
@@ -40,22 +40,13 @@
 #
 # Compilation flags for 64 bit Linux
 #
-ifneq (,$(findstring x86_64,$(XDAQ_PLATFORM)))
 CFlags			= $(DebugFlags) -O1 -fPIC
 CCFlags			= -g -O1 -Wall -fPIC
-else
-CFlags			= $(DebugFlags) -O1
-CCFlags			= -g -O1 -Wall
-endif
 
 #
 # Linking flags for 64 bit Linux
 #
-ifneq (,$(findstring x86_64,$(XDAQ_PLATFORM)))
 DynamicLinkFlags	= -shared -fPIC
-else
-DynamicLinkFlags	= -shared
-endif
 
 ExecutableLinkFlags	= 
 StaticLinkFlags		=
