diff -Naur orig.xdaq/daq/toolbox/include/toolbox/mem/Usage.h xdaq/daq/toolbox/include/toolbox/mem/Usage.h
--- orig.xdaq/daq/toolbox/include/toolbox/mem/Usage.h	2009-10-26 16:23:41.000000000 +0100
+++ xdaq/daq/toolbox/include/toolbox/mem/Usage.h	2011-03-05 17:07:01.000000000 +0100
@@ -15,6 +15,7 @@
 
 #include <map>
 #include <vector>
+#include <stddef.h>
 
 namespace toolbox {
 
diff -Naur orig.xdaq/daq/xdaq/src/common/ApplicationContextImpl.cc xdaq/daq/xdaq/src/common/ApplicationContextImpl.cc
--- orig.xdaq/daq/xdaq/src/common/ApplicationContextImpl.cc	2010-12-10 11:07:18.000000000 +0100
+++ xdaq/daq/xdaq/src/common/ApplicationContextImpl.cc	2011-03-05 18:32:55.000000000 +0100
@@ -78,6 +78,10 @@
 using namespace log4cplus::helpers;
 using namespace log4cplus::spi;
 
+extern char * optarg;
+//extern int optind, optopt, opterr;
+extern int optopt;
+
 static void configureTransportReceivers(xdaq::ApplicationContextImpl* context, DOMDocument * doc, Logger logger)
 throw (xdaq::exception::ConfigurationError)
 {
@@ -278,9 +282,6 @@
 		{0, 0, 0, 0}
 	};
 	
-	extern char *optarg;
-	//extern int optind, optopt, opterr;
-	extern int optopt;
 	int option_index = -1; // index of long option
 	while ((c = getopt_long(argc, argv, ":z:h:p:l:u:c:e:s:d",long_options, &option_index)) != -1) 
 	{
