diff --git a/config/Makefile.rules b/config/Makefile.rules
index 1903d82..1cc4bdc 100755
--- a/config/Makefile.rules
+++ b/config/Makefile.rules
@@ -100,11 +100,13 @@ UserSourcePath = $(TestUserSourcePath)
 endif
 
 
-PackageLibDir		= $(SourceDir)/$(Package)/lib/$(XDAQ_OS)/$(XDAQ_PLATFORM)
-PackageBinDir		= $(SourceDir)/$(Package)/bin/$(XDAQ_OS)/$(XDAQ_PLATFORM)
-PackageTargetDir 	= $(PackageSourceDir)/$(XDAQ_OS)/$(XDAQ_PLATFORM)
-PackageIncludeDirs	= $(ConfigDir) $(SourceDir)/$(Package)/include/$(XDAQ_OS) $(SourceDir)/$(Package)/include $(IncludeDirs) \
-			 $(InstallationIncludeDirs)
+PackageLibDir		   = $(SourceDir)/$(Package)/lib/$(XDAQ_OS)/$(XDAQ_PLATFORM)
+PackageExternalLibDirs     = $(CMS_LIBRARY_DIRS)
+PackageBinDir		   = $(SourceDir)/$(Package)/bin/$(XDAQ_OS)/$(XDAQ_PLATFORM)
+PackageTargetDir 	   = $(PackageSourceDir)/$(XDAQ_OS)/$(XDAQ_PLATFORM)
+PackageIncludeDirs	   = $(ConfigDir) $(SourceDir)/$(Package)/include/$(XDAQ_OS) $(SourceDir)/$(Package)/include $(IncludeDirs) \
+			     $(InstallationIncludeDirs)
+PackageExternalIncludeDirs = $(CMS_INCLUDE_DIRS)
 endif
 
 LibInstallDir   	= $(INSTALL_PREFIX)/$(XDAQ_PLATFORM)/lib
@@ -143,10 +145,10 @@ JavaFlags		= -classpath $(ClassPath)
 
 
 # Add -I in front of every include directory
-IncludePaths		= $(PackageIncludeDirs:%=-I%)
+IncludePaths		= $(PackageIncludeDirs:%=-I%) $(PackageExternalIncludeDirs:%=-I%)
 
 # Add -L in front of every library directory
-LibraryPaths		=-L$(PackageLibDir) $(LibraryDirs:%=-L%) $(InstallationLibraryDirs:%=-L%)
+LibraryPaths		=-L$(PackageLibDir) $(PackageExternalLibDirs:%=-L%) $(LibraryDirs:%=-L%) $(InstallationLibraryDirs:%=-L%)
 DependentLibraryPaths	=-L$(PackageLibDir) $(DependentLibraryDirs:%=-L%) $(InstallationLibraryDirs:%=-L%)
 LibraryList		= $(SystemLibraries:%=-l%) $(Libraries:%=-l%)
 DependentLibraryList	= $(DependentLibraries:%=-l%)
# bla bla
