diff --git a/daq/toolbox/src/common/hexdump.cc b/daq/toolbox/src/common/hexdump.cc
index a11b270..78dfaa7 100755
--- a/daq/toolbox/src/common/hexdump.cc
+++ b/daq/toolbox/src/common/hexdump.cc
@@ -16,6 +16,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <string.h>
+#include <stdint.h>
 
 void toolbox::hexdump(std::ostream& out, std::string fmt, void* data, size_t size)
 {
diff --git a/daq/xdata/include/xdata/UnsignedLong.i b/daq/xdata/include/xdata/UnsignedLong.i
index 8365cea..743cad6 100755
--- a/daq/xdata/include/xdata/UnsignedLong.i
+++ b/daq/xdata/include/xdata/UnsignedLong.i
@@ -10,6 +10,7 @@
  * For the list of contributors see CREDITS.   			         *
  *************************************************************************/
 #include <stdlib.h>
+#include <stdint.h>
 
 template<class T> xdata::UnsignedLongType<T>::UnsignedLongType(T value): xdata::SimpleType<T>(value)
 {		
diff --git a/daq/xdata/include/xdata/exdr/StreamBuffer.h b/daq/xdata/include/xdata/exdr/StreamBuffer.h
index bf5d6fa..d58acba 100755
--- a/daq/xdata/include/xdata/exdr/StreamBuffer.h
+++ b/daq/xdata/include/xdata/exdr/StreamBuffer.h
@@ -15,6 +15,9 @@
 
 #include <rpc/types.h>
 #include <rpc/xdr.h>
+#include <stdint.h>
+#include <climits>
+#include <cstdlib>
 
 #include "xdata/exception/Exception.h"
 
diff --git a/daq/xdata/src/common/exdr/AutoSizeOutputStreamBuffer.cc b/daq/xdata/src/common/exdr/AutoSizeOutputStreamBuffer.cc
index ab59499..d1b023b 100755
--- a/daq/xdata/src/common/exdr/AutoSizeOutputStreamBuffer.cc
+++ b/daq/xdata/src/common/exdr/AutoSizeOutputStreamBuffer.cc
@@ -12,6 +12,7 @@
 
 #include "xdata/exdr/AutoSizeOutputStreamBuffer.h"
 #include <errno.h>
+#include <stdlib.h>
 
 // It starts from 4k and it grows in LOG2 e.g. 8, 16, 32 etc.
 xdata::exdr::AutoSizeOutputStreamBuffer::AutoSizeOutputStreamBuffer()
# bla bla
