diff --git a/xdaq/daq/extern/cgicc/Makefile b/xdaq/daq/extern/cgicc/Makefile
index 66e5932..120a2c8 100755
--- a/xdaq/daq/extern/cgicc/Makefile
+++ b/xdaq/daq/extern/cgicc/Makefile
@@ -68,14 +68,16 @@ cgicc-3.2.9/configure:
 
 all: cgicc-3.2.9/configure
 	cd ./cgicc-3.2.9; \
-	./configure --prefix $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
+	sed -i -e 's/^\(SUBDIRS = .*\) doc \(.*\)/\1 \2/' Makefile.in ; \
+	./configure --prefix $(BUILD_HOME)/$(Project)/$(Package)/$(XDAQ_PLATFORM) --disable-static --enable-shared --disable-demos; \
 	make all ; make install
 
 _installall: install
 
 install:
 	cd cgicc-3.2.9; \
-	./configure --prefix $(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --disable-static --enable-shared; \
+	sed -i -e 's/^\(SUBDIRS = .*\) doc \(.*\)/\1 \2/' Makefile.in ; \
+	./configure --prefix $(INSTALL_PREFIX)/$(XDAQ_PLATFORM) --disable-static --enable-shared --disable-demos; \
 	make; make install
 
 _cleanall: clean
