diff --git a/src/xercesc/util/regx/ParserForXMLSchema.cpp b/src/xercesc/util/regx/ParserForXMLSchema.cpp
index 59a2bda..9b38bc7 100644
--- a/src/xercesc/util/regx/ParserForXMLSchema.cpp
+++ b/src/xercesc/util/regx/ParserForXMLSchema.cpp
@@ -213,11 +213,11 @@ RangeToken* ParserForXMLSchema::parseCharacterClass(const bool) {
                     || (ch == chDash && getCharData() == chCloseSquare && firstLoop))) {
                 // if regex = [-] then invalid...
                 // '[', ']', '-' not allowed and should be esacaped
-                XMLCh chStr[] = { ch, chNull };
+                XMLCh chStr[] = { static_cast<XMLCh>(ch), chNull };
                 ThrowXMLwithMemMgr2(ParseException,XMLExcepts::Parser_CC6, chStr, chStr, getMemoryManager());
             }
             if (ch == chDash && getCharData() == chDash && getState() != REGX_T_BACKSOLIDUS && !wasDecoded) {
-                XMLCh chStr[] = { ch, chNull };
+                XMLCh chStr[] = { static_cast<XMLCh>(ch), chNull };
                 ThrowXMLwithMemMgr2(ParseException,XMLExcepts::Parser_CC6, chStr, chStr, getMemoryManager());
             }
 
@@ -242,7 +242,7 @@ RangeToken* ParserForXMLSchema::parseCharacterClass(const bool) {
                 else {
 
                     XMLInt32 rangeEnd = getCharData();
-                    XMLCh rangeEndStr[] = { rangeEnd, chNull };
+                    XMLCh rangeEndStr[] = { static_cast<XMLCh>(rangeEnd), chNull };
 
                     if (type == REGX_T_CHAR) {
 
@@ -259,7 +259,7 @@ RangeToken* ParserForXMLSchema::parseCharacterClass(const bool) {
                     processNext();
 
                     if (ch > rangeEnd) {
-                        XMLCh chStr[] = { ch, chNull };
+                        XMLCh chStr[] = { static_cast<XMLCh>(ch), chNull };
                         ThrowXMLwithMemMgr2(ParseException,XMLExcepts::Parser_Ope3, rangeEndStr, chStr, getMemoryManager());
                     }
 
@@ -482,7 +482,7 @@ XMLInt32 ParserForXMLSchema::decodeEscaped() {
         break;
     default:
 		{
-        XMLCh chString[] = {chBackSlash, ch, chNull};        
+        XMLCh chString[] = {chBackSlash, static_cast<XMLCh>(ch), chNull};        
         ThrowXMLwithMemMgr1(ParseException,XMLExcepts::Parser_Process2, chString, getMemoryManager());
         }
     }
