diff --git a/xgboost/include/xgboost/tree_updater.h b/xgboost/include/xgboost/tree_updater.h
index abbc7f3..5944816 100644
--- a/xgboost/include/xgboost/tree_updater.h
+++ b/xgboost/include/xgboost/tree_updater.h
@@ -16,9 +16,7 @@
 #include "./data.h"
 #include "./tree_model.h"
 
-#ifdef _MSC_VER
 #include <functional>
-#endif
 
 namespace xgboost {
 /*!
diff --git a/xgboost/src/data/sparse_batch_page.h b/xgboost/src/data/sparse_batch_page.h
index 85f7627..b9accec 100644
--- a/xgboost/src/data/sparse_batch_page.h
+++ b/xgboost/src/data/sparse_batch_page.h
@@ -23,9 +23,7 @@
 #include <thread>
 #endif
 
-#ifdef _MSC_VER
 #include <functional>
-#endif
 
 namespace xgboost {
 namespace data {
