diff --git a/xgboost/rabit/Makefile b/xgboost/rabit/Makefile
index 95dd98f..480618f 100644
--- a/xgboost/rabit/Makefile
+++ b/xgboost/rabit/Makefile
@@ -19,8 +19,10 @@ export LDFLAGS= -Llib -Wl,-rpath=/usr/local/lib/gcc6
 endif

 export WARNFLAGS= -Wall -Wextra -Wno-unused-parameter -Wno-unknown-pragmas -std=c++0x
-export CFLAGS = -O3 -msse2 $(WARNFLAGS)
-
+export CFLAGS = -O3 $(WARNFLAGS)
+ifeq ($(USE_SSE), 1)
+        CFLAGS += -msse2
+endif
 ifndef WITH_FPIC
        WITH_FPIC = 1
 endif
