diff --git a/xgboost/Makefile b/xgboost/Makefile
index 5cbfa57..f34fa72 100644
--- a/xgboost/Makefile
+++ b/xgboost/Makefile
@@ -48,8 +48,18 @@ export CXX = $(if $(shell which g++-5),g++-5,clang-omp++)
 endif
 
 export LDFLAGS= -pthread -lm $(ADD_LDFLAGS) $(DMLC_LDFLAGS) $(PLUGIN_LDFLAGS)
-export CFLAGS=  -std=c++0x -Wall -O3 -msse2  -Wno-unknown-pragmas -funroll-loops -Iinclude $(ADD_CFLAGS) $(PLUGIN_CFLAGS)
+export CFLAGS=  -std=c++0x -Wall -O3 -Wno-unknown-pragmas -funroll-loops -Iinclude $(ADD_CFLAGS) $(PLUGIN_CFLAGS)
 CFLAGS += -I$(DMLC_CORE)/include -I$(RABIT)/include
+
+ifeq ($(UNAME),Linux)
+ARCH=$(shell arch)
+ifeq ($(ARCH),x86_64)
+	CFLAGS+=-msse2
+endif
+else
+	CFLAGS+=-msse2
+endif
+
 #java include path
 export JAVAINCFLAGS = -I${JAVA_HOME}/include -I./java
 
diff --git a/xgboost/dmlc-core/CMakeLists.txt b/xgboost/dmlc-core/CMakeLists.txt
index ef71860..e956b97 100644
--- a/xgboost/dmlc-core/CMakeLists.txt
+++ b/xgboost/dmlc-core/CMakeLists.txt
@@ -66,7 +66,10 @@ else(MSVC)
   include(CheckCXXCompilerFlag)
   check_cxx_compiler_flag("-std=c++0x"    SUPPORT_CXX0X)
   check_cxx_compiler_flag("-msse2"        SUPPORT_MSSE2)
-  set(CMAKE_C_FLAGS "-O3 -Wall -msse2 -Wno-unknown-pragmas -std=c++0x -fPIC")
+  set(CMAKE_C_FLAGS "-O3 -Wall -Wno-unknown-pragmas -std=c++0x -fPIC")
+  if(SUPPORT_MSSE2)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2")
+  endif(SUPPORT_MSSE2)
   set(CMAKE_CXX_FLAGS ${CMAKE_C_FLAGS})
 endif(MSVC)
 
diff --git a/xgboost/dmlc-core/Makefile b/xgboost/dmlc-core/Makefile
index d3a62a7..4dcc2d8 100644
--- a/xgboost/dmlc-core/Makefile
+++ b/xgboost/dmlc-core/Makefile
@@ -11,7 +11,17 @@ include make/dmlc.mk
 
 # this is the common build script for dmlc lib
 export LDFLAGS= -pthread -lm
-export CFLAGS = -O3 -Wall -msse2  -Wno-unknown-pragmas -Iinclude  -std=c++0x
+export CFLAGS = -O3 -Wall -Wno-unknown-pragmas -Iinclude  -std=c++0x
+
+ifeq ($(UNAME),Linux)
+ARCH=$(shell arch)
+ifeq ($(ARCH),x86_64)
+	CFLAGS+=-msse2
+endif
+else
+	CFLAGS+=-msse2
+endif
+
 LDFLAGS+= $(DMLC_LDFLAGS)
 CFLAGS+= $(DMLC_CFLAGS)
 
diff --git a/xgboost/rabit/Makefile b/xgboost/rabit/Makefile
index d494b9f..f504d44 100644
--- a/xgboost/rabit/Makefile
+++ b/xgboost/rabit/Makefile
@@ -4,7 +4,17 @@ endif
 export MPICXX = mpicxx
 export LDFLAGS= -Llib -lrt
 export WARNFLAGS= -Wall -Wextra -Wno-unused-parameter -Wno-unknown-pragmas -std=c++0x
-export CFLAGS = -O3 -msse2 $(WARNFLAGS)
+export CFLAGS = -O3 $(WARNFLAGS)
+
+UNAME=$(shell uname)
+ifeq ($(UNAME),Linux)
+ARCH=$(shell arch)
+ifeq ($(ARCH),x86_64)
+	CFLAGS+=-msse2
+endif
+else
+	CFLAGS+=-msse2
+endif
 
 ifndef WITH_FPIC
 	WITH_FPIC = 1
