diff --git a/src/common/hist_util.cc b/src/common/hist_util.cc
index c14da59..d95649a 100644
--- a/src/common/hist_util.cc
+++ b/src/common/hist_util.cc
@@ -17,14 +17,7 @@
 #include "quantile.h"
 #include "../data/gradient_index.h"
 
-#if defined(XGBOOST_MM_PREFETCH_PRESENT)
-  #include <xmmintrin.h>
-  #define PREFETCH_READ_T0(addr) _mm_prefetch(reinterpret_cast<const char*>(addr), _MM_HINT_T0)
-#elif defined(XGBOOST_BUILTIN_PREFETCH_PRESENT)
-  #define PREFETCH_READ_T0(addr) __builtin_prefetch(reinterpret_cast<const char*>(addr), 0, 3)
-#else  // no SW pre-fetching available; PREFETCH_READ_T0 is no-op
-  #define PREFETCH_READ_T0(addr) do {} while (0)
-#endif  // defined(XGBOOST_MM_PREFETCH_PRESENT)
+#define PREFETCH_READ_T0(addr) do {} while (0)
 
 namespace xgboost {
 namespace common {
