diff --git a/src/XrdApps.cmake b/src/XrdApps.cmake
index e175a4e..1410749 100644
--- a/src/XrdApps.cmake
+++ b/src/XrdApps.cmake
@@ -1,4 +1,4 @@
-
+include_directories( ${ZLIB_INCLUDE_DIRS} )
 include( XRootDCommon )
 
 #-------------------------------------------------------------------------------
diff --git a/src/XrdClient.cmake b/src/XrdClient.cmake
index 5bf54d5..fc65b90 100644
--- a/src/XrdClient.cmake
+++ b/src/XrdClient.cmake
@@ -1,4 +1,4 @@
-
+include_directories( ${ZLIB_INCLUDE_DIRS} )
 include( XRootDCommon )
 
 #-------------------------------------------------------------------------------
