diff -arupN xrootd-3.1.0.orig/src/XrdClient/XrdCommandLine.cc xrootd-3.1.0/src/XrdClient/XrdCommandLine.cc
--- xrootd-3.1.0.orig/src/XrdClient/XrdCommandLine.cc	2012-04-06 11:57:15.000000000 +0200
+++ xrootd-3.1.0/src/XrdClient/XrdCommandLine.cc	2012-04-06 11:59:40.000000000 +0200
@@ -37,7 +37,7 @@
 #undef NOMACROS
 #endif
 
-#define XRDCLI_VERSION "(C) 2004-2010 by the Xrootd group. Xrootd version: "XrdVSTRING
+#define XRDCLI_VERSION "(C) 2004-2010 by the Xrootd group. Xrootd version: " XrdVSTRING
 
 //------------------------------------------------------------------------------
 // Some globals
diff -arupN xrootd-3.1.0.orig/src/XrdClient/Xrdcp.cc xrootd-3.1.0/src/XrdClient/Xrdcp.cc
--- xrootd-3.1.0.orig/src/XrdClient/Xrdcp.cc	2012-04-06 11:57:15.000000000 +0200
+++ xrootd-3.1.0/src/XrdClient/Xrdcp.cc	2012-04-06 11:58:40.000000000 +0200
@@ -89,7 +89,7 @@ struct XrdCpInfo {
 
 #define XRDCP_BLOCKSIZE          (8*1024*1024)
 #define XRDCP_XRDRASIZE          (30*XRDCP_BLOCKSIZE)
-#define XRDCP_VERSION            "(C) 2004-2011 by the XRootD collaboration. Version: "XrdVSTRING
+#define XRDCP_VERSION            "(C) 2004-2011 by the XRootD collaboration. Version: " XrdVSTRING
 
 ///////////////////////////////////////////////////////////////////////
 // Coming from parameters on the cmd line
