From d10c528900539891037566b5d26c26be1c662132 Mon Sep 17 00:00:00 2001
From: Lukasz Janyst <ljanyst@cern.ch>
Date: Thu, 03 Nov 2011 12:22:33 +0100
Subject: [bug #87880] Properly unpack the incomming data in the case when the actual readv vector length modulo READV_MAXCHUNKS was 1

---
diff --git a/src/XrdClient/XrdClient.cc b/src/XrdClient/XrdClient.cc
index 963f110..44ab171 100644
--- a/src/XrdClient/XrdClient.cc
+++ b/src/XrdClient/XrdClient.cc
@@ -763,7 +763,7 @@ kXR_int64 XrdClient::ReadV(char *buf, kXR_int64 *offsets, int *lens, int nbuf)
             fCounters.ReadVSubRequests++;
             fCounters.ReadVSubChunks++;
             fCounters.ReadVBytes += reqvect[startitem].len;
-            res = Read(buf, reqvect[startitem].offset, reqvect[startitem].len);
+            res = Read(buf+bytesread, reqvect[startitem].offset, reqvect[startitem].len);
             
          } else {
             // Asynchronous, res stays the same
--
cgit v0.9
