diff --git a/src/XrdHeaders.cmake b/src/XrdHeaders.cmake
index 6402a32..08c0cf2 100644
--- a/src/XrdHeaders.cmake
+++ b/src/XrdHeaders.cmake
@@ -95,6 +95,8 @@ set( XROOTD_PRIVATE_HEADERS
   XrdClient/XrdClientPhyConnection.hh
   XrdClient/XrdClientReadCache.hh
   XrdClient/XrdClientSock.hh
+  XrdClient/XrdClientProtocol.hh
+  XrdClient/XrdClientSid.hh
   XrdNet/XrdNetPeer.hh
   XrdNet/XrdNetBuffer.hh
   XrdOfs/XrdOfs.hh
@@ -102,7 +104,6 @@ set( XROOTD_PRIVATE_HEADERS
   XrdOfs/XrdOfsHandle.hh
   XrdOfs/XrdOfsTrace.hh
   XrdSys/XrdSysPriv.hh
-
   XrdOss/XrdOssApi.hh
   XrdOss/XrdOssConfig.hh
   XrdOss/XrdOssError.hh
