diff -Naur orig.xrootd/src/XrdBwm/XrdBwmHandle.cc xrootd/src/XrdBwm/XrdBwmHandle.cc
--- orig.xrootd/src/XrdBwm/XrdBwmHandle.cc	2009-02-12 19:03:52.000000000 +0100
+++ xrootd/src/XrdBwm/XrdBwmHandle.cc	2009-05-06 02:17:50.000000000 +0200
@@ -22,6 +22,7 @@
 #include "XrdSys/XrdSysPlatform.hh"
 
 #include "XProtocol/XProtocol.hh"
+#include <cstdio>
 
 /******************************************************************************/
 /*                        S t a t i c   O b j e c t s                         */
diff -Naur orig.xrootd/src/XrdCms/XrdCmsAdmin.cc xrootd/src/XrdCms/XrdCmsAdmin.cc
--- orig.xrootd/src/XrdCms/XrdCmsAdmin.cc	2009-02-12 19:03:52.000000000 +0100
+++ xrootd/src/XrdCms/XrdCmsAdmin.cc	2009-05-05 21:31:20.000000000 +0200
@@ -17,6 +17,7 @@
 #include <limits.h>
 #include <unistd.h>
 #include <sys/types.h>
+#include <cstdio>
 
 #include "XProtocol/YProtocol.hh"
 
diff -Naur orig.xrootd/src/XrdCms/XrdCmsCache.cc xrootd/src/XrdCms/XrdCmsCache.cc
--- orig.xrootd/src/XrdCms/XrdCmsCache.cc	2009-02-12 19:03:52.000000000 +0100
+++ xrootd/src/XrdCms/XrdCmsCache.cc	2009-05-05 21:39:21.000000000 +0200
@@ -24,6 +24,7 @@
 
 #include "Xrd/XrdJob.hh"
 #include "Xrd/XrdScheduler.hh"
+#include <cstdio>
 
 namespace XrdCms
 {
diff -Naur orig.xrootd/src/XrdCms/XrdCmsParser.cc xrootd/src/XrdCms/XrdCmsParser.cc
--- orig.xrootd/src/XrdCms/XrdCmsParser.cc	2009-02-12 19:03:52.000000000 +0100
+++ xrootd/src/XrdCms/XrdCmsParser.cc	2009-05-06 01:27:31.000000000 +0200
@@ -20,6 +20,7 @@
 #include <string.h>
 #include <netinet/in.h>
 #include <sys/types.h>
+#include <cstdio>
 
 #include "XrdCms/XrdCmsParser.hh"
 #include "XrdCms/XrdCmsRRData.hh"
diff -Naur orig.xrootd/src/XrdCms/XrdCmsResp.cc xrootd/src/XrdCms/XrdCmsResp.cc
--- orig.xrootd/src/XrdCms/XrdCmsResp.cc	2009-02-12 19:03:52.000000000 +0100
+++ xrootd/src/XrdCms/XrdCmsResp.cc	2009-05-06 01:57:58.000000000 +0200
@@ -26,6 +26,7 @@
 #include "XrdNet/XrdNetBuffer.hh"
 #include "XrdSfs/XrdSfsInterface.hh"
 #include "XrdSys/XrdSysError.hh"
+#include <cstdio>
 
 using namespace XrdCms;
  
