diff -Naur yoda/1.0.4/src/Histo2D.cc yoda-fixisnan/1.0.4/src/Histo2D.cc
--- yoda/1.0.4/src/Histo2D.cc	2013-06-07 00:05:51.000000000 +0200
+++ yoda-fixisnan/1.0.4/src/Histo2D.cc	2013-12-12 19:23:13.000000000 +0100
@@ -13,10 +13,10 @@
 
 
   void Histo2D::fill(double x, double y, double weight) {
-    if ( isnan(x) ) throw RangeError("X is NaN");
-    if ( isinf(x) ) throw RangeError("X is Inf");
-    if ( isnan(y) ) throw RangeError("Y is NaN");
-    if ( isinf(y) ) throw RangeError("Y is Inf");
+    if ( std::isnan(x) ) throw RangeError("X is NaN");
+    if ( std::isinf(x) ) throw RangeError("X is Inf");
+    if ( std::isnan(y) ) throw RangeError("Y is NaN");
+    if ( std::isinf(y) ) throw RangeError("Y is Inf");
     // Fill the overall distribution
     _axis.totalDbn().fill(x, y, weight);
     // Fill the bins and overflows
# bla bla
