diff --git a/pyext/yoda/pyroot_helpers.hh b/pyext/yoda/pyroot_helpers.hh
index b299a74..3e7f4d5 100644
--- a/pyext/yoda/pyroot_helpers.hh
+++ b/pyext/yoda/pyroot_helpers.hh
@@ -1,12 +1,12 @@
 #include "YODA/ROOTCnv.h"
 #include "Python.h"
 #include "YODA/Profile1D.h"
-#include "TPython.h"
+#include "CPyCppyy/API.h"
 
 
 // Get a PyROOT object from a ROOT one
 inline PyObject* root_to_py_owned(TObject* root_obj) {
-  return TPython::ObjectProxy_FromVoidPtr(root_obj, root_obj->ClassName());
+  return CPyCppyy::Instance_FromVoidPtr(root_obj, root_obj->ClassName());
   /// @todo Different signatures in different ROOT versions?
   //return TPython::ObjectProxy_FromVoidPtr(root_obj, root_obj->ClassName(), kFALSE);
 }
@@ -14,5 +14,5 @@ inline PyObject* root_to_py_owned(TObject* root_obj) {
 
 // Get the ROOT object in a PyROOT one
 inline TObject* py_owned_to_root(PyObject* pyroot_obj) {
-  return (TObject*) TPython::ObjectProxy_AsVoidPtr(pyroot_obj);
+  return (TObject*) CPyCppyy::Instance_AsVoidPtr(pyroot_obj);
 }
