diff --git a/pyext/yoda/pyroot_helpers.hh b/pyext/yoda/pyroot_helpers.hh
index 0e6a10e..184c5cd 100644
--- a/pyext/yoda/pyroot_helpers.hh
+++ b/pyext/yoda/pyroot_helpers.hh
@@ -1,7 +1,7 @@
 #include "YODA/ROOTCnv.h"
 #include "Python.h"
 #include "YODA/Profile1D.h"
-#include "TPython.h"
+#include "CPyCppyy/API.h"
 
 #ifndef ROOT_VERSION_CODE
 #define ROOT_VERSION_CODE 397313
@@ -16,7 +16,7 @@
 inline PyObject* root_to_py_owned(TObject* root_obj) {
   // Different signatures in different ROOT versions... *sigh*
   #if ROOT_VERSION_CODE >= ROOT_VERSION(6,22,0)
-  return TPython::CPPInstance_FromVoidPtr(root_obj, root_obj->ClassName());
+  return CPyCppyy::Instance_FromVoidPtr(root_obj, root_obj->ClassName());
   #elif ROOT_VERSION_CODE >= ROOT_VERSION(6,0,0)
   return TPython::ObjectProxy_FromVoidPtr(root_obj, root_obj->ClassName());
   #else
@@ -29,7 +29,7 @@ inline PyObject* root_to_py_owned(TObject* root_obj) {
 inline TObject* py_owned_to_root(PyObject* pyroot_obj) {
   // Different signatures in different ROOT versions... *sigh*
   #if ROOT_VERSION_CODE >= ROOT_VERSION(6,22,0)
-  return (TObject*) TPython::CPPInstance_AsVoidPtr(pyroot_obj);
+  return (TObject*) CPyCppyy::Instance_AsVoidPtr(pyroot_obj);
   #else
   return (TObject*) TPython::ObjectProxy_AsVoidPtr(pyroot_obj);
   #endif
