--- poetry-plugin-export-1.0.7/pyproject.toml.orig	2022-10-10 17:51:00.629913598 +0200
+++ poetry-plugin-export-1.0.7/pyproject.toml	2022-10-10 17:51:09.979835427 +0200
@@ -18,7 +18,6 @@
 
 [tool.poetry.dependencies]
 python = "^3.7"
-poetry = "^1.2.0"
 poetry-core = "^1.1.0"
 
 [tool.poetry.dev-dependencies]
--- poetry-plugin-export-1.0.7/setup.py.orig	2022-10-10 17:50:42.686063618 +0200
+++ poetry-plugin-export-1.0.7/setup.py	2022-10-10 17:50:52.395982438 +0200
@@ -11,7 +11,7 @@
 {'': ['*']}
 
 install_requires = \
-['poetry-core>=1.1.0,<2.0.0', 'poetry>=1.2.0,<2.0.0']
+['poetry-core>=1.1.0,<2.0.0']
 
 entry_points = \
 {'poetry.application.plugin': ['export = '
