diff --git a/CMakeLists.txt b/CMakeLists.txt
index aba57b4..8e4584c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,8 +170,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/include/aotriton/config.h
 
 # Kernel Storage V2 uses xz/LZMA for compression
 if(UNIX)
-  include(FindPkgConfig)
-  pkg_search_module(LZMA REQUIRED liblzma)
+  find_package(LibLZMA REQUIRED)
   add_library(lzma_interface INTERFACE)
   target_link_libraries(lzma_interface INTERFACE ${LZMA_LIBRARIES})
   target_link_directories(lzma_interface INTERFACE ${LZMA_LIBRARY_DIRS})
