diff --git a/CMakeLists.txt b/CMakeLists.txt
index aba57b4..2c3edac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,12 +170,11 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/include/aotriton/config.h
 
 # Kernel Storage V2 uses xz/LZMA for compression
 if(UNIX)
-  include(FindPkgConfig)
-  pkg_search_module(LZMA REQUIRED liblzma)
+  find_package(LibLZMA REQUIRED)
   add_library(lzma_interface INTERFACE)
-  target_link_libraries(lzma_interface INTERFACE ${LZMA_LIBRARIES})
-  target_link_directories(lzma_interface INTERFACE ${LZMA_LIBRARY_DIRS})
-  target_include_directories(lzma_interface INTERFACE ${LZMA_INCLUDE_DIRS})
+  target_link_libraries(lzma_interface INTERFACE ${LIBLZMA_LIBRARIES})
+  target_link_directories(lzma_interface INTERFACE ${LIBLZMA_LIBRARY_DIRS})
+  target_include_directories(lzma_interface INTERFACE ${LIBLZMA_INCLUDE_DIRS})
 else()
   find_package(liblzma REQUIRED CONFIG)
   if(NOT liblzma_FOUND)
